{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3rdtry.pdf               AMPtk 2015 insects.ipynb Amy_04202017.bam\r\n"
     ]
    }
   ],
   "source": [
    "!ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[11:00:37 AM]\u001b[0m: OS: MacOSX 10.12.4, 24 cores, ~ 67 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[11:00:38 AM]\u001b[0m: amptk v.0.9.1, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[11:00:38 AM]\u001b[0m: Converting Ion Torrent BAM file to FASTQ\n",
      "\u001b[92m[11:01:56 AM]\u001b[0m: Foward primer: AGGTCAACAAATCATAAAGATATTGG,  Rev comp'd rev primer: GGATTTGGAAATTGATTAGTWCC\n",
      "\u001b[92m[11:01:56 AM]\u001b[0m: Loading FASTQ Records\n",
      "\u001b[92m[11:02:03 AM]\u001b[0m: 6,474,485 reads (3.6 GB)\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[12:24:35 PM]\u001b[0m: Concatenating Demuxed Files\n",
      "\u001b[92m[12:25:37 PM]\u001b[0m: 6,474,485 total reads\n",
      "\u001b[92m[12:25:37 PM]\u001b[0m: 4,756,733 valid Barcode\n",
      "\u001b[92m[12:25:37 PM]\u001b[0m: 4,007,883 Fwd Primer found, 2,859,265 Rev Primer found\n",
      "\u001b[92m[12:25:37 PM]\u001b[0m: 153,351 discarded too short (< 50 bp)\n",
      "\u001b[92m[12:25:37 PM]\u001b[0m: 2,705,914 valid output reads\n",
      "\u001b[92m[12:25:53 PM]\u001b[0m: Found 75 barcoded samples\n",
      "                Sample:  Count\n",
      "                 BC.47:  68954\n",
      "                 BC.16:  63988\n",
      "                 BC.41:  56381\n",
      "                 BC.32:  55649\n",
      "                 BC.57:  53568\n",
      "                 BC.67:  51313\n",
      "                 BC.28:  51250\n",
      "                 BC.35:  50228\n",
      "                 BC.60:  48365\n",
      "                 BC.73:  48236\n",
      "                 BC.63:  46915\n",
      "                 BC.56:  45847\n",
      "                  BC.5:  45737\n",
      "                 BC.64:  45709\n",
      "                 BC.30:  45221\n",
      "                  BC.6:  45178\n",
      "                 BC.58:  44187\n",
      "                 BC.52:  43992\n",
      "                 BC.51:  43690\n",
      "                 BC.61:  43140\n",
      "                 BC.48:  42333\n",
      "                 BC.66:  42182\n",
      "                 BC.62:  42143\n",
      "                 BC.21:  41825\n",
      "                 BC.59:  41778\n",
      "                 BC.65:  41400\n",
      "                 BC.70:  41084\n",
      "                 BC.39:  40112\n",
      "                 BC.36:  40057\n",
      "                 BC.53:  39928\n",
      "                 BC.43:  39680\n",
      "                 BC.55:  39672\n",
      "                 BC.29:  39581\n",
      "                 BC.17:  39265\n",
      "                 BC.14:  38928\n",
      "                 BC.26:  38091\n",
      "                 BC.45:  37732\n",
      "                 BC.72:  37577\n",
      "                 BC.25:  37574\n",
      "                 BC.34:  36470\n",
      "                 BC.22:  36226\n",
      "                 BC.49:  35750\n",
      "                  BC.2:  35445\n",
      "                 BC.54:  35387\n",
      "                 BC.23:  34533\n",
      "                 BC.46:  34469\n",
      "                 BC.33:  33934\n",
      "                 BC.19:  33645\n",
      "                 BC.69:  33541\n",
      "                 BC.40:  33444\n",
      "                 BC.24:  33322\n",
      "                 BC.37:  33275\n",
      "                  BC.8:  32828\n",
      "                 BC.12:  32421\n",
      "                 BC.31:  32138\n",
      "                 BC.38:  31978\n",
      "                 BC.42:  31966\n",
      "                 BC.20:  31341\n",
      "                 BC.44:  30553\n",
      "                 BC.10:  30105\n",
      "                 BC.13:  29272\n",
      "                  BC.7:  28318\n",
      "                 BC.15:  27942\n",
      "                 BC.50:  26008\n",
      "                 BC.11:  24650\n",
      "                  BC.3:  22951\n",
      "                 BC.71:  21777\n",
      "                 BC.18:  19180\n",
      "                  BC.4:  18468\n",
      "                  BC.9:  11935\n",
      "                 BC.68:  6249\n",
      "                  BC.1:  6043\n",
      "                 BC.27:  1858\n",
      "                 BC.82:  1\n",
      "                 BC.94:  1\n",
      "\u001b[92m[12:25:53 PM]\u001b[0m: Output file:  amy2.demux.fq (1.0 GB)\n",
      "\u001b[92m[12:25:53 PM]\u001b[0m: Mapping file: amy2.mapping_file.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd: \u001b[0mamptk cluster -i amy2.demux.fq -o out\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#process data\n",
    "!amptk ion -i Amy_04202017.bam -o amy2 --full_length -f COI-F -r COI-R --cpus 5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removed 2 samples\r\n",
      "Kept 2705912 reads out of 2705914 total reads\r\n"
     ]
    }
   ],
   "source": [
    "#drop the few samples with only 1 read\n",
    "!amptk remove -i amy2.demux.fq -l BC.82 BC.94 -o amy2.clean.fq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[12:39:15 PM]\u001b[0m: OS: MacOSX 10.12.4, 24 cores, ~ 67 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[12:39:16 PM]\u001b[0m: amptk v.0.9.3, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[12:39:16 PM]\u001b[0m: R v3.3.1; DADA2 v1.5.0\n",
      "\u001b[92m[12:39:16 PM]\u001b[0m: Loading FASTQ Records\n",
      "\u001b[92m[12:39:21 PM]\u001b[0m: 2,705,912 reads (1.0 GB)\n",
      "\u001b[92m[12:39:57 PM]\u001b[0m: Quality Filtering, expected errors < 1.0\n",
      "\u001b[92m[12:40:38 PM]\u001b[0m: 2,120,705 reads passed\n",
      "\u001b[92m[12:40:38 PM]\u001b[0m: Splitting FASTQ file by Sample into individual files\n",
      "\u001b[92m[12:44:58 PM]\u001b[0m: Running DADA2 pipeline\n",
      "\u001b[92m[04:01:02 PM]\u001b[0m: 2,020 total inferred sequences (iSeqs)\n",
      "\u001b[92m[04:01:02 PM]\u001b[0m: 378 denovo chimeras removed\n",
      "\u001b[92m[04:01:02 PM]\u001b[0m: 1,642 valid iSeqs\n",
      "\u001b[92m[04:01:02 PM]\u001b[0m: Chimera Filtering (VSEARCH) using COI DB\n",
      "\u001b[92m[04:01:14 PM]\u001b[0m: 1,534 iSeqs passed, 108 ref chimeras removed\n",
      "\u001b[92m[04:01:14 PM]\u001b[0m: Mapping reads to DADA2 iSeqs\n",
      "\u001b[92m[04:03:59 PM]\u001b[0m: 2,705,912 reads mapped to iSeqs (100%)\n",
      "\u001b[92m[04:03:59 PM]\u001b[0m: Clustering iSeqs at 97% to generate biological OTUs\n",
      "\u001b[92m[04:04:00 PM]\u001b[0m: 993 OTUs generated\n",
      "\u001b[92m[04:04:00 PM]\u001b[0m: Mapping reads to OTUs\n",
      "\u001b[92m[04:06:03 PM]\u001b[0m: 2,705,912 reads mapped to OTUs (100%)\n",
      "-------------------------------------------------------\n",
      "DADA2 Script has Finished Successfully\n",
      "-------------------------------------------------------\n",
      "Inferred iSeqs: /Users/ngs/Downloads/amy2015/amy2.iSeqs.fa\n",
      "iSeq OTU Table: /Users/ngs/Downloads/amy2015/amy2.otu_table.txt\n",
      "Clustered OTUs: /Users/ngs/Downloads/amy2015/amy2.cluster.otus.fa\n",
      "OTU Table: /Users/ngs/Downloads/amy2015/amy2.cluster.otu_table.txt\n",
      "iSeqs 2 OTUs: /Users/ngs/Downloads/amy2015/amy2.iSeqs2clusters.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk filter -i amy2.cluster.otu_table.txt -f amy2.cluster.otus.fa -b <mock barcode>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!amptk dada2 -i amy2.clean.fq --pool --uchime_ref COI -o amy2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Just a quick note that the number of Reads mapped is wrong here in the output, apparently newer version of vsearch changed the output, but we mapped something like 95% of the reads, which is good"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[05:29:39 PM]\u001b[0m: OS: MacOSX 10.12.4, 24 cores, ~ 67 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[05:29:39 PM]\u001b[0m: amptk v.0.9.3, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[05:29:39 PM]\u001b[0m: Loading OTU table: amy2.cluster.otu_table.txt\n",
      "\u001b[92m[05:29:39 PM]\u001b[0m: OTU table contains 993 OTUs\n",
      "\u001b[92m[05:29:39 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[05:29:39 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[05:29:39 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[05:29:39 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[05:29:39 PM]\u001b[0m: Index bleed, samples into mock: 2.163000%.\n",
      "\u001b[92m[05:29:39 PM]\u001b[0m: Will use value of 2.200000% for index-bleed OTU filtering.\n",
      "\u001b[92m[05:29:43 PM]\u001b[0m: Filtering OTU table down to 971 OTUs\n",
      "\u001b[92m[05:29:43 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      amy2.cluster.stats.txt\n",
      "Sorted OTU table:     amy2.cluster.sorted.txt\n",
      "Normalized/filter:    amy2.cluster.normalized.txt\n",
      "Final Binary table:   amy2.cluster.final.binary.txt\n",
      "Final OTU table:      amy2.cluster.final.txt\n",
      "Filtered OTUs:        amy2.cluster.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f amy2.cluster.filtered.otus.fa -i amy2.cluster.final.txt -m mapping_file.txt -d ITS2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#Okay, now lets filter for index bleed\n",
    "!amptk filter -i amy2.cluster.otu_table.txt -f amy2.cluster.otus.fa -b BC.13 -m CFMR_insect_mock2.fa -c in"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[05:32:30 PM]\u001b[0m: OS: MacOSX 10.12.4, 24 cores, ~ 67 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[05:32:30 PM]\u001b[0m: amptk v.0.9.3, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[05:32:30 PM]\u001b[0m: Loading OTU table: amy2.cluster.otu_table.txt\n",
      "\u001b[92m[05:32:30 PM]\u001b[0m: OTU table contains 993 OTUs\n",
      "\u001b[92m[05:32:30 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[05:32:30 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[05:32:30 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[05:32:30 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[05:32:30 PM]\u001b[0m: Index bleed, mock into samples: 81.427668%.  Index bleed, samples into mock: 2.163000%.\n",
      "\u001b[92m[05:32:30 PM]\u001b[0m: Overwriting auto detect index-bleed, setting to 0.500000%\n",
      "\u001b[92m[05:32:34 PM]\u001b[0m: Filtering OTU table down to 973 OTUs\n",
      "\u001b[92m[05:32:34 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      amy2.cluster.stats.txt\n",
      "Sorted OTU table:     amy2.cluster.sorted.txt\n",
      "Normalized/filter:    amy2.cluster.normalized.txt\n",
      "Final Binary table:   amy2.cluster.final.binary.txt\n",
      "Final OTU table:      amy2.cluster.final.txt\n",
      "Filtered OTUs:        amy2.cluster.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f amy2.cluster.filtered.otus.fa -i amy2.cluster.final.txt -m mapping_file.txt -d ITS2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#I think the index bleed here is too high, looks like one of the mocks is either missing or is now a different OTU, \n",
    "#specifically looks like OTU10 and IM12 could be paralogs and perhaps DADA2 corrected IM12 to OTU12, not sure...\n",
    "!amptk filter -i amy2.cluster.otu_table.txt -f amy2.cluster.otus.fa -b BC.13 -m CFMR_insect_mock2.fa -p 0.005"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[05:51:59 PM]\u001b[0m: OS: MacOSX 10.12.4, 24 cores, ~ 67 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[05:52:00 PM]\u001b[0m: amptk v.0.9.3, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[05:52:00 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[05:52:00 PM]\u001b[0m: 973 OTUs\n",
      "\u001b[92m[05:52:00 PM]\u001b[0m: Global alignment OTUs with usearch_global (USEARCH)\n",
      "\u001b[92m[05:52:06 PM]\u001b[0m: Classifying OTUs with UTAX (USEARCH)\n",
      "\u001b[92m[05:52:08 PM]\u001b[0m: Classifying OTUs with SINTAX (USEARCH)\n",
      "\u001b[92m[05:54:03 PM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[05:54:03 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[05:54:06 PM]\u001b[0m: Taxonomy finished: amy2.cluster.taxonomy.txt\n",
      "\u001b[92m[05:54:06 PM]\u001b[0m: Classic OTU table with taxonomy: amy2.cluster.otu_table.taxonomy.txt\n",
      "\u001b[92m[05:54:10 PM]\u001b[0m: BIOM OTU table created: amy2.cluster.biom\n",
      "\u001b[92m[05:54:10 PM]\u001b[0m: OTUs with taxonomy: amy2.cluster.otus.taxonomy.fa\n",
      "\u001b[92m[05:54:10 PM]\u001b[0m: OTU phylogeny: amy2.cluster.tree.phy\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#assign taxonomy/build biom file\n",
    "!amptk taxonomy -f amy2.cluster.filtered.otus.fa -i amy2.cluster.final.txt -m mapping_file.fixed.txt -d COI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-3\n",
      "Loading required package: RColorBrewer\n",
      "[1] ‘1.19.1’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘1.8.4’\n",
      "[1] ‘2.4.3’\n",
      "[1] ‘1.1.2’\n",
      "[1] \"LinkerPrimerSequence\" \"phinchID\"             \"date\"                \n",
      "[4] \"ReversePrimer\"        \"BarcodeSequence\"      \"site\"                \n",
      "[7] \"species\"             \n",
      "[1] \"date\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 959 taxa and 72 samples ]\n",
      "sample_data() Sample Data:       [ 72 samples by 7 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 959 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 959 tips and 958 internal nodes ]\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 959 taxa and 72 samples ]\n",
      "sample_data() Sample Data:       [ 72 samples by 7 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 959 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 959 tips and 958 internal nodes ]\n",
      "Square root transformation\n",
      "Wisconsin double standardization\n",
      "Run 0 stress 0.2908844 \n",
      "Run 1 stress 0.2844554 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.07034705  max resid 0.2105799 \n",
      "Run 2 stress 0.2889712 \n",
      "Run 3 stress 0.291047 \n",
      "Run 4 stress 0.2912086 \n",
      "Run 5 stress 0.2913266 \n",
      "Run 6 stress 0.294269 \n",
      "Run 7 stress 0.2826805 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.03140901  max resid 0.1471979 \n",
      "Run 8 stress 0.2852215 \n",
      "Run 9 stress 0.2927922 \n",
      "Run 10 stress 0.2884564 \n",
      "Run 11 stress 0.2910609 \n",
      "Run 12 stress 0.291337 \n",
      "Run 13 stress 0.2864222 \n",
      "Run 14 stress 0.2973839 \n",
      "Run 15 stress 0.294316 \n",
      "Run 16 stress 0.2888426 \n",
      "Run 17 stress 0.293231 \n",
      "Run 18 stress 0.2864987 \n",
      "Run 19 stress 0.2904921 \n",
      "Run 20 stress 0.2983094 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "     1: no. of iterations >= maxit\n",
      "    19: stress ratio > sratmax\n",
      "Run 0 stress 0.2803105 \n",
      "Run 1 stress 0.2817585 \n",
      "Run 2 stress 0.2777144 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.03665603  max resid 0.1643014 \n",
      "Run 3 stress 0.2840573 \n",
      "Run 4 stress 0.2859817 \n",
      "Run 5 stress 0.2893536 \n",
      "Run 6 stress 0.2797279 \n",
      "Run 7 stress 0.2843909 \n",
      "Run 8 stress 0.2853144 \n",
      "Run 9 stress 0.2820567 \n",
      "Run 10 stress 0.2825594 \n",
      "Run 11 stress 0.2800273 \n",
      "Run 12 stress 0.278434 \n",
      "Run 13 stress 0.2951425 \n",
      "Run 14 stress 0.2941179 \n",
      "Run 15 stress 0.2813672 \n",
      "Run 16 stress 0.2834728 \n",
      "Run 17 stress 0.2917367 \n",
      "Run 18 stress 0.2805019 \n",
      "Run 19 stress 0.2820004 \n",
      "Run 20 stress 0.2786725 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "     1: no. of iterations >= maxit\n",
      "    19: stress ratio > sratmax\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "[1] \"site\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 959 taxa and 72 samples ]\n",
      "sample_data() Sample Data:       [ 72 samples by 7 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 959 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 959 tips and 958 internal nodes ]\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 959 taxa and 72 samples ]\n",
      "sample_data() Sample Data:       [ 72 samples by 7 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 959 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 959 tips and 958 internal nodes ]\n",
      "Square root transformation\n",
      "Wisconsin double standardization\n",
      "Run 0 stress 0.2908844 \n",
      "Run 1 stress 0.2916081 \n",
      "Run 2 stress 0.2923553 \n",
      "Run 3 stress 0.2933446 \n",
      "Run 4 stress 0.2945513 \n",
      "Run 5 stress 0.2872755 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.08301563  max resid 0.2054639 \n",
      "Run 6 stress 0.2967065 \n",
      "Run 7 stress 0.2882384 \n",
      "Run 8 stress 0.2891103 \n",
      "Run 9 stress 0.2827129 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.06103307  max resid 0.1998742 \n",
      "Run 10 stress 0.292036 \n",
      "Run 11 stress 0.2960472 \n",
      "Run 12 stress 0.2873817 \n",
      "Run 13 stress 0.291838 \n",
      "Run 14 stress 0.2895156 \n",
      "Run 15 stress 0.2967102 \n",
      "Run 16 stress 0.2942143 \n",
      "Run 17 stress 0.293693 \n",
      "Run 18 stress 0.2884903 \n",
      "Run 19 stress 0.2912688 \n",
      "Run 20 stress 0.282938 \n",
      "... Procrustes: rmse 0.02659933  max resid 0.147952 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n",
      "Run 0 stress 0.2803105 \n",
      "Run 1 stress 0.280287 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.02779764  max resid 0.1233123 \n",
      "Run 2 stress 0.2834578 \n",
      "Run 3 stress 0.2838837 \n",
      "Run 4 stress 0.2822394 \n",
      "Run 5 stress 0.2903038 \n",
      "Run 6 stress 0.2793363 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.05469222  max resid 0.2412075 \n",
      "Run 7 stress 0.287964 \n",
      "Run 8 stress 0.2851868 \n",
      "Run 9 stress 0.2843536 \n",
      "Run 10 stress 0.284401 \n",
      "Run 11 stress 0.277021 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.03940107  max resid 0.2512307 \n",
      "Run 12 stress 0.2889452 \n",
      "Run 13 stress 0.2771226 \n",
      "... Procrustes: rmse 0.009549401  max resid 0.04934499 \n",
      "Run 14 stress 0.2823368 \n",
      "Run 15 stress 0.2826758 \n",
      "Run 16 stress 0.2910829 \n",
      "Run 17 stress 0.2917999 \n",
      "Run 18 stress 0.2854281 \n",
      "Run 19 stress 0.2856669 \n",
      "Run 20 stress 0.2921098 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "[1] \"species\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 959 taxa and 72 samples ]\n",
      "sample_data() Sample Data:       [ 72 samples by 7 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 959 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 959 tips and 958 internal nodes ]\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 959 taxa and 72 samples ]\n",
      "sample_data() Sample Data:       [ 72 samples by 7 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 959 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 959 tips and 958 internal nodes ]\n",
      "Square root transformation\n",
      "Wisconsin double standardization\n",
      "Run 0 stress 0.2908844 \n",
      "Run 1 stress 0.2871375 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.08279016  max resid 0.1861186 \n",
      "Run 2 stress 0.288876 \n",
      "Run 3 stress 0.2851054 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.04394969  max resid 0.1484746 \n",
      "Run 4 stress 0.2897023 \n",
      "Run 5 stress 0.2949441 \n",
      "Run 6 stress 0.2848014 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.05246583  max resid 0.1622394 \n",
      "Run 7 stress 0.2838771 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.04217622  max resid 0.147571 \n",
      "Run 8 stress 0.2907795 \n",
      "Run 9 stress 0.2916153 \n",
      "Run 10 stress 0.2912451 \n",
      "Run 11 stress 0.2826829 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.02593791  max resid 0.1336617 \n",
      "Run 12 stress 0.2906142 \n",
      "Run 13 stress 0.2824747 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.03108203  max resid 0.1396178 \n",
      "Run 14 stress 0.2869602 \n",
      "Run 15 stress 0.2885948 \n",
      "Run 16 stress 0.2904178 \n",
      "Run 17 stress 0.2917609 \n",
      "Run 18 stress 0.2958194 \n",
      "Run 19 stress 0.2926708 \n",
      "Run 20 stress 0.2942403 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "     1: no. of iterations >= maxit\n",
      "    19: stress ratio > sratmax\n",
      "Run 0 stress 0.2803105 \n",
      "Run 1 stress 0.2911648 \n",
      "Run 2 stress 0.2863582 \n",
      "Run 3 stress 0.2770082 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.03442076  max resid 0.1549406 \n",
      "Run 4 stress 0.2801889 \n",
      "Run 5 stress 0.2792663 \n",
      "Run 6 stress 0.2905 \n",
      "Run 7 stress 0.2800371 \n",
      "Run 8 stress 0.2824582 \n",
      "Run 9 stress 0.2829027 \n",
      "Run 10 stress 0.2912888 \n",
      "Run 11 stress 0.2834035 \n",
      "Run 12 stress 0.2816853 \n",
      "Run 13 stress 0.2916384 \n",
      "Run 14 stress 0.2776857 \n",
      "Run 15 stress 0.2844705 \n",
      "Run 16 stress 0.2859477 \n",
      "Run 17 stress 0.2843826 \n",
      "Run 18 stress 0.2817134 \n",
      "Run 19 stress 0.2836683 \n",
      "Run 20 stress 0.2843417 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 144 rows containing missing values (geom_errorbar). \n",
      "2: Removed 144 rows containing missing values (geom_errorbar). \n",
      "3: Removed 144 rows containing missing values (geom_errorbar). \n"
     ]
    }
   ],
   "source": [
    "#okay, loop through some very quick stats\n",
    "!Rscript --vanilla phyloseq_graphs.R amy2.cluster.biom amy2.cluster.tree.phy phyloseq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
