{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[03:24:07 PM]\u001b[0m: OS: MacOSX 10.12.4, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[03:24:08 PM]\u001b[0m: amptk v.0.9.3, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[03:24:08 PM]\u001b[0m: Foward primer: AGGGTGCTCCTCACAGCCCTGTG,  Rev comp'd rev primer: CAGGAAATKYGCYGCGGGGACA\n",
      "\u001b[92m[03:24:08 PM]\u001b[0m: Loading FASTQ Records\n",
      "\u001b[92m[03:24:12 PM]\u001b[0m: 2,840,048 reads (1.7 GB)\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[03:54:50 PM]\u001b[0m: Concatenating Demuxed Files\n",
      "\u001b[92m[03:54:54 PM]\u001b[0m: 2,840,048 total reads\n",
      "\u001b[92m[03:54:54 PM]\u001b[0m: 2,517,886 valid Barcode\n",
      "\u001b[92m[03:54:54 PM]\u001b[0m: 2,321,440 Fwd Primer found, 326,493 Rev Primer found\n",
      "\u001b[92m[03:54:54 PM]\u001b[0m: 5,127 discarded too short (< 125 bp)\n",
      "\u001b[92m[03:54:54 PM]\u001b[0m: 321,366 valid output reads\n",
      "\u001b[92m[03:54:55 PM]\u001b[0m: Found 44 barcoded samples\n",
      "                Sample:  Count\n",
      "                 NY040:  26088\n",
      "                 NY012:  25348\n",
      "                 NY047:  21492\n",
      "                 WI091:  15646\n",
      "                 NY001:  13602\n",
      "                 NY017:  13353\n",
      "                 NY008:  12801\n",
      "                NY033B:  12262\n",
      "                 NY016:  12105\n",
      "                 NY004:  10955\n",
      "                 NY048:  10885\n",
      "                 NY014:  10617\n",
      "                 NY054:  9047\n",
      "                 WI048:  8906\n",
      "                 NY002:  7797\n",
      "                NY031B:  7772\n",
      "                 NY051:  7745\n",
      "                  NY34:  7706\n",
      "                 NY019:  7519\n",
      "                 NY015:  7497\n",
      "                 NY038:  7093\n",
      "                 NY041:  6819\n",
      "                NY034B:  6040\n",
      "                 NY046:  5570\n",
      "                NY035B:  5455\n",
      "                 WI029:  5065\n",
      "                 NY007:  5020\n",
      "                 NY032:  4615\n",
      "                 NY052:  4240\n",
      "                 NY053:  4021\n",
      "                 NY003:  3647\n",
      "                 NY010:  3195\n",
      "                 NY050:  2702\n",
      "                 NY049:  2230\n",
      "                 NY011:  1615\n",
      "                 NY044:  1354\n",
      "                 NY018:  920\n",
      "                 NY020:  710\n",
      "                 NY009:  525\n",
      "                 NY005:  483\n",
      "                 NY036:  292\n",
      "                 NY006:  277\n",
      "                 NY013:  211\n",
      "                 NY045:  124\n",
      "\u001b[92m[03:54:55 PM]\u001b[0m: Output file:  MHC2a.demux.fq (185.5 MB)\n",
      "\u001b[92m[03:54:55 PM]\u001b[0m: Mapping file: ../MHC2a.mapping_file.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd: \u001b[0mamptk cluster -i MHC2a.demux.fq -o out\n",
      "\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[03:55:04 PM]\u001b[0m: OS: MacOSX 10.12.4, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[03:55:04 PM]\u001b[0m: amptk v.0.9.3, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[03:55:04 PM]\u001b[0m: Foward primer: AGGGTGCTCCTCACAGCCCTGTG,  Rev comp'd rev primer: CAGGAAATKYGCYGCGGGGACA\n",
      "\u001b[92m[03:55:04 PM]\u001b[0m: Loading FASTQ Records\n",
      "\u001b[92m[03:55:07 PM]\u001b[0m: 2,296,224 reads (1.7 GB)\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[04:24:28 PM]\u001b[0m: Concatenating Demuxed Files\n",
      "\u001b[92m[04:24:31 PM]\u001b[0m: 2,296,224 total reads\n",
      "\u001b[92m[04:24:31 PM]\u001b[0m: 1,326,954 valid Barcode\n",
      "\u001b[92m[04:24:31 PM]\u001b[0m: 1,061,137 Fwd Primer found, 142,665 Rev Primer found\n",
      "\u001b[92m[04:24:31 PM]\u001b[0m: 10,071 discarded too short (< 125 bp)\n",
      "\u001b[92m[04:24:31 PM]\u001b[0m: 132,594 valid output reads\n",
      "\u001b[92m[04:24:32 PM]\u001b[0m: Found 43 barcoded samples\n",
      "                Sample:  Count\n",
      "                 NY032:  46546\n",
      "                 NY053:  17398\n",
      "                  NY34:  5642\n",
      "                 NY051:  5413\n",
      "                 NY047:  4480\n",
      "                 WI091:  3209\n",
      "                 NY046:  2631\n",
      "                 NY020:  2554\n",
      "                NY031B:  2528\n",
      "                 NY001:  2476\n",
      "                 NY038:  2367\n",
      "                 NY015:  2098\n",
      "                 NY012:  2028\n",
      "                 WI048:  1802\n",
      "                 NY004:  1796\n",
      "                 NY016:  1795\n",
      "                 NY019:  1572\n",
      "                 NY040:  1509\n",
      "                 NY002:  1491\n",
      "                NY033B:  1467\n",
      "                 NY014:  1446\n",
      "                 NY010:  1408\n",
      "                NY034B:  1397\n",
      "                 NY017:  1388\n",
      "                 NY009:  1258\n",
      "                 NY011:  1239\n",
      "                 NY008:  1175\n",
      "                 NY007:  1160\n",
      "                 NY045:  1070\n",
      "                 NY013:  1041\n",
      "                 NY005:  1015\n",
      "                 NY054:  989\n",
      "                 NY003:  936\n",
      "                 NY018:  880\n",
      "                 NY044:  858\n",
      "                 NY048:  803\n",
      "                 WI029:  673\n",
      "                 NY050:  663\n",
      "                 NY052:  609\n",
      "                 NY041:  608\n",
      "                 NY036:  482\n",
      "                NY035B:  382\n",
      "                 NY006:  312\n",
      "\u001b[92m[04:24:32 PM]\u001b[0m: Output file:  MHC2b.demux.fq (76.0 MB)\n",
      "\u001b[92m[04:24:32 PM]\u001b[0m: Mapping file: ../MHC2b.mapping_file.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd: \u001b[0mamptk cluster -i MHC2b.demux.fq -o out\n",
      "\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[04:24:36 PM]\u001b[0m: OS: MacOSX 10.12.4, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[04:24:36 PM]\u001b[0m: amptk v.0.9.3, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[04:24:36 PM]\u001b[0m: Foward primer: AGGGTGCTCCTCACAGCCCTGTG,  Rev comp'd rev primer: CAGGAAATKYGCYGCGGGGACA\n",
      "\u001b[92m[04:24:36 PM]\u001b[0m: Loading FASTQ Records\n",
      "\u001b[92m[04:24:39 PM]\u001b[0m: 2,471,729 reads (1.5 GB)\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[04:59:19 PM]\u001b[0m: Concatenating Demuxed Files\n",
      "\u001b[92m[04:59:23 PM]\u001b[0m: 2,471,729 total reads\n",
      "\u001b[92m[04:59:23 PM]\u001b[0m: 2,072,227 valid Barcode\n",
      "\u001b[92m[04:59:23 PM]\u001b[0m: 1,886,298 Fwd Primer found, 233,396 Rev Primer found\n",
      "\u001b[92m[04:59:23 PM]\u001b[0m: 1,714 discarded too short (< 125 bp)\n",
      "\u001b[92m[04:59:23 PM]\u001b[0m: 231,682 valid output reads\n",
      "\u001b[92m[04:59:24 PM]\u001b[0m: Found 42 barcoded samples\n",
      "                Sample:  Count\n",
      "                 PA012:  17703\n",
      "                 VT006:  15094\n",
      "                 PA001:  13886\n",
      "                 VT018:  13180\n",
      "                 VT016:  12810\n",
      "                 PA018:  11673\n",
      "                 VT005:  10086\n",
      "                 VT010:  9803\n",
      "                 VT017:  9675\n",
      "                 VT014:  9482\n",
      "                 PA007:  8708\n",
      "                 VT001:  8606\n",
      "                 VT011:  7868\n",
      "                 PA002:  7715\n",
      "                 VT013:  7551\n",
      "                 PA015:  6635\n",
      "                 PA020:  5737\n",
      "                 PA011:  5573\n",
      "                 MI126:  5026\n",
      "                 PA010:  4821\n",
      "                 PA017:  4707\n",
      "                 PA019:  4564\n",
      "                 PA016:  3992\n",
      "                 PA004:  3449\n",
      "                 NY060:  3389\n",
      "                 VT004:  3200\n",
      "                 PA009:  2605\n",
      "                 PA013:  2534\n",
      "                 VT012:  2456\n",
      "                 VT003:  2275\n",
      "                 VT015:  1560\n",
      "                 VT002:  1221\n",
      "                 MI127:  1139\n",
      "                 VT007:  819\n",
      "                 NY058:  737\n",
      "                 PA003:  524\n",
      "                 PA014:  507\n",
      "                 PA006:  119\n",
      "                 PA008:  96\n",
      "                 VT019:  64\n",
      "                 VT020:  54\n",
      "                 PA005:  39\n",
      "\u001b[92m[04:59:24 PM]\u001b[0m: Output file:  MHC3.demux.fq (133.7 MB)\n",
      "\u001b[92m[04:59:24 PM]\u001b[0m: Mapping file: ../MHC3.mapping_file.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd: \u001b[0mamptk cluster -i MHC3.demux.fq -o out\n",
      "\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[04:59:28 PM]\u001b[0m: OS: MacOSX 10.12.4, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[04:59:28 PM]\u001b[0m: amptk v.0.9.3, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[04:59:28 PM]\u001b[0m: Foward primer: AGGGTGCTCCTCACAGCCCTGTG,  Rev comp'd rev primer: CAGGAAATKYGCYGCGGGGACA\n",
      "\u001b[92m[04:59:28 PM]\u001b[0m: Loading FASTQ Records\n",
      "\u001b[92m[04:59:31 PM]\u001b[0m: 2,337,635 reads (1.5 GB)\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[05:31:23 PM]\u001b[0m: Concatenating Demuxed Files\n",
      "\u001b[92m[05:31:26 PM]\u001b[0m: 2,337,635 total reads\n",
      "\u001b[92m[05:31:26 PM]\u001b[0m: 1,909,629 valid Barcode\n",
      "\u001b[92m[05:31:26 PM]\u001b[0m: 1,717,114 Fwd Primer found, 215,521 Rev Primer found\n",
      "\u001b[92m[05:31:26 PM]\u001b[0m: 1,601 discarded too short (< 125 bp)\n",
      "\u001b[92m[05:31:26 PM]\u001b[0m: 213,920 valid output reads\n",
      "\u001b[92m[05:31:26 PM]\u001b[0m: Found 49 barcoded samples\n",
      "                Sample:  Count\n",
      "                 WI054:  14924\n",
      "                 MI130:  13828\n",
      "                 WI025:  11572\n",
      "                 WI030:  11334\n",
      "                 WI033:  10243\n",
      "                 MI139:  10134\n",
      "                 WI035:  9944\n",
      "                 MI134:  9327\n",
      "                 WI026:  8491\n",
      "                NY032B:  7100\n",
      "                 WI088:  6975\n",
      "                 WI051:  6963\n",
      "                 VT009:  6738\n",
      "                 MI136:  6389\n",
      "                 MI144:  6060\n",
      "                 MI132:  5548\n",
      "                 WI049:  5228\n",
      "                 WI089:  4626\n",
      "                 WI028:  3996\n",
      "                 WI032:  3837\n",
      "                 WI087:  3818\n",
      "                 WI027:  3531\n",
      "                 MI143:  3508\n",
      "                 WI090:  3491\n",
      "                 MI128:  3085\n",
      "                 WI031:  2867\n",
      "                 MI133:  2748\n",
      "                 WI034:  2688\n",
      "                 MI129:  2594\n",
      "                 WI045:  2492\n",
      "                 MI141:  2210\n",
      "                 MI138:  2157\n",
      "                 MI140:  2090\n",
      "                 MI135:  2074\n",
      "                 MI137:  1916\n",
      "                 WI093:  1609\n",
      "                 WI053:  1443\n",
      "                 WI047:  1271\n",
      "                 MI145:  1012\n",
      "                 WI095:  913\n",
      "                 NY042:  594\n",
      "                 WI046:  580\n",
      "                 WI094:  490\n",
      "                 WI086:  472\n",
      "                 WI052:  297\n",
      "                 WI050:  279\n",
      "                 WI092:  188\n",
      "                 MI142:  130\n",
      "                 MI131:  116\n",
      "\u001b[92m[05:31:26 PM]\u001b[0m: Output file:  MHC4.demux.fq (123.3 MB)\n",
      "\u001b[92m[05:31:26 PM]\u001b[0m: Mapping file: ../MHC4.mapping_file.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd: \u001b[0mamptk cluster -i MHC4.demux.fq -o out\n",
      "\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[05:31:29 PM]\u001b[0m: OS: MacOSX 10.12.4, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[05:31:29 PM]\u001b[0m: amptk v.0.9.3, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[05:31:29 PM]\u001b[0m: Foward primer: AGGGTGCTCCTCACAGCCCTGTG,  Rev comp'd rev primer: CAGGAAATKYGCYGCGGGGACA\n",
      "\u001b[92m[05:31:29 PM]\u001b[0m: Loading FASTQ Records\n",
      "\u001b[92m[05:31:33 PM]\u001b[0m: 2,882,965 reads (2.3 GB)\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[06:38:35 PM]\u001b[0m: Concatenating Demuxed Files\n",
      "\u001b[92m[06:38:41 PM]\u001b[0m: 2,882,965 total reads\n",
      "\u001b[92m[06:38:41 PM]\u001b[0m: 2,432,690 valid Barcode\n",
      "\u001b[92m[06:38:41 PM]\u001b[0m: 2,224,115 Fwd Primer found, 282,257 Rev Primer found\n",
      "\u001b[92m[06:38:41 PM]\u001b[0m: 2,849 discarded too short (< 125 bp)\n",
      "\u001b[92m[06:38:41 PM]\u001b[0m: 279,408 valid output reads\n",
      "\u001b[92m[06:38:42 PM]\u001b[0m: Found 46 barcoded samples\n",
      "                Sample:  Count\n",
      "                 WI107:  20334\n",
      "                 WI098:  16590\n",
      "                 WI105:  14021\n",
      "                 WI169:  13544\n",
      "                 WI076:  13365\n",
      "                 WI078:  12562\n",
      "                 WI062:  12531\n",
      "                 WI166:  12342\n",
      "                 WI084:  9679\n",
      "                 WI171:  9252\n",
      "                 WI104:  8607\n",
      "                 WI103:  8474\n",
      "                 WI083:  8356\n",
      "                 WI168:  8124\n",
      "                 WI064:  7663\n",
      "                 WI081:  6724\n",
      "                 WI097:  6544\n",
      "                 WI066:  6282\n",
      "                 WI101:  6243\n",
      "                 WI163:  5717\n",
      "                 WI067:  5711\n",
      "                 WI073:  5079\n",
      "                 IL030:  4771\n",
      "                 WI057:  4703\n",
      "                 IL020:  4668\n",
      "                 WI075:  4509\n",
      "                 WI058:  4461\n",
      "                 WI072:  3717\n",
      "                 WI099:  3537\n",
      "                 WI165:  3531\n",
      "                 IL032:  3267\n",
      "                 WI106:  3190\n",
      "                 WI065:  2907\n",
      "                 IL024:  2894\n",
      "                 WI100:  2661\n",
      "                 WI068:  2433\n",
      "                 WI096:  2373\n",
      "                 WI170:  1578\n",
      "                 IL022:  1553\n",
      "                 WI080:  1395\n",
      "                 WI172:  1387\n",
      "                 WI167:  1226\n",
      "                 WI102:  394\n",
      "                 WI164:  264\n",
      "                 IL016:  171\n",
      "                 IL018:  74\n",
      "\u001b[92m[06:38:42 PM]\u001b[0m: Output file:  MHC5.demux.fq (166.0 MB)\n",
      "\u001b[92m[06:38:42 PM]\u001b[0m: Mapping file: ../MHC5.mapping_file.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd: \u001b[0mamptk cluster -i MHC5.demux.fq -o out\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#First step is to demultiplex each of the 5 runs. Will use AMPtk to demultiplex the data\n",
    "#this will keep only amplicons where both forward and reverse primers are found, as well as discard reads < 125 bp\n",
    "!for i in MHC2a MHC2b MHC3 MHC4 MHC5; \\\n",
    "    do amptk ion -i ../$i.fastq -m ../$i.mapping_file.txt -o $i --full_length --min_len 125; \\\n",
    "    done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#concatenate all files together\n",
    "!cat mhc2a.demux.fq mhc2b.demux.fq mhc3.demux.fq mhc4.demux.fq mhc5.demux.fq > mhc.all.demux.fq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----------------------------------\n",
      "Found 181 barcoded samples\n",
      "              Sample:  Count\n",
      "               NY032:  51161\n",
      "               NY040:  27597\n",
      "               NY012:  27376\n",
      "               NY047:  25972\n",
      "               NY053:  21419\n",
      "               WI107:  20334\n",
      "               WI091:  18855\n",
      "               PA012:  17703\n",
      "               WI098:  16590\n",
      "               NY001:  16078\n",
      "               VT006:  15094\n",
      "               WI054:  14924\n",
      "               NY017:  14741\n",
      "               WI105:  14021\n",
      "               NY008:  13976\n",
      "               NY016:  13900\n",
      "               PA001:  13886\n",
      "               MI130:  13828\n",
      "              NY033B:  13729\n",
      "               WI169:  13544\n",
      "               WI076:  13365\n",
      "                NY34:  13348\n",
      "               VT018:  13180\n",
      "               NY051:  13158\n",
      "               VT016:  12810\n",
      "               NY004:  12751\n",
      "               WI078:  12562\n",
      "               WI062:  12531\n",
      "               WI166:  12342\n",
      "               NY014:  12063\n",
      "               NY048:  11688\n",
      "               PA018:  11673\n",
      "               WI025:  11572\n",
      "               WI030:  11334\n",
      "               WI048:  10708\n",
      "              NY031B:  10300\n",
      "               WI033:  10243\n",
      "               MI139:  10134\n",
      "               VT005:  10086\n",
      "               NY054:  10036\n",
      "               WI035:  9944\n",
      "               VT010:  9803\n",
      "               WI084:  9679\n",
      "               VT017:  9675\n",
      "               NY015:  9595\n",
      "               VT014:  9482\n",
      "               NY038:  9460\n",
      "               MI134:  9327\n",
      "               NY002:  9288\n",
      "               WI171:  9252\n",
      "               NY019:  9091\n",
      "               PA007:  8708\n",
      "               WI104:  8607\n",
      "               VT001:  8606\n",
      "               WI026:  8491\n",
      "               WI103:  8474\n",
      "               WI083:  8356\n",
      "               NY046:  8201\n",
      "               WI168:  8124\n",
      "               VT011:  7868\n",
      "               PA002:  7715\n",
      "               WI064:  7663\n",
      "               VT013:  7551\n",
      "              NY034B:  7437\n",
      "               NY041:  7427\n",
      "              NY032B:  7100\n",
      "               WI088:  6975\n",
      "               WI051:  6963\n",
      "               VT009:  6738\n",
      "               WI081:  6724\n",
      "               PA015:  6635\n",
      "               WI097:  6544\n",
      "               MI136:  6389\n",
      "               WI066:  6282\n",
      "               WI101:  6243\n",
      "               NY007:  6180\n",
      "               MI144:  6060\n",
      "              NY035B:  5837\n",
      "               WI029:  5738\n",
      "               PA020:  5737\n",
      "               WI163:  5717\n",
      "               WI067:  5711\n",
      "               PA011:  5573\n",
      "               MI132:  5548\n",
      "               WI049:  5228\n",
      "               WI073:  5079\n",
      "               MI126:  5026\n",
      "               NY052:  4849\n",
      "               PA010:  4821\n",
      "               IL030:  4771\n",
      "               PA017:  4707\n",
      "               WI057:  4703\n",
      "               IL020:  4668\n",
      "               WI089:  4626\n",
      "               NY010:  4603\n",
      "               NY003:  4583\n",
      "               PA019:  4564\n",
      "               WI075:  4509\n",
      "               WI058:  4461\n",
      "               WI028:  3996\n",
      "               PA016:  3992\n",
      "               WI032:  3837\n",
      "               WI087:  3818\n",
      "               WI072:  3717\n",
      "               WI099:  3537\n",
      "               WI027:  3531\n",
      "               WI165:  3531\n",
      "               MI143:  3508\n",
      "               WI090:  3491\n",
      "               PA004:  3449\n",
      "               NY060:  3389\n",
      "               NY050:  3365\n",
      "               IL032:  3267\n",
      "               NY020:  3264\n",
      "               VT004:  3200\n",
      "               WI106:  3190\n",
      "               MI128:  3085\n",
      "               WI065:  2907\n",
      "               IL024:  2894\n",
      "               WI031:  2867\n",
      "               NY011:  2854\n",
      "               MI133:  2748\n",
      "               WI034:  2688\n",
      "               WI100:  2661\n",
      "               PA009:  2605\n",
      "               MI129:  2594\n",
      "               PA013:  2534\n",
      "               WI045:  2492\n",
      "               VT012:  2456\n",
      "               WI068:  2433\n",
      "               WI096:  2373\n",
      "               VT003:  2275\n",
      "               NY049:  2230\n",
      "               NY044:  2212\n",
      "               MI141:  2210\n",
      "               MI138:  2157\n",
      "               MI140:  2090\n",
      "               MI135:  2074\n",
      "               MI137:  1916\n",
      "               NY018:  1800\n",
      "               NY009:  1783\n",
      "               WI093:  1609\n",
      "               WI170:  1578\n",
      "               VT015:  1560\n",
      "               IL022:  1553\n",
      "               NY005:  1498\n",
      "               WI053:  1443\n",
      "               WI080:  1395\n",
      "               WI172:  1387\n",
      "               WI047:  1271\n",
      "               NY013:  1252\n",
      "               WI167:  1226\n",
      "               VT002:  1221\n",
      "               NY045:  1194\n",
      "               MI127:  1139\n",
      "               MI145:  1012\n",
      "               WI095:  913\n",
      "               VT007:  819\n",
      "               NY036:  774\n",
      "               NY058:  737\n",
      "               NY042:  594\n",
      "               NY006:  589\n",
      "               WI046:  580\n",
      "               PA003:  524\n",
      "               PA014:  507\n",
      "               WI094:  490\n",
      "               WI086:  472\n",
      "               WI102:  394\n",
      "               WI052:  297\n",
      "               WI050:  279\n",
      "               WI164:  264\n",
      "               WI092:  188\n",
      "               IL016:  171\n",
      "               MI142:  130\n",
      "               PA006:  119\n",
      "               MI131:  116\n",
      "               PA008:  96\n",
      "               IL018:  74\n",
      "               VT019:  64\n",
      "               VT020:  54\n",
      "               PA005:  39\n",
      "----------------------------------\n",
      "Read count: 1178970\n",
      "Read length average: 313\n",
      "Read length range: 125 - 535 bp\n",
      "----------------------------------\n"
     ]
    }
   ],
   "source": [
    "!amptk show -i mhc.all.demux.fq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Okay, so now all the data has been demultiplexed, we kept only full length reads because we need to anchor to the start of the DRB2-exon2 translation when running the HMM filter later on.  Obviously many of these samples will drop out due to there being too few reads.  I'm implemented the sample dropout in the MHC-DADA2 script."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Kept 74363 reads out of 1178970 total reads\r\n"
     ]
    }
   ],
   "source": [
    "#pull out the small subset for methods, as they likely can't be used in larger study\n",
    "!amptk select -i mhc.all.demux.fq -f ../mhc.subset.methods.txt -o mhc.methods.demux.fq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----------------------------------\n",
      "Found 17 barcoded samples\n",
      "              Sample:  Count\n",
      "               WI169:  13544\n",
      "               WI166:  12342\n",
      "               WI171:  9252\n",
      "               WI168:  8124\n",
      "               WI163:  5717\n",
      "               IL030:  4771\n",
      "               IL020:  4668\n",
      "               WI165:  3531\n",
      "               IL032:  3267\n",
      "               IL024:  2894\n",
      "               WI170:  1578\n",
      "               IL022:  1553\n",
      "               WI172:  1387\n",
      "               WI167:  1226\n",
      "               WI164:  264\n",
      "               IL016:  171\n",
      "               IL018:  74\n",
      "----------------------------------\n",
      "Read count: 74363\n",
      "Read length average: 305\n",
      "Read length range: 125 - 497 bp\n",
      "----------------------------------\n"
     ]
    }
   ],
   "source": [
    "!amptk show -i mhc.methods.demux.fq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[05:30:04 PM]\u001b[0m: OS: MacOSX 10.12.4, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[05:30:04 PM]\u001b[0m: Reverse complementing FASTQ records\n",
      "\u001b[92m[05:30:13 PM]\u001b[0m: Loading FASTQ Records\n",
      "\u001b[92m[05:30:13 PM]\u001b[0m: 74,363 reads (45.2 MB)\n",
      "\u001b[92m[05:30:13 PM]\u001b[0m: Quality Filtering, expected errors < 2.0\n",
      "\u001b[92m[05:30:14 PM]\u001b[0m: 47,638 reads passed\n",
      "\u001b[92m[05:30:14 PM]\u001b[0m: Splitting FASTQ file by Sample\n",
      "\u001b[92m[05:30:17 PM]\u001b[0m: Removed 6 samples with read number < 500 (--min_reads_sample)\n",
      "\u001b[92m[05:30:17 PM]\u001b[0m: Running DADA2 pipeline\n",
      "\u001b[92m[05:31:50 PM]\u001b[0m: R v3.3.3, DADA2 v1.3.5\n",
      "\u001b[92m[05:31:50 PM]\u001b[0m: 306 total inferred sequences (iSeqs)\n",
      "\u001b[92m[05:31:50 PM]\u001b[0m: 27 denovo chimeras removed\n",
      "\u001b[92m[05:31:50 PM]\u001b[0m: 279 valid iSeqs\n",
      "\u001b[92m[05:31:50 PM]\u001b[0m: Translating to Amino Acid Sequence in 3 frames\n",
      "\u001b[92m[05:31:50 PM]\u001b[0m: 111 total translated sequences > 75 aa\n",
      "\u001b[92m[05:31:50 PM]\u001b[0m: Running HMMER3 MHC HMM model\n",
      "\u001b[92m[05:31:50 PM]\u001b[0m: 87 pass HMM filter, 84 are unique AA sequence.\n",
      "\u001b[92m[05:31:50 PM]\u001b[0m: Mapping reads to DADA2 filtered iSeqs\n",
      "\u001b[92m[05:32:04 PM]\u001b[0m: 48,848 reads mapped to iSeqs (66%)\n",
      "\u001b[92m[05:32:05 PM]\u001b[0m: BIOM OTU table created: methods.biom\n",
      "\u001b[92m[05:32:05 PM]\u001b[0m: Generating phylogenetic tree\n",
      "-------------------------------------------------------\n",
      "MHC DADA2 Script has Finished Successfully\n",
      "-------------------------------------------------------\n",
      "11 samples\n",
      "86 total alleles\n",
      "32.270000 +/- 7.520000 average alleles per sample\n",
      "17 - 46 range of alelles\n",
      "-------------------------------------------------------\n",
      "Tmp Folder of files: methods_41298_filtered\n",
      "Intermediate files: methods_41298\n",
      "MHC Alleles: methods.alleles.fa\n",
      "MHC Protein Alleles : methods.alleles.proteins.fa\n",
      "Allelic OTU Table: methods.otu_table.txt\n",
      "Allelic BIOM table: methods.biom\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#now run through MHC DADA2 script, since only a few samples, don't remove alleles found only 1\n",
    "#try increase quality threshold to 2.0, since we have longer reads and need the 3' end \n",
    "#to serve as an anchor, i.e. that is where translation starts\n",
    "!mhc-dada2.py -i mhc.methods.demux.fq --filter_allele 1  \\\n",
    "              -e 2.0 --rev_comp -m ../mhc.mapping_file.txt -o methods --debug"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removed 17 samples\r\n",
      "Kept 1104607 reads out of 1178970 total reads\r\n"
     ]
    }
   ],
   "source": [
    "#now try DADA2 workflow with the rest of the data, drop the Illinois samples as small dataset and meta data is\n",
    "#questionable as well as Neda mine as we don't have any qPCR data for those sample\n",
    "#these samples were used for a smaller subset for methods manuscript\n",
    "!amptk remove -i mhc.all.demux.fq -f ../mhc.subset.methods.txt -o mhc.large.demux.fq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[05:44:00 PM]\u001b[0m: OS: MacOSX 10.12.4, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[05:44:00 PM]\u001b[0m: Reverse complementing FASTQ records\n",
      "\u001b[92m[05:46:03 PM]\u001b[0m: Loading FASTQ Records\n",
      "\u001b[92m[05:46:04 PM]\u001b[0m: 1,104,607 reads (639.3 MB)\n",
      "\u001b[92m[05:46:04 PM]\u001b[0m: Quality Filtering, expected errors < 2.0\n",
      "\u001b[92m[05:46:18 PM]\u001b[0m: 861,929 reads passed\n",
      "\u001b[92m[05:46:18 PM]\u001b[0m: Splitting FASTQ file by Sample\n",
      "\u001b[92m[05:47:22 PM]\u001b[0m: Removed 27 samples with read number < 500 (--min_reads_sample)\n",
      "\u001b[92m[05:47:22 PM]\u001b[0m: Running DADA2 pipeline\n",
      "\u001b[92m[09:13:26 PM]\u001b[0m: R v3.3.3, DADA2 v1.3.5\n",
      "\u001b[92m[09:13:26 PM]\u001b[0m: 1,758 total inferred sequences (iSeqs)\n",
      "\u001b[92m[09:13:26 PM]\u001b[0m: 120 denovo chimeras removed\n",
      "\u001b[92m[09:13:26 PM]\u001b[0m: 1,638 valid iSeqs\n",
      "\u001b[92m[09:13:26 PM]\u001b[0m: Translating to Amino Acid Sequence in 3 frames\n",
      "\u001b[92m[09:13:26 PM]\u001b[0m: 657 total translated sequences > 75 aa\n",
      "\u001b[92m[09:13:26 PM]\u001b[0m: Running HMMER3 MHC HMM model\n",
      "\u001b[92m[09:13:27 PM]\u001b[0m: 483 pass HMM filter, 436 are unique AA sequence.\n",
      "\u001b[92m[09:13:27 PM]\u001b[0m: Mapping reads to DADA2 filtered iSeqs\n",
      "\u001b[92m[09:17:05 PM]\u001b[0m: 917,765 reads mapped to iSeqs (83%)\n",
      "\u001b[92m[09:17:18 PM]\u001b[0m: BIOM OTU table created: all.biom\n",
      "\u001b[92m[09:17:18 PM]\u001b[0m: Generating phylogenetic tree\n",
      "-------------------------------------------------------\n",
      "MHC DADA2 Script has Finished Successfully\n",
      "-------------------------------------------------------\n",
      "130 samples\n",
      "456 total alleles\n",
      "58.400000 +/- 22.320000 average alleles per sample\n",
      "7 - 124 range of alelles\n",
      "-------------------------------------------------------\n",
      "MHC Alleles: all.alleles.fa\n",
      "MHC Protein Alleles : all.alleles.proteins.fa\n",
      "Allelic OTU Table: all.otu_table.txt\n",
      "Allelic BIOM table: all.biom\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#now run through DADA2\n",
    "!mhc-dada2.py -i mhc.large.demux.fq -e 2.0 --rev_comp -m ../mhc.mapping_file.txt -o all "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### This is somewhat discouraging, that when I try to look at the \"methods\" part of the data, that it doesn't display the same amount of allelic diversity as the larger datset.  So to check if any of these data have structure, I'll run the data through a hyopthesis test for each of the meta data variables.  For this I will use an R script that runs iteratively through vegan adonis test followed by ordination in Phyloseq to visualize the data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-3\n",
      "Loading required package: RColorBrewer\n",
      "Loading required package: gridExtra\n",
      "Loading required package: grid\n",
      "[1] ‘1.19.1’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘1.8.4’\n",
      "[1] ‘2.4.3’\n",
      "[1] ‘1.1.2’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘3.3.1’\n",
      " [1] \"CT_result\"            \"phinchID\"             \"CT_mean\"             \n",
      " [4] \"age_class\"            \"DNA_quant_result\"     \"BarcodeSequence\"     \n",
      " [7] \"site\"                 \"sex\"                  \"site_date\"           \n",
      "[10] \"LinkerPrimerSequence\" \"WNS_status\"           \"date\"                \n",
      "[13] \"DNA_quant_mean\"       \"ReversePrimer\"        \"location\"            \n",
      "[1] \"CT_result\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 456 taxa and 128 samples ]\n",
      "sample_data() Sample Data:       [ 128 samples by 15 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 456 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 456 tips and 455 internal nodes ]\n",
      "^C\n",
      "\n",
      "Execution halted\n"
     ]
    }
   ],
   "source": [
    "#This script loops through all of the treatments or meta data variables, it removes any samples that contain \"no_data\"\n",
    "#and then takes that data and does several tests. Hypothesis tests (adonis) and dispersion (betadisper() followed by\n",
    "#drawing the data in an NMDS ordination.  It does this with 4 distance metrics (RaupCrick, BrayCurtis, Unifrac, and\n",
    "#Weighted Unifrac).  The Unifrac method relies on a protein level phylogenic tree of the MHC alleles. Finally, the script\n",
    "#calculates alpha-diversity for each of the treatments.\n",
    "!$HOME/amptk/util/phyloseq_graphs.R all.update.biom all.tree.phy all"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![SiteMap](no_IL_Neda/test.location.ordinations.pdf)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "#### There seems to be very little to no structure in these data, even between Eastcoast (NY,VT,PA) vs Midwest (WI,MI).  \n",
    "##### I ran adonis test in R for each variable and then betadispersion\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                 braycurtis            raupcrick            unifrac            weighted_unifrac           \r\n",
      "                     Adonis Betadisper    Adonis Betadisper  Adonis Betadisper           Adonis Betadisper\r\n",
      "CT_result            0.8291      0.462    0.5984      0.566  0.2384      0.979           0.7584      0.985\r\n",
      "DNA_quant_mean       0.0226      0.001    0.3169      0.001  0.4694      0.001           0.2096      0.001\r\n",
      "CT_mean              0.9017      0.001    0.7921      0.045  0.5455      0.001           0.4727      0.001\r\n",
      "age_class            0.0669      0.007    0.0024      0.189  0.0029      0.409           0.3087      0.428\r\n",
      "site_date            0.5249      0.063    0.2100      0.005  0.1837      0.313           0.2051      0.307\r\n",
      "site                 0.5069      0.035    0.1437      0.007  0.1260      0.196           0.1733      0.180\r\n",
      "sex                  0.6164      0.978    0.5453      0.556  0.1387      0.068           0.8431      0.071\r\n",
      "DNA_quant_result     0.7763      0.288    0.5756      0.899  0.2636      0.900           0.7558      0.915\r\n",
      "WNS_status           0.6996      0.565    0.6077      0.730  0.0261      0.957           0.7755      0.979\r\n",
      "date                 0.2617      0.001    0.3263      0.001  0.2070      0.001           0.1460      0.001\r\n",
      "location             0.3077      0.266    0.2729      0.909  0.0510      0.862           0.4176      0.854\r\n"
     ]
    }
   ],
   "source": [
    "#This script grabs the raw data and formats into a table.\n",
    "!$HOME/amptk/util/parse_adonis.py no_IL_Neda"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![Alpha diversity by STATUS](no_IL_Neda/test.WNS_status.alpha_diversity.pdf)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Of course Michelle you can look at these data in a lot more detail than I am able to, but if there are not even site differences or geographic differences between populations.  So I think we need another genetic marker to show what is happening here, i.e. can we differientiate between groups of LBB by geographic location or not?  Are there simply too many MHC alleles?  Or is the answer something along the lines that there are huge numbers of MHC alleles, however does not seem to correlate with WNS?  Perhaps we sampled much too early, i.e. lifespan is ~20 years, may take generations before you see any bottleneck or stabilization?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### So I went back and put all the data together, i.e. the entire MHC dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[09:55:37 AM]\u001b[0m: OS: MacOSX 10.12.4, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[09:55:37 AM]\u001b[0m: Reverse complementing FASTQ records\n",
      "\u001b[92m[09:57:41 AM]\u001b[0m: Loading FASTQ Records\n",
      "\u001b[92m[09:57:41 AM]\u001b[0m: 1,178,970 reads (684.5 MB)\n",
      "\u001b[92m[09:57:41 AM]\u001b[0m: Quality Filtering, expected errors < 2.0\n",
      "\u001b[92m[09:57:56 AM]\u001b[0m: 909,567 reads passed\n",
      "\u001b[92m[09:57:56 AM]\u001b[0m: Splitting FASTQ file by Sample\n",
      "\u001b[92m[09:59:00 AM]\u001b[0m: Removed 33 samples with read number < 500 (--min_reads_sample)\n",
      "\u001b[92m[09:59:00 AM]\u001b[0m: Running DADA2 pipeline\n",
      "\u001b[92m[07:21:44 PM]\u001b[0m: R v3.3.1, DADA2 v1.5.0\n",
      "\u001b[92m[07:21:44 PM]\u001b[0m: 1,883 total inferred sequences (iSeqs)\n",
      "\u001b[92m[07:21:44 PM]\u001b[0m: 129 denovo chimeras removed\n",
      "\u001b[92m[07:21:44 PM]\u001b[0m: 1,754 valid iSeqs\n",
      "\u001b[92m[07:21:44 PM]\u001b[0m: Translating to Amino Acid Sequence in 3 frames\n",
      "\u001b[92m[07:21:45 PM]\u001b[0m: 694 total translated sequences > 75 aa\n",
      "\u001b[92m[07:21:45 PM]\u001b[0m: Running HMMER3 MHC HMM model\n",
      "\u001b[92m[07:21:45 PM]\u001b[0m: 507 pass HMM filter, 459 are unique AA sequence.\n",
      "\u001b[92m[07:21:56 PM]\u001b[0m: BIOM OTU table created: mhc.biom\n",
      "\u001b[92m[07:21:56 PM]\u001b[0m: Generating phylogenetic tree\n",
      "-------------------------------------------------------\n",
      "MHC DADA2 Script has Finished Successfully\n",
      "-------------------------------------------------------\n",
      "139 samples\n",
      "486 total alleles\n",
      "62.060000 +/- 23.400000 average alleles per sample\n",
      "9 - 129 range of alelles\n",
      "-------------------------------------------------------\n",
      "MHC Alleles: mhc.alleles.fa\n",
      "MHC Protein Alleles : mhc.alleles.proteins.fa\n",
      "Allelic OTU Table: mhc.otu_table.txt\n",
      "MHC phylogeny : mhc.tree.phy\n",
      "Allelic BIOM table: mhc.biom\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#now run through DADA2\n",
    "!mhc-dada2.py -i mhc.all.demux.fq -e 2.0 --rev_comp -m ../mhc.mapping_file.txt -o mhc "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-3\n",
      "Loading required package: RColorBrewer\n",
      "Loading required package: gridExtra\n",
      "Loading required package: grid\n",
      "[1] ‘1.19.1’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘1.8.4’\n",
      "[1] ‘2.4.3’\n",
      "[1] ‘1.1.2’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘3.3.1’\n",
      " [1] \"CT_result\"            \"phinchID\"             \"CT_mean\"             \n",
      " [4] \"age_class\"            \"DNA_quant_result\"     \"BarcodeSequence\"     \n",
      " [7] \"site\"                 \"sex\"                  \"site_date\"           \n",
      "[10] \"LinkerPrimerSequence\" \"WNS_status\"           \"date\"                \n",
      "[13] \"DNA_quant_mean\"       \"ReversePrimer\"        \"location\"            \n",
      "[1] \"CT_result\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 486 taxa and 126 samples ]\n",
      "sample_data() Sample Data:       [ 126 samples by 15 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 486 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 486 tips and 485 internal nodes ]\n",
      "Run 0 stress 0.280549 \n",
      "Run 1 stress 0.2872998 \n",
      "Run 2 stress 0.2839361 \n",
      "Run 3 stress 0.2835801 \n",
      "Run 4 stress 0.2834173 \n",
      "Run 5 stress 0.2891882 \n",
      "Run 6 stress 0.2816759 \n",
      "Run 7 stress 0.2844713 \n",
      "Run 8 stress 0.2816471 \n",
      "Run 9 stress 0.2837428 \n",
      "Run 10 stress 0.2974996 \n",
      "Run 11 stress 0.2826227 \n",
      "Run 12 stress 0.2857732 \n",
      "Run 13 stress 0.2806275 \n",
      "... Procrustes: rmse 0.006075822  max resid 0.04714598 \n",
      "Run 14 stress 0.2807302 \n",
      "... Procrustes: rmse 0.004997572  max resid 0.04672114 \n",
      "Run 15 stress 0.2867973 \n",
      "Run 16 stress 0.2805389 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.004903506  max resid 0.03652208 \n",
      "Run 17 stress 0.284596 \n",
      "Run 18 stress 0.2805958 \n",
      "... Procrustes: rmse 0.005200507  max resid 0.04469693 \n",
      "Run 19 stress 0.2805162 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.002237738  max resid 0.01602069 \n",
      "Run 20 stress 0.2836159 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "     1: no. of iterations >= maxit\n",
      "    19: stress ratio > sratmax\n",
      "Run 0 stress 0.2921121 \n",
      "Run 1 stress 0.2979195 \n",
      "Run 2 stress 0.2966817 \n",
      "Run 3 stress 0.3027055 \n",
      "Run 4 stress 0.3000005 \n",
      "Run 5 stress 0.2920908 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.05897806  max resid 0.256289 \n",
      "Run 6 stress 0.300079 \n",
      "Run 7 stress 0.2947598 \n",
      "Run 8 stress 0.290869 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0474883  max resid 0.2561264 \n",
      "Run 9 stress 0.3003309 \n",
      "Run 10 stress 0.2891925 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.03327024  max resid 0.1970284 \n",
      "Run 11 stress 0.2983998 \n",
      "Run 12 stress 0.2967367 \n",
      "Run 13 stress 0.2939257 \n",
      "Run 14 stress 0.3004349 \n",
      "Run 15 stress 0.2959483 \n",
      "Run 16 stress 0.3047107 \n",
      "Run 17 stress 0.3026803 \n",
      "Run 18 stress 0.3010517 \n",
      "Run 19 stress 0.2987271 \n",
      "Run 20 stress 0.3000469 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n",
      "Run 0 stress 0.2763757 \n",
      "Run 1 stress 0.2753179 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.02056129  max resid 0.131112 \n",
      "Run 2 stress 0.275721 \n",
      "... Procrustes: rmse 0.02891269  max resid 0.1945425 \n",
      "Run 3 stress 0.2919888 \n",
      "Run 4 stress 0.2783805 \n",
      "Run 5 stress 0.2757186 \n",
      "... Procrustes: rmse 0.02520196  max resid 0.1707333 \n",
      "Run 6 stress 0.2748364 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.01978439  max resid 0.1678504 \n",
      "Run 7 stress 0.2903393 \n",
      "Run 8 stress 0.2839139 \n",
      "Run 9 stress 0.2766955 \n",
      "Run 10 stress 0.4140086 \n",
      "Run 11 stress 0.2770645 \n",
      "Run 12 stress 0.2783688 \n",
      "Run 13 stress 0.2764489 \n",
      "Run 14 stress 0.2748546 \n",
      "... Procrustes: rmse 0.006896137  max resid 0.05615938 \n",
      "Run 15 stress 0.2780584 \n",
      "Run 16 stress 0.2767856 \n",
      "Run 17 stress 0.2761886 \n",
      "Run 18 stress 0.2774051 \n",
      "Run 19 stress 0.2870078 \n",
      "Run 20 stress 0.2761938 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "     2: no. of iterations >= maxit\n",
      "    18: stress ratio > sratmax\n",
      "Run 0 stress 0.2321884 \n",
      "Run 1 stress 0.2484494 \n",
      "Run 2 stress 0.2493723 \n",
      "Run 3 stress 0.2681395 \n",
      "Run 4 stress 0.2666709 \n",
      "Run 5 stress 0.246341 \n",
      "Run 6 stress 0.2639463 \n",
      "Run 7 stress 0.25985 \n",
      "Run 8 stress 0.2363484 \n",
      "Run 9 stress 0.260019 \n",
      "Run 10 stress 0.2794795 \n",
      "Run 11 stress 0.244824 \n",
      "Run 12 stress 0.2431495 \n",
      "Run 13 stress 0.2707269 \n",
      "Run 14 stress 0.2398567 \n",
      "Run 15 stress 0.2434594 \n",
      "Run 16 stress 0.2536722 \n",
      "Run 17 stress 0.2655656 \n",
      "Run 18 stress 0.2747112 \n",
      "Run 19 stress 0.2447428 \n",
      "Run 20 stress 0.233153 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n",
      "Saving 7 x 7 in image\n",
      "[1] \"CT_mean\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 486 taxa and 126 samples ]\n",
      "sample_data() Sample Data:       [ 126 samples by 15 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 486 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 486 tips and 485 internal nodes ]\n",
      "[1] \"age_class\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 486 taxa and 127 samples ]\n",
      "sample_data() Sample Data:       [ 127 samples by 15 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 486 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 486 tips and 485 internal nodes ]\n",
      "Run 0 stress 0.2836153 \n",
      "Run 1 stress 0.2841463 \n",
      "Run 2 stress 0.2947731 \n",
      "Run 3 stress 0.2874943 \n",
      "Run 4 stress 0.2975276 \n",
      "Run 5 stress 0.2877476 \n",
      "Run 6 stress 0.296464 \n",
      "Run 7 stress 0.2835513 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.008009308  max resid 0.04563372 \n",
      "Run 8 stress 0.2844536 \n",
      "Run 9 stress 0.284378 \n",
      "Run 10 stress 0.2977321 \n",
      "Run 11 stress 0.2836999 \n",
      "... Procrustes: rmse 0.0105853  max resid 0.08805138 \n",
      "Run 12 stress 0.2838437 \n",
      "... Procrustes: rmse 0.009771303  max resid 0.08738224 \n",
      "Run 13 stress 0.2848676 \n",
      "Run 14 stress 0.2887472 \n",
      "Run 15 stress 0.296496 \n",
      "Run 16 stress 0.2841341 \n",
      "Run 17 stress 0.2838049 \n",
      "... Procrustes: rmse 0.009702278  max resid 0.0695237 \n",
      "Run 18 stress 0.2946492 \n",
      "Run 19 stress 0.2893681 \n",
      "Run 20 stress 0.2853628 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "     1: no. of iterations >= maxit\n",
      "    19: stress ratio > sratmax\n",
      "Run 0 stress 0.2980351 \n",
      "Run 1 stress 0.303694 \n",
      "Run 2 stress 0.3064187 \n",
      "Run 3 stress 0.2992243 \n",
      "Run 4 stress 0.3027831 \n",
      "Run 5 stress 0.2960601 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.07386612  max resid 0.1674528 \n",
      "Run 6 stress 0.2931695 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.05207038  max resid 0.2930315 \n",
      "Run 7 stress 0.3028997 \n",
      "Run 8 stress 0.3039377 \n",
      "Run 9 stress 0.3053032 \n",
      "Run 10 stress 0.2931072 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.03468642  max resid 0.2180645 \n",
      "Run 11 stress 0.2999347 \n",
      "Run 12 stress 0.2971844 \n",
      "Run 13 stress 0.2905541 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.03015526  max resid 0.2100442 \n",
      "Run 14 stress 0.3039153 \n",
      "Run 15 stress 0.299245 \n",
      "Run 16 stress 0.2976108 \n",
      "Run 17 stress 0.3014125 \n",
      "Run 18 stress 0.3067473 \n",
      "Run 19 stress 0.2915941 \n",
      "Run 20 stress 0.299746 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n",
      "Run 0 stress 0.27778 \n",
      "Run 1 stress 0.2781166 \n",
      "... Procrustes: rmse 0.02834138  max resid 0.1479122 \n",
      "Run 2 stress 0.2826772 \n",
      "Run 3 stress 0.2792274 \n",
      "Run 4 stress 0.277954 \n",
      "... Procrustes: rmse 0.03459085  max resid 0.2279253 \n",
      "Run 5 stress 0.2831256 \n",
      "Run 6 stress 0.27868 \n",
      "Run 7 stress 0.2770943 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.02834019  max resid 0.1601874 \n",
      "Run 8 stress 0.2772472 \n",
      "... Procrustes: rmse 0.009224485  max resid 0.06389423 \n",
      "Run 9 stress 0.2841929 \n",
      "Run 10 stress 0.2897409 \n",
      "Run 11 stress 0.2789045 \n",
      "Run 12 stress 0.2941966 \n",
      "Run 13 stress 0.2790124 \n",
      "Run 14 stress 0.287924 \n",
      "Run 15 stress 0.2773068 \n",
      "... Procrustes: rmse 0.03046333  max resid 0.1616726 \n",
      "Run 16 stress 0.2791828 \n",
      "Run 17 stress 0.2947088 \n",
      "Run 18 stress 0.2771983 \n",
      "... Procrustes: rmse 0.01124738  max resid 0.08556778 \n",
      "Run 19 stress 0.2800645 \n",
      "Run 20 stress 0.2805374 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "     3: no. of iterations >= maxit\n",
      "    17: stress ratio > sratmax\n",
      "Run 0 stress 0.2301501 \n",
      "Run 1 stress 0.2561139 \n",
      "Run 2 stress 0.2617255 \n",
      "Run 3 stress 0.2400025 \n",
      "Run 4 stress 0.2642923 \n",
      "Run 5 stress 0.242734 \n",
      "Run 6 stress 0.2597992 \n",
      "Run 7 stress 0.2454396 \n",
      "Run 8 stress 0.260774 \n",
      "Run 9 stress 0.2508645 \n",
      "Run 10 stress 0.2348368 \n",
      "Run 11 stress 0.2712977 \n",
      "Run 12 stress 0.2393693 \n",
      "Run 13 stress 0.2539163 \n",
      "Run 14 stress 0.2371917 \n",
      "Run 15 stress 0.2477804 \n",
      "Run 16 stress 0.2406854 \n",
      "Run 17 stress 0.2366179 \n",
      "Run 18 stress 0.2545337 \n",
      "Run 19 stress 0.2510736 \n",
      "Run 20 stress 0.2288437 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.03161789  max resid 0.184493 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n",
      "Saving 7 x 7 in image\n",
      "[1] \"DNA_quant_result\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 486 taxa and 126 samples ]\n",
      "sample_data() Sample Data:       [ 126 samples by 15 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 486 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 486 tips and 485 internal nodes ]\n",
      "Run 0 stress 0.2809973 \n",
      "Run 1 stress 0.2819712 \n",
      "Run 2 stress 0.2812037 \n",
      "... Procrustes: rmse 0.006137658  max resid 0.04475065 \n",
      "Run 3 stress 0.292258 \n",
      "Run 4 stress 0.2823614 \n",
      "Run 5 stress 0.2854665 \n",
      "Run 6 stress 0.2824444 \n",
      "Run 7 stress 0.2822214 \n",
      "Run 8 stress 0.2919474 \n",
      "Run 9 stress 0.2821091 \n",
      "Run 10 stress 0.2857985 \n",
      "Run 11 stress 0.2830323 \n",
      "Run 12 stress 0.2841843 \n",
      "Run 13 stress 0.280968 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.002446666  max resid 0.02073449 \n",
      "Run 14 stress 0.2850874 \n",
      "Run 15 stress 0.2822723 \n",
      "Run 16 stress 0.2810553 \n",
      "... Procrustes: rmse 0.004529606  max resid 0.04325293 \n",
      "Run 17 stress 0.2810687 \n",
      "... Procrustes: rmse 0.004863722  max resid 0.0492252 \n",
      "Run 18 stress 0.2809729 \n",
      "... Procrustes: rmse 0.001655151  max resid 0.01520356 \n",
      "Run 19 stress 0.2824525 \n",
      "Run 20 stress 0.2878434 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "     2: no. of iterations >= maxit\n",
      "    18: stress ratio > sratmax\n",
      "Run 0 stress 0.2921121 \n",
      "Run 1 stress 0.2936816 \n",
      "Run 2 stress 0.2914627 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.07144709  max resid 0.2340745 \n",
      "Run 3 stress 0.2996894 \n",
      "Run 4 stress 0.3003128 \n",
      "Run 5 stress 0.3014408 \n",
      "Run 6 stress 0.2991282 \n",
      "Run 7 stress 0.2954677 \n",
      "Run 8 stress 0.3075912 \n",
      "Run 9 stress 0.2973221 \n",
      "Run 10 stress 0.2913643 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.02061488  max resid 0.1696476 \n",
      "Run 11 stress 0.2963007 \n",
      "Run 12 stress 0.2904426 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.03508035  max resid 0.1963672 \n",
      "Run 13 stress 0.2948664 \n",
      "Run 14 stress 0.2959391 \n",
      "Run 15 stress 0.2986974 \n",
      "Run 16 stress 0.2951625 \n",
      "Run 17 stress 0.2888426 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.02576596  max resid 0.13948 \n",
      "Run 18 stress 0.2903912 \n",
      "Run 19 stress 0.2883247 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0124605  max resid 0.0828813 \n",
      "Run 20 stress 0.2967073 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n",
      "Run 0 stress 0.2763757 \n",
      "Run 1 stress 0.2783865 \n",
      "Run 2 stress 0.2761017 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.03482467  max resid 0.2301152 \n",
      "Run 3 stress 0.4139732 \n",
      "Run 4 stress 0.2760561 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.009803542  max resid 0.05415514 \n",
      "Run 5 stress 0.2759018 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.047407  max resid 0.2293136 \n",
      "Run 6 stress 0.2759807 \n",
      "... Procrustes: rmse 0.04691981  max resid 0.2264266 \n",
      "Run 7 stress 0.2767724 \n",
      "Run 8 stress 0.2767677 \n",
      "Run 9 stress 0.2752161 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0388586  max resid 0.208003 \n",
      "Run 10 stress 0.2748359 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.01054599  max resid 0.06006298 \n",
      "Run 11 stress 0.2767098 \n",
      "Run 12 stress 0.4139047 \n",
      "Run 13 stress 0.2757203 \n",
      "Run 14 stress 0.2754352 \n",
      "Run 15 stress 0.2776426 \n",
      "Run 16 stress 0.2770278 \n",
      "Run 17 stress 0.274844 \n",
      "... Procrustes: rmse 0.0005942948  max resid 0.00433648 \n",
      "... Similar to previous best\n",
      "Run 18 stress 0.2809172 \n",
      "Run 19 stress 0.2756854 \n",
      "Run 20 stress 0.2852729 \n",
      "*** Solution reached\n",
      "Run 0 stress 0.2321884 \n",
      "Run 1 stress 0.2763151 \n",
      "Run 2 stress 0.2501099 \n",
      "Run 3 stress 0.2454397 \n",
      "Run 4 stress 0.2362864 \n",
      "Run 5 stress 0.2469133 \n",
      "Run 6 stress 0.2400479 \n",
      "Run 7 stress 0.2408877 \n",
      "Run 8 stress 0.242311 \n",
      "Run 9 stress 0.2424265 \n",
      "Run 10 stress 0.240193 \n",
      "Run 11 stress 0.2562311 \n",
      "Run 12 stress 0.2381775 \n",
      "Run 13 stress 0.24058 \n",
      "Run 14 stress 0.2397056 \n",
      "Run 15 stress 0.2791926 \n",
      "Run 16 stress 0.230331 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.02901052  max resid 0.2527359 \n",
      "Run 17 stress 0.2330456 \n",
      "Run 18 stress 0.2498377 \n",
      "Run 19 stress 0.2727718 \n",
      "Run 20 stress 0.2596732 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    19: stress ratio > sratmax\n",
      "     1: scale factor of the gradient < sfgrmin\n",
      "Saving 7 x 7 in image\n",
      "[1] \"site\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 486 taxa and 139 samples ]\n",
      "sample_data() Sample Data:       [ 139 samples by 15 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 486 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 486 tips and 485 internal nodes ]\n",
      "Run 0 stress 0.283494 \n",
      "Run 1 stress 0.2836646 \n",
      "... Procrustes: rmse 0.01326734  max resid 0.09043649 \n",
      "Run 2 stress 0.2852634 \n",
      "Run 3 stress 0.2875388 \n",
      "Run 4 stress 0.2904355 \n",
      "Run 5 stress 0.2925766 \n",
      "Run 6 stress 0.2901521 \n",
      "Run 7 stress 0.284632 \n",
      "Run 8 stress 0.2915426 \n",
      "Run 9 stress 0.29765 \n",
      "Run 10 stress 0.2844216 \n",
      "Run 11 stress 0.3231326 \n",
      "Run 12 stress 0.2943209 \n",
      "Run 13 stress 0.2859655 \n",
      "Run 14 stress 0.3159782 \n",
      "Run 15 stress 0.2914795 \n",
      "Run 16 stress 0.2851911 \n",
      "Run 17 stress 0.2838328 \n",
      "... Procrustes: rmse 0.01666933  max resid 0.1747377 \n",
      "Run 18 stress 0.2838044 \n",
      "... Procrustes: rmse 0.01073508  max resid 0.09083062 \n",
      "Run 19 stress 0.2837497 \n",
      "... Procrustes: rmse 0.0162803  max resid 0.174993 \n",
      "Run 20 stress 0.2883333 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "     1: no. of iterations >= maxit\n",
      "    19: stress ratio > sratmax\n",
      "Run 0 stress 0.2972208 \n",
      "Run 1 stress 0.3105771 \n",
      "Run 2 stress 0.3013663 \n",
      "Run 3 stress 0.2969276 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.04335638  max resid 0.2600843 \n",
      "Run 4 stress 0.3022238 \n",
      "Run 5 stress 0.2993523 \n",
      "Run 6 stress 0.296627 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.05741476  max resid 0.2156615 \n",
      "Run 7 stress 0.2969981 \n",
      "... Procrustes: rmse 0.06013406  max resid 0.2406678 \n",
      "Run 8 stress 0.3076152 \n",
      "Run 9 stress 0.2922907 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.04242632  max resid 0.2128565 \n",
      "Run 10 stress 0.3005775 \n",
      "Run 11 stress 0.3063752 \n",
      "Run 12 stress 0.303919 \n",
      "Run 13 stress 0.2949119 \n",
      "Run 14 stress 0.2928779 \n",
      "Run 15 stress 0.3150688 \n",
      "Run 16 stress 0.2917664 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0307303  max resid 0.1930421 \n",
      "Run 17 stress 0.2999808 \n",
      "Run 18 stress 0.2973336 \n",
      "Run 19 stress 0.2936227 \n",
      "Run 20 stress 0.2915725 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0130775  max resid 0.07630311 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n",
      "Run 0 stress 0.2778284 \n",
      "Run 1 stress 0.2781088 \n",
      "... Procrustes: rmse 0.02616234  max resid 0.1782222 \n",
      "Run 2 stress 0.2920521 \n",
      "Run 3 stress 0.2791696 \n",
      "Run 4 stress 0.2776955 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.008854333  max resid 0.05144144 \n",
      "Run 5 stress 0.2794912 \n",
      "Run 6 stress 0.2854847 \n",
      "Run 7 stress 0.2772371 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.02720128  max resid 0.2465045 \n",
      "Run 8 stress 0.2774459 \n",
      "... Procrustes: rmse 0.01604111  max resid 0.09608776 \n",
      "Run 9 stress 0.2776037 \n",
      "... Procrustes: rmse 0.02492529  max resid 0.1508486 \n",
      "Run 10 stress 0.2781871 \n",
      "Run 11 stress 0.279293 \n",
      "Run 12 stress 0.2793942 \n",
      "Run 13 stress 0.2798556 \n",
      "Run 14 stress 0.2779314 \n",
      "Run 15 stress 0.2774517 \n",
      "... Procrustes: rmse 0.02201484  max resid 0.1258639 \n",
      "Run 16 stress 0.2784829 \n",
      "Run 17 stress 0.2773655 \n",
      "... Procrustes: rmse 0.01494071  max resid 0.09507261 \n",
      "Run 18 stress 0.2775948 \n",
      "... Procrustes: rmse 0.0272974  max resid 0.2130449 \n",
      "Run 19 stress 0.2783772 \n",
      "Run 20 stress 0.2794114 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "     3: no. of iterations >= maxit\n",
      "    17: stress ratio > sratmax\n",
      "Run 0 stress 0.2326691 \n",
      "Run 1 stress 0.2837247 \n",
      "Run 2 stress 0.2376941 \n",
      "Run 3 stress 0.2484203 \n",
      "Run 4 stress 0.2655383 \n",
      "Run 5 stress 0.2502864 \n",
      "Run 6 stress 0.2318367 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.04194393  max resid 0.2399691 \n",
      "Run 7 stress 0.245316 \n",
      "Run 8 stress 0.2747834 \n",
      "Run 9 stress 0.2678968 \n",
      "Run 10 stress 0.2695694 \n",
      "Run 11 stress 0.2439222 \n",
      "Run 12 stress 0.2348067 \n",
      "Run 13 stress 0.2610165 \n",
      "Run 14 stress 0.2584023 \n",
      "Run 15 stress 0.2562378 \n",
      "Run 16 stress 0.2576983 \n",
      "Run 17 stress 0.2596684 \n",
      "Run 18 stress 0.2611465 \n",
      "Run 19 stress 0.2670538 \n",
      "Run 20 stress 0.2419897 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n",
      "Saving 7 x 7 in image\n",
      "[1] \"sex\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 486 taxa and 127 samples ]\n",
      "sample_data() Sample Data:       [ 127 samples by 15 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 486 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 486 tips and 485 internal nodes ]\n",
      "Run 0 stress 0.2838526 \n",
      "Run 1 stress 0.2849606 \n",
      "Run 2 stress 0.3380694 \n",
      "Run 3 stress 0.2845372 \n",
      "Run 4 stress 0.2919236 \n",
      "Run 5 stress 0.3011197 \n",
      "Run 6 stress 0.2950155 \n",
      "Run 7 stress 0.2845562 \n",
      "Run 8 stress 0.28527 \n",
      "Run 9 stress 0.302226 \n",
      "Run 10 stress 0.2848995 \n",
      "Run 11 stress 0.2838003 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.005017764  max resid 0.03138687 \n",
      "Run 12 stress 0.3002151 \n",
      "Run 13 stress 0.2982387 \n",
      "Run 14 stress 0.2846448 \n",
      "Run 15 stress 0.3078024 \n",
      "Run 16 stress 0.2974056 \n",
      "Run 17 stress 0.2846662 \n",
      "Run 18 stress 0.2940716 \n",
      "Run 19 stress 0.295659 \n",
      "Run 20 stress 0.2922803 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n",
      "Run 0 stress 0.2980351 \n",
      "Run 1 stress 0.3051241 \n",
      "Run 2 stress 0.2999519 \n",
      "Run 3 stress 0.3052713 \n",
      "Run 4 stress 0.298324 \n",
      "... Procrustes: rmse 0.07332193  max resid 0.2298308 \n",
      "Run 5 stress 0.3013185 \n",
      "Run 6 stress 0.3054124 \n",
      "Run 7 stress 0.3046926 \n",
      "Run 8 stress 0.3053534 \n",
      "Run 9 stress 0.2928521 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.07380084  max resid 0.229091 \n",
      "Run 10 stress 0.305857 \n",
      "Run 11 stress 0.3002674 \n",
      "Run 12 stress 0.3063282 \n",
      "Run 13 stress 0.3012395 \n",
      "Run 14 stress 0.2985492 \n",
      "Run 15 stress 0.2927806 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.05052167  max resid 0.2246816 \n",
      "Run 16 stress 0.3005386 \n",
      "Run 17 stress 0.2984458 \n",
      "Run 18 stress 0.2980213 \n",
      "Run 19 stress 0.3034815 \n",
      "Run 20 stress 0.2987514 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n",
      "Run 0 stress 0.27778 \n",
      "Run 1 stress 0.284504 \n",
      "Run 2 stress 0.2824179 \n",
      "Run 3 stress 0.2791993 \n",
      "Run 4 stress 0.2809235 \n",
      "Run 5 stress 0.2830638 \n",
      "Run 6 stress 0.278164 \n",
      "... Procrustes: rmse 0.02356303  max resid 0.129852 \n",
      "Run 7 stress 0.2774885 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.03018471  max resid 0.1616534 \n",
      "Run 8 stress 0.2788131 \n",
      "Run 9 stress 0.2808455 \n",
      "Run 10 stress 0.2799713 \n",
      "Run 11 stress 0.2799404 \n",
      "Run 12 stress 0.2832035 \n",
      "Run 13 stress 0.2772103 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.01152693  max resid 0.08963555 \n",
      "Run 14 stress 0.2831978 \n",
      "Run 15 stress 0.2777708 \n",
      "Run 16 stress 0.2780881 \n",
      "Run 17 stress 0.2812138 \n",
      "Run 18 stress 0.2771023 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.01011416  max resid 0.08604925 \n",
      "Run 19 stress 0.2995338 \n",
      "Run 20 stress 0.2805156 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "     2: no. of iterations >= maxit\n",
      "    18: stress ratio > sratmax\n",
      "Run 0 stress 0.2301501 \n",
      "Run 1 stress 0.2531549 \n",
      "Run 2 stress 0.2429954 \n",
      "Run 3 stress 0.2544932 \n",
      "Run 4 stress 0.2542187 \n",
      "Run 5 stress 0.239905 \n",
      "Run 6 stress 0.2622965 \n",
      "Run 7 stress 0.2601541 \n",
      "Run 8 stress 0.2605616 \n",
      "Run 9 stress 0.2532457 \n",
      "Run 10 stress 0.2732203 \n",
      "Run 11 stress 0.2590869 \n",
      "Run 12 stress 0.2490934 \n",
      "Run 13 stress 0.2688716 \n",
      "Run 14 stress 0.2716522 \n",
      "Run 15 stress 0.255827 \n",
      "Run 16 stress 0.2694904 \n",
      "Run 17 stress 0.2618671 \n",
      "Run 18 stress 0.2727873 \n",
      "Run 19 stress 0.2665378 \n",
      "Run 20 stress 0.2707647 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n",
      "Saving 7 x 7 in image\n",
      "[1] \"site_date\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 486 taxa and 139 samples ]\n",
      "sample_data() Sample Data:       [ 139 samples by 15 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 486 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 486 tips and 485 internal nodes ]\n",
      "Run 0 stress 0.284155 \n",
      "Run 1 stress 0.2853083 \n",
      "Run 2 stress 0.2877338 \n",
      "Run 3 stress 0.2938246 \n",
      "Run 4 stress 0.288308 \n",
      "Run 5 stress 0.2853844 \n",
      "Run 6 stress 0.2899112 \n",
      "Run 7 stress 0.2865742 \n",
      "Run 8 stress 0.2887533 \n",
      "Run 9 stress 0.2856753 \n",
      "Run 10 stress 0.2843976 \n",
      "... Procrustes: rmse 0.01061529  max resid 0.1012968 \n",
      "Run 11 stress 0.2912308 \n",
      "Run 12 stress 0.2952893 \n",
      "Run 13 stress 0.2851582 \n",
      "Run 14 stress 0.288973 \n",
      "Run 15 stress 0.2921377 \n",
      "Run 16 stress 0.2876938 \n",
      "Run 17 stress 0.2865664 \n",
      "Run 18 stress 0.2845101 \n",
      "... Procrustes: rmse 0.01039401  max resid 0.1018613 \n",
      "Run 19 stress 0.2880346 \n",
      "Run 20 stress 0.2843138 \n",
      "... Procrustes: rmse 0.01527278  max resid 0.1734615 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "     1: no. of iterations >= maxit\n",
      "    19: stress ratio > sratmax\n",
      "Run 0 stress 0.2972208 \n",
      "Run 1 stress 0.2934031 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.05965866  max resid 0.2379351 \n",
      "Run 2 stress 0.2922289 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.02899023  max resid 0.1799922 \n",
      "Run 3 stress 0.2948427 \n",
      "Run 4 stress 0.2982596 \n",
      "Run 5 stress 0.3070432 \n",
      "Run 6 stress 0.2927786 \n",
      "Run 7 stress 0.3022173 \n",
      "Run 8 stress 0.2968778 \n",
      "Run 9 stress 0.2967805 \n",
      "Run 10 stress 0.2953044 \n",
      "Run 11 stress 0.2966776 \n",
      "Run 12 stress 0.2973314 \n",
      "Run 13 stress 0.3043804 \n",
      "Run 14 stress 0.2957487 \n",
      "Run 15 stress 0.2995364 \n",
      "Run 16 stress 0.2997709 \n",
      "Run 17 stress 0.2968178 \n",
      "Run 18 stress 0.2929391 \n",
      "Run 19 stress 0.2953988 \n",
      "Run 20 stress 0.3021149 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "     1: no. of iterations >= maxit\n",
      "    19: stress ratio > sratmax\n",
      "Run 0 stress 0.2778284 \n",
      "Run 1 stress 0.2775685 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.02736117  max resid 0.1696416 \n",
      "Run 2 stress 0.2791935 \n",
      "Run 3 stress 0.2834464 \n",
      "Run 4 stress 0.2790317 \n",
      "Run 5 stress 0.2790497 \n",
      "Run 6 stress 0.2786531 \n",
      "Run 7 stress 0.2786254 \n",
      "Run 8 stress 0.2814286 \n",
      "Run 9 stress 0.2803568 \n",
      "Run 10 stress 0.2801711 \n",
      "Run 11 stress 0.2777742 \n",
      "... Procrustes: rmse 0.01540673  max resid 0.1371062 \n",
      "Run 12 stress 0.2787089 \n",
      "Run 13 stress 0.278784 \n",
      "Run 14 stress 0.2925599 \n",
      "Run 15 stress 0.2781455 \n",
      "Run 16 stress 0.2791407 \n",
      "Run 17 stress 0.2769304 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.03434384  max resid 0.2136887 \n",
      "Run 18 stress 0.2785435 \n",
      "Run 19 stress 0.2772822 \n",
      "... Procrustes: rmse 0.01854349  max resid 0.1783758 \n",
      "Run 20 stress 0.2818174 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "     4: no. of iterations >= maxit\n",
      "    16: stress ratio > sratmax\n",
      "Run 0 stress 0.2326691 \n",
      "Run 1 stress 0.2642627 \n",
      "Run 2 stress 0.2449667 \n",
      "Run 3 stress 0.2688225 \n",
      "Run 4 stress 0.2783378 \n",
      "Run 5 stress 0.2596421 \n",
      "Run 6 stress 0.2765718 \n",
      "Run 7 stress 0.2377352 \n",
      "Run 8 stress 0.2699471 \n",
      "Run 9 stress 0.2763152 \n",
      "Run 10 stress 0.256487 \n",
      "Run 11 stress 0.2652175 \n",
      "Run 12 stress 0.2343014 \n",
      "Run 13 stress 0.2784592 \n",
      "Run 14 stress 0.276933 \n",
      "Run 15 stress 0.2575641 \n",
      "Run 16 stress 0.2695328 \n",
      "Run 17 stress 0.244485 \n",
      "Run 18 stress 0.266318 \n",
      "Run 19 stress 0.2741726 \n",
      "Run 20 stress 0.2609095 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n",
      "Saving 7 x 7 in image\n",
      "[1] \"WNS_status\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 486 taxa and 139 samples ]\n",
      "sample_data() Sample Data:       [ 139 samples by 15 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 486 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 486 tips and 485 internal nodes ]\n",
      "Run 0 stress 0.2843494 \n",
      "Run 1 stress 0.2889186 \n",
      "Run 2 stress 0.2848453 \n",
      "... Procrustes: rmse 0.01665429  max resid 0.1759519 \n",
      "Run 3 stress 0.3044485 \n",
      "Run 4 stress 0.2922264 \n",
      "Run 5 stress 0.2895667 \n",
      "Run 6 stress 0.2854714 \n",
      "Run 7 stress 0.2842242 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.009603185  max resid 0.08775474 \n",
      "Run 8 stress 0.2934266 \n",
      "Run 9 stress 0.2917032 \n",
      "Run 10 stress 0.2859408 \n",
      "Run 11 stress 0.2889209 \n",
      "Run 12 stress 0.2938565 \n",
      "Run 13 stress 0.331985 \n",
      "Run 14 stress 0.2855454 \n",
      "Run 15 stress 0.2856432 \n",
      "Run 16 stress 0.2956112 \n",
      "Run 17 stress 0.2848895 \n",
      "Run 18 stress 0.288307 \n",
      "Run 19 stress 0.2884632 \n",
      "Run 20 stress 0.2845462 \n",
      "... Procrustes: rmse 0.009065931  max resid 0.08891238 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "     3: no. of iterations >= maxit\n",
      "    17: stress ratio > sratmax\n",
      "Run 0 stress 0.2972208 \n",
      "Run 1 stress 0.2994044 \n",
      "Run 2 stress 0.2949018 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0570936  max resid 0.2454892 \n",
      "Run 3 stress 0.2981809 \n",
      "Run 4 stress 0.2962645 \n",
      "Run 5 stress 0.2963342 \n",
      "Run 6 stress 0.3050517 \n",
      "Run 7 stress 0.3089052 \n",
      "Run 8 stress 0.3057857 \n",
      "Run 9 stress 0.2987712 \n",
      "Run 10 stress 0.2926707 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.04263451  max resid 0.2519455 \n",
      "Run 11 stress 0.2985932 \n",
      "Run 12 stress 0.3012472 \n",
      "Run 13 stress 0.2993932 \n",
      "Run 14 stress 0.2928704 \n",
      "... Procrustes: rmse 0.03113911  max resid 0.2172557 \n",
      "Run 15 stress 0.3046382 \n",
      "Run 16 stress 0.2962779 \n",
      "Run 17 stress 0.2979551 \n",
      "Run 18 stress 0.2935487 \n",
      "Run 19 stress 0.2950311 \n",
      "Run 20 stress 0.2950899 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n",
      "Run 0 stress 0.2778284 \n",
      "Run 1 stress 0.2799138 \n",
      "Run 2 stress 0.2796833 \n",
      "Run 3 stress 0.2783458 \n",
      "Run 4 stress 0.278067 \n",
      "... Procrustes: rmse 0.02144794  max resid 0.1666497 \n",
      "Run 5 stress 0.2775161 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.02414001  max resid 0.2404796 \n",
      "Run 6 stress 0.2779468 \n",
      "... Procrustes: rmse 0.03024964  max resid 0.227482 \n",
      "Run 7 stress 0.2786503 \n",
      "Run 8 stress 0.2780191 \n",
      "Run 9 stress 0.2772533 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.02750272  max resid 0.2442409 \n",
      "Run 10 stress 0.2786166 \n",
      "Run 11 stress 0.2773985 \n",
      "... Procrustes: rmse 0.01867318  max resid 0.1271303 \n",
      "Run 12 stress 0.2771934 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.02517843  max resid 0.2142492 \n",
      "Run 13 stress 0.2776589 \n",
      "... Procrustes: rmse 0.02520514  max resid 0.189975 \n",
      "Run 14 stress 0.2776706 \n",
      "... Procrustes: rmse 0.02155988  max resid 0.1799659 \n",
      "Run 15 stress 0.2796597 \n",
      "Run 16 stress 0.2779072 \n",
      "Run 17 stress 0.2797195 \n",
      "Run 18 stress 0.2829973 \n",
      "Run 19 stress 0.2780919 \n",
      "Run 20 stress 0.2769375 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.01234847  max resid 0.1171713 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "     2: no. of iterations >= maxit\n",
      "    18: stress ratio > sratmax\n",
      "Run 0 stress 0.2326691 \n",
      "Run 1 stress 0.2386941 \n",
      "Run 2 stress 0.2719006 \n",
      "Run 3 stress 0.2572616 \n",
      "Run 4 stress 0.2650005 \n",
      "Run 5 stress 0.2513683 \n",
      "Run 6 stress 0.2423869 \n",
      "Run 7 stress 0.2496754 \n",
      "Run 8 stress 0.2687994 \n",
      "Run 9 stress 0.274765 \n",
      "Run 10 stress 0.2541954 \n",
      "Run 11 stress 0.250598 \n",
      "Run 12 stress 0.2744285 \n",
      "Run 13 stress 0.2708266 \n",
      "Run 14 stress 0.2738837 \n",
      "Run 15 stress 0.2783048 \n",
      "Run 16 stress 0.2431821 \n",
      "Run 17 stress 0.2645671 \n",
      "Run 18 stress 0.2629849 \n",
      "Run 19 stress 0.2509835 \n",
      "Run 20 stress 0.2610643 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n",
      "Saving 7 x 7 in image\n",
      "[1] \"date\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 486 taxa and 139 samples ]\n",
      "sample_data() Sample Data:       [ 139 samples by 15 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 486 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 486 tips and 485 internal nodes ]\n",
      "[1] \"DNA_quant_mean\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 486 taxa and 126 samples ]\n",
      "sample_data() Sample Data:       [ 126 samples by 15 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 486 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 486 tips and 485 internal nodes ]\n",
      "[1] \"location\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 486 taxa and 139 samples ]\n",
      "sample_data() Sample Data:       [ 139 samples by 15 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 486 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 486 tips and 485 internal nodes ]\n",
      "Run 0 stress 0.2841796 \n",
      "Run 1 stress 0.2888654 \n",
      "Run 2 stress 0.2876604 \n",
      "Run 3 stress 0.2923156 \n",
      "Run 4 stress 0.2916074 \n",
      "Run 5 stress 0.2849591 \n",
      "Run 6 stress 0.2853465 \n",
      "Run 7 stress 0.2926252 \n",
      "Run 8 stress 0.2853055 \n",
      "Run 9 stress 0.2876559 \n",
      "Run 10 stress 0.291576 \n",
      "Run 11 stress 0.2908124 \n",
      "Run 12 stress 0.2851061 \n",
      "Run 13 stress 0.3026651 \n",
      "Run 14 stress 0.2847617 \n",
      "Run 15 stress 0.2891051 \n",
      "Run 16 stress 0.3083399 \n",
      "Run 17 stress 0.2911396 \n",
      "Run 18 stress 0.2881875 \n",
      "Run 19 stress 0.2885832 \n",
      "Run 20 stress 0.2851393 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "     1: no. of iterations >= maxit\n",
      "    19: stress ratio > sratmax\n",
      "Run 0 stress 0.2972208 \n",
      "Run 1 stress 0.2926897 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.05925242  max resid 0.2440471 \n",
      "Run 2 stress 0.2935088 \n",
      "Run 3 stress 0.295881 \n",
      "Run 4 stress 0.2919767 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.03908588  max resid 0.215404 \n",
      "Run 5 stress 0.3115526 \n",
      "Run 6 stress 0.3032336 \n",
      "Run 7 stress 0.3002838 \n",
      "Run 8 stress 0.3023415 \n",
      "Run 9 stress 0.2956961 \n",
      "Run 10 stress 0.2986295 \n",
      "Run 11 stress 0.3050513 \n",
      "Run 12 stress 0.3001349 \n",
      "Run 13 stress 0.297831 \n",
      "Run 14 stress 0.2987862 \n",
      "Run 15 stress 0.2994836 \n",
      "Run 16 stress 0.2973557 \n",
      "Run 17 stress 0.2953638 \n",
      "Run 18 stress 0.3079769 \n",
      "Run 19 stress 0.3049094 \n",
      "Run 20 stress 0.2957587 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n",
      "Run 0 stress 0.2778284 \n",
      "Run 1 stress 0.277117 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.02121789  max resid 0.2004457 \n",
      "Run 2 stress 0.2784583 \n",
      "Run 3 stress 0.4146859 \n",
      "Run 4 stress 0.2804555 \n",
      "Run 5 stress 0.2781929 \n",
      "Run 6 stress 0.2794045 \n",
      "Run 7 stress 0.27846 \n",
      "Run 8 stress 0.278893 \n",
      "Run 9 stress 0.2786297 \n",
      "Run 10 stress 0.2776623 \n",
      "Run 11 stress 0.2786402 \n",
      "Run 12 stress 0.2780536 \n",
      "Run 13 stress 0.2799387 \n",
      "Run 14 stress 0.278114 \n",
      "Run 15 stress 0.2781137 \n",
      "Run 16 stress 0.2767975 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.01153359  max resid 0.1158395 \n",
      "Run 17 stress 0.2871978 \n",
      "Run 18 stress 0.2779927 \n",
      "Run 19 stress 0.2782945 \n",
      "Run 20 stress 0.2805687 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "     2: no. of iterations >= maxit\n",
      "    18: stress ratio > sratmax\n",
      "Run 0 stress 0.2326691 \n",
      "Run 1 stress 0.2791086 \n",
      "Run 2 stress 0.2615669 \n",
      "Run 3 stress 0.2532719 \n",
      "Run 4 stress 0.2529873 \n",
      "Run 5 stress 0.2367913 \n",
      "Run 6 stress 0.2522389 \n",
      "Run 7 stress 0.2525448 \n",
      "Run 8 stress 0.2637164 \n",
      "Run 9 stress 0.2510048 \n",
      "Run 10 stress 0.2652542 \n",
      "Run 11 stress 0.2364105 \n",
      "Run 12 stress 0.2590142 \n",
      "Run 13 stress 0.2562406 \n",
      "Run 14 stress 0.2575768 \n",
      "Run 15 stress 0.2675787 \n",
      "Run 16 stress 0.2406929 \n",
      "Run 17 stress 0.2342704 \n",
      "Run 18 stress 0.2810198 \n",
      "Run 19 stress 0.2603631 \n",
      "Run 20 stress 0.2564059 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 252 rows containing missing values (geom_errorbar). \n",
      "2: Removed 254 rows containing missing values (geom_errorbar). \n",
      "3: Removed 252 rows containing missing values (geom_errorbar). \n",
      "4: Removed 278 rows containing missing values (geom_errorbar). \n",
      "5: Removed 254 rows containing missing values (geom_errorbar). \n",
      "6: Removed 278 rows containing missing values (geom_errorbar). \n",
      "7: Removed 278 rows containing missing values (geom_errorbar). \n",
      "8: Removed 278 rows containing missing values (geom_errorbar). \n"
     ]
    }
   ],
   "source": [
    "#run phyloseq/vegan script\n",
    "!$HOME/amptk/util/phyloseq_graphs.R mhc.biom mhc.tree.phy mhc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                 braycurtis            raupcrick            unifrac            weighted_unifrac           \r\n",
      "                     Adonis Betadisper    Adonis Betadisper  Adonis Betadisper           Adonis Betadisper\r\n",
      "CT_result            0.4170     0.2756    0.3010     0.8094  0.7191     0.5817           0.2748     0.5675\r\n",
      "DNA_quant_mean       0.0169     0.0001    0.2900     0.0001  0.8116     0.0001           0.0366     0.0001\r\n",
      "CT_mean              0.9622     0.0001    0.9296     0.0039  0.3744     0.0002           0.1261     0.0002\r\n",
      "age_class            0.0439     0.0009    0.0048     0.1028  0.0186     0.5920           0.0102     0.5967\r\n",
      "site_date            0.4262     0.0575    0.0704     0.1199  0.2292     0.0150           0.3872     0.0138\r\n",
      "site                 0.2501     0.0284    0.0337     0.1437  0.1629     0.0352           0.3001     0.0352\r\n",
      "sex                  0.6454     0.9867    0.2781     0.1570  0.9151     0.0117           0.4475     0.0104\r\n",
      "DNA_quant_result     0.3542     0.2116    0.3513     0.6831  0.6854     0.8254           0.2581     0.8290\r\n",
      "WNS_status           0.1070     0.2411    0.0767     0.3604  0.2161     0.0597           0.1640     0.0656\r\n",
      "date                 0.3072     0.0001    0.1606     0.0001  0.0658     0.0001           0.0541     0.0001\r\n",
      "location             0.0801     0.0667    0.0192     0.8026  0.4446     0.3666           0.1956     0.3616\r\n"
     ]
    }
   ],
   "source": [
    "#summarize the statistics \n",
    "!$HOME/amptk/util/parse_adonis.py phyloseq_all"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### We still have very weak structure here...\n",
    "![SiteMap](mhc_phyloseq_all/mhc.location.ordinations.pdf)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![SiteMap](mhc_phyloseq_all/mhc.location.alpha_diversity.pdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
