{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[03:24:07 PM]\u001b[0m: OS: MacOSX 10.12.4, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[03:24:08 PM]\u001b[0m: amptk v.0.9.3, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[03:24:08 PM]\u001b[0m: Foward primer: AGGGTGCTCCTCACAGCCCTGTG,  Rev comp'd rev primer: CAGGAAATKYGCYGCGGGGACA\n",
      "\u001b[92m[03:24:08 PM]\u001b[0m: Loading FASTQ Records\n",
      "\u001b[92m[03:24:12 PM]\u001b[0m: 2,840,048 reads (1.7 GB)\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[03:54:50 PM]\u001b[0m: Concatenating Demuxed Files\n",
      "\u001b[92m[03:54:54 PM]\u001b[0m: 2,840,048 total reads\n",
      "\u001b[92m[03:54:54 PM]\u001b[0m: 2,517,886 valid Barcode\n",
      "\u001b[92m[03:54:54 PM]\u001b[0m: 2,321,440 Fwd Primer found, 326,493 Rev Primer found\n",
      "\u001b[92m[03:54:54 PM]\u001b[0m: 5,127 discarded too short (< 125 bp)\n",
      "\u001b[92m[03:54:54 PM]\u001b[0m: 321,366 valid output reads\n",
      "\u001b[92m[03:54:55 PM]\u001b[0m: Found 44 barcoded samples\n",
      "                Sample:  Count\n",
      "                 NY040:  26088\n",
      "                 NY012:  25348\n",
      "                 NY047:  21492\n",
      "                 WI091:  15646\n",
      "                 NY001:  13602\n",
      "                 NY017:  13353\n",
      "                 NY008:  12801\n",
      "                NY033B:  12262\n",
      "                 NY016:  12105\n",
      "                 NY004:  10955\n",
      "                 NY048:  10885\n",
      "                 NY014:  10617\n",
      "                 NY054:  9047\n",
      "                 WI048:  8906\n",
      "                 NY002:  7797\n",
      "                NY031B:  7772\n",
      "                 NY051:  7745\n",
      "                  NY34:  7706\n",
      "                 NY019:  7519\n",
      "                 NY015:  7497\n",
      "                 NY038:  7093\n",
      "                 NY041:  6819\n",
      "                NY034B:  6040\n",
      "                 NY046:  5570\n",
      "                NY035B:  5455\n",
      "                 WI029:  5065\n",
      "                 NY007:  5020\n",
      "                 NY032:  4615\n",
      "                 NY052:  4240\n",
      "                 NY053:  4021\n",
      "                 NY003:  3647\n",
      "                 NY010:  3195\n",
      "                 NY050:  2702\n",
      "                 NY049:  2230\n",
      "                 NY011:  1615\n",
      "                 NY044:  1354\n",
      "                 NY018:  920\n",
      "                 NY020:  710\n",
      "                 NY009:  525\n",
      "                 NY005:  483\n",
      "                 NY036:  292\n",
      "                 NY006:  277\n",
      "                 NY013:  211\n",
      "                 NY045:  124\n",
      "\u001b[92m[03:54:55 PM]\u001b[0m: Output file:  MHC2a.demux.fq (185.5 MB)\n",
      "\u001b[92m[03:54:55 PM]\u001b[0m: Mapping file: ../MHC2a.mapping_file.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd: \u001b[0mamptk cluster -i MHC2a.demux.fq -o out\n",
      "\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[03:55:04 PM]\u001b[0m: OS: MacOSX 10.12.4, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[03:55:04 PM]\u001b[0m: amptk v.0.9.3, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[03:55:04 PM]\u001b[0m: Foward primer: AGGGTGCTCCTCACAGCCCTGTG,  Rev comp'd rev primer: CAGGAAATKYGCYGCGGGGACA\n",
      "\u001b[92m[03:55:04 PM]\u001b[0m: Loading FASTQ Records\n",
      "\u001b[92m[03:55:07 PM]\u001b[0m: 2,296,224 reads (1.7 GB)\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[04:24:28 PM]\u001b[0m: Concatenating Demuxed Files\n",
      "\u001b[92m[04:24:31 PM]\u001b[0m: 2,296,224 total reads\n",
      "\u001b[92m[04:24:31 PM]\u001b[0m: 1,326,954 valid Barcode\n",
      "\u001b[92m[04:24:31 PM]\u001b[0m: 1,061,137 Fwd Primer found, 142,665 Rev Primer found\n",
      "\u001b[92m[04:24:31 PM]\u001b[0m: 10,071 discarded too short (< 125 bp)\n",
      "\u001b[92m[04:24:31 PM]\u001b[0m: 132,594 valid output reads\n",
      "\u001b[92m[04:24:32 PM]\u001b[0m: Found 43 barcoded samples\n",
      "                Sample:  Count\n",
      "                 NY032:  46546\n",
      "                 NY053:  17398\n",
      "                  NY34:  5642\n",
      "                 NY051:  5413\n",
      "                 NY047:  4480\n",
      "                 WI091:  3209\n",
      "                 NY046:  2631\n",
      "                 NY020:  2554\n",
      "                NY031B:  2528\n",
      "                 NY001:  2476\n",
      "                 NY038:  2367\n",
      "                 NY015:  2098\n",
      "                 NY012:  2028\n",
      "                 WI048:  1802\n",
      "                 NY004:  1796\n",
      "                 NY016:  1795\n",
      "                 NY019:  1572\n",
      "                 NY040:  1509\n",
      "                 NY002:  1491\n",
      "                NY033B:  1467\n",
      "                 NY014:  1446\n",
      "                 NY010:  1408\n",
      "                NY034B:  1397\n",
      "                 NY017:  1388\n",
      "                 NY009:  1258\n",
      "                 NY011:  1239\n",
      "                 NY008:  1175\n",
      "                 NY007:  1160\n",
      "                 NY045:  1070\n",
      "                 NY013:  1041\n",
      "                 NY005:  1015\n",
      "                 NY054:  989\n",
      "                 NY003:  936\n",
      "                 NY018:  880\n",
      "                 NY044:  858\n",
      "                 NY048:  803\n",
      "                 WI029:  673\n",
      "                 NY050:  663\n",
      "                 NY052:  609\n",
      "                 NY041:  608\n",
      "                 NY036:  482\n",
      "                NY035B:  382\n",
      "                 NY006:  312\n",
      "\u001b[92m[04:24:32 PM]\u001b[0m: Output file:  MHC2b.demux.fq (76.0 MB)\n",
      "\u001b[92m[04:24:32 PM]\u001b[0m: Mapping file: ../MHC2b.mapping_file.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd: \u001b[0mamptk cluster -i MHC2b.demux.fq -o out\n",
      "\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[04:24:36 PM]\u001b[0m: OS: MacOSX 10.12.4, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[04:24:36 PM]\u001b[0m: amptk v.0.9.3, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[04:24:36 PM]\u001b[0m: Foward primer: AGGGTGCTCCTCACAGCCCTGTG,  Rev comp'd rev primer: CAGGAAATKYGCYGCGGGGACA\n",
      "\u001b[92m[04:24:36 PM]\u001b[0m: Loading FASTQ Records\n",
      "\u001b[92m[04:24:39 PM]\u001b[0m: 2,471,729 reads (1.5 GB)\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[04:59:19 PM]\u001b[0m: Concatenating Demuxed Files\n",
      "\u001b[92m[04:59:23 PM]\u001b[0m: 2,471,729 total reads\n",
      "\u001b[92m[04:59:23 PM]\u001b[0m: 2,072,227 valid Barcode\n",
      "\u001b[92m[04:59:23 PM]\u001b[0m: 1,886,298 Fwd Primer found, 233,396 Rev Primer found\n",
      "\u001b[92m[04:59:23 PM]\u001b[0m: 1,714 discarded too short (< 125 bp)\n",
      "\u001b[92m[04:59:23 PM]\u001b[0m: 231,682 valid output reads\n",
      "\u001b[92m[04:59:24 PM]\u001b[0m: Found 42 barcoded samples\n",
      "                Sample:  Count\n",
      "                 PA012:  17703\n",
      "                 VT006:  15094\n",
      "                 PA001:  13886\n",
      "                 VT018:  13180\n",
      "                 VT016:  12810\n",
      "                 PA018:  11673\n",
      "                 VT005:  10086\n",
      "                 VT010:  9803\n",
      "                 VT017:  9675\n",
      "                 VT014:  9482\n",
      "                 PA007:  8708\n",
      "                 VT001:  8606\n",
      "                 VT011:  7868\n",
      "                 PA002:  7715\n",
      "                 VT013:  7551\n",
      "                 PA015:  6635\n",
      "                 PA020:  5737\n",
      "                 PA011:  5573\n",
      "                 MI126:  5026\n",
      "                 PA010:  4821\n",
      "                 PA017:  4707\n",
      "                 PA019:  4564\n",
      "                 PA016:  3992\n",
      "                 PA004:  3449\n",
      "                 NY060:  3389\n",
      "                 VT004:  3200\n",
      "                 PA009:  2605\n",
      "                 PA013:  2534\n",
      "                 VT012:  2456\n",
      "                 VT003:  2275\n",
      "                 VT015:  1560\n",
      "                 VT002:  1221\n",
      "                 MI127:  1139\n",
      "                 VT007:  819\n",
      "                 NY058:  737\n",
      "                 PA003:  524\n",
      "                 PA014:  507\n",
      "                 PA006:  119\n",
      "                 PA008:  96\n",
      "                 VT019:  64\n",
      "                 VT020:  54\n",
      "                 PA005:  39\n",
      "\u001b[92m[04:59:24 PM]\u001b[0m: Output file:  MHC3.demux.fq (133.7 MB)\n",
      "\u001b[92m[04:59:24 PM]\u001b[0m: Mapping file: ../MHC3.mapping_file.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd: \u001b[0mamptk cluster -i MHC3.demux.fq -o out\n",
      "\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[04:59:28 PM]\u001b[0m: OS: MacOSX 10.12.4, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[04:59:28 PM]\u001b[0m: amptk v.0.9.3, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[04:59:28 PM]\u001b[0m: Foward primer: AGGGTGCTCCTCACAGCCCTGTG,  Rev comp'd rev primer: CAGGAAATKYGCYGCGGGGACA\n",
      "\u001b[92m[04:59:28 PM]\u001b[0m: Loading FASTQ Records\n",
      "\u001b[92m[04:59:31 PM]\u001b[0m: 2,337,635 reads (1.5 GB)\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[05:31:23 PM]\u001b[0m: Concatenating Demuxed Files\n",
      "\u001b[92m[05:31:26 PM]\u001b[0m: 2,337,635 total reads\n",
      "\u001b[92m[05:31:26 PM]\u001b[0m: 1,909,629 valid Barcode\n",
      "\u001b[92m[05:31:26 PM]\u001b[0m: 1,717,114 Fwd Primer found, 215,521 Rev Primer found\n",
      "\u001b[92m[05:31:26 PM]\u001b[0m: 1,601 discarded too short (< 125 bp)\n",
      "\u001b[92m[05:31:26 PM]\u001b[0m: 213,920 valid output reads\n",
      "\u001b[92m[05:31:26 PM]\u001b[0m: Found 49 barcoded samples\n",
      "                Sample:  Count\n",
      "                 WI054:  14924\n",
      "                 MI130:  13828\n",
      "                 WI025:  11572\n",
      "                 WI030:  11334\n",
      "                 WI033:  10243\n",
      "                 MI139:  10134\n",
      "                 WI035:  9944\n",
      "                 MI134:  9327\n",
      "                 WI026:  8491\n",
      "                NY032B:  7100\n",
      "                 WI088:  6975\n",
      "                 WI051:  6963\n",
      "                 VT009:  6738\n",
      "                 MI136:  6389\n",
      "                 MI144:  6060\n",
      "                 MI132:  5548\n",
      "                 WI049:  5228\n",
      "                 WI089:  4626\n",
      "                 WI028:  3996\n",
      "                 WI032:  3837\n",
      "                 WI087:  3818\n",
      "                 WI027:  3531\n",
      "                 MI143:  3508\n",
      "                 WI090:  3491\n",
      "                 MI128:  3085\n",
      "                 WI031:  2867\n",
      "                 MI133:  2748\n",
      "                 WI034:  2688\n",
      "                 MI129:  2594\n",
      "                 WI045:  2492\n",
      "                 MI141:  2210\n",
      "                 MI138:  2157\n",
      "                 MI140:  2090\n",
      "                 MI135:  2074\n",
      "                 MI137:  1916\n",
      "                 WI093:  1609\n",
      "                 WI053:  1443\n",
      "                 WI047:  1271\n",
      "                 MI145:  1012\n",
      "                 WI095:  913\n",
      "                 NY042:  594\n",
      "                 WI046:  580\n",
      "                 WI094:  490\n",
      "                 WI086:  472\n",
      "                 WI052:  297\n",
      "                 WI050:  279\n",
      "                 WI092:  188\n",
      "                 MI142:  130\n",
      "                 MI131:  116\n",
      "\u001b[92m[05:31:26 PM]\u001b[0m: Output file:  MHC4.demux.fq (123.3 MB)\n",
      "\u001b[92m[05:31:26 PM]\u001b[0m: Mapping file: ../MHC4.mapping_file.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd: \u001b[0mamptk cluster -i MHC4.demux.fq -o out\n",
      "\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[05:31:29 PM]\u001b[0m: OS: MacOSX 10.12.4, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[05:31:29 PM]\u001b[0m: amptk v.0.9.3, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[05:31:29 PM]\u001b[0m: Foward primer: AGGGTGCTCCTCACAGCCCTGTG,  Rev comp'd rev primer: CAGGAAATKYGCYGCGGGGACA\n",
      "\u001b[92m[05:31:29 PM]\u001b[0m: Loading FASTQ Records\n",
      "\u001b[92m[05:31:33 PM]\u001b[0m: 2,882,965 reads (2.3 GB)\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[06:38:35 PM]\u001b[0m: Concatenating Demuxed Files\n",
      "\u001b[92m[06:38:41 PM]\u001b[0m: 2,882,965 total reads\n",
      "\u001b[92m[06:38:41 PM]\u001b[0m: 2,432,690 valid Barcode\n",
      "\u001b[92m[06:38:41 PM]\u001b[0m: 2,224,115 Fwd Primer found, 282,257 Rev Primer found\n",
      "\u001b[92m[06:38:41 PM]\u001b[0m: 2,849 discarded too short (< 125 bp)\n",
      "\u001b[92m[06:38:41 PM]\u001b[0m: 279,408 valid output reads\n",
      "\u001b[92m[06:38:42 PM]\u001b[0m: Found 46 barcoded samples\n",
      "                Sample:  Count\n",
      "                 WI107:  20334\n",
      "                 WI098:  16590\n",
      "                 WI105:  14021\n",
      "                 WI169:  13544\n",
      "                 WI076:  13365\n",
      "                 WI078:  12562\n",
      "                 WI062:  12531\n",
      "                 WI166:  12342\n",
      "                 WI084:  9679\n",
      "                 WI171:  9252\n",
      "                 WI104:  8607\n",
      "                 WI103:  8474\n",
      "                 WI083:  8356\n",
      "                 WI168:  8124\n",
      "                 WI064:  7663\n",
      "                 WI081:  6724\n",
      "                 WI097:  6544\n",
      "                 WI066:  6282\n",
      "                 WI101:  6243\n",
      "                 WI163:  5717\n",
      "                 WI067:  5711\n",
      "                 WI073:  5079\n",
      "                 IL030:  4771\n",
      "                 WI057:  4703\n",
      "                 IL020:  4668\n",
      "                 WI075:  4509\n",
      "                 WI058:  4461\n",
      "                 WI072:  3717\n",
      "                 WI099:  3537\n",
      "                 WI165:  3531\n",
      "                 IL032:  3267\n",
      "                 WI106:  3190\n",
      "                 WI065:  2907\n",
      "                 IL024:  2894\n",
      "                 WI100:  2661\n",
      "                 WI068:  2433\n",
      "                 WI096:  2373\n",
      "                 WI170:  1578\n",
      "                 IL022:  1553\n",
      "                 WI080:  1395\n",
      "                 WI172:  1387\n",
      "                 WI167:  1226\n",
      "                 WI102:  394\n",
      "                 WI164:  264\n",
      "                 IL016:  171\n",
      "                 IL018:  74\n",
      "\u001b[92m[06:38:42 PM]\u001b[0m: Output file:  MHC5.demux.fq (166.0 MB)\n",
      "\u001b[92m[06:38:42 PM]\u001b[0m: Mapping file: ../MHC5.mapping_file.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd: \u001b[0mamptk cluster -i MHC5.demux.fq -o out\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#First step is to demultiplex each of the 5 runs. Will use AMPtk to demultiplex the data\n",
    "#this will keep only amplicons where both forward and reverse primers are found, as well as discard reads < 125 bp\n",
    "!for i in MHC2a MHC2b MHC3 MHC4 MHC5; \\\n",
    "    do amptk ion -i ../$i.fastq -m ../$i.mapping_file.txt -o $i --full_length --min_len 125; \\\n",
    "    done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#concatenate all files together\n",
    "!cat mhc2a.demux.fq mhc2b.demux.fq mhc3.demux.fq mhc4.demux.fq mhc5.demux.fq > mhc.all.demux.fq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----------------------------------\n",
      "Found 181 barcoded samples\n",
      "              Sample:  Count\n",
      "               NY032:  51161\n",
      "               NY040:  27597\n",
      "               NY012:  27376\n",
      "               NY047:  25972\n",
      "               NY053:  21419\n",
      "               WI107:  20334\n",
      "               WI091:  18855\n",
      "               PA012:  17703\n",
      "               WI098:  16590\n",
      "               NY001:  16078\n",
      "               VT006:  15094\n",
      "               WI054:  14924\n",
      "               NY017:  14741\n",
      "               WI105:  14021\n",
      "               NY008:  13976\n",
      "               NY016:  13900\n",
      "               PA001:  13886\n",
      "               MI130:  13828\n",
      "              NY033B:  13729\n",
      "               WI169:  13544\n",
      "               WI076:  13365\n",
      "                NY34:  13348\n",
      "               VT018:  13180\n",
      "               NY051:  13158\n",
      "               VT016:  12810\n",
      "               NY004:  12751\n",
      "               WI078:  12562\n",
      "               WI062:  12531\n",
      "               WI166:  12342\n",
      "               NY014:  12063\n",
      "               NY048:  11688\n",
      "               PA018:  11673\n",
      "               WI025:  11572\n",
      "               WI030:  11334\n",
      "               WI048:  10708\n",
      "              NY031B:  10300\n",
      "               WI033:  10243\n",
      "               MI139:  10134\n",
      "               VT005:  10086\n",
      "               NY054:  10036\n",
      "               WI035:  9944\n",
      "               VT010:  9803\n",
      "               WI084:  9679\n",
      "               VT017:  9675\n",
      "               NY015:  9595\n",
      "               VT014:  9482\n",
      "               NY038:  9460\n",
      "               MI134:  9327\n",
      "               NY002:  9288\n",
      "               WI171:  9252\n",
      "               NY019:  9091\n",
      "               PA007:  8708\n",
      "               WI104:  8607\n",
      "               VT001:  8606\n",
      "               WI026:  8491\n",
      "               WI103:  8474\n",
      "               WI083:  8356\n",
      "               NY046:  8201\n",
      "               WI168:  8124\n",
      "               VT011:  7868\n",
      "               PA002:  7715\n",
      "               WI064:  7663\n",
      "               VT013:  7551\n",
      "              NY034B:  7437\n",
      "               NY041:  7427\n",
      "              NY032B:  7100\n",
      "               WI088:  6975\n",
      "               WI051:  6963\n",
      "               VT009:  6738\n",
      "               WI081:  6724\n",
      "               PA015:  6635\n",
      "               WI097:  6544\n",
      "               MI136:  6389\n",
      "               WI066:  6282\n",
      "               WI101:  6243\n",
      "               NY007:  6180\n",
      "               MI144:  6060\n",
      "              NY035B:  5837\n",
      "               WI029:  5738\n",
      "               PA020:  5737\n",
      "               WI163:  5717\n",
      "               WI067:  5711\n",
      "               PA011:  5573\n",
      "               MI132:  5548\n",
      "               WI049:  5228\n",
      "               WI073:  5079\n",
      "               MI126:  5026\n",
      "               NY052:  4849\n",
      "               PA010:  4821\n",
      "               IL030:  4771\n",
      "               PA017:  4707\n",
      "               WI057:  4703\n",
      "               IL020:  4668\n",
      "               WI089:  4626\n",
      "               NY010:  4603\n",
      "               NY003:  4583\n",
      "               PA019:  4564\n",
      "               WI075:  4509\n",
      "               WI058:  4461\n",
      "               WI028:  3996\n",
      "               PA016:  3992\n",
      "               WI032:  3837\n",
      "               WI087:  3818\n",
      "               WI072:  3717\n",
      "               WI099:  3537\n",
      "               WI027:  3531\n",
      "               WI165:  3531\n",
      "               MI143:  3508\n",
      "               WI090:  3491\n",
      "               PA004:  3449\n",
      "               NY060:  3389\n",
      "               NY050:  3365\n",
      "               IL032:  3267\n",
      "               NY020:  3264\n",
      "               VT004:  3200\n",
      "               WI106:  3190\n",
      "               MI128:  3085\n",
      "               WI065:  2907\n",
      "               IL024:  2894\n",
      "               WI031:  2867\n",
      "               NY011:  2854\n",
      "               MI133:  2748\n",
      "               WI034:  2688\n",
      "               WI100:  2661\n",
      "               PA009:  2605\n",
      "               MI129:  2594\n",
      "               PA013:  2534\n",
      "               WI045:  2492\n",
      "               VT012:  2456\n",
      "               WI068:  2433\n",
      "               WI096:  2373\n",
      "               VT003:  2275\n",
      "               NY049:  2230\n",
      "               NY044:  2212\n",
      "               MI141:  2210\n",
      "               MI138:  2157\n",
      "               MI140:  2090\n",
      "               MI135:  2074\n",
      "               MI137:  1916\n",
      "               NY018:  1800\n",
      "               NY009:  1783\n",
      "               WI093:  1609\n",
      "               WI170:  1578\n",
      "               VT015:  1560\n",
      "               IL022:  1553\n",
      "               NY005:  1498\n",
      "               WI053:  1443\n",
      "               WI080:  1395\n",
      "               WI172:  1387\n",
      "               WI047:  1271\n",
      "               NY013:  1252\n",
      "               WI167:  1226\n",
      "               VT002:  1221\n",
      "               NY045:  1194\n",
      "               MI127:  1139\n",
      "               MI145:  1012\n",
      "               WI095:  913\n",
      "               VT007:  819\n",
      "               NY036:  774\n",
      "               NY058:  737\n",
      "               NY042:  594\n",
      "               NY006:  589\n",
      "               WI046:  580\n",
      "               PA003:  524\n",
      "               PA014:  507\n",
      "               WI094:  490\n",
      "               WI086:  472\n",
      "               WI102:  394\n",
      "               WI052:  297\n",
      "               WI050:  279\n",
      "               WI164:  264\n",
      "               WI092:  188\n",
      "               IL016:  171\n",
      "               MI142:  130\n",
      "               PA006:  119\n",
      "               MI131:  116\n",
      "               PA008:  96\n",
      "               IL018:  74\n",
      "               VT019:  64\n",
      "               VT020:  54\n",
      "               PA005:  39\n",
      "----------------------------------\n",
      "Read count: 1178970\n",
      "Read length average: 313\n",
      "Read length range: 125 - 535 bp\n",
      "----------------------------------\n"
     ]
    }
   ],
   "source": [
    "!amptk show -i mhc.all.demux.fq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Kept 74363 reads out of 1178970 total reads\r\n"
     ]
    }
   ],
   "source": [
    "#pull out the small subset for methods, as they can't be used in larger study\n",
    "!amptk select -i mhc.all.demux.fq -f ../mhc.subset.methods.txt -o mhc.methods.demux.fq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----------------------------------\n",
      "Found 17 barcoded samples\n",
      "              Sample:  Count\n",
      "               WI169:  13544\n",
      "               WI166:  12342\n",
      "               WI171:  9252\n",
      "               WI168:  8124\n",
      "               WI163:  5717\n",
      "               IL030:  4771\n",
      "               IL020:  4668\n",
      "               WI165:  3531\n",
      "               IL032:  3267\n",
      "               IL024:  2894\n",
      "               WI170:  1578\n",
      "               IL022:  1553\n",
      "               WI172:  1387\n",
      "               WI167:  1226\n",
      "               WI164:  264\n",
      "               IL016:  171\n",
      "               IL018:  74\n",
      "----------------------------------\n",
      "Read count: 74363\n",
      "Read length average: 305\n",
      "Read length range: 125 - 497 bp\n",
      "----------------------------------\n"
     ]
    }
   ],
   "source": [
    "!amptk show -i mhc.methods.demux.fq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[05:30:04 PM]\u001b[0m: OS: MacOSX 10.12.4, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[05:30:04 PM]\u001b[0m: Reverse complementing FASTQ records\n",
      "\u001b[92m[05:30:13 PM]\u001b[0m: Loading FASTQ Records\n",
      "\u001b[92m[05:30:13 PM]\u001b[0m: 74,363 reads (45.2 MB)\n",
      "\u001b[92m[05:30:13 PM]\u001b[0m: Quality Filtering, expected errors < 2.0\n",
      "\u001b[92m[05:30:14 PM]\u001b[0m: 47,638 reads passed\n",
      "\u001b[92m[05:30:14 PM]\u001b[0m: Splitting FASTQ file by Sample\n",
      "\u001b[92m[05:30:17 PM]\u001b[0m: Removed 6 samples with read number < 500 (--min_reads_sample)\n",
      "\u001b[92m[05:30:17 PM]\u001b[0m: Running DADA2 pipeline\n",
      "\u001b[92m[05:31:50 PM]\u001b[0m: R v3.3.3, DADA2 v1.3.5\n",
      "\u001b[92m[05:31:50 PM]\u001b[0m: 306 total inferred sequences (iSeqs)\n",
      "\u001b[92m[05:31:50 PM]\u001b[0m: 27 denovo chimeras removed\n",
      "\u001b[92m[05:31:50 PM]\u001b[0m: 279 valid iSeqs\n",
      "\u001b[92m[05:31:50 PM]\u001b[0m: Translating to Amino Acid Sequence in 3 frames\n",
      "\u001b[92m[05:31:50 PM]\u001b[0m: 111 total translated sequences > 75 aa\n",
      "\u001b[92m[05:31:50 PM]\u001b[0m: Running HMMER3 MHC HMM model\n",
      "\u001b[92m[05:31:50 PM]\u001b[0m: 87 pass HMM filter, 84 are unique AA sequence.\n",
      "\u001b[92m[05:31:50 PM]\u001b[0m: Mapping reads to DADA2 filtered iSeqs\n",
      "\u001b[92m[05:32:04 PM]\u001b[0m: 48,848 reads mapped to iSeqs (66%)\n",
      "\u001b[92m[05:32:05 PM]\u001b[0m: BIOM OTU table created: methods.biom\n",
      "\u001b[92m[05:32:05 PM]\u001b[0m: Generating phylogenetic tree\n",
      "-------------------------------------------------------\n",
      "MHC DADA2 Script has Finished Successfully\n",
      "-------------------------------------------------------\n",
      "11 samples\n",
      "86 total alleles\n",
      "32.270000 +/- 7.520000 average alleles per sample\n",
      "17 - 46 range of alelles\n",
      "-------------------------------------------------------\n",
      "Tmp Folder of files: methods_41298_filtered\n",
      "Intermediate files: methods_41298\n",
      "MHC Alleles: methods.alleles.fa\n",
      "MHC Protein Alleles : methods.alleles.proteins.fa\n",
      "Allelic OTU Table: methods.otu_table.txt\n",
      "Allelic BIOM table: methods.biom\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#now run through MHC DADA2 script, since only a few samples, don't remove alleles found only 1\n",
    "#try increase quality threshold to 2.0, since we have longer reads and need the 3' end \n",
    "#to serve as an anchor, i.e. that is where translation starts\n",
    "!mhc-dada2.py -i mhc.methods.demux.fq --filter_allele 1  \\\n",
    "              -e 2.0 --rev_comp -m ../mhc.mapping_file.txt -o methods --debug"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "!amptk heatmap -i methods.otu_table.txt -o methods.pdf -d tsv -m heatmap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-3\n",
      "[1] ‘1.19.1’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘1.8.4’\n",
      "[1] ‘2.4.3’\n",
      "Run 0 stress 0.1461771 \n",
      "Run 1 stress 0.1904085 \n",
      "Run 2 stress 0.1461772 \n",
      "... Procrustes: rmse 0.0001741912  max resid 0.0003391395 \n",
      "... Similar to previous best\n",
      "Run 3 stress 0.1488007 \n",
      "Run 4 stress 0.146177 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0001888352  max resid 0.0004258482 \n",
      "... Similar to previous best\n",
      "Run 5 stress 0.1461771 \n",
      "... Procrustes: rmse 9.258806e-05  max resid 0.0001961878 \n",
      "... Similar to previous best\n",
      "Run 6 stress 0.1904087 \n",
      "Run 7 stress 0.1461771 \n",
      "... Procrustes: rmse 0.0001820524  max resid 0.0003492834 \n",
      "... Similar to previous best\n",
      "Run 8 stress 0.146177 \n",
      "... New best solution\n",
      "... Procrustes: rmse 6.788699e-05  max resid 0.0001196706 \n",
      "... Similar to previous best\n",
      "Run 9 stress 0.1461769 \n",
      "... New best solution\n",
      "... Procrustes: rmse 5.505809e-05  max resid 8.666009e-05 \n",
      "... Similar to previous best\n",
      "Run 10 stress 0.1487999 \n",
      "Run 11 stress 0.1714604 \n",
      "Run 12 stress 0.1461773 \n",
      "... Procrustes: rmse 0.0002924878  max resid 0.0006413514 \n",
      "... Similar to previous best\n",
      "Run 13 stress 0.1496026 \n",
      "Run 14 stress 0.1718017 \n",
      "Run 15 stress 0.1488001 \n",
      "Run 16 stress 0.146177 \n",
      "... Procrustes: rmse 8.535423e-05  max resid 0.0001929478 \n",
      "... Similar to previous best\n",
      "Run 17 stress 0.1691356 \n",
      "Run 18 stress 0.1638121 \n",
      "Run 19 stress 0.1461769 \n",
      "... New best solution\n",
      "... Procrustes: rmse 2.23292e-05  max resid 3.550761e-05 \n",
      "... Similar to previous best\n",
      "Run 20 stress 0.3407359 \n",
      "*** Solution reached\n",
      "Run 0 stress 0.1493113 \n",
      "Run 1 stress 0.1733783 \n",
      "Run 2 stress 0.167288 \n",
      "Run 3 stress 0.1493113 \n",
      "... New best solution\n",
      "... Procrustes: rmse 4.1775e-05  max resid 7.286015e-05 \n",
      "... Similar to previous best\n",
      "Run 4 stress 0.1493113 \n",
      "... Procrustes: rmse 4.462246e-05  max resid 9.124089e-05 \n",
      "... Similar to previous best\n",
      "Run 5 stress 0.1629006 \n",
      "Run 6 stress 0.1646202 \n",
      "Run 7 stress 0.1493113 \n",
      "... Procrustes: rmse 2.642634e-05  max resid 5.026414e-05 \n",
      "... Similar to previous best\n",
      "Run 8 stress 0.1749612 \n",
      "Run 9 stress 0.1610275 \n",
      "Run 10 stress 0.1493113 \n",
      "... Procrustes: rmse 5.403321e-05  max resid 0.0001128622 \n",
      "... Similar to previous best\n",
      "Run 11 stress 0.1690174 \n",
      "Run 12 stress 0.1646194 \n",
      "Run 13 stress 0.1788681 \n",
      "Run 14 stress 0.1493113 \n",
      "... Procrustes: rmse 5.365169e-05  max resid 9.967988e-05 \n",
      "... Similar to previous best\n",
      "Run 15 stress 0.1672879 \n",
      "Run 16 stress 0.1493113 \n",
      "... Procrustes: rmse 1.277715e-05  max resid 2.338012e-05 \n",
      "... Similar to previous best\n",
      "Run 17 stress 0.1646194 \n",
      "Run 18 stress 0.1493113 \n",
      "... New best solution\n",
      "... Procrustes: rmse 2.398089e-05  max resid 5.059615e-05 \n",
      "... Similar to previous best\n",
      "Run 19 stress 0.1493113 \n",
      "... Procrustes: rmse 9.666275e-06  max resid 1.508669e-05 \n",
      "... Similar to previous best\n",
      "Run 20 stress 0.1629006 \n",
      "*** Solution reached\n",
      "Error in `contrasts<-`(`*tmp*`, value = contr.funs[1 + isOF[nn]]) : \n",
      "  contrasts can be applied only to factors with 2 or more levels\n",
      "Calls: adonis ... model.matrix -> model.matrix.default -> contrasts<-\n",
      "Execution halted\n"
     ]
    }
   ],
   "source": [
    "#run phyloseq/vegan script\n",
    "!$HOME/amptk/util/phyloseq_graphs.R methods.biom methods.tree.phy methods"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![Heatmap](Methods.pdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removed 17 samples\r\n",
      "Kept 1104607 reads out of 1178970 total reads\r\n"
     ]
    }
   ],
   "source": [
    "#now try DADA2 workflow with the rest of the data, drop the Illinois samples as small dataset and meta data is\n",
    "#questionable as well as Neda mine as we don't have any qPCR data for those sample\n",
    "#these samples were used for a smaller subset for methods manuscript\n",
    "!amptk remove -i mhc.all.demux.fq -f ../mhc.subset.methods.txt -o mhc.large.demux.fq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[05:44:00 PM]\u001b[0m: OS: MacOSX 10.12.4, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[05:44:00 PM]\u001b[0m: Reverse complementing FASTQ records\n",
      "\u001b[92m[05:46:03 PM]\u001b[0m: Loading FASTQ Records\n",
      "\u001b[92m[05:46:04 PM]\u001b[0m: 1,104,607 reads (639.3 MB)\n",
      "\u001b[92m[05:46:04 PM]\u001b[0m: Quality Filtering, expected errors < 2.0\n",
      "\u001b[92m[05:46:18 PM]\u001b[0m: 861,929 reads passed\n",
      "\u001b[92m[05:46:18 PM]\u001b[0m: Splitting FASTQ file by Sample\n",
      "\u001b[92m[05:47:22 PM]\u001b[0m: Removed 27 samples with read number < 500 (--min_reads_sample)\n",
      "\u001b[92m[05:47:22 PM]\u001b[0m: Running DADA2 pipeline\n",
      "\u001b[92m[09:13:26 PM]\u001b[0m: R v3.3.3, DADA2 v1.3.5\n",
      "\u001b[92m[09:13:26 PM]\u001b[0m: 1,758 total inferred sequences (iSeqs)\n",
      "\u001b[92m[09:13:26 PM]\u001b[0m: 120 denovo chimeras removed\n",
      "\u001b[92m[09:13:26 PM]\u001b[0m: 1,638 valid iSeqs\n",
      "\u001b[92m[09:13:26 PM]\u001b[0m: Translating to Amino Acid Sequence in 3 frames\n",
      "\u001b[92m[09:13:26 PM]\u001b[0m: 657 total translated sequences > 75 aa\n",
      "\u001b[92m[09:13:26 PM]\u001b[0m: Running HMMER3 MHC HMM model\n",
      "\u001b[92m[09:13:27 PM]\u001b[0m: 483 pass HMM filter, 436 are unique AA sequence.\n",
      "\u001b[92m[09:13:27 PM]\u001b[0m: Mapping reads to DADA2 filtered iSeqs\n",
      "\u001b[92m[09:17:05 PM]\u001b[0m: 917,765 reads mapped to iSeqs (83%)\n",
      "\u001b[92m[09:17:18 PM]\u001b[0m: BIOM OTU table created: all.biom\n",
      "\u001b[92m[09:17:18 PM]\u001b[0m: Generating phylogenetic tree\n",
      "-------------------------------------------------------\n",
      "MHC DADA2 Script has Finished Successfully\n",
      "-------------------------------------------------------\n",
      "130 samples\n",
      "456 total alleles\n",
      "58.400000 +/- 22.320000 average alleles per sample\n",
      "7 - 124 range of alelles\n",
      "-------------------------------------------------------\n",
      "MHC Alleles: all.alleles.fa\n",
      "MHC Protein Alleles : all.alleles.proteins.fa\n",
      "Allelic OTU Table: all.otu_table.txt\n",
      "Allelic BIOM table: all.biom\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#now run through DADA2\n",
    "!mhc-dada2.py -i mhc.large.demux.fq -e 2.0 --rev_comp -m ../mhc.mapping_file.txt -o all "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-3\n",
      "[1] ‘1.19.1’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘1.8.4’\n",
      "[1] ‘2.4.3’\n",
      "Run 0 stress 0.2526258 \n",
      "Run 1 stress 0.2604341 \n",
      "Run 2 stress 0.2643054 \n",
      "Run 3 stress 0.2577333 \n",
      "Run 4 stress 0.2585463 \n",
      "Run 5 stress 0.2562278 \n",
      "Run 6 stress 0.2554188 \n",
      "Run 7 stress 0.2526192 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.001918925  max resid 0.02036672 \n",
      "Run 8 stress 0.2562992 \n",
      "Run 9 stress 0.2616752 \n",
      "Run 10 stress 0.2526209 \n",
      "... Procrustes: rmse 0.007176689  max resid 0.07373238 \n",
      "Run 11 stress 0.2567007 \n",
      "Run 12 stress 0.2547139 \n",
      "Run 13 stress 0.2526505 \n",
      "... Procrustes: rmse 0.007363126  max resid 0.07308424 \n",
      "Run 14 stress 0.2525594 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.004426885  max resid 0.04305596 \n",
      "Run 15 stress 0.253885 \n",
      "Run 16 stress 0.2594483 \n",
      "Run 17 stress 0.252641 \n",
      "... Procrustes: rmse 0.00545821  max resid 0.04154393 \n",
      "Run 18 stress 0.2545161 \n",
      "Run 19 stress 0.2553638 \n",
      "Run 20 stress 0.256304 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "     3: no. of iterations >= maxit\n",
      "    17: stress ratio > sratmax\n",
      "Run 0 stress 0.2618892 \n",
      "Run 1 stress 0.2609171 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.01575773  max resid 0.1586346 \n",
      "Run 2 stress 0.2608737 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.003726487  max resid 0.03544707 \n",
      "Run 3 stress 0.2810558 \n",
      "Run 4 stress 0.319559 \n",
      "Run 5 stress 0.2688914 \n",
      "Run 6 stress 0.2608585 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.01443833  max resid 0.1304269 \n",
      "Run 7 stress 0.2615914 \n",
      "Run 8 stress 0.2608409 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.01470436  max resid 0.1308885 \n",
      "Run 9 stress 0.2635626 \n",
      "Run 10 stress 0.2728579 \n",
      "Run 11 stress 0.2606189 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.01181335  max resid 0.1301575 \n",
      "Run 12 stress 0.2636925 \n",
      "Run 13 stress 0.2654882 \n",
      "Run 14 stress 0.2773259 \n",
      "Run 15 stress 0.2759241 \n",
      "Run 16 stress 0.2636114 \n",
      "Run 17 stress 0.2783556 \n",
      "Run 18 stress 0.260927 \n",
      "... Procrustes: rmse 0.01253808  max resid 0.1312637 \n",
      "Run 19 stress 0.2605893 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.003274137  max resid 0.02631612 \n",
      "Run 20 stress 0.2663333 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "     4: no. of iterations >= maxit\n",
      "    16: stress ratio > sratmax\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "There were 31 warnings (use warnings() to see them)\n"
     ]
    }
   ],
   "source": [
    "!$HOME/amptk/util/phyloseq_graphs.R all.update.biom all.tree.phy all"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![SiteMap](phyloseq_all/all.location.raup.pdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (<ipython-input-57-fa43aa295d37>, line 2)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-57-fa43aa295d37>\"\u001b[0;36m, line \u001b[0;32m2\u001b[0m\n\u001b[0;31m    Call:\u001b[0m\n\u001b[0m        ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "#### There seems to be very little to no structure in these data, even between Eastcoast (NY,VT,PA) vs Midwest (WI,MI).  \n",
    "Call:\n",
    "adonis(formula = rp ~ metadata[[location]], permutations = 9999) \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                           Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[location]]   1     0.364 0.36404  1.7046 0.01314 0.2487\n",
    "Residuals                 128    27.335 0.21356         0.98686       \n",
    "Total                     129    27.700                 1.00000       \n",
    "\n",
    "#### There is also nothing significant when we look at WNS positive vs WNS negative\n",
    "Call:\n",
    "adonis(formula = rp ~ metadata[CT_result], permutations = 9999) \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                           Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[CT_result]]   3    0.5229 0.17429 0.80806 0.01888    0.6\n",
    "Residuals                 126   27.1767 0.21569         0.98112       \n",
    "Total                     129   27.6996                 1.00000       \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![SiteMap](phyloseq_all/all.CT_result.raup.pdf)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Of course Michelle you can look at these data in a lot more detail than I am able to, but if there are not even site differences or geographic differences between populations - there is perhaps no chance of being any correlation with WNS/Pd.  If that is the case, not sure how we want to proceed with these data...."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
