{
 "cells": [
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "###PreProcessing Reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[01:48:59 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[01:49:00 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[01:49:00 PM]\u001b[0m: Foward primer: AGTGARTCATCGAATCTTTG,  Rev comp'd rev primer: GCATATCAATAAGCGGAGGA\n",
      "\u001b[92m[01:49:00 PM]\u001b[0m: Loading FASTQ Records\n",
      "\u001b[92m[01:49:05 PM]\u001b[0m: 4,332,502 reads (3.3 GB)\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[02:32:41 PM]\u001b[0m: Concatenating Demuxed Files\n",
      "\u001b[92m[02:33:12 PM]\u001b[0m: 4,332,502 total reads\n",
      "\u001b[92m[02:33:12 PM]\u001b[0m: 3,115,600 valid Barcode\n",
      "\u001b[92m[02:33:12 PM]\u001b[0m: 2,915,936 Fwd Primer found, 2,158,790 Rev Primer found\n",
      "\u001b[92m[02:33:12 PM]\u001b[0m: 36,792 discarded too short (< 100 bp)\n",
      "\u001b[92m[02:33:12 PM]\u001b[0m: 2,879,144 valid output reads\n",
      "\u001b[92m[02:33:20 PM]\u001b[0m: Found 19 barcoded samples\n",
      "                Sample:  Count\n",
      "              m3-766-1:  205305\n",
      "       m3-mock3-32000b:  189175\n",
      "        m3-mock3-16000:  183277\n",
      "        m3-mock3-3200a:  181421\n",
      "              m3-301-2:  170879\n",
      "              m3-744-2:  160175\n",
      "              m3-500-1:  158095\n",
      "              m3-500-2:  152733\n",
      "              m3-766-2:  152560\n",
      "              m3-301-1:  151264\n",
      "              m3-755-1:  150229\n",
      "              m3-736-1:  148309\n",
      "                 755-2:  138299\n",
      "              m3-736-2:  134956\n",
      "              m3-757-2:  134727\n",
      "              m3-744-1:  130294\n",
      "              m3-712-1:  125701\n",
      "              m3-712-2:  110748\n",
      "               m3-stds:  100997\n",
      "\u001b[92m[02:33:20 PM]\u001b[0m: Output file:  ion3.demux.fq (1.4 GB)\n",
      "\u001b[92m[02:33:20 PM]\u001b[0m: Mapping file: ion3.mapping_file.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd: \u001b[0mamptk cluster -i ion3.demux.fq -o out\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!amptk ion -i mock3.ion.fastq --barcode_fasta mock3.barcodes.fa --min_len 100 -o ion3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[02:33:23 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[02:33:23 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[02:33:23 PM]\u001b[0m: Gzipped files detected, uncompressing\n",
      "\u001b[92m[02:34:48 PM]\u001b[0m: Merging Overlaping Pairs using USEARCH\n",
      "\u001b[92m[02:34:48 PM]\u001b[0m: working on sample m4A-301-1 (Read Length: 300)\n",
      "\u001b[92m[02:34:59 PM]\u001b[0m: 270,478 reads passed (100.0%)\n",
      "\u001b[92m[02:34:59 PM]\u001b[0m: working on sample m4A-301-2 (Read Length: 300)\n",
      "\u001b[92m[02:35:10 PM]\u001b[0m: 296,226 reads passed (100.0%)\n",
      "\u001b[92m[02:35:10 PM]\u001b[0m: working on sample m4A-500-1 (Read Length: 300)\n",
      "\u001b[92m[02:35:24 PM]\u001b[0m: 334,852 reads passed (100.0%)\n",
      "\u001b[92m[02:35:24 PM]\u001b[0m: working on sample m4A-500-2 (Read Length: 300)\n",
      "\u001b[92m[02:35:35 PM]\u001b[0m: 275,990 reads passed (100.0%)\n",
      "\u001b[92m[02:35:35 PM]\u001b[0m: working on sample m4A-712-1 (Read Length: 300)\n",
      "\u001b[92m[02:35:46 PM]\u001b[0m: 265,985 reads passed (100.0%)\n",
      "\u001b[92m[02:35:46 PM]\u001b[0m: working on sample m4A-712-2 (Read Length: 300)\n",
      "\u001b[92m[02:35:56 PM]\u001b[0m: 242,161 reads passed (100.0%)\n",
      "\u001b[92m[02:35:56 PM]\u001b[0m: working on sample m4A-736-1 (Read Length: 300)\n",
      "\u001b[92m[02:36:07 PM]\u001b[0m: 279,896 reads passed (100.0%)\n",
      "\u001b[92m[02:36:07 PM]\u001b[0m: working on sample m4A-736-2 (Read Length: 300)\n",
      "\u001b[92m[02:36:17 PM]\u001b[0m: 264,055 reads passed (100.0%)\n",
      "\u001b[92m[02:36:17 PM]\u001b[0m: working on sample m4A-744-1 (Read Length: 300)\n",
      "\u001b[92m[02:36:29 PM]\u001b[0m: 284,153 reads passed (100.0%)\n",
      "\u001b[92m[02:36:29 PM]\u001b[0m: working on sample m4A-744-2 (Read Length: 300)\n",
      "\u001b[92m[02:36:39 PM]\u001b[0m: 258,138 reads passed (100.0%)\n",
      "\u001b[92m[02:36:39 PM]\u001b[0m: working on sample m4A-755-1 (Read Length: 300)\n",
      "\u001b[92m[02:36:52 PM]\u001b[0m: 305,617 reads passed (100.0%)\n",
      "\u001b[92m[02:36:52 PM]\u001b[0m: working on sample m4A-755-2 (Read Length: 300)\n",
      "\u001b[92m[02:37:03 PM]\u001b[0m: 296,389 reads passed (100.0%)\n",
      "\u001b[92m[02:37:03 PM]\u001b[0m: working on sample m4A-757-1 (Read Length: 300)\n",
      "\u001b[92m[02:37:15 PM]\u001b[0m: 282,080 reads passed (100.0%)\n",
      "\u001b[92m[02:37:15 PM]\u001b[0m: working on sample m4A-757-2 (Read Length: 300)\n",
      "\u001b[92m[02:37:26 PM]\u001b[0m: 277,106 reads passed (100.0%)\n",
      "\u001b[92m[02:37:26 PM]\u001b[0m: working on sample m4A-766-1 (Read Length: 300)\n",
      "\u001b[92m[02:37:38 PM]\u001b[0m: 302,345 reads passed (100.0%)\n",
      "\u001b[92m[02:37:38 PM]\u001b[0m: working on sample m4A-766-2 (Read Length: 300)\n",
      "\u001b[92m[02:37:51 PM]\u001b[0m: 324,127 reads passed (100.0%)\n",
      "\u001b[92m[02:37:51 PM]\u001b[0m: working on sample m4A-mock3-16000 (Read Length: 300)\n",
      "\u001b[92m[02:38:04 PM]\u001b[0m: 284,316 reads passed (100.0%)\n",
      "\u001b[92m[02:38:04 PM]\u001b[0m: working on sample m4A-mock3-32000a (Read Length: 300)\n",
      "\u001b[92m[02:38:16 PM]\u001b[0m: 296,832 reads passed (100.0%)\n",
      "\u001b[92m[02:38:16 PM]\u001b[0m: working on sample m4A-mock3-32000b (Read Length: 300)\n",
      "\u001b[92m[02:38:27 PM]\u001b[0m: 290,925 reads passed (100.0%)\n",
      "\u001b[92m[02:38:27 PM]\u001b[0m: working on sample m4A-stds (Read Length: 300)\n",
      "\u001b[92m[02:38:37 PM]\u001b[0m: 237,351 reads passed (100.0%)\n",
      "\u001b[92m[02:38:37 PM]\u001b[0m: Stripping primers and trim/pad to 250 bp\n",
      "\u001b[92m[02:38:37 PM]\u001b[0m: splitting the job over 8 cpus, but this may still take awhile\n",
      "\u001b[92m[02:38:37 PM]\u001b[0m: Foward primer: GTGARTCATCGAATCTTTG,  Rev comp'd rev primer: GCATATCAATAAGCGGAGGA\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[03:23:42 PM]\u001b[0m: Concatenating Demuxed Files\n",
      "\u001b[92m[03:23:58 PM]\u001b[0m: 5,669,022 total reads\n",
      "\u001b[92m[03:23:58 PM]\u001b[0m: 5,605,244 Fwd Primer found, 5,177,782 Rev Primer found\n",
      "\u001b[92m[03:23:58 PM]\u001b[0m: 392 discarded too short (< 100 bp)\n",
      "\u001b[92m[03:23:58 PM]\u001b[0m: 5,628,414 valid output reads\n",
      "\u001b[92m[03:24:15 PM]\u001b[0m: Found 20 barcoded samples\n",
      "                        Sample:  Count\n",
      "                     m4A-500-1:  332586\n",
      "                     m4A-766-2:  321816\n",
      "                     m4A-755-1:  302871\n",
      "                     m4A-766-1:  302038\n",
      "              m4A-mock3-32000a:  294813\n",
      "                     m4A-755-2:  293758\n",
      "                     m4A-301-2:  293694\n",
      "              m4A-mock3-32000b:  288912\n",
      "               m4A-mock3-16000:  282380\n",
      "                     m4A-744-1:  281436\n",
      "                     m4A-757-1:  280341\n",
      "                     m4A-736-1:  277532\n",
      "                     m4A-757-2:  274936\n",
      "                     m4A-500-2:  274015\n",
      "                     m4A-301-1:  268266\n",
      "                     m4A-712-1:  264849\n",
      "                     m4A-736-2:  261875\n",
      "                     m4A-744-2:  256458\n",
      "                     m4A-712-2:  240173\n",
      "                      m4A-stds:  235665\n",
      "\u001b[92m[03:24:15 PM]\u001b[0m: Output file:  m4a.demux.fq (2.8 GB)\n",
      "\u001b[92m[03:24:15 PM]\u001b[0m: Mapping file: m4a.mapping_file.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd: \u001b[0mamptk cluster -i m4a.demux.fq -o out\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#Now process first illumina run containing termX samples + biomock\n",
    "!amptk illumina -i termX_miseqA -o m4a --min_len 100 --read_length 300 --cleanup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[07:17:12 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[07:17:12 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[07:17:12 PM]\u001b[0m: Gzipped files detected, uncompressing\n",
      "\u001b[92m[07:19:02 PM]\u001b[0m: Merging Overlaping Pairs using USEARCH\n",
      "\u001b[92m[07:19:02 PM]\u001b[0m: working on sample m4a-1mm-301-1 (Read Length: 300)\n",
      "\u001b[92m[07:19:14 PM]\u001b[0m: 283,491 reads passed (100.0%)\n",
      "\u001b[92m[07:19:14 PM]\u001b[0m: working on sample m4a-1mm-301-2 (Read Length: 300)\n",
      "\u001b[92m[07:19:28 PM]\u001b[0m: 310,136 reads passed (100.0%)\n",
      "\u001b[92m[07:19:28 PM]\u001b[0m: working on sample m4a-1mm-500-1 (Read Length: 300)\n",
      "\u001b[92m[07:19:44 PM]\u001b[0m: 351,169 reads passed (100.0%)\n",
      "\u001b[92m[07:19:44 PM]\u001b[0m: working on sample m4a-1mm-500-2 (Read Length: 300)\n",
      "\u001b[92m[07:19:56 PM]\u001b[0m: 289,626 reads passed (100.0%)\n",
      "\u001b[92m[07:19:56 PM]\u001b[0m: working on sample m4a-1mm-712-1 (Read Length: 300)\n",
      "\u001b[92m[07:20:08 PM]\u001b[0m: 277,133 reads passed (100.0%)\n",
      "\u001b[92m[07:20:08 PM]\u001b[0m: working on sample m4a-1mm-712-2 (Read Length: 300)\n",
      "\u001b[92m[07:20:19 PM]\u001b[0m: 259,415 reads passed (100.0%)\n",
      "\u001b[92m[07:20:19 PM]\u001b[0m: working on sample m4a-1mm-736-1 (Read Length: 300)\n",
      "\u001b[92m[07:20:32 PM]\u001b[0m: 293,626 reads passed (100.0%)\n",
      "\u001b[92m[07:20:32 PM]\u001b[0m: working on sample m4a-1mm-736-2 (Read Length: 300)\n",
      "\u001b[92m[07:20:44 PM]\u001b[0m: 279,281 reads passed (100.0%)\n",
      "\u001b[92m[07:20:44 PM]\u001b[0m: working on sample m4a-1mm-744-1 (Read Length: 300)\n",
      "\u001b[92m[07:20:57 PM]\u001b[0m: 296,929 reads passed (100.0%)\n",
      "\u001b[92m[07:20:57 PM]\u001b[0m: working on sample m4a-1mm-744-2 (Read Length: 300)\n",
      "\u001b[92m[07:21:09 PM]\u001b[0m: 276,399 reads passed (100.0%)\n",
      "\u001b[92m[07:21:09 PM]\u001b[0m: working on sample m4a-1mm-755-1 (Read Length: 300)\n",
      "\u001b[92m[07:21:23 PM]\u001b[0m: 324,459 reads passed (100.0%)\n",
      "\u001b[92m[07:21:23 PM]\u001b[0m: working on sample m4a-1mm-755-2 (Read Length: 300)\n",
      "\u001b[92m[07:21:37 PM]\u001b[0m: 309,194 reads passed (100.0%)\n",
      "\u001b[92m[07:21:37 PM]\u001b[0m: working on sample m4a-1mm-757-1 (Read Length: 300)\n",
      "\u001b[92m[07:21:49 PM]\u001b[0m: 298,960 reads passed (100.0%)\n",
      "\u001b[92m[07:21:49 PM]\u001b[0m: working on sample m4a-1mm-757-2 (Read Length: 300)\n",
      "\u001b[92m[07:22:02 PM]\u001b[0m: 288,560 reads passed (100.0%)\n",
      "\u001b[92m[07:22:02 PM]\u001b[0m: working on sample m4a-1mm-766-1 (Read Length: 300)\n",
      "\u001b[92m[07:22:16 PM]\u001b[0m: 317,527 reads passed (100.0%)\n",
      "\u001b[92m[07:22:16 PM]\u001b[0m: working on sample m4a-1mm-766-2 (Read Length: 300)\n",
      "\u001b[92m[07:22:31 PM]\u001b[0m: 342,203 reads passed (100.0%)\n",
      "\u001b[92m[07:22:31 PM]\u001b[0m: working on sample m4a-1mm-mock3-16000 (Read Length: 300)\n",
      "\u001b[92m[07:22:44 PM]\u001b[0m: 297,005 reads passed (100.0%)\n",
      "\u001b[92m[07:22:44 PM]\u001b[0m: working on sample m4a-1mm-mock3-32000a (Read Length: 300)\n",
      "\u001b[92m[07:22:58 PM]\u001b[0m: 312,477 reads passed (100.0%)\n",
      "\u001b[92m[07:22:58 PM]\u001b[0m: working on sample m4a-1mm-mock3-32000b (Read Length: 300)\n",
      "\u001b[92m[07:23:11 PM]\u001b[0m: 302,343 reads passed (100.0%)\n",
      "\u001b[92m[07:23:11 PM]\u001b[0m: working on sample m4a-1mm-stds (Read Length: 300)\n",
      "\u001b[92m[07:23:21 PM]\u001b[0m: 253,161 reads passed (100.0%)\n",
      "\u001b[92m[07:23:21 PM]\u001b[0m: Stripping primers and trim/pad to 250 bp\n",
      "\u001b[92m[07:23:21 PM]\u001b[0m: splitting the job over 8 cpus, but this may still take awhile\n",
      "\u001b[92m[07:23:21 PM]\u001b[0m: Foward primer: GTGARTCATCGAATCTTTG,  Rev comp'd rev primer: GCATATCAATAAGCGGAGGA\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[08:12:47 PM]\u001b[0m: Concatenating Demuxed Files\n",
      "\u001b[92m[08:13:01 PM]\u001b[0m: 5,963,094 total reads\n",
      "\u001b[92m[08:13:01 PM]\u001b[0m: 5,893,099 Fwd Primer found, 5,357,516 Rev Primer found\n",
      "\u001b[92m[08:13:01 PM]\u001b[0m: 408 discarded too short (< 100 bp)\n",
      "\u001b[92m[08:13:01 PM]\u001b[0m: 5,921,159 valid output reads\n",
      "\u001b[92m[08:13:20 PM]\u001b[0m: Found 20 barcoded samples\n",
      "                        Sample:  Count\n",
      "                 m4a-1mm-500-1:  348835\n",
      "                 m4a-1mm-766-2:  339808\n",
      "                 m4a-1mm-755-1:  321611\n",
      "                 m4a-1mm-766-1:  317212\n",
      "          m4a-1mm-mock3-32000a:  310372\n",
      "                 m4a-1mm-301-2:  307544\n",
      "                 m4a-1mm-755-2:  306489\n",
      "          m4a-1mm-mock3-32000b:  300285\n",
      "                 m4a-1mm-757-1:  297145\n",
      "           m4a-1mm-mock3-16000:  295022\n",
      "                 m4a-1mm-744-1:  294155\n",
      "                 m4a-1mm-736-1:  291199\n",
      "                 m4a-1mm-500-2:  287595\n",
      "                 m4a-1mm-757-2:  286344\n",
      "                 m4a-1mm-301-1:  281212\n",
      "                 m4a-1mm-736-2:  277029\n",
      "                 m4a-1mm-712-1:  275977\n",
      "                 m4a-1mm-744-2:  274620\n",
      "                 m4a-1mm-712-2:  257323\n",
      "                  m4a-1mm-stds:  251382\n",
      "\u001b[92m[08:13:20 PM]\u001b[0m: Output file:  m4a1m.demux.fq (3.0 GB)\n",
      "\u001b[92m[08:13:20 PM]\u001b[0m: Mapping file: m4a1m.mapping_file.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd: \u001b[0mamptk cluster -i m4a1m.demux.fq -o out\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#Now process first illumina run containing termX samples + biomock but run with 1 mismatch binning\n",
    "!amptk illumina -i miseqA_1mm -o m4a1m --min_len 100 --read_length 300 --cleanup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[03:24:16 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[03:24:16 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[03:24:16 PM]\u001b[0m: Merging Overlaping Pairs using USEARCH\n",
      "\u001b[92m[03:24:16 PM]\u001b[0m: working on sample m4b-744-1 (Read Length: 300)\n",
      "\u001b[92m[03:24:24 PM]\u001b[0m: 191,554 reads passed (100.0%)\n",
      "\u001b[92m[03:24:24 PM]\u001b[0m: working on sample m4b-744-2 (Read Length: 300)\n",
      "\u001b[92m[03:24:31 PM]\u001b[0m: 156,694 reads passed (100.0%)\n",
      "\u001b[92m[03:24:31 PM]\u001b[0m: working on sample m4b-766-1 (Read Length: 300)\n",
      "\u001b[92m[03:24:38 PM]\u001b[0m: 165,809 reads passed (100.0%)\n",
      "\u001b[92m[03:24:38 PM]\u001b[0m: working on sample m4b-766-2 (Read Length: 300)\n",
      "\u001b[92m[03:24:44 PM]\u001b[0m: 145,681 reads passed (100.0%)\n",
      "\u001b[92m[03:24:44 PM]\u001b[0m: Stripping primers and trim/pad to 250 bp\n",
      "\u001b[92m[03:24:44 PM]\u001b[0m: splitting the job over 8 cpus, but this may still take awhile\n",
      "\u001b[92m[03:24:44 PM]\u001b[0m: Foward primer: GTGARTCATCGAATCTTTG,  Rev comp'd rev primer: GCATATCAATAAGCGGAGGA\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[03:30:24 PM]\u001b[0m: Concatenating Demuxed Files\n",
      "\u001b[92m[03:30:25 PM]\u001b[0m: 659,738 total reads\n",
      "\u001b[92m[03:30:25 PM]\u001b[0m: 650,074 Fwd Primer found, 615,757 Rev Primer found\n",
      "\u001b[92m[03:30:25 PM]\u001b[0m: 12 discarded too short (< 100 bp)\n",
      "\u001b[92m[03:30:25 PM]\u001b[0m: 654,478 valid output reads\n",
      "\u001b[92m[03:30:27 PM]\u001b[0m: Found 4 barcoded samples\n",
      "                        Sample:  Count\n",
      "                     m4b-744-1:  189168\n",
      "                     m4b-766-1:  165624\n",
      "                     m4b-744-2:  155381\n",
      "                     m4b-766-2:  144305\n",
      "\u001b[92m[03:30:27 PM]\u001b[0m: Output file:  m4b.demux.fq (335.4 MB)\n",
      "\u001b[92m[03:30:27 PM]\u001b[0m: Mapping file: m4b.mapping_file.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd: \u001b[0mamptk cluster -i m4b.demux.fq -o out\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#process illumina run B; this was the same 4 libraries run on a different run, shows no index bleed\n",
    "!amptk illumina -i termX_miseqB -o m4b --min_len 100 --read_length 300 --cleanup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[03:30:28 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[03:30:28 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[03:30:28 PM]\u001b[0m: Gzipped files detected, uncompressing\n",
      "\u001b[92m[03:32:04 PM]\u001b[0m: Merging Overlaping Pairs using USEARCH\n",
      "\u001b[92m[03:32:04 PM]\u001b[0m: working on sample m4C-301-1 (Read Length: 300)\n",
      "\u001b[92m[03:32:18 PM]\u001b[0m: 314,816 reads passed (100.0%)\n",
      "\u001b[92m[03:32:18 PM]\u001b[0m: working on sample m4C-301-2 (Read Length: 300)\n",
      "\u001b[92m[03:32:35 PM]\u001b[0m: 388,353 reads passed (100.0%)\n",
      "\u001b[92m[03:32:35 PM]\u001b[0m: working on sample m4C-500-1 (Read Length: 300)\n",
      "\u001b[92m[03:32:48 PM]\u001b[0m: 300,488 reads passed (100.0%)\n",
      "\u001b[92m[03:32:48 PM]\u001b[0m: working on sample m4C-500-2 (Read Length: 300)\n",
      "\u001b[92m[03:33:01 PM]\u001b[0m: 298,858 reads passed (100.0%)\n",
      "\u001b[92m[03:33:01 PM]\u001b[0m: working on sample m4C-712-1 (Read Length: 300)\n",
      "\u001b[92m[03:33:14 PM]\u001b[0m: 307,694 reads passed (100.0%)\n",
      "\u001b[92m[03:33:14 PM]\u001b[0m: working on sample m4C-712-2 (Read Length: 300)\n",
      "\u001b[92m[03:33:26 PM]\u001b[0m: 302,851 reads passed (100.0%)\n",
      "\u001b[92m[03:33:26 PM]\u001b[0m: working on sample m4C-736-1 (Read Length: 300)\n",
      "\u001b[92m[03:33:38 PM]\u001b[0m: 290,077 reads passed (100.0%)\n",
      "\u001b[92m[03:33:38 PM]\u001b[0m: working on sample m4C-736-2 (Read Length: 300)\n",
      "\u001b[92m[03:33:52 PM]\u001b[0m: 316,914 reads passed (100.0%)\n",
      "\u001b[92m[03:33:52 PM]\u001b[0m: working on sample m4C-744-1 (Read Length: 300)\n",
      "\u001b[92m[03:34:02 PM]\u001b[0m: 264,152 reads passed (100.0%)\n",
      "\u001b[92m[03:34:02 PM]\u001b[0m: working on sample m4C-744-2 (Read Length: 300)\n",
      "\u001b[92m[03:34:12 PM]\u001b[0m: 221,198 reads passed (100.0%)\n",
      "\u001b[92m[03:34:12 PM]\u001b[0m: working on sample m4C-755-1 (Read Length: 300)\n",
      "\u001b[92m[03:34:24 PM]\u001b[0m: 281,982 reads passed (100.0%)\n",
      "\u001b[92m[03:34:24 PM]\u001b[0m: working on sample m4C-755-2 (Read Length: 300)\n",
      "\u001b[92m[03:34:40 PM]\u001b[0m: 354,163 reads passed (100.0%)\n",
      "\u001b[92m[03:34:40 PM]\u001b[0m: working on sample m4C-757-1 (Read Length: 300)\n",
      "\u001b[92m[03:34:53 PM]\u001b[0m: 312,291 reads passed (100.0%)\n",
      "\u001b[92m[03:34:53 PM]\u001b[0m: working on sample m4C-757-2 (Read Length: 300)\n",
      "\u001b[92m[03:35:10 PM]\u001b[0m: 392,270 reads passed (100.0%)\n",
      "\u001b[92m[03:35:10 PM]\u001b[0m: working on sample m4C-766-1 (Read Length: 300)\n",
      "\u001b[92m[03:35:23 PM]\u001b[0m: 284,461 reads passed (100.0%)\n",
      "\u001b[92m[03:35:23 PM]\u001b[0m: working on sample m4C-766-2 (Read Length: 300)\n",
      "\u001b[92m[03:35:33 PM]\u001b[0m: 256,778 reads passed (100.0%)\n",
      "\u001b[92m[03:35:33 PM]\u001b[0m: working on sample m4C-mock3-16000 (Read Length: 300)\n",
      "\u001b[92m[03:35:46 PM]\u001b[0m: 300,944 reads passed (100.0%)\n",
      "\u001b[92m[03:35:46 PM]\u001b[0m: working on sample m4C-mock3-32000a (Read Length: 300)\n",
      "\u001b[92m[03:36:00 PM]\u001b[0m: 350,381 reads passed (100.0%)\n",
      "\u001b[92m[03:36:00 PM]\u001b[0m: working on sample m4C-mock3-32000b (Read Length: 300)\n",
      "\u001b[92m[03:36:12 PM]\u001b[0m: 306,888 reads passed (100.0%)\n",
      "\u001b[92m[03:36:12 PM]\u001b[0m: working on sample m4C-stds (Read Length: 300)\n",
      "\u001b[92m[03:36:23 PM]\u001b[0m: 258,286 reads passed (100.0%)\n",
      "\u001b[92m[03:36:23 PM]\u001b[0m: Stripping primers and trim/pad to 250 bp\n",
      "\u001b[92m[03:36:23 PM]\u001b[0m: splitting the job over 8 cpus, but this may still take awhile\n",
      "\u001b[92m[03:36:23 PM]\u001b[0m: Foward primer: GTGARTCATCGAATCTTTG,  Rev comp'd rev primer: GCATATCAATAAGCGGAGGA\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[04:26:32 PM]\u001b[0m: Concatenating Demuxed Files\n",
      "\u001b[92m[04:26:49 PM]\u001b[0m: 6,103,845 total reads\n",
      "\u001b[92m[04:26:49 PM]\u001b[0m: 6,043,185 Fwd Primer found, 4,104,649 Rev Primer found\n",
      "\u001b[92m[04:26:49 PM]\u001b[0m: 1,265 discarded too short (< 100 bp)\n",
      "\u001b[92m[04:26:49 PM]\u001b[0m: 6,069,913 valid output reads\n",
      "\u001b[92m[04:27:08 PM]\u001b[0m: Found 20 barcoded samples\n",
      "                        Sample:  Count\n",
      "                     m4C-757-2:  390122\n",
      "                     m4C-301-2:  385896\n",
      "                     m4C-755-2:  351946\n",
      "              m4C-mock3-32000a:  348020\n",
      "                     m4C-736-2:  314792\n",
      "                     m4C-301-1:  312792\n",
      "                     m4C-757-1:  310890\n",
      "                     m4C-712-1:  306776\n",
      "              m4C-mock3-32000b:  305175\n",
      "                     m4C-712-2:  300925\n",
      "               m4C-mock3-16000:  299132\n",
      "                     m4C-500-1:  299131\n",
      "                     m4C-500-2:  297249\n",
      "                     m4C-736-1:  288079\n",
      "                     m4C-766-1:  284263\n",
      "                     m4C-755-1:  280033\n",
      "                     m4C-744-1:  262240\n",
      "                      m4C-stds:  256792\n",
      "                     m4C-766-2:  255491\n",
      "                     m4C-744-2:  220169\n",
      "\u001b[92m[04:27:08 PM]\u001b[0m: Output file:  m4c.demux.fq (3.0 GB)\n",
      "\u001b[92m[04:27:08 PM]\u001b[0m: Mapping file: m4c.mapping_file.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd: \u001b[0mamptk cluster -i m4c.demux.fq -o out\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#now demux miseq run 3, remade libraries from same amplicons with different barcodes\n",
    "!amptk illumina -i termX_miseqC -o m4c --min_len 100 --read_length 300 --cleanup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[01:32:43 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[01:32:43 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[01:32:43 PM]\u001b[0m: Foward primer: AGTGARTCATCGAATCTTTG,  Rev comp'd rev primer: GCATATCAATAAGCGGAGGA\n",
      "\u001b[92m[01:32:43 PM]\u001b[0m: Loading FASTQ Records\n",
      "\u001b[92m[01:32:48 PM]\u001b[0m: 4,341,392 reads (3.3 GB)\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[02:20:44 PM]\u001b[0m: Concatenating Demuxed Files\n",
      "\u001b[92m[02:21:18 PM]\u001b[0m: 4,341,392 total reads\n",
      "\u001b[92m[02:21:18 PM]\u001b[0m: 3,230,560 valid Barcode\n",
      "\u001b[92m[02:21:18 PM]\u001b[0m: 2,925,109 Fwd Primer found, 2,329,942 Rev Primer found\n",
      "\u001b[92m[02:21:18 PM]\u001b[0m: 35,658 discarded too short (< 100 bp)\n",
      "\u001b[92m[02:21:18 PM]\u001b[0m: 2,889,451 valid output reads\n",
      "\u001b[92m[02:21:27 PM]\u001b[0m: Found 24 barcoded samples\n",
      "                Sample:  Count\n",
      "   m5-SynMock1-mock3.0:  270701\n",
      "              m5-712-2:  232495\n",
      "              m5-301-2:  186706\n",
      "m5-SynMock1.736-1.736-2:  173385\n",
      "              m5-301-1:  173161\n",
      "              m5-500-1:  159088\n",
      "      m5-SynMock1-8000:  145683\n",
      "m5-SynMock2.757-1.757-2:  125919\n",
      "              m5-744-2:  115363\n",
      "              m5-736-2:  110748\n",
      "              m5-757-1:  108256\n",
      "              m5-757-2:  107142\n",
      "              m5-755-1:  104765\n",
      "              m5-736-1:  93187\n",
      "   m5-SynMock1-mock3.5:  91563\n",
      "      m5-mock3.5-16000:  86917\n",
      "              m5-766-1:  86246\n",
      "              m5-755-2:  81537\n",
      "      m5-mock3.0-16000:  80375\n",
      "              m5-766-2:  79478\n",
      "              m5-744-1:  79267\n",
      "      m5-SynMock2-8000:  71677\n",
      "              m5-712-1:  71344\n",
      "              m5-500-2:  54448\n",
      "\u001b[92m[02:21:27 PM]\u001b[0m: Output file:  ion5.demux.fq (1.5 GB)\n",
      "\u001b[92m[02:21:27 PM]\u001b[0m: Mapping file: ion5.mapping_file.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd: \u001b[0mamptk cluster -i ion5.demux.fq -o out\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#now finally one more run, where we used SynMock on same samples using Ion torrent\n",
    "!amptk ion -i mock5.ion.fastq --barcode_fasta mock5.barcodes.fa --min_len 100 -o ion5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#concatenate all miseq runs\n",
    "!cat m4a.demux.fq m4b.demux.fq m4c.demux.fq > miseq.fq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----------------------------------\n",
      "Found 64 barcoded samples\n",
      "              Sample:  Count\n",
      "           m4C-757-2:  390122\n",
      "           m4C-301-2:  385896\n",
      "           m4C-755-2:  351946\n",
      "       m4a-1mm-500-1:  348835\n",
      "    m4C-mock3-32000a:  348020\n",
      "       m4a-1mm-766-2:  339808\n",
      "           m4A-500-1:  332586\n",
      "           m4A-766-2:  321816\n",
      "       m4a-1mm-755-1:  321611\n",
      "       m4a-1mm-766-1:  317212\n",
      "           m4C-736-2:  314792\n",
      "           m4C-301-1:  312792\n",
      "           m4C-757-1:  310890\n",
      "m4a-1mm-mock3-32000a:  310372\n",
      "       m4a-1mm-301-2:  307544\n",
      "           m4C-712-1:  306776\n",
      "       m4a-1mm-755-2:  306489\n",
      "    m4C-mock3-32000b:  305175\n",
      "           m4A-755-1:  302871\n",
      "           m4A-766-1:  302038\n",
      "           m4C-712-2:  300925\n",
      "m4a-1mm-mock3-32000b:  300285\n",
      "     m4C-mock3-16000:  299132\n",
      "           m4C-500-1:  299131\n",
      "           m4C-500-2:  297249\n",
      "       m4a-1mm-757-1:  297145\n",
      " m4a-1mm-mock3-16000:  295022\n",
      "    m4A-mock3-32000a:  294813\n",
      "       m4a-1mm-744-1:  294155\n",
      "           m4A-755-2:  293758\n",
      "           m4A-301-2:  293694\n",
      "       m4a-1mm-736-1:  291199\n",
      "    m4A-mock3-32000b:  288912\n",
      "           m4C-736-1:  288079\n",
      "       m4a-1mm-500-2:  287595\n",
      "       m4a-1mm-757-2:  286344\n",
      "           m4C-766-1:  284263\n",
      "     m4A-mock3-16000:  282380\n",
      "           m4A-744-1:  281436\n",
      "       m4a-1mm-301-1:  281212\n",
      "           m4A-757-1:  280341\n",
      "           m4C-755-1:  280033\n",
      "           m4A-736-1:  277532\n",
      "       m4a-1mm-736-2:  277029\n",
      "       m4a-1mm-712-1:  275977\n",
      "           m4A-757-2:  274936\n",
      "       m4a-1mm-744-2:  274620\n",
      "           m4A-500-2:  274015\n",
      "           m4A-301-1:  268266\n",
      "           m4A-712-1:  264849\n",
      "           m4C-744-1:  262240\n",
      "           m4A-736-2:  261875\n",
      "       m4a-1mm-712-2:  257323\n",
      "            m4C-stds:  256792\n",
      "           m4A-744-2:  256458\n",
      "           m4C-766-2:  255491\n",
      "        m4a-1mm-stds:  251382\n",
      "           m4A-712-2:  240173\n",
      "            m4A-stds:  235665\n",
      "           m4C-744-2:  220169\n",
      "           m4b-744-1:  189168\n",
      "           m4b-766-1:  165624\n",
      "           m4b-744-2:  155381\n",
      "           m4b-766-2:  144305\n",
      "----------------------------------\n",
      "Read count: 18273964\n",
      "Read Length: 250 bp\n",
      "----------------------------------\n"
     ]
    }
   ],
   "source": [
    "!amptk show -i miseq.fq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[08:33:58 AM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[08:33:58 AM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[08:33:58 AM]\u001b[0m: Loading FASTQ Records\n",
      "\u001b[92m[08:34:09 AM]\u001b[0m: 12,352,805 reads (6.2 GB)\n",
      "\u001b[92m[08:34:09 AM]\u001b[0m: Quality Filtering, expected errors < 1.0\n",
      "\u001b[92m[08:38:33 AM]\u001b[0m: 9,739,836 reads passed\n",
      "\u001b[92m[08:38:33 AM]\u001b[0m: De-replication (remove duplicate reads)\n",
      "\u001b[92m[08:39:03 AM]\u001b[0m: 362,215 reads passed\n",
      "\u001b[92m[08:39:04 AM]\u001b[0m: Clustering OTUs (UPARSE)\n",
      "\u001b[92m[08:39:27 AM]\u001b[0m: 2,212 OTUs\n",
      "\u001b[92m[08:39:27 AM]\u001b[0m: Cleaning up padding from OTUs\n",
      "\u001b[92m[08:39:27 AM]\u001b[0m: Mapping Reads to OTUs and Building OTU table\n",
      "\u001b[92m[09:18:22 AM]\u001b[0m: 10,607,514 reads mapped to OTUs (86%)\n",
      "-------------------------------------------------------\n",
      "OTU Clustering Script has Finished Successfully\n",
      "-------------------------------------------------------\n",
      "Clustered OTUs: /Users/jon/projects/synmock_paper/miseq.cluster.otus.fa\n",
      "OTU Table: /Users/jon/projects/synmock_paper/miseq.otu_table.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk filter -i miseq.otu_table.txt -f miseq.cluster.otus.fa -b <mock barcode>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#cluster the miseq data\n",
    "!amptk cluster -i miseq.fq -o miseq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[09:18:46 AM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[09:18:46 AM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[09:18:46 AM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[09:18:46 AM]\u001b[0m: 2,212 OTUs\n",
      "\u001b[92m[09:18:46 AM]\u001b[0m: Global alignment OTUs with usearch_global (USEARCH)\n",
      "\u001b[92m[09:19:07 AM]\u001b[0m: Classifying OTUs with UTAX (USEARCH)\n",
      "\u001b[92m[09:19:07 AM]\u001b[0m: Classifying OTUs with SINTAX (USEARCH)\n",
      "\u001b[92m[09:24:20 AM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[09:24:29 AM]\u001b[0m: Taxonomy finished: miseq.taxonomy.txt\n",
      "\u001b[92m[09:24:29 AM]\u001b[0m: OTUs with taxonomy: miseq.otus.taxonomy.fa\n",
      "\u001b[92m[09:24:29 AM]\u001b[0m: OTU phylogeny: miseq.tree.phy\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#get taxonomy for miseq OTUs\n",
    "!amptk taxonomy -f miseq.cluster.otus.fa -o miseq -d ITS2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 232,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#concatenate all ion runs\n",
    "!cat ion3.demux.fq ion5.demux.fq > ion.fq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 233,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[02:46:03 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[02:46:03 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[02:46:03 PM]\u001b[0m: Loading FASTQ Records\n",
      "\u001b[92m[02:46:08 PM]\u001b[0m: 5,768,595 reads (2.9 GB)\n",
      "\u001b[92m[02:46:08 PM]\u001b[0m: Quality Filtering, expected errors < 1.0\n",
      "\u001b[92m[02:48:13 PM]\u001b[0m: 3,077,553 reads passed\n",
      "\u001b[92m[02:48:13 PM]\u001b[0m: De-replication (remove duplicate reads)\n",
      "\u001b[92m[02:48:23 PM]\u001b[0m: 110,375 reads passed\n",
      "\u001b[92m[02:48:24 PM]\u001b[0m: Clustering OTUs (UPARSE)\n",
      "\u001b[92m[02:48:33 PM]\u001b[0m: 1,382 OTUs\n",
      "\u001b[92m[02:48:33 PM]\u001b[0m: Cleaning up padding from OTUs\n",
      "\u001b[92m[02:48:33 PM]\u001b[0m: Mapping Reads to OTUs and Building OTU table\n",
      "\u001b[92m[02:59:43 PM]\u001b[0m: 4,739,372 reads mapped to OTUs (82%)\n",
      "-------------------------------------------------------\n",
      "OTU Clustering Script has Finished Successfully\n",
      "-------------------------------------------------------\n",
      "Clustered OTUs: /Users/jon/projects/synmock_paper/ion.cluster.otus.fa\n",
      "OTU Table: /Users/jon/projects/synmock_paper/ion.otu_table.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk filter -i ion.otu_table.txt -f ion.cluster.otus.fa -b <mock barcode>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#cluster the ion data\n",
    "!amptk cluster -i ion.fq -o ion"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 234,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[03:03:09 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[03:03:09 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[03:03:09 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[03:03:09 PM]\u001b[0m: 1,382 OTUs\n",
      "\u001b[92m[03:03:09 PM]\u001b[0m: Global alignment OTUs with usearch_global (USEARCH)\n",
      "\u001b[92m[03:03:23 PM]\u001b[0m: Classifying OTUs with UTAX (USEARCH)\n",
      "\u001b[92m[03:03:23 PM]\u001b[0m: Classifying OTUs with SINTAX (USEARCH)\n",
      "\u001b[92m[03:07:04 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[03:07:09 PM]\u001b[0m: Taxonomy finished: ion.taxonomy.txt\n",
      "\u001b[92m[03:07:09 PM]\u001b[0m: OTUs with taxonomy: ion.otus.taxonomy.fa\n",
      "\u001b[92m[03:07:09 PM]\u001b[0m: OTU phylogeny: ion.tree.phy\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#get taxonomy for ion OTUs\n",
    "!amptk taxonomy -f ion.cluster.otus.fa -o ion -d ITS2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 235,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#generate OTU table for each of the runs using the combined OTUs from either miseq or ion\n",
    "#now slice the OTU table for each of the runs, Ion3, Miseq4A, Miseq4C, Ion5\n",
    "#load original table, not the name changed version for normal processing\n",
    "import pandas as pd\n",
    "from natsort import natsorted\n",
    "\n",
    "miseqdf = pd.read_csv('miseq.otu_table.txt', sep='\\t', index_col=0)\n",
    "iondf = pd.read_csv('ion.otu_table.txt', sep='\\t', index_col=0)\n",
    "\n",
    "ion3 = ['m3-301-1', 'm3-301-2', 'm3-500-1', 'm3-500-2', 'm3-712-1', 'm3-712-2', \n",
    "        'm3-736-1', 'm3-736-2', 'm3-744-1', 'm3-744-2', 'm3-755-1', 'm3-755-2',\n",
    "        'm3-757-2', 'm3-766-1', 'm3-766-2', 'm3-mock3-16000']\n",
    "ion3df = iondf.reindex(columns=ion3, index=natsorted(iondf.index))\n",
    "ion3df.to_csv('ion3.otu_table.txt', sep='\\t')\n",
    "miseq4a = ['m4A-301-1', 'm4A-301-2', 'm4A-500-1', 'm4A-500-2', 'm4A-712-1', \n",
    "           'm4A-712-2', 'm4A-736-1', 'm4A-736-2', 'm4A-744-1', 'm4A-744-2', \n",
    "           'm4A-755-1', 'm4A-755-2', 'm4A-757-1', 'm4A-757-2', 'm4A-766-1', 'm4A-766-2', 'm4A-mock3-16000']\n",
    "miseq4adf = miseqdf.reindex(columns=miseq4a, index=natsorted(miseqdf.index))\n",
    "miseq4adf.to_csv('miseq4a.otu_table.txt', sep='\\t')\n",
    "miseq4c = ['m4C-301-1', 'm4C-301-2', 'm4C-500-1', 'm4C-500-2', 'm4C-712-1', \n",
    "           'm4C-712-2', 'm4C-736-1', 'm4C-736-2', 'm4C-744-1', 'm4C-744-2', \n",
    "           'm4C-755-1', 'm4C-755-2', 'm4C-757-1', 'm4C-757-2', 'm4C-766-1', 'm4C-766-2', 'm4C-mock3-16000']\n",
    "miseq4cdf = miseqdf.reindex(columns=miseq4c, index=natsorted(miseqdf.index))\n",
    "miseq4cdf.to_csv('miseq4c.otu_table.txt', sep='\\t')\n",
    "ion5 = ['m5-301-1', 'm5-301-2', 'm5-500-1', 'm5-500-2', 'm5-712-1', \n",
    "        'm5-712-2', 'm5-736-1', 'm5-736-2', 'm5-744-1', 'm5-744-2', \n",
    "        'm5-755-1', 'm5-755-2', 'm5-757-1', 'm5-757-2', 'm5-766-1', 'm5-766-2', 'm5-SynMock2-8000']\n",
    "ion5df = iondf.reindex(columns=ion5, index=natsorted(iondf.index))\n",
    "ion5df.to_csv('ion5.otu_table.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 236,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[03:12:23 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[03:12:23 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[03:12:23 PM]\u001b[0m: Loading OTU table: miseq4a.otu_table.txt\n",
      "\u001b[92m[03:12:23 PM]\u001b[0m: OTU table contains 2205 OTUs\n",
      "\u001b[92m[03:12:23 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[03:12:23 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[03:12:23 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[03:12:23 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[03:12:23 PM]\u001b[0m: Index bleed, samples into mock: 0.054002%.\n",
      "\u001b[92m[03:12:23 PM]\u001b[0m: Overwriting auto detect index-bleed, setting to 0.500000%\n",
      "\u001b[92m[03:12:24 PM]\u001b[0m: Filtering OTU table down to 1865 OTUs\n",
      "\u001b[92m[03:12:24 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      miseq4a.stats.txt\n",
      "Sorted OTU table:     miseq4a.sorted.txt\n",
      "Normalized/filter:    miseq4a.normalized.txt\n",
      "Final Binary table:   miseq4a.final.binary.txt\n",
      "Final OTU table:      miseq4a.final.txt\n",
      "Filtered OTUs:        miseq4a.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f miseq4a.filtered.otus.fa -i miseq4a.final.txt -m mapping_file.txt -d ITS2\n",
      "\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[03:12:25 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[03:12:25 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[03:12:25 PM]\u001b[0m: Loading OTU table: miseq4c.otu_table.txt\n",
      "\u001b[92m[03:12:25 PM]\u001b[0m: OTU table contains 2205 OTUs\n",
      "\u001b[92m[03:12:25 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[03:12:25 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[03:12:25 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[03:12:25 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[03:12:25 PM]\u001b[0m: Index bleed, samples into mock: 0.051003%.\n",
      "\u001b[92m[03:12:25 PM]\u001b[0m: Overwriting auto detect index-bleed, setting to 0.500000%\n",
      "\u001b[92m[03:12:26 PM]\u001b[0m: Filtering OTU table down to 1764 OTUs\n",
      "\u001b[92m[03:12:26 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      miseq4c.stats.txt\n",
      "Sorted OTU table:     miseq4c.sorted.txt\n",
      "Normalized/filter:    miseq4c.normalized.txt\n",
      "Final Binary table:   miseq4c.final.binary.txt\n",
      "Final OTU table:      miseq4c.final.txt\n",
      "Filtered OTUs:        miseq4c.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f miseq4c.filtered.otus.fa -i miseq4c.final.txt -m mapping_file.txt -d ITS2\n",
      "\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[03:12:27 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[03:12:27 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[03:12:27 PM]\u001b[0m: Loading OTU table: ion3.otu_table.txt\n",
      "\u001b[92m[03:12:27 PM]\u001b[0m: OTU table contains 1382 OTUs\n",
      "\u001b[92m[03:12:27 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[03:12:27 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[03:12:28 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[03:12:28 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[03:12:28 PM]\u001b[0m: Index bleed, samples into mock: 0.098997%.\n",
      "\u001b[92m[03:12:28 PM]\u001b[0m: Overwriting auto detect index-bleed, setting to 0.500000%\n",
      "\u001b[92m[03:12:28 PM]\u001b[0m: Filtering OTU table down to 1136 OTUs\n",
      "\u001b[92m[03:12:28 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      ion3.stats.txt\n",
      "Sorted OTU table:     ion3.sorted.txt\n",
      "Normalized/filter:    ion3.normalized.txt\n",
      "Final Binary table:   ion3.final.binary.txt\n",
      "Final OTU table:      ion3.final.txt\n",
      "Filtered OTUs:        ion3.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f ion3.filtered.otus.fa -i ion3.final.txt -m mapping_file.txt -d ITS2\n",
      "\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[03:12:29 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[03:12:29 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[03:12:29 PM]\u001b[0m: Loading OTU table: ion5.otu_table.txt\n",
      "\u001b[92m[03:12:29 PM]\u001b[0m: OTU table contains 1382 OTUs\n",
      "\u001b[92m[03:12:29 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[03:12:30 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[03:12:30 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[03:12:30 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[03:12:30 PM]\u001b[0m: Index bleed, mock into samples: 0.173787%.  Index bleed, samples into mock: 1.198024%.\n",
      "\u001b[92m[03:12:30 PM]\u001b[0m: Overwriting auto detect index-bleed, setting to 0.500000%\n",
      "\u001b[92m[03:12:30 PM]\u001b[0m: Filtering OTU table down to 1155 OTUs\n",
      "\u001b[92m[03:12:30 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      ion5.stats.txt\n",
      "Sorted OTU table:     ion5.sorted.txt\n",
      "Normalized/filter:    ion5.normalized.txt\n",
      "Final Binary table:   ion5.final.binary.txt\n",
      "Final OTU table:      ion5.final.txt\n",
      "Filtered OTUs:        ion5.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f ion5.filtered.otus.fa -i ion5.final.txt -m mapping_file.txt -d ITS2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#now filter these datasets for raw and filtered (no subtraction, index-bleed 0.5% for all - because extra mocks in each run)\n",
    "!amptk filter -f miseq.cluster.otus.fa -i miseq4a.otu_table.txt -o miseq4a -b m4A-mock3-16000 -m mock3 -p 0.005 -c in\n",
    "!amptk filter -f miseq.cluster.otus.fa -i miseq4c.otu_table.txt -o miseq4c -b m4C-mock3-16000 -m mock3 -p 0.005 -c in\n",
    "!amptk filter -f ion.cluster.otus.fa -i ion3.otu_table.txt -o ion3 -b m3-mock3-16000 -m mock3 -p 0.005 -c in\n",
    "!amptk filter -f ion.cluster.otus.fa -i ion5.otu_table.txt -o ion5 -b m5-SynMock2-8000 -m synmock -p 0.005"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1367 16 1227 15\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 212,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#now generate the raw OTU table from the sorted (original output)\n",
    "import pandas as pd\n",
    "from natsort import natsorted\n",
    "import subprocess\n",
    "#pull out header for indexing later\n",
    "base = 'ion3'\n",
    "platform = 'ion'\n",
    "    \n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "sort = pd.read_csv(base+'.sorted.txt', sep='\\t', index_col=0)\n",
    "\n",
    "#now we want to drop last column\n",
    "header = ['m3-301-1', 'm3-301-2', 'm3-500-1', 'm3-500-2', 'm3-712-1', 'm3-712-2', \n",
    "        'm3-736-1', 'm3-736-2', 'm3-744-1', 'm3-744-2', 'm3-755-1', 'm3-755-2',\n",
    "        'm3-757-2', 'm3-766-1', 'm3-766-2']\n",
    "cleaned = sort.reindex(columns=header, index=natsorted(sort.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "fs = cleaned.sum(axis=0)\n",
    "filtered = pd.DataFrame(cleaned, columns=fs.index)\n",
    "filt2 = filtered.loc[(filtered != 0).any(1)]\n",
    "print len(sort), len(sort.columns.values), len(filt2), len(filt2.columns.values)\n",
    "filt2.to_csv(base+'.sorted.nomock.txt', sep='\\t')\n",
    "\n",
    "#now convert to biom\n",
    "cmd = ['amptk', 'taxonomy', '-f', platform+'.cluster.otus.fa', '-o', base+'_raw', '-i', base+'.sorted.nomock.txt', '-d', 'ITS2','-m','mapping/'+base+'.mapping_file.txt', '-t', platform+'.taxonomy.txt']\n",
    "subprocess.call(cmd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 237,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1382 17 1235 16\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 237,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#now generate the raw OTU table from the sorted (original output)\n",
    "import pandas as pd\n",
    "from natsort import natsorted\n",
    "import subprocess\n",
    "#pull out header for indexing later\n",
    "base = 'ion5'\n",
    "platform = 'ion'\n",
    "    \n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "sort = pd.read_csv(base+'.sorted.txt', sep='\\t', index_col=0)\n",
    "\n",
    "#now we want to drop last column\n",
    "header = ['m5-301-1', 'm5-301-2', 'm5-500-1', 'm5-500-2', 'm5-712-1', \n",
    "        'm5-712-2', 'm5-736-1', 'm5-736-2', 'm5-744-1', 'm5-744-2', \n",
    "        'm5-755-1', 'm5-755-2', 'm5-757-1', 'm5-757-2', 'm5-766-1', 'm5-766-2']\n",
    "cleaned = sort.reindex(columns=header, index=natsorted(sort.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "fs = cleaned.sum(axis=0)\n",
    "filtered = pd.DataFrame(cleaned, columns=fs.index)\n",
    "filt2 = filtered.loc[(filtered != 0).any(1)]\n",
    "print len(sort), len(sort.columns.values), len(filt2), len(filt2.columns.values)\n",
    "filt2.to_csv(base+'.sorted.nomock.txt', sep='\\t')\n",
    "\n",
    "#now convert to biom\n",
    "cmd = ['amptk', 'taxonomy', '-f', platform+'.cluster.otus.fa', '-o', base+'_raw', '-i', base+'.sorted.nomock.txt', '-d', 'ITS2','-m','mapping/'+base+'.mapping_file.txt', '-t', platform+'.taxonomy.txt']\n",
    "subprocess.call(cmd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 214,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2205 17 2094 16\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 214,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#now generate the raw OTU table from the sorted (original output)\n",
    "import pandas as pd\n",
    "from natsort import natsorted\n",
    "import subprocess\n",
    "#pull out header for indexing later\n",
    "base = 'miseq4a'\n",
    "platform = 'miseq'\n",
    "    \n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "sort = pd.read_csv(base+'.sorted.txt', sep='\\t', index_col=0)\n",
    "\n",
    "#now we want to drop last column\n",
    "header = ['m4A-301-1', 'm4A-301-2', 'm4A-500-1', 'm4A-500-2', 'm4A-712-1', \n",
    "           'm4A-712-2', 'm4A-736-1', 'm4A-736-2', 'm4A-744-1', 'm4A-744-2', \n",
    "           'm4A-755-1', 'm4A-755-2', 'm4A-757-1', 'm4A-757-2', 'm4A-766-1', 'm4A-766-2']\n",
    "cleaned = sort.reindex(columns=header, index=natsorted(sort.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "fs = cleaned.sum(axis=0)\n",
    "filtered = pd.DataFrame(cleaned, columns=fs.index)\n",
    "filt2 = filtered.loc[(filtered != 0).any(1)]\n",
    "print len(sort), len(sort.columns.values), len(filt2), len(filt2.columns.values)\n",
    "filt2.to_csv(base+'.sorted.nomock.txt', sep='\\t')\n",
    "\n",
    "#now convert to biom\n",
    "cmd = ['amptk', 'taxonomy', '-f', platform+'.cluster.otus.fa', '-o', base+'_raw', '-i', base+'.sorted.nomock.txt', '-d', 'ITS2','-m','mapping/'+base+'.mapping_file.txt', '-t', platform+'.taxonomy.txt']\n",
    "subprocess.call(cmd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 215,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2205 17 2035 16\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 215,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#now generate the raw OTU table from the sorted (original output)\n",
    "import pandas as pd\n",
    "from natsort import natsorted\n",
    "import subprocess\n",
    "#pull out header for indexing later\n",
    "base = 'miseq4c'\n",
    "platform = 'miseq'\n",
    "    \n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "sort = pd.read_csv(base+'.sorted.txt', sep='\\t', index_col=0)\n",
    "\n",
    "#now we want to drop last column\n",
    "header = ['m4C-301-1', 'm4C-301-2', 'm4C-500-1', 'm4C-500-2', 'm4C-712-1', \n",
    "           'm4C-712-2', 'm4C-736-1', 'm4C-736-2', 'm4C-744-1', 'm4C-744-2', \n",
    "           'm4C-755-1', 'm4C-755-2', 'm4C-757-1', 'm4C-757-2', 'm4C-766-1', 'm4C-766-2']\n",
    "cleaned = sort.reindex(columns=header, index=natsorted(sort.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "fs = cleaned.sum(axis=0)\n",
    "filtered = pd.DataFrame(cleaned, columns=fs.index)\n",
    "filt2 = filtered.loc[(filtered != 0).any(1)]\n",
    "print len(sort), len(sort.columns.values), len(filt2), len(filt2.columns.values)\n",
    "filt2.to_csv(base+'.sorted.nomock.txt', sep='\\t')\n",
    "\n",
    "#now convert to biom\n",
    "cmd = ['amptk', 'taxonomy', '-f', platform+'.cluster.otus.fa', '-o', base+'_raw', '-i', base+'.sorted.nomock.txt', '-d', 'ITS2','-m','mapping/'+base+'.mapping_file.txt', '-t', platform+'.taxonomy.txt']\n",
    "subprocess.call(cmd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[03:16:15 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[03:16:15 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[03:16:15 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[03:16:15 PM]\u001b[0m: 1,136 OTUs\n",
      "\u001b[92m[03:16:15 PM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[03:16:15 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[03:16:18 PM]\u001b[0m: Taxonomy finished: ion.taxonomy.txt\n",
      "\u001b[92m[03:16:18 PM]\u001b[0m: Classic OTU table with taxonomy: ion3.otu_table.taxonomy.txt\n",
      "\u001b[92m[03:16:19 PM]\u001b[0m: BIOM OTU table created: ion3.biom\n",
      "\u001b[92m[03:16:19 PM]\u001b[0m: OTUs with taxonomy: ion3.otus.taxonomy.fa\n",
      "\u001b[92m[03:16:19 PM]\u001b[0m: OTU phylogeny: ion3.tree.phy\n",
      "-------------------------------------------------------\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[03:16:20 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[03:16:20 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[03:16:20 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[03:16:20 PM]\u001b[0m: 1,155 OTUs\n",
      "\u001b[92m[03:16:20 PM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[03:16:20 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[03:16:23 PM]\u001b[0m: Taxonomy finished: ion.taxonomy.txt\n",
      "\u001b[92m[03:16:23 PM]\u001b[0m: Classic OTU table with taxonomy: ion5.otu_table.taxonomy.txt\n",
      "\u001b[92m[03:16:24 PM]\u001b[0m: BIOM OTU table created: ion5.biom\n",
      "\u001b[92m[03:16:24 PM]\u001b[0m: OTUs with taxonomy: ion5.otus.taxonomy.fa\n",
      "\u001b[92m[03:16:24 PM]\u001b[0m: OTU phylogeny: ion5.tree.phy\n",
      "-------------------------------------------------------\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[03:16:25 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[03:16:25 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[03:16:25 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[03:16:25 PM]\u001b[0m: 1,865 OTUs\n",
      "\u001b[92m[03:16:25 PM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[03:16:25 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[03:16:32 PM]\u001b[0m: Taxonomy finished: miseq.taxonomy.txt\n",
      "\u001b[92m[03:16:32 PM]\u001b[0m: Classic OTU table with taxonomy: miseq4a.otu_table.taxonomy.txt\n",
      "\u001b[92m[03:16:33 PM]\u001b[0m: BIOM OTU table created: miseq4a.biom\n",
      "\u001b[92m[03:16:33 PM]\u001b[0m: OTUs with taxonomy: miseq4a.otus.taxonomy.fa\n",
      "\u001b[92m[03:16:33 PM]\u001b[0m: OTU phylogeny: miseq4a.tree.phy\n",
      "-------------------------------------------------------\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[03:16:34 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[03:16:34 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[03:16:34 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[03:16:34 PM]\u001b[0m: 1,764 OTUs\n",
      "\u001b[92m[03:16:34 PM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[03:16:34 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[03:16:40 PM]\u001b[0m: Taxonomy finished: miseq.taxonomy.txt\n",
      "\u001b[92m[03:16:40 PM]\u001b[0m: Classic OTU table with taxonomy: miseq4c.otu_table.taxonomy.txt\n",
      "\u001b[92m[03:16:42 PM]\u001b[0m: BIOM OTU table created: miseq4c.biom\n",
      "\u001b[92m[03:16:42 PM]\u001b[0m: OTUs with taxonomy: miseq4c.otus.taxonomy.fa\n",
      "\u001b[92m[03:16:42 PM]\u001b[0m: OTU phylogeny: miseq4c.tree.phy\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#run taxonomy command on all output\n",
    "!amptk taxonomy -f ion3.filtered.otus.fa -i ion3.final.txt -o ion3 -d ITS2 -t ion.taxonomy.txt -m mapping/ion3.mapping_file.txt\n",
    "!amptk taxonomy -f ion5.filtered.otus.fa -i ion5.final.txt -o ion5 -d ITS2 -t ion.taxonomy.txt -m mapping/ion5.mapping_file.txt\n",
    "!amptk taxonomy -f miseq4a.filtered.otus.fa -i miseq4a.final.txt -o miseq4a -d ITS2 -t miseq.taxonomy.txt -m mapping/m4a.mapping_file.txt\n",
    "!amptk taxonomy -f miseq4c.filtered.otus.fa -i miseq4c.final.txt -o miseq4c -d ITS2 -t miseq.taxonomy.txt -m mapping/m4c.mapping_file.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 239,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Working on complete OTU table: found 1227 OTUs\n",
      "ERROR m4C-301-1, m4C-301-2, m4C-500-1, m4C-500-2, m4C-712-1, m4C-712-2, m4C-736-1, m4C-736-2, m4C-744-1, m4C-744-2, m4C-755-1, m4C-755-2, m4C-757-1, m4C-757-2, m4C-766-1, m4C-766-2, m4C-mock3-16000, m4C-mock3-32000a, m4C-mock3-32000b, m4C-stds, m4A-301-1, m4A-301-2, m4A-500-1, m4A-500-2, m4A-712-1, m4A-712-2, m4A-736-1, m4A-736-2, m4A-744-1, m4A-744-2, m4A-755-1, m4A-755-2, m4A-757-1, m4A-757-2, m4A-766-1, m4A-766-2, m4A-mock3-16000, m4A-mock3-32000a, m4A-mock3-32000b, m4A-stds, m3-mock3-3200a, m3-mock3-16000, m3-mock3-32000b, m3-stds, m5-301-1, m5-301-2, m5-500-1, m5-500-2, m5-712-1, m5-712-2, m5-736-1, m5-736-2, m5-744-1, m5-744-2, m5-755-1, m5-755-2, m5-757-1, m5-757-2, m5-766-1, m5-766-2, m5-SynMock1-8000, m5-SynMock2-8000, m5-mock3.0-16000, m5-mock3.5-16000: not found in OTU table. Names must max exactly.\n",
      "Working on complete OTU table: found 1136 OTUs\n",
      "ERROR m4C-301-1, m4C-301-2, m4C-500-1, m4C-500-2, m4C-712-1, m4C-712-2, m4C-736-1, m4C-736-2, m4C-744-1, m4C-744-2, m4C-755-1, m4C-755-2, m4C-757-1, m4C-757-2, m4C-766-1, m4C-766-2, m4C-mock3-16000, m4C-mock3-32000a, m4C-mock3-32000b, m4C-stds, m4A-301-1, m4A-301-2, m4A-500-1, m4A-500-2, m4A-712-1, m4A-712-2, m4A-736-1, m4A-736-2, m4A-744-1, m4A-744-2, m4A-755-1, m4A-755-2, m4A-757-1, m4A-757-2, m4A-766-1, m4A-766-2, m4A-mock3-16000, m4A-mock3-32000a, m4A-mock3-32000b, m4A-stds, m3-mock3-3200a, m3-mock3-16000, m3-mock3-32000b, m3-stds, m5-301-1, m5-301-2, m5-500-1, m5-500-2, m5-712-1, m5-712-2, m5-736-1, m5-736-2, m5-744-1, m5-744-2, m5-755-1, m5-755-2, m5-757-1, m5-757-2, m5-766-1, m5-766-2, m5-SynMock1-8000, m5-SynMock2-8000, m5-mock3.0-16000, m5-mock3.5-16000: not found in OTU table. Names must max exactly.\n",
      "Working on complete OTU table: found 1235 OTUs\n",
      "ERROR m4C-301-1, m4C-301-2, m4C-500-1, m4C-500-2, m4C-712-1, m4C-712-2, m4C-736-1, m4C-736-2, m4C-744-1, m4C-744-2, m4C-755-1, m4C-755-2, m4C-757-1, m4C-757-2, m4C-766-1, m4C-766-2, m4C-mock3-16000, m4C-mock3-32000a, m4C-mock3-32000b, m4C-stds, m4A-301-1, m4A-301-2, m4A-500-1, m4A-500-2, m4A-712-1, m4A-712-2, m4A-736-1, m4A-736-2, m4A-744-1, m4A-744-2, m4A-755-1, m4A-755-2, m4A-757-1, m4A-757-2, m4A-766-1, m4A-766-2, m4A-mock3-16000, m4A-mock3-32000a, m4A-mock3-32000b, m4A-stds, m3-301-1, m3-301-2, m3-500-1, m3-500-2, m3-712-1, m3-712-2, m3-736-1, m3-736-2, m3-744-1, m3-744-2, m3-755-1, m3-755-2, m3-757-2, m3-766-1, m3-766-2, m3-mock3-3200a, m3-mock3-16000, m3-mock3-32000b, m3-stds, m5-SynMock1-8000, m5-SynMock2-8000, m5-mock3.0-16000, m5-mock3.5-16000: not found in OTU table. Names must max exactly.\n",
      "Working on complete OTU table: found 1155 OTUs\n",
      "ERROR m4C-301-1, m4C-301-2, m4C-500-1, m4C-500-2, m4C-712-1, m4C-712-2, m4C-736-1, m4C-736-2, m4C-744-1, m4C-744-2, m4C-755-1, m4C-755-2, m4C-757-1, m4C-757-2, m4C-766-1, m4C-766-2, m4C-mock3-16000, m4C-mock3-32000a, m4C-mock3-32000b, m4C-stds, m4A-301-1, m4A-301-2, m4A-500-1, m4A-500-2, m4A-712-1, m4A-712-2, m4A-736-1, m4A-736-2, m4A-744-1, m4A-744-2, m4A-755-1, m4A-755-2, m4A-757-1, m4A-757-2, m4A-766-1, m4A-766-2, m4A-mock3-16000, m4A-mock3-32000a, m4A-mock3-32000b, m4A-stds, m3-301-1, m3-301-2, m3-500-1, m3-500-2, m3-712-1, m3-712-2, m3-736-1, m3-736-2, m3-744-1, m3-744-2, m3-755-1, m3-755-2, m3-757-2, m3-766-1, m3-766-2, m3-mock3-3200a, m3-mock3-16000, m3-mock3-32000b, m3-stds, m5-SynMock1-8000, m5-SynMock2-8000, m5-mock3.0-16000, m5-mock3.5-16000: not found in OTU table. Names must max exactly.\n",
      "Working on complete OTU table: found 2094 OTUs\n",
      "ERROR m4C-301-1, m4C-301-2, m4C-500-1, m4C-500-2, m4C-712-1, m4C-712-2, m4C-736-1, m4C-736-2, m4C-744-1, m4C-744-2, m4C-755-1, m4C-755-2, m4C-757-1, m4C-757-2, m4C-766-1, m4C-766-2, m4C-mock3-16000, m4C-mock3-32000a, m4C-mock3-32000b, m4C-stds, m4A-mock3-16000, m4A-mock3-32000a, m4A-mock3-32000b, m4A-stds, m3-301-1, m3-301-2, m3-500-1, m3-500-2, m3-712-1, m3-712-2, m3-736-1, m3-736-2, m3-744-1, m3-744-2, m3-755-1, m3-755-2, m3-757-2, m3-766-1, m3-766-2, m3-mock3-3200a, m3-mock3-16000, m3-mock3-32000b, m3-stds, m5-301-1, m5-301-2, m5-500-1, m5-500-2, m5-712-1, m5-712-2, m5-736-1, m5-736-2, m5-744-1, m5-744-2, m5-755-1, m5-755-2, m5-757-1, m5-757-2, m5-766-1, m5-766-2, m5-SynMock1-8000, m5-SynMock2-8000, m5-mock3.0-16000, m5-mock3.5-16000: not found in OTU table. Names must max exactly.\n",
      "Working on complete OTU table: found 1865 OTUs\n",
      "ERROR m4C-301-1, m4C-301-2, m4C-500-1, m4C-500-2, m4C-712-1, m4C-712-2, m4C-736-1, m4C-736-2, m4C-744-1, m4C-744-2, m4C-755-1, m4C-755-2, m4C-757-1, m4C-757-2, m4C-766-1, m4C-766-2, m4C-mock3-16000, m4C-mock3-32000a, m4C-mock3-32000b, m4C-stds, m4A-mock3-16000, m4A-mock3-32000a, m4A-mock3-32000b, m4A-stds, m3-301-1, m3-301-2, m3-500-1, m3-500-2, m3-712-1, m3-712-2, m3-736-1, m3-736-2, m3-744-1, m3-744-2, m3-755-1, m3-755-2, m3-757-2, m3-766-1, m3-766-2, m3-mock3-3200a, m3-mock3-16000, m3-mock3-32000b, m3-stds, m5-301-1, m5-301-2, m5-500-1, m5-500-2, m5-712-1, m5-712-2, m5-736-1, m5-736-2, m5-744-1, m5-744-2, m5-755-1, m5-755-2, m5-757-1, m5-757-2, m5-766-1, m5-766-2, m5-SynMock1-8000, m5-SynMock2-8000, m5-mock3.0-16000, m5-mock3.5-16000: not found in OTU table. Names must max exactly.\n",
      "Working on complete OTU table: found 2035 OTUs\n",
      "ERROR m4C-mock3-16000, m4C-mock3-32000a, m4C-mock3-32000b, m4C-stds, m4A-301-1, m4A-301-2, m4A-500-1, m4A-500-2, m4A-712-1, m4A-712-2, m4A-736-1, m4A-736-2, m4A-744-1, m4A-744-2, m4A-755-1, m4A-755-2, m4A-757-1, m4A-757-2, m4A-766-1, m4A-766-2, m4A-mock3-16000, m4A-mock3-32000a, m4A-mock3-32000b, m4A-stds, m3-301-1, m3-301-2, m3-500-1, m3-500-2, m3-712-1, m3-712-2, m3-736-1, m3-736-2, m3-744-1, m3-744-2, m3-755-1, m3-755-2, m3-757-2, m3-766-1, m3-766-2, m3-mock3-3200a, m3-mock3-16000, m3-mock3-32000b, m3-stds, m5-301-1, m5-301-2, m5-500-1, m5-500-2, m5-712-1, m5-712-2, m5-736-1, m5-736-2, m5-744-1, m5-744-2, m5-755-1, m5-755-2, m5-757-1, m5-757-2, m5-766-1, m5-766-2, m5-SynMock1-8000, m5-SynMock2-8000, m5-mock3.0-16000, m5-mock3.5-16000: not found in OTU table. Names must max exactly.\n",
      "Working on complete OTU table: found 1764 OTUs\n",
      "ERROR m4C-mock3-16000, m4C-mock3-32000a, m4C-mock3-32000b, m4C-stds, m4A-301-1, m4A-301-2, m4A-500-1, m4A-500-2, m4A-712-1, m4A-712-2, m4A-736-1, m4A-736-2, m4A-744-1, m4A-744-2, m4A-755-1, m4A-755-2, m4A-757-1, m4A-757-2, m4A-766-1, m4A-766-2, m4A-mock3-16000, m4A-mock3-32000a, m4A-mock3-32000b, m4A-stds, m3-301-1, m3-301-2, m3-500-1, m3-500-2, m3-712-1, m3-712-2, m3-736-1, m3-736-2, m3-744-1, m3-744-2, m3-755-1, m3-755-2, m3-757-2, m3-766-1, m3-766-2, m3-mock3-3200a, m3-mock3-16000, m3-mock3-32000b, m3-stds, m5-301-1, m5-301-2, m5-500-1, m5-500-2, m5-712-1, m5-712-2, m5-736-1, m5-736-2, m5-744-1, m5-744-2, m5-755-1, m5-755-2, m5-757-1, m5-757-2, m5-766-1, m5-766-2, m5-SynMock1-8000, m5-SynMock2-8000, m5-mock3.0-16000, m5-mock3.5-16000: not found in OTU table. Names must max exactly.\n"
     ]
    }
   ],
   "source": [
    "#now convert all to meta transposition for R/michelle's scripts\n",
    "!amptk meta -i ion3_raw.otu_table.taxonomy.txt -m all_data.meta.csv -o ion3_raw.meta.csv\n",
    "!amptk meta -i ion3.otu_table.taxonomy.txt -m all_data.meta.csv -o ion3_filtered.meta.csv\n",
    "!amptk meta -i ion5_raw.otu_table.taxonomy.txt -m all_data.meta.csv -o ion5_raw.meta.csv\n",
    "!amptk meta -i ion5.otu_table.taxonomy.txt -m all_data.meta.csv -o ion5_filtered.meta.csv\n",
    "!amptk meta -i miseq4a_raw.otu_table.taxonomy.txt -m all_data.meta.csv -o miseq4a_raw.meta.csv\n",
    "!amptk meta -i miseq4a.otu_table.taxonomy.txt -m all_data.meta.csv -o miseq4a_filtered.meta.csv\n",
    "!amptk meta -i miseq4c_raw.otu_table.taxonomy.txt -m all_data.meta.csv -o miseq4c_raw.meta.csv\n",
    "!amptk meta -i miseq4c.otu_table.taxonomy.txt -m all_data.meta.csv -o miseq4c_filtered.meta.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Kept 2965759 reads out of 17459832 total reads\r\n"
     ]
    }
   ],
   "source": [
    "#to look at mock communities across all runs, pull out reads and map to mock2 sequences\n",
    "!amptk select -i all.reads.fq -o mock.samples.demux.fa --format fasta -l m3-stds m3-mock3-16000 m3-mock3-3200a m3-mock3-32000b m4A-stds m4A-mock3-16000 m4A-mock3-32000a m4A-mock3-32000b m4C-stds m4C-mock3-16000 m4C-mock3-32000a m4C-mock3-32000b"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "vsearch v2.4.2_osx_x86_64, 16.0GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file /Users/jon/amptk/DB/amptk_mock2.fa 100%  \n",
      "14847 nt in 26 seqs, min 335, max 680, avg 571\n",
      "Masking 100% \n",
      "Counting unique k-mers 100% \n",
      "Creating index of unique k-mers 100% \n",
      "Searching 100%                                                                                                                                                                                                                                                                                                                                                                                                              \n",
      "Matching query sequences: 2464948 of 2965759 (83.11%)\n",
      "Writing OTU table (classic) 100%  \n"
     ]
    }
   ],
   "source": [
    "#map these reads to mock community sequences\n",
    "!vsearch --usearch_global mock.samples.demux.fa --id 0.97 --db /Users/jon/amptk/DB/amptk_mock2.fa --otutabout figure1.mock.otu_table.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 303,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[08:45:30 AM]\u001b[0m: Searching for primers, this may take awhile: Fwd: fITS7  Rev: ITS4\n",
      "\u001b[92m[08:45:30 AM]\u001b[0m: 35 records loaded\n",
      "\u001b[92m[08:45:30 AM]\u001b[0m: Using 8 cpus to process data\n",
      "\u001b[92m[08:45:30 AM]\u001b[0m: 35 records passed (100.00%)\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#now look at synmock and biomock from run 5, it was run several different ways, multiple combinatons of plasmids\n",
    "#as well as mixing plasmids of the mocks together and also mixing with other samples\n",
    "#asking question here about what happens when complexity of community is greater than just 12 \n",
    "\n",
    "#first get both synmock and biomock3 together to use as database, get only ITS2 portion\n",
    "!cat $HOME/amptk/DB/amptk_mock3.fa $HOME/amptk/DB/amptk_synmock.fa > mock3_synmock.fa\n",
    "#little hack here but use amptk database to trim primers\n",
    "!amptk database -i mock3_synmock.fa --format off --keep_all --primer_mismatch 2 -o mock3_synmock.trimmed\n",
    "#change synmock names\n",
    "!gsed 's/>mock_/>Synmock/g' $HOME/amptk/DB/mock3_synmock.trimmed.extracted.fa > mock3_synmock.trim.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 305,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Kept 1046220 reads out of 2889451 total reads\n",
      "vsearch v2.4.2_osx_x86_64, 16.0GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file mock3_synmock.trim.fa 100%  \n",
      "10010 nt in 35 seqs, min 161, max 411, avg 286\n",
      "Masking 100% \n",
      "Counting unique k-mers 100% \n",
      "Creating index of unique k-mers 100% \n",
      "Searching 100%                                                                                                                                                                                                                                                                                                                                                                                                                \n",
      "Matching query sequences: 705289 of 1046220 (67.41%)\n",
      "Writing OTU table (classic) 100%  \n"
     ]
    }
   ],
   "source": [
    "#ok now map ion reads to the combined mock sequences\n",
    "!amptk select -i ion5.demux.fq -o ion5.mocks.demux.fa --format fasta -l m5-SynMock1-8000 m5-SynMock1-mock3.0 m5-SynMock1-mock3.5 m5-SynMock1.736-1.736-2 m5-SynMock2-8000 m5-SynMock2.757-1.757-2 m5-mock3.0-16000 m5-mock3.5-16000\n",
    "!vsearch --usearch_global ion5.mocks.demux.fa --db mock3_synmock.trim.fa --id 0.97 --strand plus --otutabout ion5.mock.otu_table.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 331,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "m5-SynMock1-8000            26\n",
      "m5-SynMock1.736-1.736-2    213\n",
      "m5-SynMock2-8000            28\n",
      "m5-SynMock2.757-1.757-2    209\n",
      "Name: Number of OTUs, dtype: int64\n",
      "                            m5-SynMock1-8000  m5-SynMock1.736-1.736-2  \\\n",
      "#OTU ID                                                                 \n",
      "mock_1_pident=100.0_OTU141              3499                     1012   \n",
      "mock_2_pident=100.0_OTU46              12252                     3156   \n",
      "mock_3_pident=100.0_OTU23              11937                     3629   \n",
      "mock_4_pident=100.0_OTU31              11339                     4041   \n",
      "mock_5_pident=100.0_OTU27               6720                     2272   \n",
      "mock_6_pident=100.0_OTU30               4915                     1475   \n",
      "mock_7_pident=100.0_OTU36              13725                     6725   \n",
      "mock_8_pident=100.0_OTU12              16065                     5802   \n",
      "mock_9_pident=100.0_OTU34               5449                     1516   \n",
      "mock_10_pident=100.0_OTU37              3635                      816   \n",
      "mock_11_pident=100.0_OTU16             13471                     4532   \n",
      "mock_12_pident=100.0_OTU54             10022                     3947   \n",
      "Other OTUs                              1187                   106379   \n",
      "\n",
      "                            m5-SynMock2-8000  m5-SynMock2.757-1.757-2  \n",
      "#OTU ID                                                                \n",
      "mock_1_pident=100.0_OTU141              2198                     2879  \n",
      "mock_2_pident=100.0_OTU46               5557                     5020  \n",
      "mock_3_pident=100.0_OTU23               5981                     6622  \n",
      "mock_4_pident=100.0_OTU31               6074                     5389  \n",
      "mock_5_pident=100.0_OTU27               3148                     2476  \n",
      "mock_6_pident=100.0_OTU30               2912                     2049  \n",
      "mock_7_pident=100.0_OTU36               7251                     8388  \n",
      "mock_8_pident=100.0_OTU12               8281                    11022  \n",
      "mock_9_pident=100.0_OTU34               3202                     2072  \n",
      "mock_10_pident=100.0_OTU37              2395                     1557  \n",
      "mock_11_pident=100.0_OTU16              7509                     7665  \n",
      "mock_12_pident=100.0_OTU54              5641                     4455  \n",
      "Other OTUs                               730                    42056  \n",
      "m5-SynMock1-8000           114216\n",
      "m5-SynMock1.736-1.736-2    145302\n",
      "m5-SynMock2-8000            60879\n",
      "m5-SynMock2.757-1.757-2    101650\n",
      "dtype: int64\n",
      "                            m5-SynMock1-8000  m5-SynMock1.736-1.736-2  \\\n",
      "#OTU ID                                                                 \n",
      "mock_1_pident=100.0_OTU141              3499                     1012   \n",
      "mock_2_pident=100.0_OTU46              12252                     3156   \n",
      "mock_3_pident=100.0_OTU23              11937                     3629   \n",
      "mock_4_pident=100.0_OTU31              11339                     4041   \n",
      "mock_5_pident=100.0_OTU27               6720                     2272   \n",
      "mock_6_pident=100.0_OTU30               4915                     1475   \n",
      "mock_7_pident=100.0_OTU36              13725                     6725   \n",
      "mock_8_pident=100.0_OTU12              16065                     5802   \n",
      "mock_9_pident=100.0_OTU34               5449                     1516   \n",
      "mock_10_pident=100.0_OTU37              3635                      816   \n",
      "mock_11_pident=100.0_OTU16             13471                     4532   \n",
      "mock_12_pident=100.0_OTU54             10022                     3947   \n",
      "Other OTUs                              1187                   106379   \n",
      "\n",
      "                            m5-SynMock2-8000  m5-SynMock2.757-1.757-2  \n",
      "#OTU ID                                                                \n",
      "mock_1_pident=100.0_OTU141              2198                     2879  \n",
      "mock_2_pident=100.0_OTU46               5557                     5020  \n",
      "mock_3_pident=100.0_OTU23               5981                     6622  \n",
      "mock_4_pident=100.0_OTU31               6074                     5389  \n",
      "mock_5_pident=100.0_OTU27               3148                     2476  \n",
      "mock_6_pident=100.0_OTU30               2912                     2049  \n",
      "mock_7_pident=100.0_OTU36               7251                     8388  \n",
      "mock_8_pident=100.0_OTU12               8281                    11022  \n",
      "mock_9_pident=100.0_OTU34               3202                     2072  \n",
      "mock_10_pident=100.0_OTU37              2395                     1557  \n",
      "mock_11_pident=100.0_OTU16              7509                     7665  \n",
      "mock_12_pident=100.0_OTU54              5641                     4455  \n",
      "Other OTUs                               730                    42056  \n",
      "m3-301-1                   114149\n",
      "m3-301-2                   129285\n",
      "m3-500-1                   122693\n",
      "m3-500-2                   128350\n",
      "m3-712-1                    96421\n",
      "m3-712-2                    90163\n",
      "m3-736-1                    97752\n",
      "m3-736-2                   117016\n",
      "m3-744-1                   111627\n",
      "m3-744-2                   137269\n",
      "m3-755-1                   130869\n",
      "m3-755-2                   122726\n",
      "m3-757-2                   110666\n",
      "m3-766-1                   173901\n",
      "m3-766-2                   123351\n",
      "m3-mock3-16000             134353\n",
      "m3-mock3-32000b            136580\n",
      "m3-mock3-3200a             132572\n",
      "m3-stds                     89054\n",
      "m5-301-1                   147483\n",
      "m5-301-2                   144164\n",
      "m5-500-1                   139177\n",
      "m5-500-2                    46478\n",
      "m5-712-1                    57453\n",
      "m5-712-2                   198855\n",
      "m5-736-1                    73247\n",
      "m5-736-2                   103822\n",
      "m5-744-1                    72195\n",
      "m5-744-2                   103325\n",
      "m5-755-1                    93031\n",
      "m5-755-2                    76862\n",
      "m5-757-1                    96531\n",
      "m5-757-2                    94206\n",
      "m5-766-1                    71354\n",
      "m5-766-2                    68240\n",
      "m5-SynMock1-8000           114216\n",
      "m5-SynMock1-mock3.0        218564\n",
      "m5-SynMock1-mock3.5         74075\n",
      "m5-SynMock1.736-1.736-2    145302\n",
      "m5-SynMock2-8000            60879\n",
      "m5-SynMock2.757-1.757-2    101650\n",
      "m5-mock3.0-16000            68010\n",
      "m5-mock3.5-16000            71456\n",
      "Name: Read_counts, dtype: int64\n",
      "                            m5-SynMock1-8000  m5-SynMock1.736-1.736-2  \\\n",
      "#OTU ID                                                                 \n",
      "mock_1_pident=100.0_OTU141              3065                      887   \n",
      "mock_2_pident=100.0_OTU46              10733                     2765   \n",
      "mock_3_pident=100.0_OTU23              10457                     3179   \n",
      "mock_4_pident=100.0_OTU31               9934                     3540   \n",
      "mock_5_pident=100.0_OTU27               5887                     1990   \n",
      "mock_6_pident=100.0_OTU30               4306                     1292   \n",
      "mock_7_pident=100.0_OTU36              12024                     5891   \n",
      "mock_8_pident=100.0_OTU12              14074                     5083   \n",
      "mock_9_pident=100.0_OTU34               4774                     1328   \n",
      "mock_10_pident=100.0_OTU37              3184                      715   \n",
      "mock_11_pident=100.0_OTU16             11801                     3970   \n",
      "mock_12_pident=100.0_OTU54              8780                     3458   \n",
      "Other OTUs                              1040                    93193   \n",
      "\n",
      "                            m5-SynMock2-8000  m5-SynMock2.757-1.757-2  \n",
      "#OTU ID                                                                \n",
      "mock_1_pident=100.0_OTU141              1926                     2522  \n",
      "mock_2_pident=100.0_OTU46               4868                     4398  \n",
      "mock_3_pident=100.0_OTU23               5240                     5801  \n",
      "mock_4_pident=100.0_OTU31               5321                     4721  \n",
      "mock_5_pident=100.0_OTU27               2758                     2169  \n",
      "mock_6_pident=100.0_OTU30               2551                     1795  \n",
      "mock_7_pident=100.0_OTU36               6352                     7348  \n",
      "mock_8_pident=100.0_OTU12               7255                     9656  \n",
      "mock_9_pident=100.0_OTU34               2805                     1815  \n",
      "mock_10_pident=100.0_OTU37              2098                     1364  \n",
      "mock_11_pident=100.0_OTU16              6578                     6715  \n",
      "mock_12_pident=100.0_OTU54              4942                     3903  \n",
      "Other OTUs                               640                    36843  \n"
     ]
    }
   ],
   "source": [
    "#reorder and sort table\n",
    "import pandas as pd\n",
    "from natsort import natsorted\n",
    "\n",
    "#load table\n",
    "df = pd.read_csv('syn.final.txt', sep='\\t', index_col=0)\n",
    "headers = ['m5-SynMock1-8000', 'm5-SynMock1.736-1.736-2','m5-SynMock2-8000', 'm5-SynMock2.757-1.757-2']\n",
    "df2 = df.reindex(columns=headers, index=natsorted(df.index))\n",
    "mocks = []\n",
    "for x in df2.index:\n",
    "    if x.startswith('mock'):\n",
    "        mocks.append(x)\n",
    "otus = [item for item in df2.index if item not in mocks]\n",
    "\n",
    "df3 = df2.reindex(columns=headers, index=mocks)\n",
    "df3.to_csv('ion5.onlymocks.txt', sep='\\t')\n",
    "#draw heatmap here of only the mock members, add the rest manually?\n",
    "!amptk heatmap -i ion5.onlymocks.txt -o ion5.only.mock.pdf -m heatmap -d tsv --color YlOrRd --annotate\n",
    "df4 = df2.reindex(columns=headers, index=otus)\n",
    "#get number of OTUs from df4 as well as total reads\n",
    "reads = df4.sum(axis=0)\n",
    "reads.name =\"Other OTUs\"\n",
    "otu_count = df4[df4 > 0].count(axis=0, numeric_only=True)\n",
    "otu_count.name = \"Number of OTUs\"\n",
    "#print reads\n",
    "print otu_count\n",
    "#now append the read numbers to df3 dataframe\n",
    "df3 = df3.append(reads)\n",
    "print df3\n",
    "total_reads = df3.sum(axis=0)\n",
    "print total_reads\n",
    "df3.to_csv('ion5.mock.clean.otu_table.txt', sep='\\t')\n",
    "\n",
    "#regular heatmap\n",
    "!amptk heatmap -i ion5.mock.clean.otu_table.txt -o ion5.mock.pdf -m heatmap -d tsv --color YlOrRd --annotate\n",
    "#normalized to 100,000 reads heatmap\n",
    "!amptk heatmap -i ion5.mock.clean.otu_table.txt -o ion5.mock.normalized.pdf --debug --normalize syn.stats.txt -m heatmap -d tsv --color YlOrRd --annotate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 269,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[04:29:53 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[04:29:54 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[04:29:54 PM]\u001b[0m: Loading OTU table: miseq.otu_table.txt\n",
      "\u001b[92m[04:29:54 PM]\u001b[0m: OTU table contains 2205 OTUs\n",
      "\u001b[92m[04:29:54 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[04:29:54 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[04:29:54 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[04:29:54 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[04:29:54 PM]\u001b[0m: Index bleed, mock into samples: 87.777681%.  Index bleed, samples into mock: 0.372015%.\n",
      "\u001b[92m[04:29:54 PM]\u001b[0m: Overwriting auto detect index-bleed, setting to 0.000000%\n",
      "\u001b[92m[04:29:57 PM]\u001b[0m: Filtering OTU table down to 2139 OTUs\n",
      "\u001b[92m[04:29:57 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      miseq.stats.txt\n",
      "Sorted OTU table:     miseq.sorted.txt\n",
      "Normalized/filter:    miseq.normalized.txt\n",
      "Final Binary table:   miseq.final.binary.txt\n",
      "Final OTU table:      miseq.final.txt\n",
      "Filtered OTUs:        miseq.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f miseq.filtered.otus.fa -i miseq.final.txt -m mapping_file.txt -d ITS2\n",
      "\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[04:29:58 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[04:29:58 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[04:29:58 PM]\u001b[0m: Loading OTU table: ion.otu_table.txt\n",
      "\u001b[92m[04:29:58 PM]\u001b[0m: OTU table contains 1382 OTUs\n",
      "\u001b[92m[04:29:58 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[04:29:58 PM]\u001b[0m: Mock members not found: mock26\n",
      "\u001b[92m[04:29:58 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[04:29:58 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[04:29:58 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[04:29:58 PM]\u001b[0m: Index bleed, mock into samples: 85.010499%.  Index bleed, samples into mock: 0.057003%.\n",
      "\u001b[92m[04:29:58 PM]\u001b[0m: Overwriting auto detect index-bleed, setting to 0.000000%\n",
      "\u001b[92m[04:30:00 PM]\u001b[0m: Filtering OTU table down to 1382 OTUs\n",
      "\u001b[92m[04:30:00 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      ion.stats.txt\n",
      "Sorted OTU table:     ion.sorted.txt\n",
      "Normalized/filter:    ion.normalized.txt\n",
      "Final Binary table:   ion.final.binary.txt\n",
      "Final OTU table:      ion.final.txt\n",
      "Filtered OTUs:        ion.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f ion.filtered.otus.fa -i ion.final.txt -m mapping_file.txt -d ITS2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#clean up the miseq and ion combined runs to look for index bleed, filter without removing anything\n",
    "!amptk filter -i miseq.otu_table.txt -f miseq.cluster.otus.fa -p 0 -o miseq -b m4A-stds -m mock2 --keep_mock\n",
    "!amptk filter -i ion.otu_table.txt -f ion.cluster.otus.fa -p 0 -o ion -b m3-stds -m mock2 --keep_mock"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 240,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "vsearch v2.4.2_osx_x86_64, 16.0GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading input file 100%                                                                                                                                                                                                                                                                                                                                                                                                              \n",
      "5921159 sequences kept (of which 0 truncated), 0 sequences discarded.\n",
      "vsearch v2.4.2_osx_x86_64, 16.0GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file miseq.cluster.otus.fa 100%  \n",
      "542602 nt in 2212 seqs, min 100, max 250, avg 245\n",
      "Masking 100%  \n",
      "Counting unique k-mers 100%  \n",
      "Creating index of unique k-mers 100%  \n",
      "Searching 100%                                                                                                                                                                                                                                                                                                                                                                                                                \n",
      "Matching query sequences: 5589169 of 5921159 (94.39%)\n",
      "Writing OTU table (classic) 100%  \n"
     ]
    }
   ],
   "source": [
    "#want to measure index bleed, also have 1 mismatch dataset, so generate OTU table for that dataset\n",
    "!vsearch --fastq_filter m4a1m.demux.fq -fastaout m4a1m.demux.fa\n",
    "!vsearch --usearch_global m4a1m.demux.fa --db miseq.cluster.otus.fa --id 0.97 --strand plus --otutabout miseq4a1m.otu_table.txt\n",
    "!rm m4a1m.demux.fa\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 274,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[04:37:45 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[04:37:45 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[04:37:45 PM]\u001b[0m: Loading OTU table: miseq4a1m.otu_table.txt\n",
      "\u001b[92m[04:37:45 PM]\u001b[0m: OTU table contains 2113 OTUs\n",
      "\u001b[92m[04:37:45 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[04:37:45 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[04:37:45 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[04:37:45 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[04:37:45 PM]\u001b[0m: Index bleed, mock into samples: 75.585583%.  Index bleed, samples into mock: 0.402020%.\n",
      "\u001b[92m[04:37:45 PM]\u001b[0m: Overwriting auto detect index-bleed, setting to 0.000000%\n",
      "\u001b[92m[04:37:46 PM]\u001b[0m: Filtering OTU table down to 1904 OTUs\n",
      "\u001b[92m[04:37:46 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      miseq4a1m.stats.txt\n",
      "Sorted OTU table:     miseq4a1m.sorted.txt\n",
      "Normalized/filter:    miseq4a1m.normalized.txt\n",
      "Final Binary table:   miseq4a1m.final.binary.txt\n",
      "Final OTU table:      miseq4a1m.final.txt\n",
      "Filtered OTUs:        miseq4a1m.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f miseq4a1m.filtered.otus.fa -i miseq4a1m.final.txt -m mapping_file.txt -d ITS2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#run through filtering so it is sorted etc\n",
    "!amptk filter -i miseq4a1m.otu_table.txt -f miseq.cluster.otus.fa -p 0 -b m4a-1mm-stds -m mock2 --keep_mock -o miseq4a1m"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 288,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index-bleed out of mock: 0.025328% (114 out of 450086)\n",
      "Index-bleed into mock: 0.009703% (115 out of 1185245)\n",
      "Total index-bleed: 0.035031%\n"
     ]
    }
   ],
   "source": [
    "#need to measure index bleed, lets start with ion dataset\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from natsort import natsorted\n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "df = pd.read_csv('ion.final.txt', sep='\\t', index_col=0)\n",
    "#print list(df.columns.values)\n",
    "#print len(df)\n",
    "#okay, so 1367 OTUs in entire data set, lets trim this down to only run 3, as that one is labeled here with biomock\n",
    "keepers = ['m3-301-1', 'm3-301-2', 'm3-500-1', 'm3-500-2', 'm3-712-1', 'm3-712-2', \n",
    "           'm3-736-1', 'm3-736-2', 'm3-744-1', 'm3-744-2', 'm3-755-1', 'm3-755-2', \n",
    "           'm3-757-2', 'm3-766-1', 'm3-766-2', 'm3-mock3-3200a', 'm3-mock3-16000', 'm3-mock3-32000b', 'm3-stds']\n",
    "mock_samples = ['m3-mock3-3200a', 'm3-mock3-16000', 'm3-mock3-32000b', 'm3-stds']\n",
    "samples = [item for item in keepers if item not in mock_samples]\n",
    "#print samples\n",
    "df2 = df.reindex(columns=keepers, index=natsorted(df.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "df3 = df2.loc[(df2 != 0).any(1)]\n",
    "mock_otus = []\n",
    "for x in df3.index:\n",
    "    if x.startswith('mock'):\n",
    "        if not x.startswith('mock3'): #mock3 is actually in some of the samples so can't use for index-bleed measurment\n",
    "            mock_otus.append(x)\n",
    "#now slice into two tables containing only mocks and calculate index-bleed\n",
    "bleed_counts = df3.reindex(columns=samples, index=mock_otus)\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "real_counts = df3.reindex(columns=mock_samples, index=mock_otus)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed out of mock: %f%% (%i out of %i)\" % (index_bleed, bleed_sum, real_sum)\n",
    "\n",
    "#now try to to do the opposite, count number of times that samples move into mock\n",
    "#first generate list of mock otus\n",
    "sample_otus = []\n",
    "for x in df3.index:\n",
    "    if not x.startswith('mock'):\n",
    "        sample_otus.append(x)\n",
    "#slice first to samples, further trim list of only things found in mocks, would be spurious or chimeras\n",
    "sampledf = df3.reindex(columns=samples, index=sample_otus)\n",
    "sampledf = sampledf.loc[(sampledf != 0).any(1)]\n",
    "cleaned_otus = sampledf.index\n",
    "#print len(cleaned_otus), len(sample_otus)\n",
    "\n",
    "#now using new list grab counts in mock\n",
    "bleed_counts = df3.reindex(columns=mock_samples, index=cleaned_otus)\n",
    "bleed_counts = bleed_counts.loc[(bleed_counts != 0).any(1)]\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "#print len(bleed_counts.index), bleed_sum\n",
    "\n",
    "#finally go back to samples and get read counts for the ovlerlapping otus\n",
    "real_counts = df3.reindex(columns=samples, index=bleed_counts.index)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed2 = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed into mock: %f%% (%i out of %i)\" % (index_bleed2, bleed_sum, real_sum)\n",
    "print \"Total index-bleed: %f%%\" % (index_bleed + index_bleed2)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 289,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[05:26:11 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[05:26:11 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[05:26:11 PM]\u001b[0m: Loading OTU table: ion.otu_table.txt\n",
      "\u001b[92m[05:26:11 PM]\u001b[0m: OTU table contains 1382 OTUs\n",
      "\u001b[92m[05:26:11 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[05:26:11 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[05:26:11 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[05:26:12 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[05:26:12 PM]\u001b[0m: Index bleed, mock into samples: 76.549668%.  Index bleed, samples into mock: 1.201976%.\n",
      "\u001b[92m[05:26:12 PM]\u001b[0m: Overwriting auto detect index-bleed, setting to 0.000000%\n",
      "\u001b[92m[05:26:13 PM]\u001b[0m: Filtering OTU table down to 1382 OTUs\n",
      "\u001b[92m[05:26:13 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      syn.stats.txt\n",
      "Sorted OTU table:     syn.sorted.txt\n",
      "Normalized/filter:    syn.normalized.txt\n",
      "Final Binary table:   syn.final.binary.txt\n",
      "Final OTU table:      syn.final.txt\n",
      "Filtered OTUs:        syn.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f syn.filtered.otus.fa -i syn.final.txt -m mapping_file.txt -d ITS2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#map synthetic mock onto this dataset\n",
    "!amptk filter -i ion.otu_table.txt -f ion.cluster.otus.fa -p 0 -o syn -b m5-SynMock2-8000 -m synmock --keep_mock"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 297,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index-bleed out of mock: 0.043358% (224 out of 516625)\n",
      "Index-bleed into mock: 0.029789% (87 out of 292054)\n",
      "Total index-bleed: 0.073147%\n"
     ]
    }
   ],
   "source": [
    "#now try on the ion5 dataset\n",
    "#need to measure index bleed, lets start with ion dataset\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from natsort import natsorted\n",
    "import sys\n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "df = pd.read_csv('syn.final.txt', sep='\\t', index_col=0)\n",
    "#print list(df.columns.values)\n",
    "#sys.exit(1)\n",
    "#print len(df)\n",
    "#okay, so 1367 OTUs in entire data set, lets trim this down to only run 3, as that one is labeled here with biomock\n",
    "keepers = ['m5-301-1', 'm5-301-2', 'm5-500-1', 'm5-500-2', 'm5-712-1', 'm5-712-2', 'm5-736-1', 'm5-736-2', \n",
    "           'm5-744-1', 'm5-744-2', 'm5-755-1', 'm5-755-2', 'm5-757-1', 'm5-757-2', 'm5-766-1', 'm5-766-2', \n",
    "           'm5-SynMock1-8000', 'm5-SynMock1-mock3.0', 'm5-SynMock1-mock3.5', 'm5-SynMock1.736-1.736-2', \n",
    "           'm5-SynMock2-8000', 'm5-SynMock2.757-1.757-2', 'm5-mock3.0-16000', 'm5-mock3.5-16000']\n",
    "mock_samples = ['m5-SynMock2-8000', 'm5-SynMock2.757-1.757-2', 'm5-SynMock1-8000', \n",
    "                'm5-SynMock1-mock3.0', 'm5-SynMock1-mock3.5', 'm5-SynMock1.736-1.736-2']\n",
    "samples = [item for item in keepers if item not in mock_samples]\n",
    "#print samples\n",
    "df2 = df.reindex(columns=keepers, index=natsorted(df.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "df3 = df2.loc[(df2 != 0).any(1)]\n",
    "mock_otus = []\n",
    "for x in df3.index:\n",
    "    if 'mock' in x: #note this is different here as I'm also including mock chimeras as based on synthetic mock\n",
    "        mock_otus.append(x)\n",
    "#now slice into two tables containing only mocks and calculate index-bleed\n",
    "bleed_counts = df3.reindex(columns=samples, index=mock_otus)\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "real_counts = df3.reindex(columns=mock_samples, index=mock_otus)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed out of mock: %f%% (%i out of %i)\" % (index_bleed, bleed_sum, real_sum)\n",
    "\n",
    "#now try to to do the opposite, count number of times that samples move into mock\n",
    "#first generate list of mock otus\n",
    "sample_otus = []\n",
    "for x in df3.index:\n",
    "    if not 'mock' in x: #note this is different here as I'm also including mock chimeras as based on synthetic mock\n",
    "        sample_otus.append(x)\n",
    "clean_mocks = ['m5-SynMock2-8000', 'm5-SynMock1-8000',]\n",
    "#slice first to samples, further trim list of only things found in mocks, would be spurious or chimeras\n",
    "sampledf = df3.reindex(columns=samples, index=sample_otus)\n",
    "sampledf = sampledf.loc[(sampledf != 0).any(1)]\n",
    "cleaned_otus = sampledf.index\n",
    "#print len(cleaned_otus), len(sample_otus)\n",
    "\n",
    "#now using new list grab counts in mock\n",
    "bleed_counts = df3.reindex(columns=clean_mocks, index=cleaned_otus)\n",
    "bleed_counts = bleed_counts.loc[(bleed_counts != 0).any(1)]\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "#print len(bleed_counts.index), bleed_sum\n",
    "\n",
    "#finally go back to samples and get read counts for the ovlerlapping otus\n",
    "real_counts = df3.reindex(columns=samples, index=bleed_counts.index)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed2 = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed into mock: %f%% (%i out of %i)\" % (index_bleed2, bleed_sum, real_sum)\n",
    "print \"Total index-bleed: %f%%\" % (index_bleed + index_bleed2)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 287,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index-bleed out of mock: 0.329377% (3281 out of 996124)\n",
      "Index-bleed into mock: 0.024575% (734 out of 2986781)\n",
      "Total index-bleed: 0.353952%\n"
     ]
    }
   ],
   "source": [
    "#need to measure index bleed of miseq data now, trying 1 mismatch dataset first\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from natsort import natsorted\n",
    "import sys\n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "df = pd.read_csv('miseq4a1m.final.txt', sep='\\t', index_col=0)\n",
    "#print list(df.columns.values)\n",
    "#sys.exit(1)\n",
    "#print len(df)\n",
    "#okay, so 1367 OTUs in entire data set, lets trim this down to only run 3, as that one is labeled here with biomock\n",
    "keepers = ['m4a-1mm-301-1', 'm4a-1mm-301-2', 'm4a-1mm-500-1', 'm4a-1mm-500-2', 'm4a-1mm-712-1', \n",
    "           'm4a-1mm-712-2', 'm4a-1mm-736-1', 'm4a-1mm-736-2', 'm4a-1mm-744-1', 'm4a-1mm-744-2', \n",
    "           'm4a-1mm-755-1', 'm4a-1mm-755-2', 'm4a-1mm-757-1', 'm4a-1mm-757-2', 'm4a-1mm-766-1', \n",
    "           'm4a-1mm-766-2', 'm4a-1mm-mock3-16000', 'm4a-1mm-mock3-32000a', 'm4a-1mm-mock3-32000b', 'm4a-1mm-stds']\n",
    "mock_samples = ['m4a-1mm-mock3-16000', 'm4a-1mm-mock3-32000a', 'm4a-1mm-mock3-32000b', 'm4a-1mm-stds']\n",
    "samples = [item for item in keepers if item not in mock_samples]\n",
    "#print samples\n",
    "df2 = df.reindex(columns=keepers, index=natsorted(df.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "df3 = df2.loc[(df2 != 0).any(1)]\n",
    "mock_otus = []\n",
    "for x in df3.index:\n",
    "    if x.startswith('mock'):\n",
    "        if not x.startswith('mock3'): #mock3 is actually in some of the samples so can't use for index-bleed measurment\n",
    "            mock_otus.append(x)\n",
    "#now slice into two tables containing only mocks and calculate index-bleed\n",
    "bleed_counts = df3.reindex(columns=samples, index=mock_otus)\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "real_counts = df3.reindex(columns=mock_samples, index=mock_otus)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed out of mock: %f%% (%i out of %i)\" % (index_bleed, bleed_sum, real_sum)\n",
    "\n",
    "#now try to to do the opposite, count number of times that samples move into mock\n",
    "#first generate list of mock otus\n",
    "sample_otus = []\n",
    "for x in df3.index:\n",
    "    if not x.startswith('mock'):\n",
    "        sample_otus.append(x)\n",
    "#slice first to samples, further trim list of only things found in mocks, would be spurious or chimeras\n",
    "sampledf = df3.reindex(columns=samples, index=sample_otus)\n",
    "sampledf = sampledf.loc[(sampledf != 0).any(1)]\n",
    "cleaned_otus = sampledf.index\n",
    "#print len(cleaned_otus), len(sample_otus)\n",
    "\n",
    "#now using new list grab counts in mock\n",
    "bleed_counts = df3.reindex(columns=mock_samples, index=cleaned_otus)\n",
    "bleed_counts = bleed_counts.loc[(bleed_counts != 0).any(1)]\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "#print len(bleed_counts.index), bleed_sum\n",
    "\n",
    "#finally go back to samples and get read counts for the ovlerlapping otus\n",
    "real_counts = df3.reindex(columns=samples, index=bleed_counts.index)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed2 = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed into mock: %f%% (%i out of %i)\" % (index_bleed2, bleed_sum, real_sum)\n",
    "print \"Total index-bleed: %f%%\" % (index_bleed + index_bleed2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 286,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index-bleed out of mock: 0.253212% (2436 out of 962041)\n",
      "Index-bleed into mock: 0.015920% (434 out of 2726206)\n",
      "Total index-bleed: 0.269131%\n"
     ]
    }
   ],
   "source": [
    "#need to measure index bleed of miseq data now, trying 0 mismatch miseqA dataset first\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from natsort import natsorted\n",
    "import sys\n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "df = pd.read_csv('miseq.final.txt', sep='\\t', index_col=0)\n",
    "#print list(df.columns.values)\n",
    "#sys.exit(1)\n",
    "#print len(df)\n",
    "#okay, so 1367 OTUs in entire data set, lets trim this down to only run 3, as that one is labeled here with biomock\n",
    "keepers = ['m4A-301-1', 'm4A-301-2', 'm4A-500-1', 'm4A-500-2', 'm4A-712-1', 'm4A-712-2', 'm4A-736-1', \n",
    "           'm4A-736-2', 'm4A-744-1', 'm4A-744-2', 'm4A-755-1', 'm4A-755-2', 'm4A-757-1', 'm4A-757-2', \n",
    "           'm4A-766-1', 'm4A-766-2', 'm4A-mock3-16000', 'm4A-mock3-32000a', 'm4A-mock3-32000b', 'm4A-stds']\n",
    "mock_samples = ['m4A-mock3-16000', 'm4A-mock3-32000a', 'm4A-mock3-32000b', 'm4A-stds']\n",
    "samples = [item for item in keepers if item not in mock_samples]\n",
    "#print samples\n",
    "df2 = df.reindex(columns=keepers, index=natsorted(df.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "df3 = df2.loc[(df2 != 0).any(1)]\n",
    "mock_otus = []\n",
    "for x in df3.index:\n",
    "    if x.startswith('mock'):\n",
    "        if not x.startswith('mock3'): #mock3 is actually in some of the samples so can't use for index-bleed measurment\n",
    "            mock_otus.append(x)\n",
    "#now slice into two tables containing only mocks and calculate index-bleed\n",
    "bleed_counts = df3.reindex(columns=samples, index=mock_otus)\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "real_counts = df3.reindex(columns=mock_samples, index=mock_otus)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed out of mock: %f%% (%i out of %i)\" % (index_bleed, bleed_sum, real_sum)\n",
    "\n",
    "#now try to to do the opposite, count number of times that samples move into mock\n",
    "#first generate list of mock otus\n",
    "sample_otus = []\n",
    "for x in df3.index:\n",
    "    if not x.startswith('mock'):\n",
    "        sample_otus.append(x)\n",
    "#slice first to samples, further trim list of only things found in mocks, would be spurious or chimeras\n",
    "sampledf = df3.reindex(columns=samples, index=sample_otus)\n",
    "sampledf = sampledf.loc[(sampledf != 0).any(1)]\n",
    "cleaned_otus = sampledf.index\n",
    "#print len(cleaned_otus), len(sample_otus)\n",
    "\n",
    "#now using new list grab counts in mock\n",
    "bleed_counts = df3.reindex(columns=mock_samples, index=cleaned_otus)\n",
    "bleed_counts = bleed_counts.loc[(bleed_counts != 0).any(1)]\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "#print len(bleed_counts.index), bleed_sum\n",
    "\n",
    "#finally go back to samples and get read counts for the ovlerlapping otus\n",
    "real_counts = df3.reindex(columns=samples, index=bleed_counts.index)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed2 = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed into mock: %f%% (%i out of %i)\" % (index_bleed2, bleed_sum, real_sum)\n",
    "print \"Total index-bleed: %f%%\" % (index_bleed + index_bleed2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 285,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index-bleed out of mock: 0.223127% (1889 out of 846602)\n",
      "Index-bleed into mock: 0.012798% (329 out of 2570715)\n",
      "Total index-bleed: 0.235925%\n"
     ]
    }
   ],
   "source": [
    "#need to measure index bleed of miseq data now, trying 0 mismatch miseqC dataset first\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from natsort import natsorted\n",
    "import sys\n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "df = pd.read_csv('miseq.final.txt', sep='\\t', index_col=0)\n",
    "#print list(df.columns.values)\n",
    "#sys.exit(1)\n",
    "#print len(df)\n",
    "#okay, so 1367 OTUs in entire data set, lets trim this down to only run 3, as that one is labeled here with biomock\n",
    "keepers = ['m4C-301-1', 'm4C-301-2', 'm4C-500-1', 'm4C-500-2', 'm4C-712-1', 'm4C-712-2', \n",
    "           'm4C-736-1', 'm4C-736-2', 'm4C-744-1', 'm4C-744-2', 'm4C-755-1', 'm4C-755-2', \n",
    "           'm4C-757-1', 'm4C-757-2', 'm4C-766-1', 'm4C-766-2', \n",
    "           'm4C-mock3-16000', 'm4C-mock3-32000a', 'm4C-mock3-32000b', 'm4C-stds']\n",
    "mock_samples = ['m4C-mock3-16000', 'm4C-mock3-32000a', 'm4C-mock3-32000b', 'm4C-stds']\n",
    "samples = [item for item in keepers if item not in mock_samples]\n",
    "#print samples\n",
    "df2 = df.reindex(columns=keepers, index=natsorted(df.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "df3 = df2.loc[(df2 != 0).any(1)]\n",
    "mock_otus = []\n",
    "for x in df3.index:\n",
    "    if x.startswith('mock'):\n",
    "        if not x.startswith('mock3'): #mock3 is actually in some of the samples so can't use for index-bleed measurment\n",
    "            mock_otus.append(x)\n",
    "#now slice into two tables containing only mocks and calculate index-bleed\n",
    "bleed_counts = df3.reindex(columns=samples, index=mock_otus)\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "real_counts = df3.reindex(columns=mock_samples, index=mock_otus)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed out of mock: %f%% (%i out of %i)\" % (index_bleed, bleed_sum, real_sum)\n",
    "\n",
    "#now try to to do the opposite, count number of times that samples move into mock\n",
    "#first generate list of mock otus\n",
    "sample_otus = []\n",
    "for x in df3.index:\n",
    "    if not x.startswith('mock'):\n",
    "        sample_otus.append(x)\n",
    "#slice first to samples, further trim list of only things found in mocks, would be spurious or chimeras\n",
    "sampledf = df3.reindex(columns=samples, index=sample_otus)\n",
    "sampledf = sampledf.loc[(sampledf != 0).any(1)]\n",
    "cleaned_otus = sampledf.index\n",
    "#print len(cleaned_otus), len(sample_otus)\n",
    "\n",
    "#now using new list grab counts in mock\n",
    "bleed_counts = df3.reindex(columns=mock_samples, index=cleaned_otus)\n",
    "bleed_counts = bleed_counts.loc[(bleed_counts != 0).any(1)]\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "#print len(bleed_counts.index), bleed_sum\n",
    "\n",
    "#finally go back to samples and get read counts for the ovlerlapping otus\n",
    "real_counts = df3.reindex(columns=samples, index=bleed_counts.index)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed2 = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed into mock: %f%% (%i out of %i)\" % (index_bleed2, bleed_sum, real_sum)\n",
    "print \"Total index-bleed: %f%%\" % (index_bleed + index_bleed2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 284,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index-bleed out of mock: 0.0%%\n"
     ]
    },
    {
     "ename": "SystemExit",
     "evalue": "1",
     "output_type": "error",
     "traceback": [
      "An exception has occurred, use %tb to see the full traceback.\n",
      "\u001b[0;31mSystemExit\u001b[0m\u001b[0;31m:\u001b[0m 1\n"
     ]
    }
   ],
   "source": [
    "#need to measure index bleed of miseqB dataset, this one slightly differnt as can only measure index out of mock\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from natsort import natsorted\n",
    "import sys\n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "df = pd.read_csv('miseq.final.txt', sep='\\t', index_col=0)\n",
    "#print list(df.columns.values)\n",
    "#sys.exit(1)\n",
    "#print len(df)\n",
    "#okay, so 1367 OTUs in entire data set, lets trim this down to only run 3, as that one is labeled here with biomock\n",
    "keepers = ['m4b-744-1', 'm4b-744-2', 'm4b-766-1', 'm4b-766-2']\n",
    "mock_samples = ['m4C-mock3-16000', 'm4C-mock3-32000a', 'm4C-mock3-32000b', 'm4C-stds']\n",
    "samples = [item for item in keepers if item not in mock_samples]\n",
    "#print samples\n",
    "df2 = df.reindex(columns=keepers, index=natsorted(df.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "df3 = df2.loc[(df2 != 0).any(1)]\n",
    "mock_otus = []\n",
    "for x in df3.index: #get index from total table\n",
    "    if x.startswith('mock'):\n",
    "        if not x.startswith('mock3'): #mock3 is actually in some of the samples so can't use for index-bleed measurment\n",
    "            mock_otus.append(x)\n",
    "if len(mock_otus) < 1: #this means that no read counts from mock members were found, so index-bleed is 0\n",
    "    print \"Index-bleed out of mock: 0.0%\"\n",
    "    sys.exit(1)\n",
    "#now slice into two tables containing only mocks and calculate index-bleed\n",
    "bleed_counts = df3.reindex(columns=samples, index=mock_otus)\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "real_counts = df3.reindex(columns=mock_samples, index=mock_otus)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed out of mock: %f%% (%i out of %i)\" % (index_bleed, bleed_sum, real_sum)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Kept 6452482 reads out of 17459832 total reads\r\n"
     ]
    }
   ],
   "source": [
    "#while I calculated index bleed from scripts above, create simplier solution for graphing the data by simply\n",
    "#mapping the reads to the mock community members\n",
    "!amptk select -i all.reads.fq -o figure2.reads.demux.fa --format fasta -f figure2.samples.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "vsearch v2.4.2_osx_x86_64, 16.0GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file /Users/jon/amptk/DB/amptk_mock3.fa 100%  \n",
      "13072 nt in 23 seqs, min 335, max 680, avg 568\n",
      "Masking 100% \n",
      "Counting unique k-mers 100% \n",
      "Creating index of unique k-mers 100% \n",
      "Searching 100%                                                                                                                                                                                                                                                                                                                                                                                                              \n",
      "Matching query sequences: 2434153 of 6452482 (37.72%)\n",
      "Writing OTU table (classic) 100%  \n"
     ]
    }
   ],
   "source": [
    "#now map to mock3.fa\n",
    "!vsearch --usearch_global figure2.reads.demux.fa --db $HOME/amptk/DB/amptk_mock3.fa --id 0.97 --strand plus --otutabout fig2.otu_table.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[07:15:34 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[07:15:34 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[07:15:34 PM]\u001b[0m: Loading OTU table: fig2.otu_table.txt\n",
      "\u001b[92m[07:15:34 PM]\u001b[0m: OTU table contains 23 OTUs\n",
      "\u001b[92m[07:15:34 PM]\u001b[0m: Sorting OTU table by user defined order (--col_order)\n",
      "\u001b[92m[07:15:34 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[07:15:34 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[07:15:34 PM]\u001b[0m: Overwriting auto detect index-bleed, setting to 0.000000%\n",
      "\u001b[92m[07:15:34 PM]\u001b[0m: Filtering OTU table down to 23 OTUs\n",
      "\u001b[92m[07:15:34 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      fig2.stats.txt\n",
      "Sorted OTU table:     fig2.sorted.txt\n",
      "Normalized/filter:    fig2.normalized.txt\n",
      "Final Binary table:   fig2.final.binary.txt\n",
      "Final OTU table:      fig2.final.txt\n",
      "Filtered OTUs:        fig2.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f fig2.filtered.otus.fa -i fig2.final.txt -m mapping_file.txt -d ITS2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#reorder OTU table and draw heatmap\n",
    "!amptk filter -i fig2.otu_table.txt -f $HOME/amptk/DB/amptk_mock3.fa -p 0 --col_order m3-stds,m3-mock3-16000,m3-mock3-3200a,m3-mock3-32000b,m3-744-1,m3-744-2,m3-766-1,m3-766-2,m4A-stds,m4A-mock3-16000,m4A-mock3-32000a,m4A-mock3-32000b,m4A-744-1,m4A-744-2,m4A-766-1,m4A-766-2,m4b-744-1,m4b-744-2,m4b-766-1,m4b-766-2,m4C-stds,m4C-mock3-16000,m4C-mock3-32000a,m4C-mock3-32000b,m4C-744-1,m4C-744-2,m4C-766-1,m4C-766-2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 196,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "vsearch v2.4.2_osx_x86_64, 16.0GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading input file 100%                                                                                                                                                                                                                                                                                                                                                                                                              \n",
      "18273964 sequences kept (of which 0 truncated), 0 sequences discarded.\n",
      "vsearch v2.4.2_osx_x86_64, 16.0GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file /Users/jon/amptk/DB/amptk_mock2.fa 100%  \n",
      "14847 nt in 26 seqs, min 335, max 680, avg 571\n",
      "Masking 100% \n",
      "Counting unique k-mers 100% \n",
      "Creating index of unique k-mers 100% \n",
      "Searching 100%                                                                                                                                                                                                                                                                                                                                                                                                                \n",
      "Matching query sequences: 3182927 of 18273964 (17.42%)\n",
      "Writing OTU table (classic) 100%  \n"
     ]
    }
   ],
   "source": [
    "#map the miseq reads to the mock community fasta file\n",
    "!vsearch --fastq_filter miseq.fq --fastaout miseq.fa\n",
    "!vsearch --usearch_global miseq.fa --id 0.97 --db /Users/jon/amptk/DB/amptk_mock2.fa --otutabout miseq.mock.otu_table.txt\n",
    "!rm miseq.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!amptk heatmap -i fig2.final.txt -o fig2.svg -m heatmap --color YlOrRd -d tsv --annotate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "m4A-stds             34\n",
      "m4A-mock3-16000      16\n",
      "m4A-mock3-32000a     28\n",
      "m4A-mock3-32000b     34\n",
      "m4A-744-1           272\n",
      "m4A-744-2           354\n",
      "m4A-766-1           423\n",
      "m4A-766-2           263\n",
      "m4A-755-1           361\n",
      "m4A-755-2           340\n",
      "m4b-744-1           234\n",
      "m4b-744-2           315\n",
      "m4b-766-1           342\n",
      "m4b-766-2           220\n",
      "Name: Number of OTUs, dtype: int64\n",
      "                                                    m4A-stds  m4A-mock3-16000  \\\n",
      "#OTU ID                                                                         \n",
      "mock1 |FP170182|Phialocephala fusca_pident=99.6...      8655              727   \n",
      "mock2 |RF10JR|unknown Ascomycete_pident=100.0_O...      9175            20768   \n",
      "mock3 |FP170134|Phialocephala lagerbergii_piden...      8652            28520   \n",
      "mock4 |FP170235|unknown Ascomycete_pident=100.0...      9051            27726   \n",
      "mock5 |RF1JR|Aspergillus candidus_pident=100.0_...      9054              148   \n",
      "mock6 |RF3JR|Bjerkandara adusta_pident=100.0_OTU22     13625            13119   \n",
      "mock7 |GDL-1|Laetiporus caribensis_pident=100.0...      9424            15059   \n",
      "mock8 |RF5JR|Trametes gibbosa_pident=100.0_OTU25        8162            13856   \n",
      "mock9 |OR2|Laetiporus gilbertsonii_pident=100.0...      8756            22830   \n",
      "mock10 |MR5-1|Gloeoprus pannocinctus_pident=100...      9704            11131   \n",
      "mock11 |FP72612|Wolfiporia dilatohypha_pident=1...      8844              242   \n",
      "mock12 |FP170198|Schizopora sp_pident=100.0_OTU28       8725            12904   \n",
      "mock13 |FP170231|Fomitopsis ochracea_pident=100...      9493             5544   \n",
      "mock14 |L34-2|Laetiporus cremeiporus_pident=100...      8289            16400   \n",
      "mock15 |RF9JR|Phanerochaete laevis_pident=100.0...      9263            15698   \n",
      "mock16 |DA37|Laetiporus cincinnatus_pident=100....      7745            16660   \n",
      "mock17 |RF7JR|Puntularia strigooszonata_pident=...      7689            10704   \n",
      "mock18 |IN4-1|Phellinus cinereus_pident=100.0_O...      9843             6327   \n",
      "mock19 |MR5-3|Antrodiellla semisupina_pident=10...      9347             9363   \n",
      "mock20 |TJV93-174|Leptoporus mollis_pident=100....      9050             8861   \n",
      "mock21 |RLG7163|Leptoporus mollis_pident=99.2_O...      9309              402   \n",
      "mock22 |FP170186|Mortierelllales sp_pident=100....      9194            10927   \n",
      "mock23 |HHB9564|Laetiporus percisinus_pident=10...      6796              507   \n",
      "mock24 |FP170215|Penicillium nothofagi_pident=1...      8317                0   \n",
      "mock25 |FP170177|Pochonia suchlasporia_pident=1...      2067                0   \n",
      "mock26 |MD275|Wolfiporia cocos_pident=100.0_OTU127      6746                1   \n",
      "Other OTUs                                               846              150   \n",
      "\n",
      "                                                    m4A-mock3-32000a  \\\n",
      "#OTU ID                                                                \n",
      "mock1 |FP170182|Phialocephala fusca_pident=99.6...               331   \n",
      "mock2 |RF10JR|unknown Ascomycete_pident=100.0_O...             26137   \n",
      "mock3 |FP170134|Phialocephala lagerbergii_piden...             29490   \n",
      "mock4 |FP170235|unknown Ascomycete_pident=100.0...             32579   \n",
      "mock5 |RF1JR|Aspergillus candidus_pident=100.0_...               122   \n",
      "mock6 |RF3JR|Bjerkandara adusta_pident=100.0_OTU22             13907   \n",
      "mock7 |GDL-1|Laetiporus caribensis_pident=100.0...             16618   \n",
      "mock8 |RF5JR|Trametes gibbosa_pident=100.0_OTU25               14640   \n",
      "mock9 |OR2|Laetiporus gilbertsonii_pident=100.0...             21666   \n",
      "mock10 |MR5-1|Gloeoprus pannocinctus_pident=100...             11803   \n",
      "mock11 |FP72612|Wolfiporia dilatohypha_pident=1...               229   \n",
      "mock12 |FP170198|Schizopora sp_pident=100.0_OTU28              14007   \n",
      "mock13 |FP170231|Fomitopsis ochracea_pident=100...              5448   \n",
      "mock14 |L34-2|Laetiporus cremeiporus_pident=100...             16418   \n",
      "mock15 |RF9JR|Phanerochaete laevis_pident=100.0...             15549   \n",
      "mock16 |DA37|Laetiporus cincinnatus_pident=100....             16086   \n",
      "mock17 |RF7JR|Puntularia strigooszonata_pident=...             11613   \n",
      "mock18 |IN4-1|Phellinus cinereus_pident=100.0_O...              5974   \n",
      "mock19 |MR5-3|Antrodiellla semisupina_pident=10...              9006   \n",
      "mock20 |TJV93-174|Leptoporus mollis_pident=100....              8728   \n",
      "mock21 |RLG7163|Leptoporus mollis_pident=99.2_O...               307   \n",
      "mock22 |FP170186|Mortierelllales sp_pident=100....              9781   \n",
      "mock23 |HHB9564|Laetiporus percisinus_pident=10...               443   \n",
      "mock24 |FP170215|Penicillium nothofagi_pident=1...                 0   \n",
      "mock25 |FP170177|Pochonia suchlasporia_pident=1...                 0   \n",
      "mock26 |MD275|Wolfiporia cocos_pident=100.0_OTU127                 1   \n",
      "Other OTUs                                                       113   \n",
      "\n",
      "                                                    m4A-mock3-32000b  \\\n",
      "#OTU ID                                                                \n",
      "mock1 |FP170182|Phialocephala fusca_pident=99.6...              3352   \n",
      "mock2 |RF10JR|unknown Ascomycete_pident=100.0_O...             18343   \n",
      "mock3 |FP170134|Phialocephala lagerbergii_piden...             21271   \n",
      "mock4 |FP170235|unknown Ascomycete_pident=100.0...             24280   \n",
      "mock5 |RF1JR|Aspergillus candidus_pident=100.0_...               275   \n",
      "mock6 |RF3JR|Bjerkandara adusta_pident=100.0_OTU22             15090   \n",
      "mock7 |GDL-1|Laetiporus caribensis_pident=100.0...             16429   \n",
      "mock8 |RF5JR|Trametes gibbosa_pident=100.0_OTU25               14869   \n",
      "mock9 |OR2|Laetiporus gilbertsonii_pident=100.0...             20637   \n",
      "mock10 |MR5-1|Gloeoprus pannocinctus_pident=100...             14284   \n",
      "mock11 |FP72612|Wolfiporia dilatohypha_pident=1...               198   \n",
      "mock12 |FP170198|Schizopora sp_pident=100.0_OTU28              14897   \n",
      "mock13 |FP170231|Fomitopsis ochracea_pident=100...              6919   \n",
      "mock14 |L34-2|Laetiporus cremeiporus_pident=100...             16817   \n",
      "mock15 |RF9JR|Phanerochaete laevis_pident=100.0...             18182   \n",
      "mock16 |DA37|Laetiporus cincinnatus_pident=100....             14576   \n",
      "mock17 |RF7JR|Puntularia strigooszonata_pident=...             11720   \n",
      "mock18 |IN4-1|Phellinus cinereus_pident=100.0_O...              7522   \n",
      "mock19 |MR5-3|Antrodiellla semisupina_pident=10...             11618   \n",
      "mock20 |TJV93-174|Leptoporus mollis_pident=100....             10017   \n",
      "mock21 |RLG7163|Leptoporus mollis_pident=99.2_O...               410   \n",
      "mock22 |FP170186|Mortierelllales sp_pident=100....             13425   \n",
      "mock23 |HHB9564|Laetiporus percisinus_pident=10...               551   \n",
      "mock24 |FP170215|Penicillium nothofagi_pident=1...                 8   \n",
      "mock25 |FP170177|Pochonia suchlasporia_pident=1...                 2   \n",
      "mock26 |MD275|Wolfiporia cocos_pident=100.0_OTU127                 2   \n",
      "Other OTUs                                                       147   \n",
      "\n",
      "                                                    m4A-744-1  m4A-744-2  \\\n",
      "#OTU ID                                                                    \n",
      "mock1 |FP170182|Phialocephala fusca_pident=99.6...          9         24   \n",
      "mock2 |RF10JR|unknown Ascomycete_pident=100.0_O...         42         25   \n",
      "mock3 |FP170134|Phialocephala lagerbergii_piden...         74         18   \n",
      "mock4 |FP170235|unknown Ascomycete_pident=100.0...         60         17   \n",
      "mock5 |RF1JR|Aspergillus candidus_pident=100.0_...          2         17   \n",
      "mock6 |RF3JR|Bjerkandara adusta_pident=100.0_OTU22         33         22   \n",
      "mock7 |GDL-1|Laetiporus caribensis_pident=100.0...         42         28   \n",
      "mock8 |RF5JR|Trametes gibbosa_pident=100.0_OTU25           31         13   \n",
      "mock9 |OR2|Laetiporus gilbertsonii_pident=100.0...         52         24   \n",
      "mock10 |MR5-1|Gloeoprus pannocinctus_pident=100...         32         19   \n",
      "mock11 |FP72612|Wolfiporia dilatohypha_pident=1...          0         15   \n",
      "mock12 |FP170198|Schizopora sp_pident=100.0_OTU28          42         17   \n",
      "mock13 |FP170231|Fomitopsis ochracea_pident=100...         16         19   \n",
      "mock14 |L34-2|Laetiporus cremeiporus_pident=100...         42         17   \n",
      "mock15 |RF9JR|Phanerochaete laevis_pident=100.0...         32         18   \n",
      "mock16 |DA37|Laetiporus cincinnatus_pident=100....         28         20   \n",
      "mock17 |RF7JR|Puntularia strigooszonata_pident=...         33         10   \n",
      "mock18 |IN4-1|Phellinus cinereus_pident=100.0_O...         28         12   \n",
      "mock19 |MR5-3|Antrodiellla semisupina_pident=10...         27         24   \n",
      "mock20 |TJV93-174|Leptoporus mollis_pident=100....         28         19   \n",
      "mock21 |RLG7163|Leptoporus mollis_pident=99.2_O...          1         18   \n",
      "mock22 |FP170186|Mortierelllales sp_pident=100....         24         24   \n",
      "mock23 |HHB9564|Laetiporus percisinus_pident=10...          0         19   \n",
      "mock24 |FP170215|Penicillium nothofagi_pident=1...          0         13   \n",
      "mock25 |FP170177|Pochonia suchlasporia_pident=1...          0          6   \n",
      "mock26 |MD275|Wolfiporia cocos_pident=100.0_OTU127          0         14   \n",
      "Other OTUs                                             267898     247863   \n",
      "\n",
      "                                                    m4A-766-1  m4A-766-2  \\\n",
      "#OTU ID                                                                    \n",
      "mock1 |FP170182|Phialocephala fusca_pident=99.6...          2          0   \n",
      "mock2 |RF10JR|unknown Ascomycete_pident=100.0_O...         57         55   \n",
      "mock3 |FP170134|Phialocephala lagerbergii_piden...         63        579   \n",
      "mock4 |FP170235|unknown Ascomycete_pident=100.0...         54         65   \n",
      "mock5 |RF1JR|Aspergillus candidus_pident=100.0_...          0          0   \n",
      "mock6 |RF3JR|Bjerkandara adusta_pident=100.0_OTU22         27         31   \n",
      "mock7 |GDL-1|Laetiporus caribensis_pident=100.0...         33         35   \n",
      "mock8 |RF5JR|Trametes gibbosa_pident=100.0_OTU25           27         33   \n",
      "mock9 |OR2|Laetiporus gilbertsonii_pident=100.0...         59         43   \n",
      "mock10 |MR5-1|Gloeoprus pannocinctus_pident=100...         22         20   \n",
      "mock11 |FP72612|Wolfiporia dilatohypha_pident=1...          1          0   \n",
      "mock12 |FP170198|Schizopora sp_pident=100.0_OTU28          23         25   \n",
      "mock13 |FP170231|Fomitopsis ochracea_pident=100...          7         11   \n",
      "mock14 |L34-2|Laetiporus cremeiporus_pident=100...         44         43   \n",
      "mock15 |RF9JR|Phanerochaete laevis_pident=100.0...         37         26   \n",
      "mock16 |DA37|Laetiporus cincinnatus_pident=100....         32         35   \n",
      "mock17 |RF7JR|Puntularia strigooszonata_pident=...         22         24   \n",
      "mock18 |IN4-1|Phellinus cinereus_pident=100.0_O...         16         16   \n",
      "mock19 |MR5-3|Antrodiellla semisupina_pident=10...         14         28   \n",
      "mock20 |TJV93-174|Leptoporus mollis_pident=100....         18         23   \n",
      "mock21 |RLG7163|Leptoporus mollis_pident=99.2_O...          1          1   \n",
      "mock22 |FP170186|Mortierelllales sp_pident=100....          9         30   \n",
      "mock23 |HHB9564|Laetiporus percisinus_pident=10...          1          0   \n",
      "mock24 |FP170215|Penicillium nothofagi_pident=1...          0          0   \n",
      "mock25 |FP170177|Pochonia suchlasporia_pident=1...          0          0   \n",
      "mock26 |MD275|Wolfiporia cocos_pident=100.0_OTU127          0          0   \n",
      "Other OTUs                                             292463     302850   \n",
      "\n",
      "                                                    m4A-755-1  m4A-755-2  \\\n",
      "#OTU ID                                                                    \n",
      "mock1 |FP170182|Phialocephala fusca_pident=99.6...          0          0   \n",
      "mock2 |RF10JR|unknown Ascomycete_pident=100.0_O...          0          0   \n",
      "mock3 |FP170134|Phialocephala lagerbergii_piden...          6          0   \n",
      "mock4 |FP170235|unknown Ascomycete_pident=100.0...          0          0   \n",
      "mock5 |RF1JR|Aspergillus candidus_pident=100.0_...          0          0   \n",
      "mock6 |RF3JR|Bjerkandara adusta_pident=100.0_OTU22          0          1   \n",
      "mock7 |GDL-1|Laetiporus caribensis_pident=100.0...          0          0   \n",
      "mock8 |RF5JR|Trametes gibbosa_pident=100.0_OTU25            0          0   \n",
      "mock9 |OR2|Laetiporus gilbertsonii_pident=100.0...          0          0   \n",
      "mock10 |MR5-1|Gloeoprus pannocinctus_pident=100...          0          0   \n",
      "mock11 |FP72612|Wolfiporia dilatohypha_pident=1...          0          0   \n",
      "mock12 |FP170198|Schizopora sp_pident=100.0_OTU28           0          0   \n",
      "mock13 |FP170231|Fomitopsis ochracea_pident=100...          0          0   \n",
      "mock14 |L34-2|Laetiporus cremeiporus_pident=100...          0          0   \n",
      "mock15 |RF9JR|Phanerochaete laevis_pident=100.0...          0          0   \n",
      "mock16 |DA37|Laetiporus cincinnatus_pident=100....          0          0   \n",
      "mock17 |RF7JR|Puntularia strigooszonata_pident=...          0          0   \n",
      "mock18 |IN4-1|Phellinus cinereus_pident=100.0_O...          0          2   \n",
      "mock19 |MR5-3|Antrodiellla semisupina_pident=10...          0          0   \n",
      "mock20 |TJV93-174|Leptoporus mollis_pident=100....          0          0   \n",
      "mock21 |RLG7163|Leptoporus mollis_pident=99.2_O...          0          0   \n",
      "mock22 |FP170186|Mortierelllales sp_pident=100....          0          0   \n",
      "mock23 |HHB9564|Laetiporus percisinus_pident=10...          0          0   \n",
      "mock24 |FP170215|Penicillium nothofagi_pident=1...          0          0   \n",
      "mock25 |FP170177|Pochonia suchlasporia_pident=1...          0          0   \n",
      "mock26 |MD275|Wolfiporia cocos_pident=100.0_OTU127          0          0   \n",
      "Other OTUs                                             292475     284485   \n",
      "\n",
      "                                                    m4b-744-1  m4b-744-2  \\\n",
      "#OTU ID                                                                    \n",
      "mock1 |FP170182|Phialocephala fusca_pident=99.6...          0          0   \n",
      "mock2 |RF10JR|unknown Ascomycete_pident=100.0_O...          0          0   \n",
      "mock3 |FP170134|Phialocephala lagerbergii_piden...         12          4   \n",
      "mock4 |FP170235|unknown Ascomycete_pident=100.0...          0          0   \n",
      "mock5 |RF1JR|Aspergillus candidus_pident=100.0_...          0          0   \n",
      "mock6 |RF3JR|Bjerkandara adusta_pident=100.0_OTU22          0          0   \n",
      "mock7 |GDL-1|Laetiporus caribensis_pident=100.0...          0          0   \n",
      "mock8 |RF5JR|Trametes gibbosa_pident=100.0_OTU25            0          0   \n",
      "mock9 |OR2|Laetiporus gilbertsonii_pident=100.0...          0          0   \n",
      "mock10 |MR5-1|Gloeoprus pannocinctus_pident=100...          0          0   \n",
      "mock11 |FP72612|Wolfiporia dilatohypha_pident=1...          0          0   \n",
      "mock12 |FP170198|Schizopora sp_pident=100.0_OTU28           0          0   \n",
      "mock13 |FP170231|Fomitopsis ochracea_pident=100...          0          0   \n",
      "mock14 |L34-2|Laetiporus cremeiporus_pident=100...          0          0   \n",
      "mock15 |RF9JR|Phanerochaete laevis_pident=100.0...          0          0   \n",
      "mock16 |DA37|Laetiporus cincinnatus_pident=100....          0          0   \n",
      "mock17 |RF7JR|Puntularia strigooszonata_pident=...          0          0   \n",
      "mock18 |IN4-1|Phellinus cinereus_pident=100.0_O...          0          0   \n",
      "mock19 |MR5-3|Antrodiellla semisupina_pident=10...          0          0   \n",
      "mock20 |TJV93-174|Leptoporus mollis_pident=100....          0          0   \n",
      "mock21 |RLG7163|Leptoporus mollis_pident=99.2_O...          0          0   \n",
      "mock22 |FP170186|Mortierelllales sp_pident=100....          0          0   \n",
      "mock23 |HHB9564|Laetiporus percisinus_pident=10...          0          0   \n",
      "mock24 |FP170215|Penicillium nothofagi_pident=1...          0          0   \n",
      "mock25 |FP170177|Pochonia suchlasporia_pident=1...          0          0   \n",
      "mock26 |MD275|Wolfiporia cocos_pident=100.0_OTU127          0          0   \n",
      "Other OTUs                                             182064     152433   \n",
      "\n",
      "                                                    m4b-766-1  m4b-766-2  \n",
      "#OTU ID                                                                   \n",
      "mock1 |FP170182|Phialocephala fusca_pident=99.6...          0          0  \n",
      "mock2 |RF10JR|unknown Ascomycete_pident=100.0_O...          0          0  \n",
      "mock3 |FP170134|Phialocephala lagerbergii_piden...          0        246  \n",
      "mock4 |FP170235|unknown Ascomycete_pident=100.0...          0          0  \n",
      "mock5 |RF1JR|Aspergillus candidus_pident=100.0_...          0          0  \n",
      "mock6 |RF3JR|Bjerkandara adusta_pident=100.0_OTU22          0          0  \n",
      "mock7 |GDL-1|Laetiporus caribensis_pident=100.0...          0          0  \n",
      "mock8 |RF5JR|Trametes gibbosa_pident=100.0_OTU25            0          0  \n",
      "mock9 |OR2|Laetiporus gilbertsonii_pident=100.0...          0          0  \n",
      "mock10 |MR5-1|Gloeoprus pannocinctus_pident=100...          0          0  \n",
      "mock11 |FP72612|Wolfiporia dilatohypha_pident=1...          0          0  \n",
      "mock12 |FP170198|Schizopora sp_pident=100.0_OTU28           0          0  \n",
      "mock13 |FP170231|Fomitopsis ochracea_pident=100...          0          0  \n",
      "mock14 |L34-2|Laetiporus cremeiporus_pident=100...          0          0  \n",
      "mock15 |RF9JR|Phanerochaete laevis_pident=100.0...          0          0  \n",
      "mock16 |DA37|Laetiporus cincinnatus_pident=100....          0          0  \n",
      "mock17 |RF7JR|Puntularia strigooszonata_pident=...          0          0  \n",
      "mock18 |IN4-1|Phellinus cinereus_pident=100.0_O...          0          0  \n",
      "mock19 |MR5-3|Antrodiellla semisupina_pident=10...          0          0  \n",
      "mock20 |TJV93-174|Leptoporus mollis_pident=100....          0          0  \n",
      "mock21 |RLG7163|Leptoporus mollis_pident=99.2_O...          0          0  \n",
      "mock22 |FP170186|Mortierelllales sp_pident=100....          0          0  \n",
      "mock23 |HHB9564|Laetiporus percisinus_pident=10...          0          0  \n",
      "mock24 |FP170215|Penicillium nothofagi_pident=1...          0          0  \n",
      "mock25 |FP170177|Pochonia suchlasporia_pident=1...          0          0  \n",
      "mock26 |MD275|Wolfiporia cocos_pident=100.0_OTU127          0          0  \n",
      "Other OTUs                                             162453     138778  \n",
      "m4A-stds            225821\n",
      "m4A-mock3-16000     268574\n",
      "m4A-mock3-32000a    280996\n",
      "m4A-mock3-32000b    275841\n",
      "m4A-744-1           268576\n",
      "m4A-744-2           248335\n",
      "m4A-766-1           293032\n",
      "m4A-766-2           303973\n",
      "m4A-755-1           292481\n",
      "m4A-755-2           284488\n",
      "m4b-744-1           182076\n",
      "m4b-744-2           152437\n",
      "m4b-766-1           162453\n",
      "m4b-766-2           139024\n",
      "dtype: int64\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "from natsort import natsorted\n",
    "#load the miseq table, no filtering version\n",
    "df = pd.read_csv('miseq.sorted.txt', sep='\\t', index_col=0)\n",
    "#Figure 2 should be about index bleed, so generate a table containing the samples that were run in all runs\n",
    "figureTwo = ['m4A-stds','m4A-mock3-16000','m4A-mock3-32000a','m4A-mock3-32000b','m4A-744-1','m4A-744-2',\n",
    "             'm4A-766-1','m4A-766-2','m4A-755-1','m4A-755-2','m4b-744-1','m4b-744-2','m4b-766-1','m4b-766-2' ]\n",
    "df2 = df.reindex(columns=figureTwo, index=natsorted(df.index))\n",
    "mocks = []\n",
    "for x in df2.index:\n",
    "    if x.startswith('mock'):\n",
    "        mocks.append(x)\n",
    "otus = [item for item in df2.index if item not in mocks]\n",
    "\n",
    "df3 = df2.reindex(columns=figureTwo, index=mocks)\n",
    "df3.to_csv('figure2.onlymocks.txt', sep='\\t')\n",
    "df4 = df2.reindex(columns=figureTwo, index=otus)\n",
    "#get number of OTUs from df4 as well as total reads\n",
    "reads = df4.sum(axis=0)\n",
    "reads.name =\"Other OTUs\"\n",
    "otu_count = df4[df4 > 0].count(axis=0, numeric_only=True)\n",
    "otu_count.name = \"Number of OTUs\"\n",
    "#print reads\n",
    "print otu_count\n",
    "#now append the read numbers to df3 dataframe\n",
    "df3 = df3.append(reads)\n",
    "print df3\n",
    "total_reads = df3.sum(axis=0)\n",
    "print total_reads\n",
    "df3.to_csv('figure2.miseqAB_only2.txt', sep='\\t')\n",
    "\n",
    "\n",
    "!amptk heatmap -i figure2.miseqAB_only2.txt -d tsv -o figure2-2.pdf --format pdf -m heatmap --annotate --color YlOrBr --vmax 30000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['m4A-301-1' 'm4A-301-2' 'm4A-500-1' 'm4A-500-2' 'm4A-712-1' 'm4A-712-2'\n",
      " 'm4A-736-1' 'm4A-736-2' 'm4A-744-1' 'm4A-744-2' 'm4A-755-1' 'm4A-755-2'\n",
      " 'm4A-757-1' 'm4A-757-2' 'm4A-766-1' 'm4A-766-2' 'm4A-mock3-16000'\n",
      " 'm4A-mock3-32000a' 'm4A-mock3-32000b' 'm4A-stds' 'm4C-301-1' 'm4C-301-2'\n",
      " 'm4C-500-1' 'm4C-500-2' 'm4C-712-1' 'm4C-712-2' 'm4C-736-1' 'm4C-736-2'\n",
      " 'm4C-744-1' 'm4C-744-2' 'm4C-755-2' 'm4C-757-1' 'm4C-757-2' 'm4C-766-1'\n",
      " 'm4C-766-2' 'm4C-mock3-16000' 'm4C-mock3-32000a' 'm4C-mock3-32000b'\n",
      " 'm4C-stds' 'm4a-1mm-301-1' 'm4a-1mm-301-2' 'm4a-1mm-500-1' 'm4a-1mm-500-2'\n",
      " 'm4a-1mm-712-1' 'm4a-1mm-712-2' 'm4a-1mm-736-1' 'm4a-1mm-736-2'\n",
      " 'm4a-1mm-744-1' 'm4a-1mm-744-2' 'm4a-1mm-755-1' 'm4a-1mm-755-2'\n",
      " 'm4a-1mm-757-1' 'm4a-1mm-757-2' 'm4a-1mm-766-1' 'm4a-1mm-766-2'\n",
      " 'm4a-1mm-mock3-16000' 'm4a-1mm-mock3-32000a' 'm4a-1mm-mock3-32000b'\n",
      " 'm4a-1mm-stds' 'm4b-744-1' 'm4b-744-2' 'm4b-766-1' 'm4b-766-2']\n"
     ]
    }
   ],
   "source": [
    "#now split OTU table into the different parts/segments needed for different figures\n",
    "#lets do this in python/pandas\n",
    "import pandas as pd\n",
    "from natsort import natsorted\n",
    "\n",
    "#now load OTU table into dataframe\n",
    "df = pd.read_csv('miseq.mock.otu_table.txt', sep='\\t', index_col=0)\n",
    "print df.columns.values\n",
    "#headers for \n",
    "figureOne = ['m3-stds','m3-mock3-16000','m3-mock3-3200a','m3-mock3-32000b',\n",
    "             'm4A-stds','m4A-mock3-16000','m4A-mock3-32000a','m4A-mock3-32000b',\n",
    "             'm4C-stds','m4C-mock3-16000','m4C-mock3-32000a','m4C-mock3-32000b']\n",
    "#"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['m3-755-2', 'm3-301-1', 'm3-301-2', 'm3-500-1', 'm3-500-2', 'm3-712-1', 'm3-712-2', 'm3-736-1', 'm3-736-2', 'm3-744-1', 'm3-744-2', 'm3-755-1', 'm3-757-2', 'm3-766-1', 'm3-766-2', 'm3-mock3-16000', 'm3-mock3-32000b', 'm3-mock3-3200a', 'm3-stds', 'm4A-301-1', 'm4A-301-2', 'm4A-500-1', 'm4A-500-2', 'm4A-712-1', 'm4A-712-2', 'm4A-736-1', 'm4A-736-2', 'm4A-744-1', 'm4A-744-2', 'm4A-755-1', 'm4A-755-2', 'm4A-757-1', 'm4A-757-2', 'm4A-766-1', 'm4A-766-2', 'm4A-mock3-16000', 'm4A-mock3-32000a', 'm4A-mock3-32000b', 'm4A-stds', 'm4C-301-1', 'm4C-301-2', 'm4C-500-1', 'm4C-500-2', 'm4C-712-1', 'm4C-712-2', 'm4C-736-1', 'm4C-736-2', 'm4C-744-1', 'm4C-744-2', 'm4C-755-1', 'm4C-755-2', 'm4C-757-1', 'm4C-757-2', 'm4C-766-1', 'm4C-766-2', 'm4C-mock3-16000', 'm4C-mock3-32000a', 'm4C-mock3-32000b', 'm4C-stds', 'm4b-744-1', 'm4b-744-2', 'm4b-766-1', 'm4b-766-2', 'm5-301-1', 'm5-301-2', 'm5-500-1', 'm5-500-2', 'm5-712-1', 'm5-712-2', 'm5-736-1', 'm5-736-2', 'm5-744-1', 'm5-744-2', 'm5-755-1', 'm5-755-2', 'm5-757-1', 'm5-757-2', 'm5-766-1', 'm5-766-2', 'm5-SynMock1-8000', 'm5-SynMock2-8000', 'm5-mock3.0-16000', 'm5-mock3.5-16000']\n"
     ]
    }
   ],
   "source": [
    "#okay, now OTU table in dataframe and files have been renamed according to mock members\n",
    "#first figure is to compare the stds to the mock communities, set indexes in list and pull from dataframe\n",
    "print list(df.columns.values)\n",
    "from natsort import natsorted\n",
    "figureOne = ['m3-stds','m3-mock3-16000','m3-mock3-3200a','m3-mock3-32000b',\n",
    "             'm4A-stds','m4A-mock3-16000','m4A-mock3-32000a','m4A-mock3-32000b',\n",
    "             'm4C-stds','m4C-mock3-16000','m4C-mock3-32000a','m4C-mock3-32000b']\n",
    "mocks = []\n",
    "synmocks = []\n",
    "for k,v in annotation.items():\n",
    "    if not v in mocks:\n",
    "        if v.startswith('mock'):\n",
    "            mocks.append(v)\n",
    "        else:\n",
    "            synmocks.append(v)\n",
    "fig1 = df.reindex(columns=figureOne, index=natsorted(mocks))\n",
    "#print fig1\n",
    "#write to file\n",
    "fig1.to_csv('synmock.figure1.abundances.txt', sep='\\t')\n",
    "#then generate heatmap\n",
    "!amptk heatmap -i synmock.figure1.abundances.txt -d tsv -o figure1.pdf -m heatmap --annotate --color YlOrRd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         stds-expected  m3-stds  m3-mock3-16000  m3-mock3-3200a  \\\n",
      "#OTU ID                                                           \n",
      "mock1             1000     4705              19               6   \n",
      "mock2             1000     4732           11241           10358   \n",
      "mock3             1000     4532           12930           11600   \n",
      "mock4             1000     3941           14526           12488   \n",
      "mock5             1000     2679              31              23   \n",
      "mock6             1000     3801            7925            6978   \n",
      "mock7             1000     3138            7573            6283   \n",
      "mock8             1000     3573            7390            7075   \n",
      "mock9             1000     3861            8703           10196   \n",
      "mock10            1000     3328            3923            4075   \n",
      "mock11            1000     3730              90             111   \n",
      "mock12            1000     3782            6738            6816   \n",
      "mock13            1000     3583            2946            2879   \n",
      "mock14            1000     3699            9926           11498   \n",
      "mock15            1000     3523            6603            7313   \n",
      "mock16            1000     3495            6784            9090   \n",
      "mock17            1000     3779            8014            7474   \n",
      "mock18            1000     3415            2820            2871   \n",
      "mock19            1000     3216            3265            3675   \n",
      "mock20            1000     3321            4866            4198   \n",
      "mock21            1000     3681             221             379   \n",
      "mock22            1000     3087            4611            4258   \n",
      "mock23            1000     2031            2571            2306   \n",
      "mock24            1000     3462               0               0   \n",
      "mock25            1000     2691               0               0   \n",
      "mock26            1000        8               0               0   \n",
      "\n",
      "         m3-mock3-32000b  m4A-stds  m4A-mock3-16000  m4A-mock3-32000a  \\\n",
      "#OTU ID                                                                 \n",
      "mock1                  1      8655              727               331   \n",
      "mock2              11424      9175            20768             26137   \n",
      "mock3              12927      8652            28520             29490   \n",
      "mock4              14306      9051            27726             32579   \n",
      "mock5                  3      9054              148               122   \n",
      "mock6               7658     13625            13119             13907   \n",
      "mock7               6136      9424            15059             16618   \n",
      "mock8               7026      8162            13856             14640   \n",
      "mock9              12397      8756            22830             21666   \n",
      "mock10              3481      9704            11131             11803   \n",
      "mock11               110      8844              242               229   \n",
      "mock12              6503      8725            12904             14007   \n",
      "mock13              2685      9493             5544              5448   \n",
      "mock14             12049      8289            16400             16418   \n",
      "mock15              6611      9263            15698             15549   \n",
      "mock16              9199      7745            16660             16086   \n",
      "mock17              7987      7689            10704             11613   \n",
      "mock18              2506      9843             6327              5974   \n",
      "mock19              2971      9347             9363              9006   \n",
      "mock20              3946      9050             8861              8728   \n",
      "mock21               263      9309              402               307   \n",
      "mock22              3776      9194            10927              9781   \n",
      "mock23              1991      6782              506               443   \n",
      "mock24                 0      8317                0                 0   \n",
      "mock25                 0      2067                0                 0   \n",
      "mock26                 0      6746                1                 1   \n",
      "\n",
      "         m4A-mock3-32000b  m4C-stds  m4C-mock3-16000  m4C-mock3-32000a  \\\n",
      "#OTU ID                                                                  \n",
      "mock1                3352      8106              586               306   \n",
      "mock2               18343      9331            19433             24759   \n",
      "mock3               21271      8504            25274             27414   \n",
      "mock4               24280      8743            24813             31365   \n",
      "mock5                 275      7739              112                79   \n",
      "mock6               15090     12024            11439             12609   \n",
      "mock7               16429      8768            12283             15233   \n",
      "mock8               14869      7292            11410             13630   \n",
      "mock9               20637      7550            19030             19987   \n",
      "mock10              14284      8571             8999             10410   \n",
      "mock11                198      7783              181               212   \n",
      "mock12              14897      7428            10384             12517   \n",
      "mock13               6919      8725             4995              5123   \n",
      "mock14              16817      7191            13748             15575   \n",
      "mock15              18182      8194            13093             15015   \n",
      "mock16              14576      6791            13510             14762   \n",
      "mock17              11720      6943             8838             11574   \n",
      "mock18               7522      8876             5321              5839   \n",
      "mock19              11618      8577             8061              8618   \n",
      "mock20              10017      8102             7309              8606   \n",
      "mock21                410      7967              302               287   \n",
      "mock22              13425      7687             9328              9488   \n",
      "mock23                551      4690              320               345   \n",
      "mock24                  8      6729                0                 0   \n",
      "mock25                  2      1747                0                 0   \n",
      "mock26                  2      3859                1                 1   \n",
      "\n",
      "         m4C-mock3-32000b  \n",
      "#OTU ID                    \n",
      "mock1                2801  \n",
      "mock2               16555  \n",
      "mock3               18046  \n",
      "mock4               21158  \n",
      "mock5                 229  \n",
      "mock6               12828  \n",
      "mock7               14657  \n",
      "mock8               13041  \n",
      "mock9               18474  \n",
      "mock10              10547  \n",
      "mock11                167  \n",
      "mock12              12300  \n",
      "mock13               6021  \n",
      "mock14              13971  \n",
      "mock15              14786  \n",
      "mock16              12131  \n",
      "mock17               9751  \n",
      "mock18               6601  \n",
      "mock19              10458  \n",
      "mock20               8393  \n",
      "mock21                334  \n",
      "mock22              11768  \n",
      "mock23                363  \n",
      "mock24                  2  \n",
      "mock25                  2  \n",
      "mock26                  5  \n",
      "m3-301-1            114362\n",
      "m3-301-2            129396\n",
      "m3-500-1            122735\n",
      "m3-500-2            128356\n",
      "m3-712-1             96342\n",
      "m3-712-2             89754\n",
      "m3-736-1             97849\n",
      "m3-736-2            118031\n",
      "m3-744-1            111726\n",
      "m3-744-2            137995\n",
      "m3-755-1            131754\n",
      "m3-755-2            123169\n",
      "m3-757-2            110713\n",
      "m3-766-1            175512\n",
      "m3-766-2            123845\n",
      "m3-mock3-16000      133805\n",
      "m3-mock3-32000b     136068\n",
      "m3-mock3-3200a      132059\n",
      "m3-stds              88847\n",
      "m4A-301-1           257814\n",
      "m4A-301-2           283494\n",
      "m4A-500-1           316955\n",
      "m4A-500-2           263047\n",
      "m4A-712-1           251009\n",
      "m4A-712-2           228693\n",
      "m4A-736-1           265654\n",
      "m4A-736-2           250885\n",
      "m4A-744-1           268577\n",
      "m4A-744-2           248338\n",
      "m4A-755-1           292492\n",
      "                     ...  \n",
      "m4C-766-2           176818\n",
      "m4C-mock3-16000     228896\n",
      "m4C-mock3-32000a    263860\n",
      "m4C-mock3-32000b    235527\n",
      "m4C-stds            198674\n",
      "m4b-744-1           182078\n",
      "m4b-744-2           152438\n",
      "m4b-766-1           162451\n",
      "m4b-766-2           139108\n",
      "m5-301-1            147722\n",
      "m5-301-2            144223\n",
      "m5-500-1            138816\n",
      "m5-500-2             46479\n",
      "m5-712-1             57363\n",
      "m5-712-2            198236\n",
      "m5-736-1             73437\n",
      "m5-736-2            104364\n",
      "m5-744-1             72249\n",
      "m5-744-2            103864\n",
      "m5-755-1             93530\n",
      "m5-755-2             77063\n",
      "m5-757-1             96526\n",
      "m5-757-2             94298\n",
      "m5-766-1             72026\n",
      "m5-766-2             68795\n",
      "m5-SynMock1-8000    114097\n",
      "m5-SynMock2-8000     60827\n",
      "m5-mock3.0-16000     67824\n",
      "m5-mock3.5-16000     71324\n",
      "stds-expected        26000\n",
      "Name: Read_counts, dtype: int64\n",
      "         stds-expected   m3-stds  m3-mock3-16000  m3-mock3-3200a  \\\n",
      "#OTU ID                                                            \n",
      "mock1         0.008744  0.041141        0.000166        0.000052   \n",
      "mock2         0.008744  0.041377        0.098293        0.090572   \n",
      "mock3         0.008744  0.039629        0.113062        0.101432   \n",
      "mock4         0.008744  0.034461        0.127018        0.109197   \n",
      "mock5         0.008744  0.023426        0.000271        0.000201   \n",
      "mock6         0.008744  0.033237        0.069297        0.061017   \n",
      "mock7         0.008744  0.027439        0.066220        0.054940   \n",
      "mock8         0.008744  0.031243        0.064619        0.061865   \n",
      "mock9         0.008744  0.033761        0.076100        0.089155   \n",
      "mock10        0.008744  0.029101        0.034303        0.035632   \n",
      "mock11        0.008744  0.032616        0.000787        0.000971   \n",
      "mock12        0.008744  0.033070        0.058918        0.059600   \n",
      "mock13        0.008744  0.031330        0.025760        0.025174   \n",
      "mock14        0.008744  0.032345        0.086795        0.100540   \n",
      "mock15        0.008744  0.030806        0.057738        0.063946   \n",
      "mock16        0.008744  0.030561        0.059320        0.079484   \n",
      "mock17        0.008744  0.033044        0.070076        0.065354   \n",
      "mock18        0.008744  0.029861        0.024659        0.025104   \n",
      "mock19        0.008744  0.028121        0.028550        0.032135   \n",
      "mock20        0.008744  0.029039        0.042549        0.036708   \n",
      "mock21        0.008744  0.032187        0.001932        0.003314   \n",
      "mock22        0.008744  0.026993        0.040319        0.037233   \n",
      "mock23        0.008744  0.017759        0.022481        0.020164   \n",
      "mock24        0.008744  0.030272        0.000000        0.000000   \n",
      "mock25        0.008744  0.023531        0.000000        0.000000   \n",
      "mock26        0.008744  0.000070        0.000000        0.000000   \n",
      "\n",
      "         m3-mock3-32000b  m4A-stds  m4A-mock3-16000  m4A-mock3-32000a  \\\n",
      "#OTU ID                                                                 \n",
      "mock1           0.000009  0.075681         0.006357          0.002894   \n",
      "mock2           0.099893  0.080228         0.181599          0.228546   \n",
      "mock3           0.113036  0.075655         0.249384          0.257865   \n",
      "mock4           0.125094  0.079143         0.242441          0.284876   \n",
      "mock5           0.000026  0.079170         0.001294          0.001067   \n",
      "mock6           0.066963  0.119139         0.114715          0.121605   \n",
      "mock7           0.053654  0.082405         0.131678          0.145311   \n",
      "mock8           0.061436  0.071370         0.121159          0.128015   \n",
      "mock9           0.108401  0.076564         0.199629          0.189451   \n",
      "mock10          0.030438  0.084853         0.097331          0.103207   \n",
      "mock11          0.000962  0.077333         0.002116          0.002002   \n",
      "mock12          0.056863  0.076293         0.112835          0.122479   \n",
      "mock13          0.023478  0.083008         0.048478          0.047638   \n",
      "mock14          0.105358  0.072480         0.143404          0.143562   \n",
      "mock15          0.057808  0.080997         0.137266          0.135963   \n",
      "mock16          0.080438  0.067724         0.145678          0.140659   \n",
      "mock17          0.069840  0.067234         0.093598          0.101546   \n",
      "mock18          0.021913  0.086069         0.055324          0.052238   \n",
      "mock19          0.025979  0.081732         0.081872          0.078750   \n",
      "mock20          0.034504  0.079135         0.077482          0.076319   \n",
      "mock21          0.002300  0.081399         0.003515          0.002684   \n",
      "mock22          0.033018  0.080394         0.095547          0.085527   \n",
      "mock23          0.017410  0.059303         0.004425          0.003874   \n",
      "mock24          0.000000  0.072725         0.000000          0.000000   \n",
      "mock25          0.000000  0.018074         0.000000          0.000000   \n",
      "mock26          0.000000  0.058988         0.000009          0.000009   \n",
      "\n",
      "         m4A-mock3-32000b  m4C-stds  m4C-mock3-16000  m4C-mock3-32000a  \\\n",
      "#OTU ID                                                                  \n",
      "mock1            0.029310  0.070880         0.005124          0.002676   \n",
      "mock2            0.160394  0.081592         0.169925          0.216497   \n",
      "mock3            0.185997  0.074360         0.221000          0.239712   \n",
      "mock4            0.212308  0.076450         0.216969          0.274261   \n",
      "mock5            0.002405  0.067671         0.000979          0.000691   \n",
      "mock6            0.131949  0.105140         0.100024          0.110255   \n",
      "mock7            0.143658  0.076669         0.107405          0.133200   \n",
      "mock8            0.130017  0.063762         0.099771          0.119183   \n",
      "mock9            0.180453  0.066018         0.166401          0.174770   \n",
      "mock10           0.124902  0.074946         0.078689          0.091027   \n",
      "mock11           0.001731  0.068056         0.001583          0.001854   \n",
      "mock12           0.130262  0.064952         0.090799          0.109451   \n",
      "mock13           0.060501  0.076293         0.043677          0.044796   \n",
      "mock14           0.147051  0.062879         0.120215          0.136190   \n",
      "mock15           0.158986  0.071650         0.114487          0.131294   \n",
      "mock16           0.127455  0.059382         0.118134          0.129081   \n",
      "mock17           0.102482  0.060711         0.077281          0.101205   \n",
      "mock18           0.065774  0.077613         0.046528          0.051057   \n",
      "mock19           0.101590  0.074999         0.070487          0.075357   \n",
      "mock20           0.087590  0.070845         0.063911          0.075252   \n",
      "mock21           0.003585  0.069665         0.002641          0.002510   \n",
      "mock22           0.117390  0.067216         0.081566          0.082965   \n",
      "mock23           0.004818  0.041010         0.002798          0.003017   \n",
      "mock24           0.000070  0.058839         0.000000          0.000000   \n",
      "mock25           0.000017  0.015276         0.000000          0.000000   \n",
      "mock26           0.000017  0.033744         0.000009          0.000009   \n",
      "\n",
      "         m4C-mock3-32000b  \n",
      "#OTU ID                    \n",
      "mock1            0.024492  \n",
      "mock2            0.144760  \n",
      "mock3            0.157797  \n",
      "mock4            0.185009  \n",
      "mock5            0.002002  \n",
      "mock6            0.112170  \n",
      "mock7            0.128163  \n",
      "mock8            0.114033  \n",
      "mock9            0.161540  \n",
      "mock10           0.092225  \n",
      "mock11           0.001460  \n",
      "mock12           0.107553  \n",
      "mock13           0.052649  \n",
      "mock14           0.122165  \n",
      "mock15           0.129291  \n",
      "mock16           0.106075  \n",
      "mock17           0.085264  \n",
      "mock18           0.057720  \n",
      "mock19           0.091446  \n",
      "mock20           0.073390  \n",
      "mock21           0.002921  \n",
      "mock22           0.102901  \n",
      "mock23           0.003174  \n",
      "mock24           0.000017  \n",
      "mock25           0.000017  \n",
      "mock26           0.000044  \n"
     ]
    }
   ],
   "source": [
    "#try to create normalized figure\n",
    "!amptk heatmap -i synmock.figure1.abundances.expected.txt --debug -d tsv -o figure1.normal.pdf --normalize otudb.stats.txt -m heatmap --color YlOrRd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"figure1.pdf\">"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Figure 2 should be about index bleed, so generate a table containing the samples that were run in all runs\n",
    "figureTwo = ['m3-stds','m3-mock3-16000','m3-744-1','m3-744-2','m3-766-1','m3-766-2',\n",
    "             'm4A-stds','m4A-mock3-16000','m4A-744-1','m4A-744-2','m4A-766-1','m4A-766-2',\n",
    "             'm4b-744-1','m4b-744-2','m4b-766-1','m4b-766-2',\n",
    "             'm4C-stds','m4C-mock3-16000','m4C-744-1','m4C-744-2','m4C-766-1','m4C-766-2']\n",
    "#only has the mock OTUs\n",
    "fig2 = df.reindex(columns=figureTwo, index=natsorted(mocks))\n",
    "fig2.to_csv('synmock.figure2.index-bleed.txt', sep='\\t')\n",
    "#has all OTUs, dropping those that are not found in any of these samples\n",
    "fig2b = df.reindex(columns=figureTwo)\n",
    "fig2b = fig2b.loc[~(fig2b==0).all(axis=1)]\n",
    "fig2BB = fig2b.reindex(index=natsorted(fig2b.index))\n",
    "fig2BB.to_csv('synmock.figure2b.index-bleed.txt', sep='\\t')\n",
    "!amptk heatmap -i synmock.figure2.index-bleed.txt -d tsv -o figure2.pdf -m heatmap --annotate --color BuPu"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"figure2.pdf\">"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#now slice the OTU table for each of the runs, Ion3, Miseq4A, Miseq4C, Ion5\n",
    "#load original table, not the name changed version for normal processing\n",
    "df2 = pd.read_csv('otudb.otu_table.txt', sep='\\t')\n",
    "df2.set_index(OTUhead, inplace=True)\n",
    "ion3 = ['m3-301-1', 'm3-301-2', 'm3-500-1', 'm3-500-2', 'm3-712-1', 'm3-712-2', \n",
    "        'm3-736-1', 'm3-736-2', 'm3-744-1', 'm3-744-2', 'm3-755-1', 'm3-755-2',\n",
    "        'm3-757-2', 'm3-766-1', 'm3-766-2', 'm3-mock3-16000']\n",
    "ion3df = df2.reindex(columns=ion3, index=natsorted(df2.index))\n",
    "ion3df.to_csv('ion3.otu_table.txt', sep='\\t')\n",
    "miseq4a = ['m4A-301-1', 'm4A-301-2', 'm4A-500-1', 'm4A-500-2', 'm4A-712-1', \n",
    "           'm4A-712-2', 'm4A-736-1', 'm4A-736-2', 'm4A-744-1', 'm4A-744-2', \n",
    "           'm4A-755-1', 'm4A-755-2', 'm4A-757-1', 'm4A-757-2', 'm4A-766-1', 'm4A-766-2', 'm4A-mock3-16000']\n",
    "miseq4adf = df2.reindex(columns=miseq4a, index=natsorted(df2.index))\n",
    "miseq4adf.to_csv('miseq4a.otu_table.txt', sep='\\t')\n",
    "miseq4c = ['m4C-301-1', 'm4C-301-2', 'm4C-500-1', 'm4C-500-2', 'm4C-712-1', \n",
    "           'm4C-712-2', 'm4C-736-1', 'm4C-736-2', 'm4C-744-1', 'm4C-744-2', \n",
    "           'm4C-755-1', 'm4C-755-2', 'm4C-757-1', 'm4C-757-2', 'm4C-766-1', 'm4C-766-2', 'm4C-mock3-16000']\n",
    "miseq4cdf = df2.reindex(columns=miseq4c, index=natsorted(df2.index))\n",
    "miseq4cdf.to_csv('miseq4c.otu_table.txt', sep='\\t')\n",
    "ion5 = ['m5-301-1', 'm5-301-2', 'm5-500-1', 'm5-500-2', 'm5-712-1', \n",
    "        'm5-712-2', 'm5-736-1', 'm5-736-2', 'm5-744-1', 'm5-744-2', \n",
    "        'm5-755-1', 'm5-755-2', 'm5-757-1', 'm5-757-2', 'm5-766-1', 'm5-766-2', 'm5-SynMock2-8000']\n",
    "ion5df = df2.reindex(columns=ion5, index=natsorted(df2.index))\n",
    "ion5df.to_csv('ion5.otu_table.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[09:30:44 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[09:30:44 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[09:30:44 PM]\u001b[0m: Loading OTU table: ion3.otu_table.txt\n",
      "\u001b[92m[09:30:44 PM]\u001b[0m: OTU table contains 2473 OTUs\n",
      "\u001b[92m[09:30:44 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[09:30:45 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[09:30:45 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[09:30:45 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[09:30:45 PM]\u001b[0m: Index bleed, samples into mock: 0.067999%.\n",
      "\u001b[92m[09:30:45 PM]\u001b[0m: Will use value of 0.100000% for index-bleed OTU filtering.\n",
      "\u001b[92m[09:30:45 PM]\u001b[0m: Auto subtract filter set to 26\n",
      "\u001b[92m[09:30:45 PM]\u001b[0m: Subtracting 26 from OTU table\n",
      "\u001b[92m[09:30:45 PM]\u001b[0m: Filtering OTU table down to 397 OTUs\n",
      "\u001b[92m[09:30:45 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      ion3.stats.txt\n",
      "Sorted OTU table:     ion3.sorted.txt\n",
      "Normalized/filter:    ion3.normalized.txt\n",
      "Final Binary table:   ion3.final.binary.txt\n",
      "Final OTU table:      ion3.final.txt\n",
      "Filtered OTUs:        ion3.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f ion3.filtered.otus.fa -i ion3.final.txt -m mapping_file.txt -d ITS2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#now run filtering on these data\n",
    "!amptk filter -i ion3.otu_table.txt -f otudb.cluster.otus.fa -o ion3 -b m3-mock3-16000 -m mock3 -c in -s auto"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2473 16 1654 15\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[01:04:13 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[01:04:14 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[01:04:14 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[01:04:14 PM]\u001b[0m: 2,480 OTUs\n",
      "\u001b[92m[01:04:14 PM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[01:04:14 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[01:04:34 PM]\u001b[0m: Taxonomy finished: otudb.cluster.otus.taxonomy.txt\n",
      "\u001b[92m[01:04:34 PM]\u001b[0m: Classic OTU table with taxonomy: ion3_raw.otu_table.taxonomy.txt\n",
      "\u001b[92m[01:04:37 PM]\u001b[0m: BIOM OTU table created: ion3_raw.biom\n",
      "\u001b[92m[01:04:37 PM]\u001b[0m: OTUs with taxonomy: ion3_raw.otus.taxonomy.fa\n",
      "\u001b[92m[01:04:37 PM]\u001b[0m: OTU phylogeny: ion3_raw.tree.phy\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#get biom of unfiltered data\n",
    "#first trim sorted OTU table, will remove last column which is mock\n",
    "#!gsed -r 's/\\s+\\S+$//' ion3.sorted.txt > ion3.sorted.nomock.txt\n",
    "\n",
    "import pandas as pd\n",
    "from natsort import natsorted\n",
    "#pull out header for indexing later\n",
    "with open('ion3.sorted.txt', 'rU') as f:\n",
    "    first_line = f.readline()\n",
    "    OTUhead = first_line.split('\\t')[0]\n",
    "    \n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "ion3sorted = pd.read_csv('ion3.sorted.txt', sep='\\t')\n",
    "ion3sorted.set_index(OTUhead, inplace=True)\n",
    "\n",
    "#now we want to drop last column\n",
    "ion3 = ['m3-301-1', 'm3-301-2', 'm3-500-1', 'm3-500-2', 'm3-712-1', 'm3-712-2', \n",
    "        'm3-736-1', 'm3-736-2', 'm3-744-1', 'm3-744-2', 'm3-755-1', 'm3-755-2',\n",
    "        'm3-757-2', 'm3-766-1', 'm3-766-2']\n",
    "ion3cleaned = ion3sorted.reindex(columns=ion3, index=natsorted(ion3sorted.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "fs = ion3cleaned.sum(axis=0)\n",
    "filtered = pd.DataFrame(ion3cleaned, columns=fs.index)\n",
    "filt2 = filtered.loc[(filtered != 0).any(1)]\n",
    "print len(ion3sorted), len(ion3sorted.columns.values), len(filt2), len(filt2.columns.values)\n",
    "filt2.to_csv('ion3.sorted.nomock.txt', sep='\\t')\n",
    "\n",
    "#now convert to biom\n",
    "!amptk taxonomy -f otudb.cluster.otus.fa -o ion3_raw -i ion3.sorted.nomock.txt -d ITS2 -m ion3.mapping_file.txt -t otudb.cluster.otus.taxonomy.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-2\n",
      "[1] ‘1.19.1’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘1.8.4’\n",
      "[1] ‘2.4.2’\n",
      "There were 22 warnings (use warnings() to see them)\n",
      "Square root transformation\n",
      "Wisconsin double standardization\n",
      "Run 0 stress 0.1561485 \n",
      "Run 1 stress 0.1601276 \n",
      "Run 2 stress 0.1617538 \n",
      "Run 3 stress 0.1638083 \n",
      "Run 4 stress 0.1720631 \n",
      "Run 5 stress 0.1561485 \n",
      "... Procrustes: rmse 0.0002629328  max resid 0.0005522638 \n",
      "... Similar to previous best\n",
      "Run 6 stress 0.1640377 \n",
      "Run 7 stress 0.1601277 \n",
      "Run 8 stress 0.1720647 \n",
      "Run 9 stress 0.1658563 \n",
      "Run 10 stress 0.1561486 \n",
      "... Procrustes: rmse 0.0004754189  max resid 0.0009814214 \n",
      "... Similar to previous best\n",
      "Run 11 stress 0.1561483 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0002990067  max resid 0.0006326395 \n",
      "... Similar to previous best\n",
      "Run 12 stress 0.1771299 \n",
      "Run 13 stress 0.1601276 \n",
      "Run 14 stress 0.1771307 \n",
      "Run 15 stress 0.1561483 \n",
      "... New best solution\n",
      "... Procrustes: rmse 8.583865e-05  max resid 0.0001644541 \n",
      "... Similar to previous best\n",
      "Run 16 stress 0.1561485 \n",
      "... Procrustes: rmse 0.0002284932  max resid 0.0004560382 \n",
      "... Similar to previous best\n",
      "Run 17 stress 0.1601276 \n",
      "Run 18 stress 0.1601278 \n",
      "Run 19 stress 0.1633748 \n",
      "Run 20 stress 0.1624509 \n",
      "*** Solution reached\n",
      "Warning message:\n",
      "In postMDS(out$points, dis, plot = max(0, plot - 1), ...) :\n",
      "  skipping half-change scaling: too few points below threshold\n",
      "Run 0 stress 0.07661907 \n",
      "Run 1 stress 0.1727885 \n",
      "Run 2 stress 0.1111457 \n",
      "Run 3 stress 0.1665964 \n",
      "Run 4 stress 0.149725 \n",
      "Run 5 stress 0.1751834 \n",
      "Run 6 stress 0.1474644 \n",
      "Run 7 stress 0.1179652 \n",
      "Run 8 stress 0.1210712 \n",
      "Run 9 stress 0.168182 \n",
      "Run 10 stress 0.1561774 \n",
      "Run 11 stress 0.2088535 \n",
      "Run 12 stress 0.1502892 \n",
      "Run 13 stress 0.1261844 \n",
      "Run 14 stress 0.1454083 \n",
      "Run 15 stress 0.1449507 \n",
      "Run 16 stress 0.1636221 \n",
      "Run 17 stress 0.05286263 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1505901  max resid 0.3397553 \n",
      "Run 18 stress 0.1807622 \n",
      "Run 19 stress 0.05289515 \n",
      "... Procrustes: rmse 0.002917498  max resid 0.007781903 \n",
      "... Similar to previous best\n",
      "Run 20 stress 0.05286284 \n",
      "... Procrustes: rmse 0.003638329  max resid 0.008129529 \n",
      "... Similar to previous best\n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 30 rows containing missing values (geom_errorbar). \n",
      "2: Removed 30 rows containing missing values (geom_errorbar). \n"
     ]
    }
   ],
   "source": [
    "#run the phyloseq scripts\n",
    "!/Users/jon/amptk/util/phyloseq_graphs.R ion3_raw.biom ion3_raw.tree.phy ion3_raw"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Treatment adonis using raupcrick\n",
    "```\n",
    " Call:\n",
    "adonis(formula = rp ~ metadata[[treatments[y]]], permutations = 9999) \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1   0.37159 0.37159  3.1052 0.19281 0.1173\n",
    "Residuals                 13   1.55567 0.11967         0.80719       \n",
    "Total                     14   1.92726                 1.00000       \n",
    "```\n",
    "###NMDS using raup\n",
    "![alt text](ion3_raw.Treatment.raup.pdf)\n",
    "###Treatment adonis using braycurtis\n",
    "```\n",
    "Call:\n",
    "adonis(formula = voturaw ~ metadata[[treatments[y]]], permutations = 9999,      method = \"bray\") \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1    0.4609 0.46093  1.0299 0.07341 0.4272\n",
    "Residuals                 13    5.8178 0.44752         0.92659       \n",
    "Total                     14    6.2787                 1.00000        \n",
    "```\n",
    "###NMDS using braycurtis\n",
    "![alt text](ion3_raw.Treatment.braycurtis.pdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[09:59:49 AM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[09:59:49 AM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[09:59:49 AM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[09:59:49 AM]\u001b[0m: 397 OTUs\n",
      "\u001b[92m[09:59:49 AM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[09:59:49 AM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[09:59:50 AM]\u001b[0m: Taxonomy finished: otudb.cluster.otus.taxonomy.txt\n",
      "\u001b[92m[09:59:50 AM]\u001b[0m: Classic OTU table with taxonomy: ion3.otu_table.taxonomy.txt\n",
      "\u001b[92m[09:59:52 AM]\u001b[0m: BIOM OTU table created: ion3.biom\n",
      "\u001b[92m[09:59:52 AM]\u001b[0m: OTUs with taxonomy: ion3.otus.taxonomy.fa\n",
      "\u001b[92m[09:59:52 AM]\u001b[0m: OTU phylogeny: ion3.tree.phy\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#get biom of filtered data\n",
    "!amptk taxonomy -f ion3.filtered.otus.fa -i ion3.final.txt -d ITS2 -m ion3.mapping_file.txt -t otudb.cluster.otus.taxonomy.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-2\n",
      "[1] ‘1.19.1’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘1.8.4’\n",
      "[1] ‘2.4.2’\n",
      "Warning messages:\n",
      "1: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "2: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "Square root transformation\n",
      "Wisconsin double standardization\n",
      "Run 0 stress 0.1824396 \n",
      "Run 1 stress 0.2047899 \n",
      "Run 2 stress 0.2014881 \n",
      "Run 3 stress 0.2057218 \n",
      "Run 4 stress 0.2025887 \n",
      "Run 5 stress 0.1824394 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0003814994  max resid 0.0008967817 \n",
      "... Similar to previous best\n",
      "Run 6 stress 0.2274207 \n",
      "Run 7 stress 0.2057195 \n",
      "Run 8 stress 0.2042308 \n",
      "Run 9 stress 0.2043601 \n",
      "Run 10 stress 0.1811953 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1910247  max resid 0.6364945 \n",
      "Run 11 stress 0.198515 \n",
      "Run 12 stress 0.2047901 \n",
      "Run 13 stress 0.1811953 \n",
      "... New best solution\n",
      "... Procrustes: rmse 6.469609e-05  max resid 0.000149608 \n",
      "... Similar to previous best\n",
      "Run 14 stress 0.1926884 \n",
      "Run 15 stress 0.3547019 \n",
      "Run 16 stress 0.2274219 \n",
      "Run 17 stress 0.3473573 \n",
      "Run 18 stress 0.22044 \n",
      "Run 19 stress 0.2041265 \n",
      "Run 20 stress 0.1984717 \n",
      "*** Solution reached\n",
      "Warning message:\n",
      "In postMDS(out$points, dis, plot = max(0, plot - 1), ...) :\n",
      "  skipping half-change scaling: too few points below threshold\n",
      "Run 0 stress 0.1976074 \n",
      "Run 1 stress 0.190196 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1815964  max resid 0.4051678 \n",
      "Run 2 stress 0.1874114 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.07049346  max resid 0.2278205 \n",
      "Run 3 stress 0.214824 \n",
      "Run 4 stress 0.1874129 \n",
      "... Procrustes: rmse 0.0003140676  max resid 0.0007033366 \n",
      "... Similar to previous best\n",
      "Run 5 stress 0.1943383 \n",
      "Run 6 stress 0.2243163 \n",
      "Run 7 stress 0.2397072 \n",
      "Run 8 stress 0.1934095 \n",
      "Run 9 stress 0.2184773 \n",
      "Run 10 stress 0.1943383 \n",
      "Run 11 stress 0.1874109 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.001792779  max resid 0.003501297 \n",
      "... Similar to previous best\n",
      "Run 12 stress 0.1874114 \n",
      "... Procrustes: rmse 0.0017054  max resid 0.003311927 \n",
      "... Similar to previous best\n",
      "Run 13 stress 0.1976088 \n",
      "Run 14 stress 0.1874142 \n",
      "... Procrustes: rmse 0.001424903  max resid 0.003056358 \n",
      "... Similar to previous best\n",
      "Run 15 stress 0.2115909 \n",
      "Run 16 stress 0.2020111 \n",
      "Run 17 stress 0.1943383 \n",
      "Run 18 stress 0.2063595 \n",
      "Run 19 stress 0.1934131 \n",
      "Run 20 stress 0.1999629 \n",
      "*** Solution reached\n",
      "Warning message:\n",
      "In postMDS(out$points, dis, plot = max(0, plot - 1), ...) :\n",
      "  skipping half-change scaling: too few points below threshold\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: In estimate_richness(physeq, split = TRUE, measures = measures) :\n",
      "  The data you have provided does not have\n",
      "any singletons. This is highly suspicious. Results of richness\n",
      "estimates (for example) are probably unreliable, or wrong, if you have already\n",
      "trimmed low-abundance taxa from the data.\n",
      "\n",
      "We recommended that you find the un-trimmed data and retry.\n",
      "2: Removed 30 rows containing missing values (geom_errorbar). \n",
      "3: In estimate_richness(physeq, split = TRUE, measures = measures) :\n",
      "  The data you have provided does not have\n",
      "any singletons. This is highly suspicious. Results of richness\n",
      "estimates (for example) are probably unreliable, or wrong, if you have already\n",
      "trimmed low-abundance taxa from the data.\n",
      "\n",
      "We recommended that you find the un-trimmed data and retry.\n",
      "4: Removed 30 rows containing missing values (geom_errorbar). \n"
     ]
    }
   ],
   "source": [
    "#run the phyloseq scripts\n",
    "!/Users/jon/amptk/util/phyloseq_graphs.R ion3.biom ion3.tree.phy ion3_filtered"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Treatment adonis using raupcrick\n",
    "```\n",
    "Call:\n",
    "adonis(formula = rp ~ metadata[[treatments[y]]], permutations = 9999) \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1   0.45024 0.45024  2.4021 0.15596 0.1156\n",
    "Residuals                 13   2.43661 0.18743         0.84404       \n",
    "Total                     14   2.88685                 1.00000       \n",
    "\n",
    "```\n",
    "###NMDS using jaccard\n",
    "![alt text](ion3_filtered.Treatment.jaccard.pdf)\n",
    "###Treatment adonis using braycurtis\n",
    "```\n",
    "Call:\n",
    "adonis(formula = voturaw ~ metadata[[treatments[y]]], permutations = 9999,      method = \"bray\") \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1    0.4617 0.46174  1.0246 0.07306 0.4396\n",
    "Residuals                 13    5.8587 0.45067         0.92694       \n",
    "Total                     14    6.3205                 1.00000       \n",
    "```\n",
    "###NMDS using braycurtis\n",
    "![alt text](ion3_filtered.Treatment.braycurtis.pdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[02:09:11 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[02:09:11 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[02:09:11 PM]\u001b[0m: Loading OTU table: ion3.otu_table.txt\n",
      "\u001b[92m[02:09:11 PM]\u001b[0m: OTU table contains 2473 OTUs\n",
      "\u001b[92m[02:09:11 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[02:09:11 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[02:09:11 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[02:09:11 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[02:09:11 PM]\u001b[0m: Index bleed, samples into mock: 0.067999%.\n",
      "\u001b[92m[02:09:11 PM]\u001b[0m: Will use value of 0.100000% for index-bleed OTU filtering.\n",
      "\u001b[92m[02:09:12 PM]\u001b[0m: Filtering OTU table down to 1353 OTUs\n",
      "\u001b[92m[02:09:12 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      ion3_nosub.stats.txt\n",
      "Sorted OTU table:     ion3_nosub.sorted.txt\n",
      "Normalized/filter:    ion3_nosub.normalized.txt\n",
      "Final Binary table:   ion3_nosub.final.binary.txt\n",
      "Final OTU table:      ion3_nosub.final.txt\n",
      "Filtered OTUs:        ion3_nosub.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f ion3_nosub.filtered.otus.fa -i ion3_nosub.final.txt -m mapping_file.txt -d ITS2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#now run filtering without subtraction\n",
    "!amptk filter -i ion3.otu_table.txt -f otudb.cluster.otus.fa -o ion3_nosub -b m3-mock3-16000 -m mock3 -c in"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[02:14:03 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[02:14:03 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[02:14:03 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[02:14:03 PM]\u001b[0m: 1,353 OTUs\n",
      "\u001b[92m[02:14:03 PM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[02:14:03 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[02:14:07 PM]\u001b[0m: Taxonomy finished: otudb.cluster.otus.taxonomy.txt\n",
      "\u001b[92m[02:14:07 PM]\u001b[0m: Classic OTU table with taxonomy: ion3_nosub.otu_table.taxonomy.txt\n",
      "\u001b[92m[02:14:08 PM]\u001b[0m: BIOM OTU table created: ion3_nosub.biom\n",
      "\u001b[92m[02:14:08 PM]\u001b[0m: OTUs with taxonomy: ion3_nosub.otus.taxonomy.fa\n",
      "\u001b[92m[02:14:08 PM]\u001b[0m: OTU phylogeny: ion3_nosub.tree.phy\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#create no subtraction biom file\n",
    "!amptk taxonomy -i ion3_nosub.final.txt -f ion3_nosub.filtered.otus.fa -t otudb.cluster.otus.taxonomy.txt -d ITS2 -m ion3.mapping_file.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-2\n",
      "[1] ‘1.19.1’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘1.8.4’\n",
      "[1] ‘2.4.2’\n",
      "Warning messages:\n",
      "1: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "2: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "3: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "4: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "5: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "6: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "Square root transformation\n",
      "Wisconsin double standardization\n",
      "Run 0 stress 0.1442957 \n",
      "Run 1 stress 0.1816228 \n",
      "Run 2 stress 0.1442957 \n",
      "... Procrustes: rmse 4.648106e-05  max resid 9.667871e-05 \n",
      "... Similar to previous best\n",
      "Run 3 stress 0.1678744 \n",
      "Run 4 stress 0.1696964 \n",
      "Run 5 stress 0.1684179 \n",
      "Run 6 stress 0.1441266 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.01698781  max resid 0.04674827 \n",
      "Run 7 stress 0.1445979 \n",
      "... Procrustes: rmse 0.02156739  max resid 0.06207322 \n",
      "Run 8 stress 0.1617605 \n",
      "Run 9 stress 0.2015939 \n",
      "Run 10 stress 0.1624567 \n",
      "Run 11 stress 0.1637265 \n",
      "Run 12 stress 0.1674994 \n",
      "Run 13 stress 0.1602755 \n",
      "Run 14 stress 0.1624532 \n",
      "Run 15 stress 0.1624535 \n",
      "Run 16 stress 0.1684185 \n",
      "Run 17 stress 0.1733977 \n",
      "Run 18 stress 0.1636648 \n",
      "Run 19 stress 0.1617544 \n",
      "Run 20 stress 0.1441266 \n",
      "... Procrustes: rmse 5.93947e-05  max resid 0.0001605118 \n",
      "... Similar to previous best\n",
      "*** Solution reached\n",
      "Run 0 stress 0.07100021 \n",
      "Run 1 stress 0.1277355 \n",
      "Run 2 stress 0.1451166 \n",
      "Run 3 stress 0.1532003 \n",
      "Run 4 stress 0.09983277 \n",
      "Run 5 stress 0.1755584 \n",
      "Run 6 stress 0.1712634 \n",
      "Run 7 stress 0.1660745 \n",
      "Run 8 stress 0.0998325 \n",
      "Run 9 stress 0.195861 \n",
      "Run 10 stress 0.1927192 \n",
      "Run 11 stress 0.1033029 \n",
      "Run 12 stress 0.1927148 \n",
      "Run 13 stress 0.1277349 \n",
      "Run 14 stress 0.1894482 \n",
      "Run 15 stress 0.1592287 \n",
      "Run 16 stress 0.1141248 \n",
      "Run 17 stress 0.09639017 \n",
      "Run 18 stress 0.1939167 \n",
      "Run 19 stress 0.1852363 \n",
      "Run 20 stress 0.1842674 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    18: stress ratio > sratmax\n",
      "     2: scale factor of the gradient < sfgrmin\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 30 rows containing missing values (geom_errorbar). \n",
      "2: Removed 30 rows containing missing values (geom_errorbar). \n"
     ]
    }
   ],
   "source": [
    "#run the phyloseq scripts\n",
    "!/Users/jon/amptk/util/phyloseq_graphs.R ion3_nosub.biom ion3_nosub.tree.phy ion3_nosub"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Treatment adonis using raupcrick\n",
    "```\n",
    "Call:\n",
    "adonis(formula = rp ~ metadata[[treatments[y]]], permutations = 9999) \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1   0.40146 0.40146  3.0324 0.18914 0.1058\n",
    "Residuals                 13   1.72107 0.13239         0.81086       \n",
    "Total                     14   2.12252                 1.00000    \n",
    "```\n",
    "###NMDS using Raup\n",
    "![alt text](ion3_nosub.Treatment.raup.pdf)\n",
    "###Treatment adonis using braycurtis\n",
    "```\n",
    "Call:\n",
    "adonis(formula = voturaw ~ metadata[[treatments[y]]], permutations = 9999,      method = \"bray\") \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1    0.4601 0.46011  1.0269 0.07321 0.4411\n",
    "Residuals                 13    5.8249 0.44807         0.92679       \n",
    "Total                     14    6.2850                 1.00000\n",
    "```\n",
    "###NMDS using braycurtis\n",
    "![alt text](ion3_nosub.Treatment.braycurtis.pdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[10:03:37 AM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[10:03:37 AM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[10:03:37 AM]\u001b[0m: Loading OTU table: miseq4a.otu_table.txt\n",
      "\u001b[92m[10:03:37 AM]\u001b[0m: OTU table contains 2473 OTUs\n",
      "\u001b[92m[10:03:37 AM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[10:03:38 AM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[10:03:38 AM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[10:03:38 AM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[10:03:38 AM]\u001b[0m: Index bleed, samples into mock: 0.054002%.\n",
      "\u001b[92m[10:03:38 AM]\u001b[0m: Will use value of 0.100000% for index-bleed OTU filtering.\n",
      "\u001b[92m[10:03:38 AM]\u001b[0m: Auto subtract filter set to 25\n",
      "\u001b[92m[10:03:38 AM]\u001b[0m: Subtracting 25 from OTU table\n",
      "\u001b[92m[10:03:38 AM]\u001b[0m: Filtering OTU table down to 446 OTUs\n",
      "\u001b[92m[10:03:38 AM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      miseq4a.stats.txt\n",
      "Sorted OTU table:     miseq4a.sorted.txt\n",
      "Normalized/filter:    miseq4a.normalized.txt\n",
      "Final Binary table:   miseq4a.final.binary.txt\n",
      "Final OTU table:      miseq4a.final.txt\n",
      "Filtered OTUs:        miseq4a.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f miseq4a.filtered.otus.fa -i miseq4a.final.txt -m mapping_file.txt -d ITS2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#now run MiseqA dataset, first filter\n",
    "!amptk filter -i miseq4a.otu_table.txt -f otudb.cluster.otus.fa -o miseq4a -b m4A-mock3-16000 -m mock3 -c in -s auto"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2473 17 2179 16\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[01:13:44 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[01:13:45 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[01:13:45 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[01:13:45 PM]\u001b[0m: 2,480 OTUs\n",
      "\u001b[92m[01:13:45 PM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[01:13:45 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[01:13:56 PM]\u001b[0m: Taxonomy finished: otudb.cluster.otus.taxonomy.txt\n",
      "\u001b[92m[01:13:56 PM]\u001b[0m: Classic OTU table with taxonomy: miseq4a_raw.otu_table.taxonomy.txt\n",
      "\u001b[92m[01:13:58 PM]\u001b[0m: BIOM OTU table created: miseq4a_raw.biom\n",
      "\u001b[92m[01:13:58 PM]\u001b[0m: OTUs with taxonomy: miseq4a_raw.otus.taxonomy.fa\n",
      "\u001b[92m[01:13:58 PM]\u001b[0m: OTU phylogeny: miseq4a_raw.tree.phy\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "from natsort import natsorted\n",
    "#pull out header for indexing later\n",
    "with open('miseq4a.sorted.txt', 'rU') as f:\n",
    "    first_line = f.readline()\n",
    "    OTUhead = first_line.split('\\t')[0]\n",
    "    \n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "dfsorted = pd.read_csv('miseq4a.sorted.txt', sep='\\t')\n",
    "dfsorted.set_index(OTUhead, inplace=True)\n",
    "\n",
    "#now we want to drop last column\n",
    "header = ['m4A-301-1', 'm4A-301-2', 'm4A-500-1', 'm4A-500-2', 'm4A-712-1', \n",
    "           'm4A-712-2', 'm4A-736-1', 'm4A-736-2', 'm4A-744-1', 'm4A-744-2', \n",
    "           'm4A-755-1', 'm4A-755-2', 'm4A-757-1', 'm4A-757-2', 'm4A-766-1', 'm4A-766-2']\n",
    "dfcleaned = dfsorted.reindex(columns=header, index=natsorted(dfsorted.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "fs = dfcleaned.sum(axis=0)\n",
    "filtered = pd.DataFrame(dfcleaned, columns=fs.index)\n",
    "filt2 = filtered.loc[(filtered != 0).any(1)]\n",
    "print len(dfsorted), len(dfsorted.columns.values), len(filt2), len(filt2.columns.values)\n",
    "filt2.to_csv('miseq4a.sorted.nomock.txt', sep='\\t')\n",
    "\n",
    "#now convert to biom\n",
    "!amptk taxonomy -f otudb.cluster.otus.fa -o miseq4a_raw -i miseq4a.sorted.nomock.txt -d ITS2 -m m4a.mapping_file.txt -t otudb.cluster.otus.taxonomy.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-2\n",
      "[1] ‘1.19.1’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘1.8.4’\n",
      "[1] ‘2.4.2’\n",
      "There were 27 warnings (use warnings() to see them)\n",
      "Square root transformation\n",
      "Wisconsin double standardization\n",
      "Run 0 stress 0.1571083 \n",
      "Run 1 stress 0.1506067 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1073115  max resid 0.3556422 \n",
      "Run 2 stress 0.1544389 \n",
      "Run 3 stress 0.1581899 \n",
      "Run 4 stress 0.1544389 \n",
      "Run 5 stress 0.154439 \n",
      "Run 6 stress 0.1506066 \n",
      "... New best solution\n",
      "... Procrustes: rmse 2.99924e-05  max resid 6.735118e-05 \n",
      "... Similar to previous best\n",
      "Run 7 stress 0.154439 \n",
      "Run 8 stress 0.1571084 \n",
      "Run 9 stress 0.1581898 \n",
      "Run 10 stress 0.1715844 \n",
      "Run 11 stress 0.1502516 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1841402  max resid 0.3639254 \n",
      "Run 12 stress 0.1567063 \n",
      "Run 13 stress 0.1567065 \n",
      "Run 14 stress 0.1567066 \n",
      "Run 15 stress 0.1502505 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0003621347  max resid 0.001024676 \n",
      "... Similar to previous best\n",
      "Run 16 stress 0.1625812 \n",
      "Run 17 stress 0.1506066 \n",
      "... Procrustes: rmse 0.1841722  max resid 0.3609517 \n",
      "Run 18 stress 0.1985202 \n",
      "Run 19 stress 0.2260002 \n",
      "Run 20 stress 0.1506066 \n",
      "... Procrustes: rmse 0.184173  max resid 0.3609523 \n",
      "*** Solution reached\n",
      "Warning message:\n",
      "In postMDS(out$points, dis, plot = max(0, plot - 1), ...) :\n",
      "  skipping half-change scaling: too few points below threshold\n",
      "Run 0 stress 0 \n",
      "Run 1 stress 0 \n",
      "... Procrustes: rmse 0.1677546  max resid 0.3257493 \n",
      "Run 2 stress 9.648007e-05 \n",
      "... Procrustes: rmse 0.2063643  max resid 0.4538865 \n",
      "Run 3 stress 4.822269e-05 \n",
      "... Procrustes: rmse 0.1593487  max resid 0.2842769 \n",
      "Run 4 stress 9.251842e-05 \n",
      "... Procrustes: rmse 0.2077653  max resid 0.5029932 \n",
      "Run 5 stress 9.56381e-05 \n",
      "... Procrustes: rmse 0.2184664  max resid 0.5361894 \n",
      "Run 6 stress 5.614732e-05 \n",
      "... Procrustes: rmse 0.2138654  max resid 0.4960524 \n",
      "Run 7 stress 9.850491e-05 \n",
      "... Procrustes: rmse 0.2166644  max resid 0.4893868 \n",
      "Run 8 stress 8.700799e-05 \n",
      "... Procrustes: rmse 0.1960203  max resid 0.4071967 \n",
      "Run 9 stress 6.435472e-05 \n",
      "... Procrustes: rmse 0.2110644  max resid 0.5141616 \n",
      "Run 10 stress 9.675426e-05 \n",
      "... Procrustes: rmse 0.2025424  max resid 0.5179274 \n",
      "Run 11 stress 8.301004e-05 \n",
      "... Procrustes: rmse 0.2188295  max resid 0.4846465 \n",
      "Run 12 stress 9.422406e-05 \n",
      "... Procrustes: rmse 0.2083507  max resid 0.4882009 \n",
      "Run 13 stress 8.035598e-05 \n",
      "... Procrustes: rmse 0.1884617  max resid 0.4439826 \n",
      "Run 14 stress 5.768641e-05 \n",
      "... Procrustes: rmse 0.2001567  max resid 0.4900689 \n",
      "Run 15 stress 9.399801e-05 \n",
      "... Procrustes: rmse 0.2151354  max resid 0.5436717 \n",
      "Run 16 stress 9.309904e-05 \n",
      "... Procrustes: rmse 0.2155624  max resid 0.4756161 \n",
      "Run 17 stress 9.580221e-05 \n",
      "... Procrustes: rmse 0.215524  max resid 0.5086177 \n",
      "Run 18 stress 8.257985e-05 \n",
      "... Procrustes: rmse 0.18015  max resid 0.3843424 \n",
      "Run 19 stress 9.389642e-05 \n",
      "... Procrustes: rmse 0.2220803  max resid 0.5366571 \n",
      "Run 20 stress 9.758251e-05 \n",
      "... Procrustes: rmse 0.2068963  max resid 0.4826969 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress < smin\n",
      "Warning message:\n",
      "In metaMDS(veganifyOTU(physeq), distance, ...) :\n",
      "  Stress is (nearly) zero - you may have insufficient data\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 32 rows containing missing values (geom_errorbar). \n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n"
     ]
    }
   ],
   "source": [
    "#run phyloseq\n",
    "!/Users/jon/amptk/util/phyloseq_graphs.R miseq4a_raw.biom miseq4a_raw.tree.phy m4a_raw"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Treatment adonis using raupcrick\n",
    "```\n",
    "Call:\n",
    "adonis(formula = rp ~ metadata[[treatments[y]]], permutations = 9999) \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs  MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1   0.30725 0.307245  4.0315 0.22358 0.1106\n",
    "Residuals                 14   1.06697 0.076212         0.77642       \n",
    "Total                     15   1.37421                  1.00000     \n",
    "```\n",
    "###NMDS using Raup\n",
    "![alt text](m4a_raw.Treatment.raup.pdf)\n",
    "###Treatment adonis using braycurtis\n",
    "```\n",
    "Call:\n",
    "adonis(formula = voturaw ~ metadata[[treatments[y]]], permutations = 9999,      method = \"bray\") \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1    0.4526 0.45255   1.016 0.06766 0.4572\n",
    "Residuals                 14    6.2357 0.44541         0.93234       \n",
    "Total                     15    6.6882                 1.00000        \n",
    "```\n",
    "###NMDS using braycurtis\n",
    "![alt text](m4a_raw.Treatment.braycurtis.pdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[10:08:28 AM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[10:08:29 AM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[10:08:29 AM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[10:08:29 AM]\u001b[0m: 446 OTUs\n",
      "\u001b[92m[10:08:29 AM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[10:08:29 AM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[10:08:30 AM]\u001b[0m: Taxonomy finished: otudb.cluster.otus.taxonomy.txt\n",
      "\u001b[92m[10:08:30 AM]\u001b[0m: Classic OTU table with taxonomy: miseq4a.otu_table.taxonomy.txt\n",
      "\u001b[92m[10:08:31 AM]\u001b[0m: BIOM OTU table created: miseq4a.biom\n",
      "\u001b[92m[10:08:31 AM]\u001b[0m: OTUs with taxonomy: miseq4a.otus.taxonomy.fa\n",
      "\u001b[92m[10:08:31 AM]\u001b[0m: OTU phylogeny: miseq4a.tree.phy\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#get biom of filtered data\n",
    "!amptk taxonomy -f miseq4a.filtered.otus.fa -i miseq4a.final.txt -d ITS2 -m m4a.mapping_file.txt -t otudb.cluster.otus.taxonomy.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-2\n",
      "[1] ‘1.19.1’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘1.8.4’\n",
      "[1] ‘2.4.2’\n",
      "Warning messages:\n",
      "1: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "2: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "3: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "Square root transformation\n",
      "Wisconsin double standardization\n",
      "Run 0 stress 0.2111013 \n",
      "Run 1 stress 0.2124962 \n",
      "Run 2 stress 0.2458501 \n",
      "Run 3 stress 0.2244359 \n",
      "Run 4 stress 0.2076407 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1588691  max resid 0.3446601 \n",
      "Run 5 stress 0.2449464 \n",
      "Run 6 stress 0.2182043 \n",
      "Run 7 stress 0.2100472 \n",
      "Run 8 stress 0.2111013 \n",
      "Run 9 stress 0.2507159 \n",
      "Run 10 stress 0.2076404 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0001285247  max resid 0.0002424704 \n",
      "... Similar to previous best\n",
      "Run 11 stress 0.2204901 \n",
      "Run 12 stress 0.2472935 \n",
      "Run 13 stress 0.2091359 \n",
      "Run 14 stress 0.2484377 \n",
      "Run 15 stress 0.2141733 \n",
      "Run 16 stress 0.2217016 \n",
      "Run 17 stress 0.215054 \n",
      "Run 18 stress 0.2157181 \n",
      "Run 19 stress 0.209077 \n",
      "Run 20 stress 0.2100471 \n",
      "*** Solution reached\n",
      "Warning message:\n",
      "In postMDS(out$points, dis, plot = max(0, plot - 1), ...) :\n",
      "  skipping half-change scaling: too few points below threshold\n",
      "Run 0 stress 0.1972144 \n",
      "Run 1 stress 0.193817 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1051851  max resid 0.2842083 \n",
      "Run 2 stress 0.2156074 \n",
      "Run 3 stress 0.2231654 \n",
      "Run 4 stress 0.2073469 \n",
      "Run 5 stress 0.2044928 \n",
      "Run 6 stress 0.2044924 \n",
      "Run 7 stress 0.1938175 \n",
      "... Procrustes: rmse 0.0003751651  max resid 0.0007882545 \n",
      "... Similar to previous best\n",
      "Run 8 stress 0.1995771 \n",
      "Run 9 stress 0.2930237 \n",
      "Run 10 stress 0.2114362 \n",
      "Run 11 stress 0.1995771 \n",
      "Run 12 stress 0.2576823 \n",
      "Run 13 stress 0.2063703 \n",
      "Run 14 stress 0.1938171 \n",
      "... Procrustes: rmse 0.0001390791  max resid 0.0003253783 \n",
      "... Similar to previous best\n",
      "Run 15 stress 0.2326785 \n",
      "Run 16 stress 0.2287837 \n",
      "Run 17 stress 0.2268246 \n",
      "Run 18 stress 0.2386985 \n",
      "Run 19 stress 0.2267231 \n",
      "Run 20 stress 0.2964626 \n",
      "*** Solution reached\n",
      "Warning message:\n",
      "In postMDS(out$points, dis, plot = max(0, plot - 1), ...) :\n",
      "  skipping half-change scaling: too few points below threshold\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: In estimate_richness(physeq, split = TRUE, measures = measures) :\n",
      "  The data you have provided does not have\n",
      "any singletons. This is highly suspicious. Results of richness\n",
      "estimates (for example) are probably unreliable, or wrong, if you have already\n",
      "trimmed low-abundance taxa from the data.\n",
      "\n",
      "We recommended that you find the un-trimmed data and retry.\n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n",
      "3: In estimate_richness(physeq, split = TRUE, measures = measures) :\n",
      "  The data you have provided does not have\n",
      "any singletons. This is highly suspicious. Results of richness\n",
      "estimates (for example) are probably unreliable, or wrong, if you have already\n",
      "trimmed low-abundance taxa from the data.\n",
      "\n",
      "We recommended that you find the un-trimmed data and retry.\n",
      "4: Removed 32 rows containing missing values (geom_errorbar). \n"
     ]
    }
   ],
   "source": [
    "#run phyloseq\n",
    "!/Users/jon/amptk/util/phyloseq_graphs.R miseq4a.biom miseq4a.tree.phy miseq4a_filtered"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Treatment adonis using raupcrick\n",
    "```\n",
    "Call:\n",
    "adonis(formula = rp ~ metadata[[treatments[y]]], permutations = 9999) \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model     R2 Pr(>F)  \n",
    "metadata[[treatments[y]]]  1   0.55147 0.55147  3.0899 0.1808 0.0564 .\n",
    "Residuals                 14   2.49866 0.17848         0.8192         \n",
    "Total                     15   3.05013                 1.0000         \n",
    "---\n",
    "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1        \n",
    "\n",
    "```\n",
    "###NMDS using raup\n",
    "![alt text](miseq4a_filtered.Treatment.raup.pdf)\n",
    "###Treatment adonis using braycurtis\n",
    "```\n",
    "Call:\n",
    "adonis(formula = voturaw ~ metadata[[treatments[y]]], permutations = 9999,      method = \"bray\") \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1    0.4535 0.45350  1.0097 0.06727  0.466\n",
    "Residuals                 14    6.2882 0.44916         0.93273       \n",
    "Total                     15    6.7417                 1.00000           \n",
    "```\n",
    "###NMDS using braycurtis\n",
    "![alt text](miseq4a_filtered.Treatment.braycurtis.pdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[02:19:45 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[02:19:45 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[02:19:45 PM]\u001b[0m: Loading OTU table: miseq4a.otu_table.txt\n",
      "\u001b[92m[02:19:45 PM]\u001b[0m: OTU table contains 2473 OTUs\n",
      "\u001b[92m[02:19:45 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[02:19:45 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[02:19:45 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[02:19:45 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[02:19:45 PM]\u001b[0m: Index bleed, samples into mock: 0.054002%.\n",
      "\u001b[92m[02:19:45 PM]\u001b[0m: Will use value of 0.100000% for index-bleed OTU filtering.\n",
      "\u001b[92m[02:19:46 PM]\u001b[0m: Filtering OTU table down to 1912 OTUs\n",
      "\u001b[92m[02:19:46 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      miseq4a_nosub.stats.txt\n",
      "Sorted OTU table:     miseq4a_nosub.sorted.txt\n",
      "Normalized/filter:    miseq4a_nosub.normalized.txt\n",
      "Final Binary table:   miseq4a_nosub.final.binary.txt\n",
      "Final OTU table:      miseq4a_nosub.final.txt\n",
      "Filtered OTUs:        miseq4a_nosub.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f miseq4a_nosub.filtered.otus.fa -i miseq4a_nosub.final.txt -m mapping_file.txt -d ITS2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#now run miseqA without subtraction filter\n",
    "!amptk filter -i miseq4a.otu_table.txt -f otudb.cluster.otus.fa -o miseq4a_nosub -b m4A-mock3-16000 -m mock3 -c in "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[02:21:31 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[02:21:31 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[02:21:31 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[02:21:31 PM]\u001b[0m: 1,912 OTUs\n",
      "\u001b[92m[02:21:31 PM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[02:21:31 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[02:21:38 PM]\u001b[0m: Taxonomy finished: otudb.cluster.otus.taxonomy.txt\n",
      "\u001b[92m[02:21:38 PM]\u001b[0m: Classic OTU table with taxonomy: miseq4a_nosub.otu_table.taxonomy.txt\n",
      "\u001b[92m[02:21:40 PM]\u001b[0m: BIOM OTU table created: miseq4a_nosub.biom\n",
      "\u001b[92m[02:21:40 PM]\u001b[0m: OTUs with taxonomy: miseq4a_nosub.otus.taxonomy.fa\n",
      "\u001b[92m[02:21:40 PM]\u001b[0m: OTU phylogeny: miseq4a_nosub.tree.phy\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#create no subtraction biom file\n",
    "!amptk taxonomy -i miseq4a_nosub.final.txt -f miseq4a_nosub.filtered.otus.fa -t otudb.cluster.otus.taxonomy.txt -d ITS2 -m m4a.mapping_file.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-2\n",
      "[1] ‘1.19.1’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘1.8.4’\n",
      "[1] ‘2.4.2’\n",
      "There were 27 warnings (use warnings() to see them)\n",
      "Square root transformation\n",
      "Wisconsin double standardization\n",
      "Run 0 stress 0.1539122 \n",
      "Run 1 stress 0.1623708 \n",
      "Run 2 stress 0.1633079 \n",
      "Run 3 stress 0.2182797 \n",
      "Run 4 stress 0.1767905 \n",
      "Run 5 stress 0.1623708 \n",
      "Run 6 stress 0.162914 \n",
      "Run 7 stress 0.1539122 \n",
      "... Procrustes: rmse 9.288234e-06  max resid 2.484461e-05 \n",
      "... Similar to previous best\n",
      "Run 8 stress 0.163308 \n",
      "Run 9 stress 0.1767905 \n",
      "Run 10 stress 0.1642 \n",
      "Run 11 stress 0.1539122 \n",
      "... Procrustes: rmse 9.123212e-06  max resid 2.105123e-05 \n",
      "... Similar to previous best\n",
      "Run 12 stress 0.1539122 \n",
      "... Procrustes: rmse 1.368471e-05  max resid 2.837103e-05 \n",
      "... Similar to previous best\n",
      "Run 13 stress 0.1539122 \n",
      "... Procrustes: rmse 3.154677e-05  max resid 6.161888e-05 \n",
      "... Similar to previous best\n",
      "Run 14 stress 0.1623708 \n",
      "Run 15 stress 0.1642001 \n",
      "Run 16 stress 0.1633079 \n",
      "Run 17 stress 0.1642 \n",
      "Run 18 stress 0.1539122 \n",
      "... New best solution\n",
      "... Procrustes: rmse 7.260616e-06  max resid 1.407689e-05 \n",
      "... Similar to previous best\n",
      "Run 19 stress 0.1642001 \n",
      "Run 20 stress 0.2148898 \n",
      "*** Solution reached\n",
      "Warning message:\n",
      "In postMDS(out$points, dis, plot = max(0, plot - 1), ...) :\n",
      "  skipping half-change scaling: too few points below threshold\n",
      "Run 0 stress 9.943553e-05 \n",
      "Run 1 stress 0.1162352 \n",
      "Run 2 stress 0.1616612 \n",
      "Run 3 stress 0.1366339 \n",
      "Run 4 stress 0.1654804 \n",
      "Run 5 stress 0.1471724 \n",
      "Run 6 stress 0.1180188 \n",
      "Run 7 stress 0.09984999 \n",
      "Run 8 stress 0.07855492 \n",
      "Run 9 stress 0.10148 \n",
      "Run 10 stress 0.1579072 \n",
      "Run 11 stress 0.07664251 \n",
      "Run 12 stress 0.1523424 \n",
      "Run 13 stress 0.1105138 \n",
      "Run 14 stress 0.129165 \n",
      "Run 15 stress 0.1523734 \n",
      "Run 16 stress 0.109899 \n",
      "Run 17 stress 0.1633614 \n",
      "Run 18 stress 0.0773965 \n",
      "Run 19 stress 0.1216028 \n",
      "Run 20 stress 0.1469812 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "     1: no. of iterations >= maxit\n",
      "    19: stress ratio > sratmax\n",
      "Warning message:\n",
      "In metaMDS(veganifyOTU(physeq), distance, ...) :\n",
      "  Stress is (nearly) zero - you may have insufficient data\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: In estimate_richness(physeq, split = TRUE, measures = measures) :\n",
      "  The data you have provided does not have\n",
      "any singletons. This is highly suspicious. Results of richness\n",
      "estimates (for example) are probably unreliable, or wrong, if you have already\n",
      "trimmed low-abundance taxa from the data.\n",
      "\n",
      "We recommended that you find the un-trimmed data and retry.\n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n",
      "3: In estimate_richness(physeq, split = TRUE, measures = measures) :\n",
      "  The data you have provided does not have\n",
      "any singletons. This is highly suspicious. Results of richness\n",
      "estimates (for example) are probably unreliable, or wrong, if you have already\n",
      "trimmed low-abundance taxa from the data.\n",
      "\n",
      "We recommended that you find the un-trimmed data and retry.\n",
      "4: Removed 32 rows containing missing values (geom_errorbar). \n"
     ]
    }
   ],
   "source": [
    "#run phyloseq\n",
    "!/Users/jon/amptk/util/phyloseq_graphs.R miseq4a_nosub.biom miseq4a_nosub.tree.phy miseq4a_nosub"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Treatment adonis using raupcrick\n",
    "```\n",
    "Call:\n",
    "adonis(formula = rp ~ metadata[[treatments[y]]], permutations = 9999) \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)  \n",
    "metadata[[treatments[y]]]  1   0.46981 0.46981  4.4274 0.24026 0.0667 .\n",
    "Residuals                 14   1.48560 0.10611         0.75974         \n",
    "Total                     15   1.95540                 1.00000         \n",
    "---\n",
    "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1   \n",
    "```\n",
    "###NMDS using raup\n",
    "![alt text](miseq4a_nosub.Treatment.raup.pdf)\n",
    "###Treatment adonis using braycurtis\n",
    "```\n",
    "Call:\n",
    "adonis(formula = voturaw ~ metadata[[treatments[y]]], permutations = 9999,      method = \"bray\") \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1    0.4522 0.45220  1.0131 0.06748 0.4511\n",
    "Residuals                 14    6.2486 0.44633         0.93252       \n",
    "Total                     15    6.7008                 1.00000 \n",
    "```\n",
    "###NMDS using braycurtis\n",
    "![alt text](miseq4a_nosub.Treatment.braycurtis.pdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[10:10:58 AM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[10:10:58 AM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[10:10:58 AM]\u001b[0m: Loading OTU table: miseq4c.otu_table.txt\n",
      "\u001b[92m[10:10:58 AM]\u001b[0m: OTU table contains 2473 OTUs\n",
      "\u001b[92m[10:10:58 AM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[10:10:58 AM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[10:10:58 AM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[10:10:58 AM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[10:10:58 AM]\u001b[0m: Index bleed, samples into mock: 0.051003%.\n",
      "\u001b[92m[10:10:58 AM]\u001b[0m: Will use value of 0.100000% for index-bleed OTU filtering.\n",
      "\u001b[92m[10:10:58 AM]\u001b[0m: Auto subtract filter set to 17\n",
      "\u001b[92m[10:10:58 AM]\u001b[0m: Subtracting 17 from OTU table\n",
      "\u001b[92m[10:10:59 AM]\u001b[0m: Filtering OTU table down to 502 OTUs\n",
      "\u001b[92m[10:10:59 AM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      miseq4c.stats.txt\n",
      "Sorted OTU table:     miseq4c.sorted.txt\n",
      "Normalized/filter:    miseq4c.normalized.txt\n",
      "Final Binary table:   miseq4c.final.binary.txt\n",
      "Final OTU table:      miseq4c.final.txt\n",
      "Filtered OTUs:        miseq4c.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f miseq4c.filtered.otus.fa -i miseq4c.final.txt -m mapping_file.txt -d ITS2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#now process miSeqC dataset\n",
    "!amptk filter -i miseq4c.otu_table.txt -f otudb.cluster.otus.fa -o miseq4c -b m4C-mock3-16000 -m mock3 -c in -s auto"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2473 17 2109 16\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[01:19:16 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[01:19:16 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[01:19:16 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[01:19:16 PM]\u001b[0m: 2,480 OTUs\n",
      "\u001b[92m[01:19:16 PM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[01:19:17 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[01:19:28 PM]\u001b[0m: Taxonomy finished: otudb.cluster.otus.taxonomy.txt\n",
      "\u001b[92m[01:19:28 PM]\u001b[0m: Classic OTU table with taxonomy: miseq4c_raw.otu_table.taxonomy.txt\n",
      "\u001b[92m[01:19:29 PM]\u001b[0m: BIOM OTU table created: miseq4c_raw.biom\n",
      "\u001b[92m[01:19:29 PM]\u001b[0m: OTUs with taxonomy: miseq4c_raw.otus.taxonomy.fa\n",
      "\u001b[92m[01:19:29 PM]\u001b[0m: OTU phylogeny: miseq4c_raw.tree.phy\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "from natsort import natsorted\n",
    "#pull out header for indexing later\n",
    "with open('miseq4c.sorted.txt', 'rU') as f:\n",
    "    first_line = f.readline()\n",
    "    OTUhead = first_line.split('\\t')[0]\n",
    "    \n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "dfsorted = pd.read_csv('miseq4c.sorted.txt', sep='\\t')\n",
    "dfsorted.set_index(OTUhead, inplace=True)\n",
    "\n",
    "#now we want to drop last column\n",
    "header = ['m4C-301-1', 'm4C-301-2', 'm4C-500-1', 'm4C-500-2', 'm4C-712-1', \n",
    "           'm4C-712-2', 'm4C-736-1', 'm4C-736-2', 'm4C-744-1', 'm4C-744-2', \n",
    "           'm4C-755-1', 'm4C-755-2', 'm4C-757-1', 'm4C-757-2', 'm4C-766-1', 'm4C-766-2']\n",
    "dfcleaned = dfsorted.reindex(columns=header, index=natsorted(dfsorted.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "fs = dfcleaned.sum(axis=0)\n",
    "filtered = pd.DataFrame(dfcleaned, columns=fs.index)\n",
    "filt2 = filtered.loc[(filtered != 0).any(1)]\n",
    "print len(dfsorted), len(dfsorted.columns.values), len(filt2), len(filt2.columns.values)\n",
    "filt2.to_csv('miseq4c.sorted.nomock.txt', sep='\\t')\n",
    "\n",
    "#now convert to biom\n",
    "!amptk taxonomy -f otudb.cluster.otus.fa -o miseq4c_raw -i miseq4c.sorted.nomock.txt -d ITS2 \\\n",
    "-m m4c.mapping_file.txt -t otudb.cluster.otus.taxonomy.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-2\n",
      "[1] ‘1.19.1’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘1.8.4’\n",
      "[1] ‘2.4.2’\n",
      "There were 27 warnings (use warnings() to see them)\n",
      "Square root transformation\n",
      "Wisconsin double standardization\n",
      "Run 0 stress 0.1437543 \n",
      "Run 1 stress 0.1459038 \n",
      "Run 2 stress 0.3525223 \n",
      "Run 3 stress 0.1437543 \n",
      "... New best solution\n",
      "... Procrustes: rmse 6.969975e-06  max resid 1.272703e-05 \n",
      "... Similar to previous best\n",
      "Run 4 stress 0.1495874 \n",
      "Run 5 stress 0.1503506 \n",
      "Run 6 stress 0.1459042 \n",
      "Run 7 stress 0.1437544 \n",
      "... Procrustes: rmse 0.0001113067  max resid 0.0002100703 \n",
      "... Similar to previous best\n",
      "Run 8 stress 0.1537847 \n",
      "Run 9 stress 0.1459037 \n",
      "Run 10 stress 0.1692988 \n",
      "Run 11 stress 0.1437543 \n",
      "... Procrustes: rmse 5.174389e-05  max resid 9.608152e-05 \n",
      "... Similar to previous best\n",
      "Run 12 stress 0.1495874 \n",
      "Run 13 stress 0.1495874 \n",
      "Run 14 stress 0.1827455 \n",
      "Run 15 stress 0.1459038 \n",
      "Run 16 stress 0.1437544 \n",
      "... Procrustes: rmse 2.901831e-05  max resid 6.008559e-05 \n",
      "... Similar to previous best\n",
      "Run 17 stress 0.2013344 \n",
      "Run 18 stress 0.1459041 \n",
      "Run 19 stress 0.1437543 \n",
      "... Procrustes: rmse 1.886444e-05  max resid 3.532684e-05 \n",
      "... Similar to previous best\n",
      "Run 20 stress 0.1437543 \n",
      "... Procrustes: rmse 1.871274e-05  max resid 3.589085e-05 \n",
      "... Similar to previous best\n",
      "*** Solution reached\n",
      "Warning message:\n",
      "In postMDS(out$points, dis, plot = max(0, plot - 1), ...) :\n",
      "  skipping half-change scaling: too few points below threshold\n",
      "Run 0 stress 0 \n",
      "Run 1 stress 9.768647e-05 \n",
      "... Procrustes: rmse 0.2169867  max resid 0.5323091 \n",
      "Run 2 stress 9.270306e-05 \n",
      "... Procrustes: rmse 0.1938858  max resid 0.4188229 \n",
      "Run 3 stress 9.228393e-05 \n",
      "... Procrustes: rmse 0.2037416  max resid 0.4632179 \n",
      "Run 4 stress 4.826428e-05 \n",
      "... Procrustes: rmse 0.1871629  max resid 0.3880056 \n",
      "Run 5 stress 0.0003050643 \n",
      "... Procrustes: rmse 0.1827921  max resid 0.4685656 \n",
      "Run 6 stress 9.237787e-05 \n",
      "... Procrustes: rmse 0.2186655  max resid 0.4702651 \n",
      "Run 7 stress 9.690631e-05 \n",
      "... Procrustes: rmse 0.2171478  max resid 0.5206595 \n",
      "Run 8 stress 8.978215e-05 \n",
      "... Procrustes: rmse 0.1879818  max resid 0.4350626 \n",
      "Run 9 stress 9.974465e-05 \n",
      "... Procrustes: rmse 0.2069189  max resid 0.4952489 \n",
      "Run 10 stress 5.022893e-05 \n",
      "... Procrustes: rmse 0.2163664  max resid 0.4953276 \n",
      "Run 11 stress 9.650356e-05 \n",
      "... Procrustes: rmse 0.1744173  max resid 0.3890655 \n",
      "Run 12 stress 9.186506e-05 \n",
      "... Procrustes: rmse 0.2097097  max resid 0.5100159 \n",
      "Run 13 stress 8.37638e-05 \n",
      "... Procrustes: rmse 0.2109637  max resid 0.4466517 \n",
      "Run 14 stress 9.687958e-05 \n",
      "... Procrustes: rmse 0.2049158  max resid 0.4562569 \n",
      "Run 15 stress 9.74303e-05 \n",
      "... Procrustes: rmse 0.1785541  max resid 0.3327219 \n",
      "Run 16 stress 9.706533e-05 \n",
      "... Procrustes: rmse 0.2006371  max resid 0.4232534 \n",
      "Run 17 stress 3.592116e-05 \n",
      "... Procrustes: rmse 0.2206375  max resid 0.4964698 \n",
      "Run 18 stress 7.453722e-05 \n",
      "... Procrustes: rmse 0.1758678  max resid 0.3443172 \n",
      "Run 19 stress 9.415661e-05 \n",
      "... Procrustes: rmse 0.1931225  max resid 0.4097249 \n",
      "Run 20 stress 0.0001762247 \n",
      "... Procrustes: rmse 0.1994366  max resid 0.4355411 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "     2: no. of iterations >= maxit\n",
      "    18: stress < smin\n",
      "Warning message:\n",
      "In metaMDS(veganifyOTU(physeq), distance, ...) :\n",
      "  Stress is (nearly) zero - you may have insufficient data\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 32 rows containing missing values (geom_errorbar). \n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n"
     ]
    }
   ],
   "source": [
    "#run phyloseq\n",
    "!/Users/jon/amptk/util/phyloseq_graphs.R miseq4c_raw.biom miseq4c_raw.tree.phy m4c_raw"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Treatment adonis using raupcrick\n",
    "```   \n",
    "Call:\n",
    "adonis(formula = rp ~ metadata[[treatments[y]]], permutations = 9999) \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)   \n",
    "metadata[[treatments[y]]]  1   0.61753 0.61753  12.067 0.46291 0.0079 **\n",
    "Residuals                 14   0.71647 0.05118         0.53709          \n",
    "Total                     15   1.33400                 1.00000          \n",
    "---\n",
    "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
    "```\n",
    "###NMDS using jaccard\n",
    "![alt text](m4c_raw.Treatment.raup.pdf)\n",
    "###Treatment adonis using braycurtis\n",
    "```\n",
    "Call:\n",
    "adonis(formula = voturaw ~ metadata[[treatments[y]]], permutations = 9999,      method = \"bray\") \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model     R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1    0.4518 0.45180  1.0086 0.0672 0.4753\n",
    "Residuals                 14    6.2712 0.44795         0.9328       \n",
    "Total                     15    6.7230                 1.0000    \n",
    "```\n",
    "###NMDS using braycurtis\n",
    "![alt text](m4c_raw.Treatment.braycurtis.pdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[10:14:31 AM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[10:14:31 AM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[10:14:31 AM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[10:14:31 AM]\u001b[0m: 502 OTUs\n",
      "\u001b[92m[10:14:31 AM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[10:14:32 AM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[10:14:33 AM]\u001b[0m: Taxonomy finished: otudb.cluster.otus.taxonomy.txt\n",
      "\u001b[92m[10:14:33 AM]\u001b[0m: Classic OTU table with taxonomy: miseq4c.otu_table.taxonomy.txt\n",
      "\u001b[92m[10:14:35 AM]\u001b[0m: BIOM OTU table created: miseq4c.biom\n",
      "\u001b[92m[10:14:35 AM]\u001b[0m: OTUs with taxonomy: miseq4c.otus.taxonomy.fa\n",
      "\u001b[92m[10:14:35 AM]\u001b[0m: OTU phylogeny: miseq4c.tree.phy\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#get biom of filtered data\n",
    "!amptk taxonomy -f miseq4c.filtered.otus.fa -i miseq4c.final.txt -d ITS2 -m m4c.mapping_file.txt -t otudb.cluster.otus.taxonomy.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-2\n",
      "[1] ‘1.19.1’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘1.8.4’\n",
      "[1] ‘2.4.2’\n",
      "Warning messages:\n",
      "1: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "2: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "3: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "4: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "Square root transformation\n",
      "Wisconsin double standardization\n",
      "Run 0 stress 0.2089407 \n",
      "Run 1 stress 0.2230954 \n",
      "Run 2 stress 0.2172407 \n",
      "Run 3 stress 0.2230953 \n",
      "Run 4 stress 0.237226 \n",
      "Run 5 stress 0.2230954 \n",
      "Run 6 stress 0.2110097 \n",
      "Run 7 stress 0.2161311 \n",
      "Run 8 stress 0.2226551 \n",
      "Run 9 stress 0.2205502 \n",
      "Run 10 stress 0.223542 \n",
      "Run 11 stress 0.2372284 \n",
      "Run 12 stress 0.2218741 \n",
      "Run 13 stress 0.216619 \n",
      "Run 14 stress 0.2399424 \n",
      "Run 15 stress 0.2215176 \n",
      "Run 16 stress 0.2205308 \n",
      "Run 17 stress 0.2975344 \n",
      "Run 18 stress 0.2080389 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.03870193  max resid 0.1077947 \n",
      "Run 19 stress 0.2220562 \n",
      "Run 20 stress 0.220461 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n",
      "Warning message:\n",
      "In postMDS(out$points, dis, plot = max(0, plot - 1), ...) :\n",
      "  skipping half-change scaling: too few points below threshold\n",
      "Run 0 stress 0.2093042 \n",
      "Run 1 stress 0.2630617 \n",
      "Run 2 stress 0.214291 \n",
      "Run 3 stress 0.2225038 \n",
      "Run 4 stress 0.22604 \n",
      "Run 5 stress 0.2198753 \n",
      "Run 6 stress 0.210782 \n",
      "Run 7 stress 0.209304 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0002183455  max resid 0.0006228795 \n",
      "... Similar to previous best\n",
      "Run 8 stress 0.2093041 \n",
      "... Procrustes: rmse 0.0001098027  max resid 0.0002134669 \n",
      "... Similar to previous best\n",
      "Run 9 stress 0.2083493 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.06478224  max resid 0.1568165 \n",
      "Run 10 stress 0.2093354 \n",
      "Run 11 stress 0.2176103 \n",
      "Run 12 stress 0.240249 \n",
      "Run 13 stress 0.2044992 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1218911  max resid 0.3945293 \n",
      "Run 14 stress 0.2169541 \n",
      "Run 15 stress 0.2684573 \n",
      "Run 16 stress 0.2083494 \n",
      "Run 17 stress 0.2119393 \n",
      "Run 18 stress 0.219876 \n",
      "Run 19 stress 0.2067958 \n",
      "Run 20 stress 0.2044991 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0001159577  max resid 0.0002194094 \n",
      "... Similar to previous best\n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: In estimate_richness(physeq, split = TRUE, measures = measures) :\n",
      "  The data you have provided does not have\n",
      "any singletons. This is highly suspicious. Results of richness\n",
      "estimates (for example) are probably unreliable, or wrong, if you have already\n",
      "trimmed low-abundance taxa from the data.\n",
      "\n",
      "We recommended that you find the un-trimmed data and retry.\n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n",
      "3: In estimate_richness(physeq, split = TRUE, measures = measures) :\n",
      "  The data you have provided does not have\n",
      "any singletons. This is highly suspicious. Results of richness\n",
      "estimates (for example) are probably unreliable, or wrong, if you have already\n",
      "trimmed low-abundance taxa from the data.\n",
      "\n",
      "We recommended that you find the un-trimmed data and retry.\n",
      "4: Removed 32 rows containing missing values (geom_errorbar). \n"
     ]
    }
   ],
   "source": [
    "#run phyloseq\n",
    "!/Users/jon/amptk/util/phyloseq_graphs.R miseq4c.biom miseq4c.tree.phy miseq4c_filtered"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Treatment adonis using raupcrick\n",
    "```\n",
    "Call:\n",
    "adonis(formula = rp ~ metadata[[treatments[y]]], permutations = 9999) \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)  \n",
    "metadata[[treatments[y]]]  1   0.52965 0.52965  3.3617 0.19363 0.0562 .\n",
    "Residuals                 14   2.20575 0.15755         0.80637         \n",
    "Total                     15   2.73540                 1.00000         \n",
    "---\n",
    "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1 \n",
    "\n",
    "```\n",
    "###NMDS using jaccard\n",
    "![alt text](miseq4c_filtered.Treatment.jaccard.pdf)\n",
    "###Treatment adonis using braycurtis\n",
    "```\n",
    "Call:\n",
    "adonis(formula = voturaw ~ metadata[[treatments[y]]], permutations = 9999,      method = \"bray\") \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1    0.4521 0.45211  1.0027 0.06683  0.486\n",
    "Residuals                 14    6.3125 0.45089         0.93317       \n",
    "Total                     15    6.7646                 1.00000          \n",
    "```\n",
    "###NMDS using braycurtis\n",
    "![alt text](miseq4c_filtered.Treatment.braycurtis.pdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[02:25:18 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[02:25:19 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[02:25:19 PM]\u001b[0m: Loading OTU table: miseq4c.otu_table.txt\n",
      "\u001b[92m[02:25:19 PM]\u001b[0m: OTU table contains 2473 OTUs\n",
      "\u001b[92m[02:25:19 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[02:25:19 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[02:25:19 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[02:25:19 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[02:25:19 PM]\u001b[0m: Index bleed, samples into mock: 0.051003%.\n",
      "\u001b[92m[02:25:19 PM]\u001b[0m: Will use value of 0.100000% for index-bleed OTU filtering.\n",
      "\u001b[92m[02:25:20 PM]\u001b[0m: Filtering OTU table down to 1809 OTUs\n",
      "\u001b[92m[02:25:20 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      miseq4c_nosub.stats.txt\n",
      "Sorted OTU table:     miseq4c_nosub.sorted.txt\n",
      "Normalized/filter:    miseq4c_nosub.normalized.txt\n",
      "Final Binary table:   miseq4c_nosub.final.binary.txt\n",
      "Final OTU table:      miseq4c_nosub.final.txt\n",
      "Filtered OTUs:        miseq4c_nosub.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f miseq4c_nosub.filtered.otus.fa -i miseq4c_nosub.final.txt -m mapping_file.txt -d ITS2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#generate no subtraction filter\n",
    "!amptk filter -i miseq4c.otu_table.txt -f otudb.cluster.otus.fa -o miseq4c_nosub -b m4C-mock3-16000 -m mock3 -c in"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[02:26:28 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[02:26:28 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[02:26:28 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[02:26:28 PM]\u001b[0m: 1,809 OTUs\n",
      "\u001b[92m[02:26:28 PM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[02:26:28 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[02:26:34 PM]\u001b[0m: Taxonomy finished: otudb.cluster.otus.taxonomy.txt\n",
      "\u001b[92m[02:26:34 PM]\u001b[0m: Classic OTU table with taxonomy: miseq4c_nosub.otu_table.taxonomy.txt\n",
      "\u001b[92m[02:26:35 PM]\u001b[0m: BIOM OTU table created: miseq4c_nosub.biom\n",
      "\u001b[92m[02:26:35 PM]\u001b[0m: OTUs with taxonomy: miseq4c_nosub.otus.taxonomy.fa\n",
      "\u001b[92m[02:26:35 PM]\u001b[0m: OTU phylogeny: miseq4c_nosub.tree.phy\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#generate biom\n",
    "!amptk taxonomy -f miseq4c_nosub.filtered.otus.fa -i miseq4c_nosub.final.txt -d ITS2 \\\n",
    "-m m4c.mapping_file.txt -t otudb.cluster.otus.taxonomy.txt\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-2\n",
      "[1] ‘1.19.1’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘1.8.4’\n",
      "[1] ‘2.4.2’\n",
      "There were 26 warnings (use warnings() to see them)\n",
      "Square root transformation\n",
      "Wisconsin double standardization\n",
      "Run 0 stress 0.1647859 \n",
      "Run 1 stress 0.181755 \n",
      "Run 2 stress 0.1658236 \n",
      "Run 3 stress 0.1613274 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.06408241  max resid 0.1751406 \n",
      "Run 4 stress 0.1647859 \n",
      "Run 5 stress 0.1613273 \n",
      "... New best solution\n",
      "... Procrustes: rmse 9.368663e-05  max resid 0.0001859158 \n",
      "... Similar to previous best\n",
      "Run 6 stress 0.279312 \n",
      "Run 7 stress 0.1613272 \n",
      "... New best solution\n",
      "... Procrustes: rmse 4.893853e-05  max resid 7.774388e-05 \n",
      "... Similar to previous best\n",
      "Run 8 stress 0.1647859 \n",
      "Run 9 stress 0.1798793 \n",
      "Run 10 stress 0.2128599 \n",
      "Run 11 stress 0.1647859 \n",
      "Run 12 stress 0.1658529 \n",
      "Run 13 stress 0.2151283 \n",
      "Run 14 stress 0.2084887 \n",
      "Run 15 stress 0.1815138 \n",
      "Run 16 stress 0.1721475 \n",
      "Run 17 stress 0.1658236 \n",
      "Run 18 stress 0.1815138 \n",
      "Run 19 stress 0.2160037 \n",
      "Run 20 stress 0.2143637 \n",
      "*** Solution reached\n",
      "Warning message:\n",
      "In postMDS(out$points, dis, plot = max(0, plot - 1), ...) :\n",
      "  skipping half-change scaling: too few points below threshold\n",
      "Run 0 stress 9.83069e-05 \n",
      "Run 1 stress 0.1272839 \n",
      "Run 2 stress 0.1325219 \n",
      "Run 3 stress 0.1561675 \n",
      "Run 4 stress 0.1427647 \n",
      "Run 5 stress 0.1289501 \n",
      "Run 6 stress 0.1159562 \n",
      "Run 7 stress 0.0947587 \n",
      "Run 8 stress 0.1412432 \n",
      "Run 9 stress 0.1334502 \n",
      "Run 10 stress 0.1087235 \n",
      "Run 11 stress 0.1231275 \n",
      "Run 12 stress 0.107974 \n",
      "Run 13 stress 0.08841234 \n",
      "Run 14 stress 0.1514957 \n",
      "Run 15 stress 0.07821941 \n",
      "Run 16 stress 0.0966197 \n",
      "Run 17 stress 0.09835812 \n",
      "Run 18 stress 0.1434851 \n",
      "Run 19 stress 0.1343448 \n",
      "Run 20 stress 0.0822243 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "     1: no. of iterations >= maxit\n",
      "    19: stress ratio > sratmax\n",
      "Warning message:\n",
      "In metaMDS(veganifyOTU(physeq), distance, ...) :\n",
      "  Stress is (nearly) zero - you may have insufficient data\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 32 rows containing missing values (geom_errorbar). \n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n"
     ]
    }
   ],
   "source": [
    "#run phyloseq\n",
    "!/Users/jon/amptk/util/phyloseq_graphs.R miseq4c_nosub.biom miseq4c_nosub.tree.phy miseq4c_nosub"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Treatment adonis using raupcrick\n",
    "```\n",
    "Call:\n",
    "adonis(formula = rp ~ metadata[[treatments[y]]], permutations = 9999) \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)  \n",
    "metadata[[treatments[y]]]  1   0.47376 0.47376  4.3466 0.23692 0.0576 .\n",
    "Residuals                 14   1.52594 0.10900         0.76308         \n",
    "Total                     15   1.99970                 1.00000         \n",
    "---\n",
    "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
    "```\n",
    "###NMDS using raup\n",
    "![alt text](miseq4c_nosub.Treatment.raup.pdf)\n",
    "###Treatment adonis using braycurtis\n",
    "```\n",
    "Call:\n",
    "adonis(formula = voturaw ~ metadata[[treatments[y]]], permutations = 9999,      method = \"bray\") \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1    0.4509 0.45092  1.0046 0.06695 0.4769\n",
    "Residuals                 14    6.2841 0.44886         0.93305       \n",
    "Total                     15    6.7350                 1.00000  \n",
    "```\n",
    "###NMDS using braycurtis\n",
    "![alt text](miseq4c_nosub.Treatment.braycurtis.pdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[10:17:07 AM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[10:17:07 AM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[10:17:07 AM]\u001b[0m: Loading OTU table: ion5.otu_table.txt\n",
      "\u001b[92m[10:17:07 AM]\u001b[0m: OTU table contains 2473 OTUs\n",
      "\u001b[92m[10:17:07 AM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[10:17:07 AM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[10:17:07 AM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[10:17:07 AM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[10:17:07 AM]\u001b[0m: Index bleed, mock into samples: 0.174878%.  Index bleed, samples into mock: 0.103002%.\n",
      "\u001b[92m[10:17:07 AM]\u001b[0m: Will use value of 0.200000% for index-bleed OTU filtering.\n",
      "\u001b[92m[10:17:07 AM]\u001b[0m: Auto subtract filter set to 36\n",
      "\u001b[92m[10:17:07 AM]\u001b[0m: Subtracting 36 from OTU table\n",
      "\u001b[92m[10:17:08 AM]\u001b[0m: Filtering OTU table down to 369 OTUs\n",
      "\u001b[92m[10:17:08 AM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      ion5.stats.txt\n",
      "Sorted OTU table:     ion5.sorted.txt\n",
      "Normalized/filter:    ion5.normalized.txt\n",
      "Final Binary table:   ion5.final.binary.txt\n",
      "Final OTU table:      ion5.final.txt\n",
      "Filtered OTUs:        ion5.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f ion5.filtered.otus.fa -i ion5.final.txt -m mapping_file.txt -d ITS2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#finally run ion5 data\n",
    "!amptk filter -i ion5.otu_table.txt -f otudb.cluster.otus.fa -o ion5 -b m5-SynMock2-8000 -m synmock -s auto"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2473 17 1584 16\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[01:23:48 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[01:23:48 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[01:23:48 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[01:23:48 PM]\u001b[0m: 2,480 OTUs\n",
      "\u001b[92m[01:23:48 PM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[01:23:48 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[01:23:59 PM]\u001b[0m: Taxonomy finished: otudb.cluster.otus.taxonomy.txt\n",
      "\u001b[92m[01:23:59 PM]\u001b[0m: Classic OTU table with taxonomy: ion5_raw.otu_table.taxonomy.txt\n",
      "\u001b[92m[01:24:01 PM]\u001b[0m: BIOM OTU table created: ion5_raw.biom\n",
      "\u001b[92m[01:24:01 PM]\u001b[0m: OTUs with taxonomy: ion5_raw.otus.taxonomy.fa\n",
      "\u001b[92m[01:24:01 PM]\u001b[0m: OTU phylogeny: ion5_raw.tree.phy\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "from natsort import natsorted\n",
    "#pull out header for indexing later\n",
    "with open('ion5.sorted.txt', 'rU') as f:\n",
    "    first_line = f.readline()\n",
    "    OTUhead = first_line.split('\\t')[0]\n",
    "    \n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "dfsorted = pd.read_csv('ion5.sorted.txt', sep='\\t')\n",
    "dfsorted.set_index(OTUhead, inplace=True)\n",
    "\n",
    "#now we want to drop last column\n",
    "header = ['m5-301-1', 'm5-301-2', 'm5-500-1', 'm5-500-2', 'm5-712-1', \n",
    "        'm5-712-2', 'm5-736-1', 'm5-736-2', 'm5-744-1', 'm5-744-2', \n",
    "        'm5-755-1', 'm5-755-2', 'm5-757-1', 'm5-757-2', 'm5-766-1', 'm5-766-2']\n",
    "dfcleaned = dfsorted.reindex(columns=header, index=natsorted(dfsorted.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "fs = dfcleaned.sum(axis=0)\n",
    "filtered = pd.DataFrame(dfcleaned, columns=fs.index)\n",
    "filt2 = filtered.loc[(filtered != 0).any(1)]\n",
    "print len(dfsorted), len(dfsorted.columns.values), len(filt2), len(filt2.columns.values)\n",
    "filt2.to_csv('ion5.sorted.nomock.txt', sep='\\t')\n",
    "\n",
    "#now convert to biom\n",
    "!amptk taxonomy -f otudb.cluster.otus.fa -o ion5_raw -i ion5.sorted.nomock.txt -d ITS2 \\\n",
    "-m ion5.mapping_file.txt -t otudb.cluster.otus.taxonomy.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-2\n",
      "[1] ‘1.19.1’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘1.8.4’\n",
      "[1] ‘2.4.2’\n",
      "There were 12 warnings (use warnings() to see them)\n",
      "Square root transformation\n",
      "Wisconsin double standardization\n",
      "Run 0 stress 0.1517762 \n",
      "Run 1 stress 0.2013717 \n",
      "Run 2 stress 0.1553845 \n",
      "Run 3 stress 0.1553846 \n",
      "Run 4 stress 0.2038471 \n",
      "Run 5 stress 0.1539358 \n",
      "Run 6 stress 0.1539353 \n",
      "Run 7 stress 0.1539353 \n",
      "Run 8 stress 0.1539354 \n",
      "Run 9 stress 0.1902329 \n",
      "Run 10 stress 0.2142988 \n",
      "Run 11 stress 0.1539359 \n",
      "Run 12 stress 0.2093007 \n",
      "Run 13 stress 0.1553844 \n",
      "Run 14 stress 0.2157074 \n",
      "Run 15 stress 0.153936 \n",
      "Run 16 stress 0.1539355 \n",
      "Run 17 stress 0.1553844 \n",
      "Run 18 stress 0.1911059 \n",
      "Run 19 stress 0.2157075 \n",
      "Run 20 stress 0.2038472 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n",
      "Warning message:\n",
      "In postMDS(out$points, dis, plot = max(0, plot - 1), ...) :\n",
      "  skipping half-change scaling: too few points below threshold\n",
      "Run 0 stress 7.500051e-05 \n",
      "Run 1 stress 0.1697149 \n",
      "Run 2 stress 0.07857724 \n",
      "Run 3 stress 0.08237217 \n",
      "Run 4 stress 0.1323855 \n",
      "Run 5 stress 0.1163478 \n",
      "Run 6 stress 0.05152296 \n",
      "Run 7 stress 0.1244288 \n",
      "Run 8 stress 0.1119889 \n",
      "Run 9 stress 0.1228514 \n",
      "Run 10 stress 0.001471346 \n",
      "Run 11 stress 0.08100865 \n",
      "Run 12 stress 0.1051792 \n",
      "Run 13 stress 0.1573692 \n",
      "Run 14 stress 0.1392436 \n",
      "Run 15 stress 0.06506838 \n",
      "Run 16 stress 9.685366e-05 \n",
      "... Procrustes: rmse 0.0789228  max resid 0.2018479 \n",
      "Run 17 stress 0.1578212 \n",
      "Run 18 stress 0.1048241 \n",
      "Run 19 stress 0.01858521 \n",
      "Run 20 stress 0.1072835 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "     2: no. of iterations >= maxit\n",
      "     1: stress < smin\n",
      "    17: stress ratio > sratmax\n",
      "Warning message:\n",
      "In metaMDS(veganifyOTU(physeq), distance, ...) :\n",
      "  Stress is (nearly) zero - you may have insufficient data\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 32 rows containing missing values (geom_errorbar). \n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n"
     ]
    }
   ],
   "source": [
    "#run phyloseq\n",
    "!/Users/jon/amptk/util/phyloseq_graphs.R ion5_raw.biom ion5_raw.tree.phy ion5_raw"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Treatment adonis using raupcrick\n",
    "```\n",
    "Call:\n",
    "adonis(formula = rp ~ metadata[[treatments[y]]], permutations = 9999) \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)  \n",
    "metadata[[treatments[y]]]  1   0.36862 0.36862  4.3565 0.23733 0.0761 .\n",
    "Residuals                 14   1.18458 0.08461         0.76267         \n",
    "Total                     15   1.55319                 1.00000         \n",
    "---\n",
    "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
    "```\n",
    "###NMDS using jaccard\n",
    "![alt text](ion5_raw.Treatment.raup.pdf)\n",
    "###Treatment adonis using braycurtis\n",
    "```\n",
    "Call:\n",
    "adonis(formula = voturaw ~ metadata[[treatments[y]]], permutations = 9999,      method = \"bray\") \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1    0.4697 0.46971  1.0284 0.06843 0.4451\n",
    "Residuals                 14    6.3946 0.45676         0.93157       \n",
    "Total                     15    6.8643                 1.00000         \n",
    "```\n",
    "###NMDS using braycurtis\n",
    "![alt text](ion5_raw.Treatment.braycurtis.pdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[10:23:54 AM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[10:23:55 AM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[10:23:55 AM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[10:23:55 AM]\u001b[0m: 369 OTUs\n",
      "\u001b[92m[10:23:55 AM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[10:23:55 AM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[10:23:55 AM]\u001b[0m: Taxonomy finished: otudb.cluster.otus.taxonomy.txt\n",
      "\u001b[92m[10:23:55 AM]\u001b[0m: Classic OTU table with taxonomy: ion5.otu_table.taxonomy.txt\n",
      "\u001b[92m[10:23:57 AM]\u001b[0m: BIOM OTU table created: ion5.biom\n",
      "\u001b[92m[10:23:57 AM]\u001b[0m: OTUs with taxonomy: ion5.otus.taxonomy.fa\n",
      "\u001b[92m[10:23:57 AM]\u001b[0m: OTU phylogeny: ion5.tree.phy\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#now get filtered BIOM\n",
    "!amptk taxonomy -f ion5.filtered.otus.fa -i ion5.final.txt -d ITS2 -m ion5.mapping_file.txt -t otudb.cluster.otus.taxonomy.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-2\n",
      "[1] ‘1.19.1’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘1.8.4’\n",
      "[1] ‘2.4.2’\n",
      "Square root transformation\n",
      "Wisconsin double standardization\n",
      "Run 0 stress 0.2088984 \n",
      "Run 1 stress 0.2122824 \n",
      "Run 2 stress 0.2085693 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.03378664  max resid 0.08989316 \n",
      "Run 3 stress 0.1932874 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1187702  max resid 0.269162 \n",
      "Run 4 stress 0.2298542 \n",
      "Run 5 stress 0.2219253 \n",
      "Run 6 stress 0.2020158 \n",
      "Run 7 stress 0.2461529 \n",
      "Run 8 stress 0.2098963 \n",
      "Run 9 stress 0.2296379 \n",
      "Run 10 stress 0.2201646 \n",
      "Run 11 stress 0.1929808 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.02307756  max resid 0.06341198 \n",
      "Run 12 stress 0.2122822 \n",
      "Run 13 stress 0.2158179 \n",
      "Run 14 stress 0.2181837 \n",
      "Run 15 stress 0.2122818 \n",
      "Run 16 stress 0.2932266 \n",
      "Run 17 stress 0.2511093 \n",
      "Run 18 stress 0.2122818 \n",
      "Run 19 stress 0.2181837 \n",
      "Run 20 stress 0.2241199 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n",
      "Warning message:\n",
      "In postMDS(out$points, dis, plot = max(0, plot - 1), ...) :\n",
      "  skipping half-change scaling: too few points below threshold\n",
      "Run 0 stress 0.2040139 \n",
      "Run 1 stress 0.2017639 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0796262  max resid 0.1517646 \n",
      "Run 2 stress 0.201739 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.04033385  max resid 0.1209669 \n",
      "Run 3 stress 0.2389955 \n",
      "Run 4 stress 0.2017389 \n",
      "... New best solution\n",
      "... Procrustes: rmse 7.143671e-05  max resid 0.000141251 \n",
      "... Similar to previous best\n",
      "Run 5 stress 0.2020292 \n",
      "... Procrustes: rmse 0.08300194  max resid 0.1685495 \n",
      "Run 6 stress 0.2336436 \n",
      "Run 7 stress 0.2370642 \n",
      "Run 8 stress 0.2743821 \n",
      "Run 9 stress 0.2116787 \n",
      "Run 10 stress 0.2021339 \n",
      "... Procrustes: rmse 0.074234  max resid 0.1595995 \n",
      "Run 11 stress 0.2021343 \n",
      "... Procrustes: rmse 0.07425199  max resid 0.1595545 \n",
      "Run 12 stress 0.2021352 \n",
      "... Procrustes: rmse 0.07412189  max resid 0.1599088 \n",
      "Run 13 stress 0.2107427 \n",
      "Run 14 stress 0.3657433 \n",
      "Run 15 stress 0.2021342 \n",
      "... Procrustes: rmse 0.07406869  max resid 0.1587481 \n",
      "Run 16 stress 0.2017387 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0006610358  max resid 0.002036369 \n",
      "... Similar to previous best\n",
      "Run 17 stress 0.2116781 \n",
      "Run 18 stress 0.2452037 \n",
      "Run 19 stress 0.2017386 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0002735151  max resid 0.0008518535 \n",
      "... Similar to previous best\n",
      "Run 20 stress 0.2464819 \n",
      "*** Solution reached\n",
      "Warning message:\n",
      "In postMDS(out$points, dis, plot = max(0, plot - 1), ...) :\n",
      "  skipping half-change scaling: too few points below threshold\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: In estimate_richness(physeq, split = TRUE, measures = measures) :\n",
      "  The data you have provided does not have\n",
      "any singletons. This is highly suspicious. Results of richness\n",
      "estimates (for example) are probably unreliable, or wrong, if you have already\n",
      "trimmed low-abundance taxa from the data.\n",
      "\n",
      "We recommended that you find the un-trimmed data and retry.\n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n",
      "3: In estimate_richness(physeq, split = TRUE, measures = measures) :\n",
      "  The data you have provided does not have\n",
      "any singletons. This is highly suspicious. Results of richness\n",
      "estimates (for example) are probably unreliable, or wrong, if you have already\n",
      "trimmed low-abundance taxa from the data.\n",
      "\n",
      "We recommended that you find the un-trimmed data and retry.\n",
      "4: Removed 32 rows containing missing values (geom_errorbar). \n"
     ]
    }
   ],
   "source": [
    "#run phyloseq\n",
    "!/Users/jon/amptk/util/phyloseq_graphs.R ion5.biom ion5.tree.phy ion5_filtered"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Treatment adonis using raupcrick\n",
    "```\n",
    "Call:\n",
    "adonis(formula = rp ~ metadata[[treatments[y]]], permutations = 9999) \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1    0.4253 0.42529   1.889 0.11889  0.188\n",
    "Residuals                 14    3.1520 0.22514         0.88111       \n",
    "Total                     15    3.5773                 1.00000 \n",
    "```\n",
    "###NMDS using raup\n",
    "![alt text](ion5_filtered.Treatment.raup.pdf)\n",
    "###Treatment adonis using braycurtis\n",
    "```\n",
    "Call:\n",
    "adonis(formula = voturaw ~ metadata[[treatments[y]]], permutations = 9999,      method = \"bray\") \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1    0.4713 0.47133  1.0239 0.06815 0.4518\n",
    "Residuals                 14    6.4444 0.46032         0.93185       \n",
    "Total                     15    6.9157                 1.00000             \n",
    "```\n",
    "###NMDS using braycurtis\n",
    "![alt text](ion5_filtered.Treatment.braycurtis.pdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[02:29:30 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[02:29:30 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[02:29:30 PM]\u001b[0m: Loading OTU table: ion5.otu_table.txt\n",
      "\u001b[92m[02:29:30 PM]\u001b[0m: OTU table contains 2473 OTUs\n",
      "\u001b[92m[02:29:30 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[02:29:30 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[02:29:30 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[02:29:30 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[02:29:30 PM]\u001b[0m: Index bleed, mock into samples: 0.174878%.  Index bleed, samples into mock: 0.103002%.\n",
      "\u001b[92m[02:29:30 PM]\u001b[0m: Will use value of 0.200000% for index-bleed OTU filtering.\n",
      "\u001b[92m[02:29:31 PM]\u001b[0m: Filtering OTU table down to 1358 OTUs\n",
      "\u001b[92m[02:29:31 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      ion5_nosub.stats.txt\n",
      "Sorted OTU table:     ion5_nosub.sorted.txt\n",
      "Normalized/filter:    ion5_nosub.normalized.txt\n",
      "Final Binary table:   ion5_nosub.final.binary.txt\n",
      "Final OTU table:      ion5_nosub.final.txt\n",
      "Filtered OTUs:        ion5_nosub.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f ion5_nosub.filtered.otus.fa -i ion5_nosub.final.txt -m mapping_file.txt -d ITS2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#run ion5 no subtraction filter\n",
    "!amptk filter -i ion5.otu_table.txt -f otudb.cluster.otus.fa -o ion5_nosub -b m5-SynMock2-8000 -m synmock"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[02:30:08 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[02:30:09 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[02:30:09 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[02:30:09 PM]\u001b[0m: 1,358 OTUs\n",
      "\u001b[92m[02:30:09 PM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[02:30:09 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[02:30:12 PM]\u001b[0m: Taxonomy finished: otudb.cluster.otus.taxonomy.txt\n",
      "\u001b[92m[02:30:12 PM]\u001b[0m: Classic OTU table with taxonomy: ion5_nosub.otu_table.taxonomy.txt\n",
      "\u001b[92m[02:30:14 PM]\u001b[0m: BIOM OTU table created: ion5_nosub.biom\n",
      "\u001b[92m[02:30:14 PM]\u001b[0m: OTUs with taxonomy: ion5_nosub.otus.taxonomy.fa\n",
      "\u001b[92m[02:30:14 PM]\u001b[0m: OTU phylogeny: ion5_nosub.tree.phy\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#now get filtered BIOM\n",
    "!amptk taxonomy -f ion5_nosub.filtered.otus.fa -i ion5_nosub.final.txt -d ITS2 -m ion5.mapping_file.txt -t otudb.cluster.otus.taxonomy.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-2\n",
      "[1] ‘1.19.1’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘1.8.4’\n",
      "[1] ‘2.4.2’\n",
      "Square root transformation\n",
      "Wisconsin double standardization\n",
      "Run 0 stress 0.149634 \n",
      "Run 1 stress 0.1533488 \n",
      "Run 2 stress 0.1937315 \n",
      "Run 3 stress 0.1504872 \n",
      "Run 4 stress 0.1504872 \n",
      "Run 5 stress 0.1771024 \n",
      "Run 6 stress 0.149634 \n",
      "... Procrustes: rmse 7.505712e-05  max resid 0.0001744291 \n",
      "... Similar to previous best\n",
      "Run 7 stress 0.1801135 \n",
      "Run 8 stress 0.1504872 \n",
      "Run 9 stress 0.1801135 \n",
      "Run 10 stress 0.1937314 \n",
      "Run 11 stress 0.149634 \n",
      "... New best solution\n",
      "... Procrustes: rmse 1.063378e-05  max resid 2.461795e-05 \n",
      "... Similar to previous best\n",
      "Run 12 stress 0.1504872 \n",
      "Run 13 stress 0.153349 \n",
      "Run 14 stress 0.1504872 \n",
      "Run 15 stress 0.149634 \n",
      "... Procrustes: rmse 2.440302e-05  max resid 5.273925e-05 \n",
      "... Similar to previous best\n",
      "Run 16 stress 0.1801135 \n",
      "Run 17 stress 0.1993237 \n",
      "Run 18 stress 0.1832015 \n",
      "Run 19 stress 0.1533488 \n",
      "Run 20 stress 0.1504872 \n",
      "*** Solution reached\n",
      "Warning message:\n",
      "In postMDS(out$points, dis, plot = max(0, plot - 1), ...) :\n",
      "  skipping half-change scaling: too few points below threshold\n",
      "Run 0 stress 0.08729864 \n",
      "Run 1 stress 0.182982 \n",
      "Run 2 stress 0.1998824 \n",
      "Run 3 stress 0.07961566 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1354051  max resid 0.2810598 \n",
      "Run 4 stress 0.1533932 \n",
      "Run 5 stress 0.1156391 \n",
      "Run 6 stress 0.1539733 \n",
      "Run 7 stress 0.1152752 \n",
      "Run 8 stress 0.07589596 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.09069849  max resid 0.2834531 \n",
      "Run 9 stress 0.07589228 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.001699567  max resid 0.00462971 \n",
      "... Similar to previous best\n",
      "Run 10 stress 0.1679324 \n",
      "Run 11 stress 0.1715548 \n",
      "Run 12 stress 0.1622773 \n",
      "Run 13 stress 0.1746743 \n",
      "Run 14 stress 0.1163007 \n",
      "Run 15 stress 0.1160921 \n",
      "Run 16 stress 0.1486479 \n",
      "Run 17 stress 0.1390891 \n",
      "Run 18 stress 0.1493274 \n",
      "Run 19 stress 0.1690413 \n",
      "Run 20 stress 0.1463911 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 32 rows containing missing values (geom_errorbar). \n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n"
     ]
    }
   ],
   "source": [
    "#run phyloseq\n",
    "!/Users/jon/amptk/util/phyloseq_graphs.R ion5_nosub.biom ion5_nosub.tree.phy ion5_nosub"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Treatment adonis using raupcrick\n",
    "```\n",
    "Call:\n",
    "adonis(formula = rp ~ metadata[[treatments[y]]], permutations = 9999) \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)  \n",
    "metadata[[treatments[y]]]  1    0.4751 0.47510  4.9297 0.26042 0.0373 *\n",
    "Residuals                 14    1.3493 0.09638         0.73958         \n",
    "Total                     15    1.8244                 1.00000         \n",
    "---\n",
    "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
    "```\n",
    "###NMDS using raup\n",
    "![alt text](ion5_nosub.Treatment.raup.pdf)\n",
    "###Treatment adonis using braycurtis\n",
    "```\n",
    "Call:\n",
    "adonis(formula = voturaw ~ metadata[[treatments[y]]], permutations = 9999,      method = \"bray\") \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1    0.4702 0.47019  1.0286 0.06844 0.4414\n",
    "Residuals                 14    6.3996 0.45711         0.93156       \n",
    "Total                     15    6.8697                 1.00000            \n",
    "```\n",
    "###NMDS using braycurtis\n",
    "![alt text](ion5_nosub.Treatment.braycurtis.pdf)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![alt text](summary.png)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Working on complete OTU table: found 1654 OTUs\n",
      "ERROR m4C-301-1, m4C-301-2, m4C-500-1, m4C-500-2, m4C-712-1, m4C-712-2, m4C-736-1, m4C-736-2, m4C-744-1, m4C-744-2, m4C-755-1, m4C-755-2, m4C-757-1, m4C-757-2, m4C-766-1, m4C-766-2, m4C-mock3-16000, m4C-mock3-32000a, m4C-mock3-32000b, m4C-stds, m4A-301-1, m4A-301-2, m4A-500-1, m4A-500-2, m4A-712-1, m4A-712-2, m4A-736-1, m4A-736-2, m4A-744-1, m4A-744-2, m4A-755-1, m4A-755-2, m4A-757-1, m4A-757-2, m4A-766-1, m4A-766-2, m4A-mock3-16000, m4A-mock3-32000a, m4A-mock3-32000b, m4A-stds, m3-mock3-3200a, m3-mock3-16000, m3-mock3-32000b, m3-stds, m5-301-1, m5-301-2, m5-500-1, m5-500-2, m5-712-1, m5-712-2, m5-736-1, m5-736-2, m5-744-1, m5-744-2, m5-755-1, m5-755-2, m5-757-1, m5-757-2, m5-766-1, m5-766-2, m5-SynMock1-8000, m5-SynMock2-8000, m5-mock3.0-16000, m5-mock3.5-16000: not found in OTU table. Names must max exactly.\n"
     ]
    }
   ],
   "source": [
    "#convert OTU tables for Michelle's R method, \n",
    "!amptk meta -i ion3.sorted.nomock.txt -m all_data.meta.csv -o ion3_raw.meta.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Working on complete OTU table: found 397 OTUs\r\n",
      "ERROR m4C-301-1, m4C-301-2, m4C-500-1, m4C-500-2, m4C-712-1, m4C-712-2, m4C-736-1, m4C-736-2, m4C-744-1, m4C-744-2, m4C-755-1, m4C-755-2, m4C-757-1, m4C-757-2, m4C-766-1, m4C-766-2, m4C-mock3-16000, m4C-mock3-32000a, m4C-mock3-32000b, m4C-stds, m4A-301-1, m4A-301-2, m4A-500-1, m4A-500-2, m4A-712-1, m4A-712-2, m4A-736-1, m4A-736-2, m4A-744-1, m4A-744-2, m4A-755-1, m4A-755-2, m4A-757-1, m4A-757-2, m4A-766-1, m4A-766-2, m4A-mock3-16000, m4A-mock3-32000a, m4A-mock3-32000b, m4A-stds, m3-mock3-3200a, m3-mock3-16000, m3-mock3-32000b, m3-stds, m5-301-1, m5-301-2, m5-500-1, m5-500-2, m5-712-1, m5-712-2, m5-736-1, m5-736-2, m5-744-1, m5-744-2, m5-755-1, m5-755-2, m5-757-1, m5-757-2, m5-766-1, m5-766-2, m5-SynMock1-8000, m5-SynMock2-8000, m5-mock3.0-16000, m5-mock3.5-16000: not found in OTU table. Names must max exactly.\r\n"
     ]
    }
   ],
   "source": [
    "!amptk meta -i ion3.final.txt -m all_data.meta.csv -o ion3_filtered.meta.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Working on complete OTU table: found 2179 OTUs\r\n",
      "ERROR m4C-301-1, m4C-301-2, m4C-500-1, m4C-500-2, m4C-712-1, m4C-712-2, m4C-736-1, m4C-736-2, m4C-744-1, m4C-744-2, m4C-755-1, m4C-755-2, m4C-757-1, m4C-757-2, m4C-766-1, m4C-766-2, m4C-mock3-16000, m4C-mock3-32000a, m4C-mock3-32000b, m4C-stds, m4A-mock3-16000, m4A-mock3-32000a, m4A-mock3-32000b, m4A-stds, m3-301-1, m3-301-2, m3-500-1, m3-500-2, m3-712-1, m3-712-2, m3-736-1, m3-736-2, m3-744-1, m3-744-2, m3-755-1, m3-755-2, m3-757-2, m3-766-1, m3-766-2, m3-mock3-3200a, m3-mock3-16000, m3-mock3-32000b, m3-stds, m5-301-1, m5-301-2, m5-500-1, m5-500-2, m5-712-1, m5-712-2, m5-736-1, m5-736-2, m5-744-1, m5-744-2, m5-755-1, m5-755-2, m5-757-1, m5-757-2, m5-766-1, m5-766-2, m5-SynMock1-8000, m5-SynMock2-8000, m5-mock3.0-16000, m5-mock3.5-16000: not found in OTU table. Names must max exactly.\r\n"
     ]
    }
   ],
   "source": [
    "!amptk meta -i miseq4a.sorted.nomock.txt -m all_data.meta.csv -o miseq4a_raw.meta.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Working on complete OTU table: found 446 OTUs\r\n",
      "ERROR m4C-301-1, m4C-301-2, m4C-500-1, m4C-500-2, m4C-712-1, m4C-712-2, m4C-736-1, m4C-736-2, m4C-744-1, m4C-744-2, m4C-755-1, m4C-755-2, m4C-757-1, m4C-757-2, m4C-766-1, m4C-766-2, m4C-mock3-16000, m4C-mock3-32000a, m4C-mock3-32000b, m4C-stds, m4A-mock3-16000, m4A-mock3-32000a, m4A-mock3-32000b, m4A-stds, m3-301-1, m3-301-2, m3-500-1, m3-500-2, m3-712-1, m3-712-2, m3-736-1, m3-736-2, m3-744-1, m3-744-2, m3-755-1, m3-755-2, m3-757-2, m3-766-1, m3-766-2, m3-mock3-3200a, m3-mock3-16000, m3-mock3-32000b, m3-stds, m5-301-1, m5-301-2, m5-500-1, m5-500-2, m5-712-1, m5-712-2, m5-736-1, m5-736-2, m5-744-1, m5-744-2, m5-755-1, m5-755-2, m5-757-1, m5-757-2, m5-766-1, m5-766-2, m5-SynMock1-8000, m5-SynMock2-8000, m5-mock3.0-16000, m5-mock3.5-16000: not found in OTU table. Names must max exactly.\r\n"
     ]
    }
   ],
   "source": [
    "!amptk meta -i miseq4a.final.txt -m all_data.meta.csv -o miseq4a_filtered.meta.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Working on complete OTU table: found 2109 OTUs\n",
      "ERROR m4C-mock3-16000, m4C-mock3-32000a, m4C-mock3-32000b, m4C-stds, m4A-301-1, m4A-301-2, m4A-500-1, m4A-500-2, m4A-712-1, m4A-712-2, m4A-736-1, m4A-736-2, m4A-744-1, m4A-744-2, m4A-755-1, m4A-755-2, m4A-757-1, m4A-757-2, m4A-766-1, m4A-766-2, m4A-mock3-16000, m4A-mock3-32000a, m4A-mock3-32000b, m4A-stds, m3-301-1, m3-301-2, m3-500-1, m3-500-2, m3-712-1, m3-712-2, m3-736-1, m3-736-2, m3-744-1, m3-744-2, m3-755-1, m3-755-2, m3-757-2, m3-766-1, m3-766-2, m3-mock3-3200a, m3-mock3-16000, m3-mock3-32000b, m3-stds, m5-301-1, m5-301-2, m5-500-1, m5-500-2, m5-712-1, m5-712-2, m5-736-1, m5-736-2, m5-744-1, m5-744-2, m5-755-1, m5-755-2, m5-757-1, m5-757-2, m5-766-1, m5-766-2, m5-SynMock1-8000, m5-SynMock2-8000, m5-mock3.0-16000, m5-mock3.5-16000: not found in OTU table. Names must max exactly.\n"
     ]
    }
   ],
   "source": [
    "!amptk meta -i miseq4c.sorted.nomock.txt -m all_data.meta.csv -o miseq4c_raw.meta.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Working on complete OTU table: found 502 OTUs\r\n",
      "ERROR m4C-mock3-16000, m4C-mock3-32000a, m4C-mock3-32000b, m4C-stds, m4A-301-1, m4A-301-2, m4A-500-1, m4A-500-2, m4A-712-1, m4A-712-2, m4A-736-1, m4A-736-2, m4A-744-1, m4A-744-2, m4A-755-1, m4A-755-2, m4A-757-1, m4A-757-2, m4A-766-1, m4A-766-2, m4A-mock3-16000, m4A-mock3-32000a, m4A-mock3-32000b, m4A-stds, m3-301-1, m3-301-2, m3-500-1, m3-500-2, m3-712-1, m3-712-2, m3-736-1, m3-736-2, m3-744-1, m3-744-2, m3-755-1, m3-755-2, m3-757-2, m3-766-1, m3-766-2, m3-mock3-3200a, m3-mock3-16000, m3-mock3-32000b, m3-stds, m5-301-1, m5-301-2, m5-500-1, m5-500-2, m5-712-1, m5-712-2, m5-736-1, m5-736-2, m5-744-1, m5-744-2, m5-755-1, m5-755-2, m5-757-1, m5-757-2, m5-766-1, m5-766-2, m5-SynMock1-8000, m5-SynMock2-8000, m5-mock3.0-16000, m5-mock3.5-16000: not found in OTU table. Names must max exactly.\r\n"
     ]
    }
   ],
   "source": [
    "!amptk meta -i miseq4c.final.txt -m all_data.meta.csv -o miseq4c_filtered.meta.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Working on complete OTU table: found 1584 OTUs\r\n",
      "ERROR m4C-301-1, m4C-301-2, m4C-500-1, m4C-500-2, m4C-712-1, m4C-712-2, m4C-736-1, m4C-736-2, m4C-744-1, m4C-744-2, m4C-755-1, m4C-755-2, m4C-757-1, m4C-757-2, m4C-766-1, m4C-766-2, m4C-mock3-16000, m4C-mock3-32000a, m4C-mock3-32000b, m4C-stds, m4A-301-1, m4A-301-2, m4A-500-1, m4A-500-2, m4A-712-1, m4A-712-2, m4A-736-1, m4A-736-2, m4A-744-1, m4A-744-2, m4A-755-1, m4A-755-2, m4A-757-1, m4A-757-2, m4A-766-1, m4A-766-2, m4A-mock3-16000, m4A-mock3-32000a, m4A-mock3-32000b, m4A-stds, m3-301-1, m3-301-2, m3-500-1, m3-500-2, m3-712-1, m3-712-2, m3-736-1, m3-736-2, m3-744-1, m3-744-2, m3-755-1, m3-755-2, m3-757-2, m3-766-1, m3-766-2, m3-mock3-3200a, m3-mock3-16000, m3-mock3-32000b, m3-stds, m5-SynMock1-8000, m5-SynMock2-8000, m5-mock3.0-16000, m5-mock3.5-16000: not found in OTU table. Names must max exactly.\r\n"
     ]
    }
   ],
   "source": [
    "!amptk meta -i ion5.sorted.nomock.txt -m all_data.meta.csv -o ion5_raw.meta.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Working on complete OTU table: found 369 OTUs\n",
      "ERROR m4C-301-1, m4C-301-2, m4C-500-1, m4C-500-2, m4C-712-1, m4C-712-2, m4C-736-1, m4C-736-2, m4C-744-1, m4C-744-2, m4C-755-1, m4C-755-2, m4C-757-1, m4C-757-2, m4C-766-1, m4C-766-2, m4C-mock3-16000, m4C-mock3-32000a, m4C-mock3-32000b, m4C-stds, m4A-301-1, m4A-301-2, m4A-500-1, m4A-500-2, m4A-712-1, m4A-712-2, m4A-736-1, m4A-736-2, m4A-744-1, m4A-744-2, m4A-755-1, m4A-755-2, m4A-757-1, m4A-757-2, m4A-766-1, m4A-766-2, m4A-mock3-16000, m4A-mock3-32000a, m4A-mock3-32000b, m4A-stds, m3-301-1, m3-301-2, m3-500-1, m3-500-2, m3-712-1, m3-712-2, m3-736-1, m3-736-2, m3-744-1, m3-744-2, m3-755-1, m3-755-2, m3-757-2, m3-766-1, m3-766-2, m3-mock3-3200a, m3-mock3-16000, m3-mock3-32000b, m3-stds, m5-SynMock1-8000, m5-SynMock2-8000, m5-mock3.0-16000, m5-mock3.5-16000: not found in OTU table. Names must max exactly.\n"
     ]
    }
   ],
   "source": [
    "!amptk meta -i ion5.final.txt -m all_data.meta.csv -o ion5_filtered.meta.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removed 3 samples\n",
      "Kept 2407551 reads out of 2879144 total reads\n",
      "----------------------------------\n",
      "Found 16 barcoded samples\n",
      "              Sample:  Count\n",
      "            m3-766-1:  205305\n",
      "      m3-mock3-16000:  183277\n",
      "            m3-301-2:  170879\n",
      "            m3-744-2:  160175\n",
      "            m3-500-1:  158095\n",
      "            m3-500-2:  152733\n",
      "            m3-766-2:  152560\n",
      "            m3-301-1:  151264\n",
      "            m3-755-1:  150229\n",
      "            m3-736-1:  148309\n",
      "            m3-755-2:  138299\n",
      "            m3-736-2:  134956\n",
      "            m3-757-2:  134727\n",
      "            m3-744-1:  130294\n",
      "            m3-712-1:  125701\n",
      "            m3-712-2:  110748\n",
      "----------------------------------\n",
      "Read count: 2407551\n",
      "Read Length: 250 bp\n",
      "----------------------------------\n"
     ]
    }
   ],
   "source": [
    "#now lets go back and process each of these runs separately, i.e. akin to how you would normally run amptk\n",
    "#first select the reads you want from original demuxing\n",
    "!amptk remove -i ion3.demux.fq -l m3-mock3-32000b m3-mock3-3200a m3-stds -o ion3.cleaned.demux.fq\n",
    "!amptk show -i ion3.cleaned.demux.fq\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[03:01:11 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[03:01:11 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[03:01:11 PM]\u001b[0m: Loading FASTQ Records\n",
      "\u001b[92m[03:01:13 PM]\u001b[0m: 2,407,551 reads (1.2 GB)\n",
      "\u001b[92m[03:01:13 PM]\u001b[0m: Quality Filtering, expected errors < 1.0\n",
      "\u001b[92m[03:02:09 PM]\u001b[0m: 1,416,710 reads passed\n",
      "\u001b[92m[03:02:09 PM]\u001b[0m: De-replication (remove duplicate reads)\n",
      "\u001b[92m[03:02:14 PM]\u001b[0m: 57,855 reads passed\n",
      "\u001b[92m[03:02:14 PM]\u001b[0m: Clustering OTUs (UPARSE)\n",
      "\u001b[92m[03:02:21 PM]\u001b[0m: 1,057 OTUs\n",
      "\u001b[92m[03:02:21 PM]\u001b[0m: Cleaning up padding from OTUs\n",
      "\u001b[92m[03:02:21 PM]\u001b[0m: Mapping Reads to OTUs and Building OTU table\n",
      "\u001b[92m[03:06:49 PM]\u001b[0m: 1,938,679 reads mapped to OTUs (81%)\n",
      "-------------------------------------------------------\n",
      "OTU Clustering Script has Finished Successfully\n",
      "-------------------------------------------------------\n",
      "Clustered OTUs: /Users/jon/projects/synmock_paper/i3.cluster.otus.fa\n",
      "OTU Table: /Users/jon/projects/synmock_paper/i3.otu_table.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk filter -i i3.otu_table.txt -f i3.cluster.otus.fa -b <mock barcode>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#now cluster the cleaned up data\n",
    "!amptk cluster -i ion3.cleaned.demux.fq -o i3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[03:07:47 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[03:07:47 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[03:07:47 PM]\u001b[0m: Loading OTU table: i3.otu_table.txt\n",
      "\u001b[92m[03:07:47 PM]\u001b[0m: OTU table contains 1057 OTUs\n",
      "\u001b[92m[03:07:47 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[03:07:47 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[03:07:47 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[03:07:47 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[03:07:47 PM]\u001b[0m: Index bleed, samples into mock: 0.047999%.\n",
      "\u001b[92m[03:07:47 PM]\u001b[0m: Will use value of 0.100000% for index-bleed OTU filtering.\n",
      "\u001b[92m[03:07:48 PM]\u001b[0m: Filtering OTU table down to 1035 OTUs\n",
      "\u001b[92m[03:07:48 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      i3_nosub.stats.txt\n",
      "Sorted OTU table:     i3_nosub.sorted.txt\n",
      "Normalized/filter:    i3_nosub.normalized.txt\n",
      "Final Binary table:   i3_nosub.final.binary.txt\n",
      "Final OTU table:      i3_nosub.final.txt\n",
      "Filtered OTUs:        i3_nosub.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f i3_nosub.filtered.otus.fa -i i3_nosub.final.txt -m mapping_file.txt -d ITS2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#run filtering with no subtraction first\n",
    "!amptk filter -i i3.otu_table.txt -f i3.cluster.otus.fa -b m3-mock3-16000 -m mock3 -c in -o i3_nosub"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1057 16 1051 15\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[03:10:29 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[03:10:29 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[03:10:29 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[03:10:29 PM]\u001b[0m: 1,057 OTUs\n",
      "\u001b[92m[03:10:29 PM]\u001b[0m: Global alignment OTUs with usearch_global (USEARCH)\n",
      "\u001b[92m[03:10:42 PM]\u001b[0m: Classifying OTUs with UTAX (USEARCH)\n",
      "\u001b[92m[03:10:42 PM]\u001b[0m: Classifying OTUs with SINTAX (USEARCH)\n",
      "\u001b[92m[03:13:40 PM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[03:13:40 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[03:13:43 PM]\u001b[0m: Taxonomy finished: i3_raw.taxonomy.txt\n",
      "\u001b[92m[03:13:43 PM]\u001b[0m: Classic OTU table with taxonomy: i3_raw.otu_table.taxonomy.txt\n",
      "\u001b[92m[03:13:44 PM]\u001b[0m: BIOM OTU table created: i3_raw.biom\n",
      "\u001b[92m[03:13:44 PM]\u001b[0m: OTUs with taxonomy: i3_raw.otus.taxonomy.fa\n",
      "\u001b[92m[03:13:44 PM]\u001b[0m: OTU phylogeny: i3_raw.tree.phy\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#get raw data into biom format\n",
    "import pandas as pd\n",
    "from natsort import natsorted\n",
    "#pull out header for indexing later\n",
    "with open('i3_nosub.sorted.txt', 'rU') as f:\n",
    "    first_line = f.readline()\n",
    "    OTUhead = first_line.split('\\t')[0]\n",
    "    \n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "dfsorted = pd.read_csv('i3_nosub.sorted.txt', sep='\\t')\n",
    "dfsorted.set_index(OTUhead, inplace=True)\n",
    "\n",
    "#now we want to drop last column\n",
    "header = ['m3-301-1', 'm3-301-2', 'm3-500-1', 'm3-500-2', 'm3-712-1', 'm3-712-2', \n",
    "        'm3-736-1', 'm3-736-2', 'm3-744-1', 'm3-744-2', 'm3-755-1', 'm3-755-2',\n",
    "        'm3-757-2', 'm3-766-1', 'm3-766-2']\n",
    "dfcleaned = dfsorted.reindex(columns=header, index=natsorted(dfsorted.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "fs = dfcleaned.sum(axis=0)\n",
    "filtered = pd.DataFrame(dfcleaned, columns=fs.index)\n",
    "filt2 = filtered.loc[(filtered != 0).any(1)]\n",
    "print len(dfsorted), len(dfsorted.columns.values), len(filt2), len(filt2.columns.values)\n",
    "filt2.to_csv('i3.sorted.nomock.txt', sep='\\t')\n",
    "\n",
    "#now convert to biom\n",
    "!amptk taxonomy -f i3.cluster.otus.fa -o i3_raw -i i3.sorted.nomock.txt -d ITS2 -m ion3.mapping_file.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-2\n",
      "[1] ‘1.19.1’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘1.8.4’\n",
      "[1] ‘2.4.2’\n",
      "There were 21 warnings (use warnings() to see them)\n",
      "Square root transformation\n",
      "Wisconsin double standardization\n",
      "Run 0 stress 0.1519626 \n",
      "Run 1 stress 0.1722504 \n",
      "Run 2 stress 0.1520057 \n",
      "... Procrustes: rmse 0.005564915  max resid 0.01840075 \n",
      "Run 3 stress 0.1621586 \n",
      "Run 4 stress 0.1704118 \n",
      "Run 5 stress 0.1709248 \n",
      "Run 6 stress 0.1519513 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.002580261  max resid 0.008318369 \n",
      "... Similar to previous best\n",
      "Run 7 stress 0.1706664 \n",
      "Run 8 stress 0.1599569 \n",
      "Run 9 stress 0.1519513 \n",
      "... Procrustes: rmse 4.442832e-05  max resid 0.0001306989 \n",
      "... Similar to previous best\n",
      "Run 10 stress 0.1617357 \n",
      "Run 11 stress 0.1627957 \n",
      "Run 12 stress 0.1519614 \n",
      "... Procrustes: rmse 0.002304336  max resid 0.007413293 \n",
      "... Similar to previous best\n",
      "Run 13 stress 0.1803962 \n",
      "Run 14 stress 0.1879302 \n",
      "Run 15 stress 0.1519513 \n",
      "... Procrustes: rmse 3.556894e-05  max resid 9.959337e-05 \n",
      "... Similar to previous best\n",
      "Run 16 stress 0.1709243 \n",
      "Run 17 stress 0.1619448 \n",
      "Run 18 stress 0.1617356 \n",
      "Run 19 stress 0.1519513 \n",
      "... Procrustes: rmse 4.144049e-05  max resid 0.0001197292 \n",
      "... Similar to previous best\n",
      "Run 20 stress 0.1619452 \n",
      "*** Solution reached\n",
      "Run 0 stress 0.09784604 \n",
      "Run 1 stress 0.1899021 \n",
      "Run 2 stress 0.1567988 \n",
      "Run 3 stress 0.168149 \n",
      "Run 4 stress 0.1897043 \n",
      "Run 5 stress 0.1923292 \n",
      "Run 6 stress 0.1515514 \n",
      "Run 7 stress 0.1719854 \n",
      "Run 8 stress 0.08009251 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1355862  max resid 0.3202882 \n",
      "Run 9 stress 0.189676 \n",
      "Run 10 stress 0.1957344 \n",
      "Run 11 stress 0.1697107 \n",
      "Run 12 stress 0.1288628 \n",
      "Run 13 stress 0.1165386 \n",
      "Run 14 stress 0.1168816 \n",
      "Run 15 stress 0.1974168 \n",
      "Run 16 stress 0.08009375 \n",
      "... Procrustes: rmse 0.0003667386  max resid 0.0007369447 \n",
      "... Similar to previous best\n",
      "Run 17 stress 0.1665984 \n",
      "Run 18 stress 0.1806417 \n",
      "Run 19 stress 0.1680685 \n",
      "Run 20 stress 0.1761956 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 30 rows containing missing values (geom_errorbar). \n",
      "2: Removed 30 rows containing missing values (geom_errorbar). \n"
     ]
    }
   ],
   "source": [
    "#run phyloseq stats\n",
    "!/Users/jon/amptk/util/phyloseq_graphs.R i3_raw.biom i3_raw.tree.phy i3_raw"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Treatment adonis using raupcrick\n",
    "```\n",
    "Call:\n",
    "adonis(formula = rp ~ metadata[[treatments[y]]], permutations = 9999) \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1   0.20506 0.20506  1.7094 0.11621 0.3175\n",
    "Residuals                 13   1.55944 0.11996         0.88379       \n",
    "Total                     14   1.76449                 1.00000  \n",
    "```\n",
    "###NMDS using raup\n",
    "![alt text](i3_raw.Treatment.raup.pdf)\n",
    "###Treatment adonis using braycurtis\n",
    "```\n",
    "Call:\n",
    "adonis(formula = voturaw ~ metadata[[treatments[y]]], permutations = 9999,      method = \"bray\") \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1    0.4533 0.45329  1.0128 0.07228 0.4644\n",
    "Residuals                 13    5.8185 0.44757         0.92772       \n",
    "Total                     14    6.2717                 1.00000\n",
    "```\n",
    "###NMDS using braycurtis\n",
    "![alt text](i3_raw.Treatment.braycurtis.pdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[03:17:52 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[03:17:52 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[03:17:52 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[03:17:52 PM]\u001b[0m: 1,035 OTUs\n",
      "\u001b[92m[03:17:52 PM]\u001b[0m: Global alignment OTUs with usearch_global (USEARCH)\n",
      "\u001b[92m[03:18:02 PM]\u001b[0m: Classifying OTUs with UTAX (USEARCH)\n",
      "\u001b[92m[03:18:03 PM]\u001b[0m: Classifying OTUs with SINTAX (USEARCH)\n",
      "\u001b[92m[03:21:00 PM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[03:21:01 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[03:21:03 PM]\u001b[0m: Taxonomy finished: i3_nosub.taxonomy.txt\n",
      "\u001b[92m[03:21:03 PM]\u001b[0m: Classic OTU table with taxonomy: i3_nosub.otu_table.taxonomy.txt\n",
      "\u001b[92m[03:21:05 PM]\u001b[0m: BIOM OTU table created: i3_nosub.biom\n",
      "\u001b[92m[03:21:05 PM]\u001b[0m: OTUs with taxonomy: i3_nosub.otus.taxonomy.fa\n",
      "\u001b[92m[03:21:05 PM]\u001b[0m: OTU phylogeny: i3_nosub.tree.phy\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#convert no subtraction to biom\n",
    "!amptk taxonomy -f i3_nosub.filtered.otus.fa -o i3_nosub -i i3_nosub.final.txt -d ITS2 -m ion3.mapping_file.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-2\n",
      "[1] ‘1.19.1’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘1.8.4’\n",
      "[1] ‘2.4.2’\n",
      "Square root transformation\n",
      "Wisconsin double standardization\n",
      "Run 0 stress 0.1482358 \n",
      "Run 1 stress 0.1591481 \n",
      "Run 2 stress 0.1482358 \n",
      "... Procrustes: rmse 8.489064e-06  max resid 1.375638e-05 \n",
      "... Similar to previous best\n",
      "Run 3 stress 0.1655535 \n",
      "Run 4 stress 0.2988611 \n",
      "Run 5 stress 0.1694109 \n",
      "Run 6 stress 0.1591476 \n",
      "Run 7 stress 0.165558 \n",
      "Run 8 stress 0.1675681 \n",
      "Run 9 stress 0.1675665 \n",
      "Run 10 stress 0.1675664 \n",
      "Run 11 stress 0.1667055 \n",
      "Run 12 stress 0.1482358 \n",
      "... Procrustes: rmse 0.000125241  max resid 0.0003721117 \n",
      "... Similar to previous best\n",
      "Run 13 stress 0.1630948 \n",
      "Run 14 stress 0.1719454 \n",
      "Run 15 stress 0.1591471 \n",
      "Run 16 stress 0.1630945 \n",
      "Run 17 stress 0.1482358 \n",
      "... Procrustes: rmse 1.381914e-05  max resid 3.85609e-05 \n",
      "... Similar to previous best\n",
      "Run 18 stress 0.1563584 \n",
      "Run 19 stress 0.1482358 \n",
      "... New best solution\n",
      "... Procrustes: rmse 1.961794e-05  max resid 5.803114e-05 \n",
      "... Similar to previous best\n",
      "Run 20 stress 0.1591472 \n",
      "*** Solution reached\n",
      "Run 0 stress 0.09732912 \n",
      "Run 1 stress 0.2139516 \n",
      "Run 2 stress 0.1821734 \n",
      "Run 3 stress 0.1990502 \n",
      "Run 4 stress 0.1980635 \n",
      "Run 5 stress 0.1751946 \n",
      "Run 6 stress 0.1875651 \n",
      "Run 7 stress 0.1962009 \n",
      "Run 8 stress 0.152476 \n",
      "Run 9 stress 0.2147076 \n",
      "Run 10 stress 0.1292834 \n",
      "Run 11 stress 0.129283 \n",
      "Run 12 stress 0.119263 \n",
      "Run 13 stress 0.20847 \n",
      "Run 14 stress 0.1956204 \n",
      "Run 15 stress 0.1837913 \n",
      "Run 16 stress 0.2266988 \n",
      "Run 17 stress 0.1581709 \n",
      "Run 18 stress 0.1860317 \n",
      "Run 19 stress 0.1935291 \n",
      "Run 20 stress 0.2110564 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    12: stress ratio > sratmax\n",
      "     8: scale factor of the gradient < sfgrmin\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 30 rows containing missing values (geom_errorbar). \n",
      "2: Removed 30 rows containing missing values (geom_errorbar). \n"
     ]
    }
   ],
   "source": [
    "#run phyloseq stats\n",
    "!/Users/jon/amptk/util/phyloseq_graphs.R i3_nosub.biom i3_nosub.tree.phy i3_nosub"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Treatment adonis using raupcrick\n",
    "```\n",
    "Call:\n",
    "adonis(formula = rp ~ metadata[[treatments[y]]], permutations = 9999) \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1   0.26727 0.26727  2.1073 0.13949 0.2208\n",
    "Residuals                 13   1.64879 0.12683         0.86051       \n",
    "Total                     14   1.91606                 1.00000  \n",
    "```\n",
    "###NMDS using raup\n",
    "![alt text](i3_raw.Treatment.raup.pdf)\n",
    "###Treatment adonis using braycurtis\n",
    "```\n",
    "Call:\n",
    "adonis(formula = voturaw ~ metadata[[treatments[y]]], permutations = 9999,      method = \"bray\") \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1    0.4524 0.45240  1.0095 0.07206 0.4718\n",
    "Residuals                 13    5.8260 0.44815         0.92794       \n",
    "Total                     14    6.2783                 1.00000       \n",
    "\n",
    "```\n",
    "###NMDS using braycurtis\n",
    "![alt text](i3_raw.Treatment.braycurtis.pdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[03:24:56 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[03:24:56 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[03:24:56 PM]\u001b[0m: Loading OTU table: i3.otu_table.txt\n",
      "\u001b[92m[03:24:56 PM]\u001b[0m: OTU table contains 1057 OTUs\n",
      "\u001b[92m[03:24:56 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[03:24:57 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[03:24:57 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[03:24:57 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[03:24:57 PM]\u001b[0m: Index bleed, samples into mock: 0.047999%.\n",
      "\u001b[92m[03:24:57 PM]\u001b[0m: Will use value of 0.100000% for index-bleed OTU filtering.\n",
      "\u001b[92m[03:24:57 PM]\u001b[0m: Auto subtract filter set to 31\n",
      "\u001b[92m[03:24:57 PM]\u001b[0m: Subtracting 31 from OTU table\n",
      "\u001b[92m[03:24:57 PM]\u001b[0m: Filtering OTU table down to 362 OTUs\n",
      "\u001b[92m[03:24:57 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      i3.stats.txt\n",
      "Sorted OTU table:     i3.sorted.txt\n",
      "Normalized/filter:    i3.normalized.txt\n",
      "Final Binary table:   i3.final.binary.txt\n",
      "Final OTU table:      i3.final.txt\n",
      "Filtered OTUs:        i3.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f i3.filtered.otus.fa -i i3.final.txt -m mapping_file.txt -d ITS2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#now run filtering with auto subtraction filter\n",
    "#run filtering with no subtraction first\n",
    "!amptk filter -i i3.otu_table.txt -f i3.cluster.otus.fa -b m3-mock3-16000 -m mock3 -c in -o i3 -s auto"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[03:25:34 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[03:25:34 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[03:25:34 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[03:25:34 PM]\u001b[0m: 362 OTUs\n",
      "\u001b[92m[03:25:34 PM]\u001b[0m: Global alignment OTUs with usearch_global (USEARCH)\n",
      "\u001b[92m[03:25:39 PM]\u001b[0m: Classifying OTUs with UTAX (USEARCH)\n",
      "\u001b[92m[03:25:39 PM]\u001b[0m: Classifying OTUs with SINTAX (USEARCH)\n",
      "\u001b[92m[03:27:11 PM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[03:27:11 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[03:27:11 PM]\u001b[0m: Taxonomy finished: i3.taxonomy.txt\n",
      "\u001b[92m[03:27:11 PM]\u001b[0m: Classic OTU table with taxonomy: i3.otu_table.taxonomy.txt\n",
      "\u001b[92m[03:27:12 PM]\u001b[0m: BIOM OTU table created: i3.biom\n",
      "\u001b[92m[03:27:12 PM]\u001b[0m: OTUs with taxonomy: i3.otus.taxonomy.fa\n",
      "\u001b[92m[03:27:12 PM]\u001b[0m: OTU phylogeny: i3.tree.phy\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "!amptk taxonomy -f i3.filtered.otus.fa -i i3.final.txt -m ion3.mapping_file.txt -d ITS2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-2\n",
      "[1] ‘1.19.1’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘1.8.4’\n",
      "[1] ‘2.4.2’\n",
      "Square root transformation\n",
      "Wisconsin double standardization\n",
      "Run 0 stress 0.2033973 \n",
      "Run 1 stress 0.1957146 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1952427  max resid 0.6547199 \n",
      "Run 2 stress 0.2118633 \n",
      "Run 3 stress 0.2039372 \n",
      "Run 4 stress 0.2173894 \n",
      "Run 5 stress 0.2114433 \n",
      "Run 6 stress 0.215529 \n",
      "Run 7 stress 0.2144408 \n",
      "Run 8 stress 0.2355701 \n",
      "Run 9 stress 0.290856 \n",
      "Run 10 stress 0.1990067 \n",
      "Run 11 stress 0.1957148 \n",
      "... Procrustes: rmse 9.06666e-05  max resid 0.0001956514 \n",
      "... Similar to previous best\n",
      "Run 12 stress 0.2237672 \n",
      "Run 13 stress 0.1957148 \n",
      "... Procrustes: rmse 0.0002491816  max resid 0.0005394005 \n",
      "... Similar to previous best\n",
      "Run 14 stress 0.1957148 \n",
      "... Procrustes: rmse 0.000301615  max resid 0.0008246696 \n",
      "... Similar to previous best\n",
      "Run 15 stress 0.2215594 \n",
      "Run 16 stress 0.2120979 \n",
      "Run 17 stress 0.2192804 \n",
      "Run 18 stress 0.2033974 \n",
      "Run 19 stress 0.2014188 \n",
      "Run 20 stress 0.2072964 \n",
      "*** Solution reached\n",
      "Warning message:\n",
      "In postMDS(out$points, dis, plot = max(0, plot - 1), ...) :\n",
      "  skipping half-change scaling: too few points below threshold\n",
      "Run 0 stress 0.2099401 \n",
      "Run 1 stress 0.2743025 \n",
      "Run 2 stress 0.2112683 \n",
      "Run 3 stress 0.2459972 \n",
      "Run 4 stress 0.2112604 \n",
      "Run 5 stress 0.2411984 \n",
      "Run 6 stress 0.2490113 \n",
      "Run 7 stress 0.2473437 \n",
      "Run 8 stress 0.245435 \n",
      "Run 9 stress 0.2336774 \n",
      "Run 10 stress 0.2322687 \n",
      "Run 11 stress 0.2451355 \n",
      "Run 12 stress 0.2428385 \n",
      "Run 13 stress 0.2099401 \n",
      "... Procrustes: rmse 0.0001400397  max resid 0.0003716665 \n",
      "... Similar to previous best\n",
      "Run 14 stress 0.2483312 \n",
      "Run 15 stress 0.2368555 \n",
      "Run 16 stress 0.2463488 \n",
      "Run 17 stress 0.2124582 \n",
      "Run 18 stress 0.2112595 \n",
      "Run 19 stress 0.2362564 \n",
      "Run 20 stress 0.2132611 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: In estimate_richness(physeq, split = TRUE, measures = measures) :\n",
      "  The data you have provided does not have\n",
      "any singletons. This is highly suspicious. Results of richness\n",
      "estimates (for example) are probably unreliable, or wrong, if you have already\n",
      "trimmed low-abundance taxa from the data.\n",
      "\n",
      "We recommended that you find the un-trimmed data and retry.\n",
      "2: Removed 30 rows containing missing values (geom_errorbar). \n",
      "3: In estimate_richness(physeq, split = TRUE, measures = measures) :\n",
      "  The data you have provided does not have\n",
      "any singletons. This is highly suspicious. Results of richness\n",
      "estimates (for example) are probably unreliable, or wrong, if you have already\n",
      "trimmed low-abundance taxa from the data.\n",
      "\n",
      "We recommended that you find the un-trimmed data and retry.\n",
      "4: Removed 30 rows containing missing values (geom_errorbar). \n"
     ]
    }
   ],
   "source": [
    "#run phyloseq stats\n",
    "!/Users/jon/amptk/util/phyloseq_graphs.R i3.biom i3.tree.phy i3"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Treatment adonis using raupcrick\n",
    "```\n",
    "Call:\n",
    "adonis(formula = rp ~ metadata[[treatments[y]]], permutations = 9999) \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model     R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1   0.41328 0.41328  2.1657 0.1428 0.1381\n",
    "Residuals                 13   2.48086 0.19084         0.8572       \n",
    "Total                     14   2.89415                 1.0000\n",
    "```\n",
    "###NMDS using raup\n",
    "![alt text](i3.Treatment.raup.pdf)\n",
    "###Treatment adonis using braycurtis\n",
    "```\n",
    "Call:\n",
    "adonis(formula = voturaw ~ metadata[[treatments[y]]], permutations = 9999,      method = \"bray\") \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1    0.4537 0.45371  1.0057 0.07181 0.4748\n",
    "Residuals                 13    5.8648 0.45114         0.92819       \n",
    "Total                     14    6.3185                 1.00000   \n",
    "```\n",
    "###NMDS using braycurtis\n",
    "![alt text](i3.Treatment.braycurtis.pdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removed 3 samples\n",
      "Kept 4809024 reads out of 5628414 total reads\n",
      "----------------------------------\n",
      "Found 17 barcoded samples\n",
      "              Sample:  Count\n",
      "           m4A-500-1:  332586\n",
      "           m4A-766-2:  321816\n",
      "           m4A-755-1:  302871\n",
      "           m4A-766-1:  302038\n",
      "           m4A-755-2:  293758\n",
      "           m4A-301-2:  293694\n",
      "     m4A-mock3-16000:  282380\n",
      "           m4A-744-1:  281436\n",
      "           m4A-757-1:  280341\n",
      "           m4A-736-1:  277532\n",
      "           m4A-757-2:  274936\n",
      "           m4A-500-2:  274015\n",
      "           m4A-301-1:  268266\n",
      "           m4A-712-1:  264849\n",
      "           m4A-736-2:  261875\n",
      "           m4A-744-2:  256458\n",
      "           m4A-712-2:  240173\n",
      "----------------------------------\n",
      "Read count: 4809024\n",
      "Read Length: 250 bp\n",
      "----------------------------------\n"
     ]
    }
   ],
   "source": [
    "#Ok, move onto miseqA run\n",
    "#now lets go back and process each of these runs separately, i.e. akin to how you would normally run amptk\n",
    "#first select the reads you want from original demuxing\n",
    "!amptk remove -i m4a.demux.fq -l m4A-mock3-32000b m4A-mock3-32000a m4A-stds -o miseq4a.cleaned.demux.fq\n",
    "!amptk show -i miseq4a.cleaned.demux.fq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[03:46:30 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[03:46:30 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[03:46:30 PM]\u001b[0m: Loading FASTQ Records\n",
      "\u001b[92m[03:46:33 PM]\u001b[0m: 4,809,024 reads (2.4 GB)\n",
      "\u001b[92m[03:46:33 PM]\u001b[0m: Quality Filtering, expected errors < 1.0\n",
      "\u001b[92m[03:48:30 PM]\u001b[0m: 4,462,394 reads passed\n",
      "\u001b[92m[03:48:30 PM]\u001b[0m: De-replication (remove duplicate reads)\n",
      "\u001b[92m[03:48:44 PM]\u001b[0m: 190,777 reads passed\n",
      "\u001b[92m[03:48:45 PM]\u001b[0m: Clustering OTUs (UPARSE)\n",
      "\u001b[92m[03:49:01 PM]\u001b[0m: 1,803 OTUs\n",
      "\u001b[92m[03:49:01 PM]\u001b[0m: Cleaning up padding from OTUs\n",
      "\u001b[92m[03:49:01 PM]\u001b[0m: Mapping Reads to OTUs and Building OTU table\n",
      "\u001b[92m[03:56:43 PM]\u001b[0m: 4,611,275 reads mapped to OTUs (96%)\n",
      "-------------------------------------------------------\n",
      "OTU Clustering Script has Finished Successfully\n",
      "-------------------------------------------------------\n",
      "Clustered OTUs: /Users/jon/projects/synmock_paper/m4a.cluster.otus.fa\n",
      "OTU Table: /Users/jon/projects/synmock_paper/m4a.otu_table.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk filter -i m4a.otu_table.txt -f m4a.cluster.otus.fa -b <mock barcode>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#now cluster the cleaned up data\n",
    "!amptk cluster -i miseq4a.cleaned.demux.fq -o m4a"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[05:07:35 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[05:07:35 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[05:07:35 PM]\u001b[0m: Loading OTU table: m4a.otu_table.txt\n",
      "\u001b[92m[05:07:35 PM]\u001b[0m: OTU table contains 1798 OTUs\n",
      "\u001b[92m[05:07:35 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[05:07:35 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[05:07:35 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[05:07:35 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[05:07:35 PM]\u001b[0m: Index bleed, samples into mock: 0.054001%.\n",
      "\u001b[92m[05:07:35 PM]\u001b[0m: Overwriting auto detect index-bleed, setting to 0.500000%\n",
      "\u001b[92m[05:07:36 PM]\u001b[0m: Filtering OTU table down to 1752 OTUs\n",
      "\u001b[92m[05:07:36 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      m4a_nosub.stats.txt\n",
      "Sorted OTU table:     m4a_nosub.sorted.txt\n",
      "Normalized/filter:    m4a_nosub.normalized.txt\n",
      "Final Binary table:   m4a_nosub.final.binary.txt\n",
      "Final OTU table:      m4a_nosub.final.txt\n",
      "Filtered OTUs:        m4a_nosub.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f m4a_nosub.filtered.otus.fa -i m4a_nosub.final.txt -m mapping_file.txt -d ITS2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#run filtering with no subtraction first\n",
    "!amptk filter -i m4a.otu_table.txt -f m4a.cluster.otus.fa -b m4A-mock3-16000 -m mock3 -c in -o m4a_nosub -p 0.005"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1798 17 1797 16\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[03:57:02 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[03:57:02 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[03:57:02 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[03:57:02 PM]\u001b[0m: 1,803 OTUs\n",
      "\u001b[92m[03:57:02 PM]\u001b[0m: Global alignment OTUs with usearch_global (USEARCH)\n",
      "\u001b[92m[03:57:20 PM]\u001b[0m: Classifying OTUs with UTAX (USEARCH)\n",
      "\u001b[92m[03:57:21 PM]\u001b[0m: Classifying OTUs with SINTAX (USEARCH)\n",
      "\u001b[92m[04:01:48 PM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[04:01:49 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[04:01:55 PM]\u001b[0m: Taxonomy finished: m4a_raw.taxonomy.txt\n",
      "\u001b[92m[04:01:55 PM]\u001b[0m: Classic OTU table with taxonomy: m4a_raw.otu_table.taxonomy.txt\n",
      "\u001b[92m[04:01:57 PM]\u001b[0m: BIOM OTU table created: m4a_raw.biom\n",
      "\u001b[92m[04:01:57 PM]\u001b[0m: OTUs with taxonomy: m4a_raw.otus.taxonomy.fa\n",
      "\u001b[92m[04:01:57 PM]\u001b[0m: OTU phylogeny: m4a_raw.tree.phy\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#get raw data into biom format\n",
    "import pandas as pd\n",
    "from natsort import natsorted\n",
    "#pull out header for indexing later\n",
    "with open('m4a_nosub.sorted.txt', 'rU') as f:\n",
    "    first_line = f.readline()\n",
    "    OTUhead = first_line.split('\\t')[0]\n",
    "    \n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "dfsorted = pd.read_csv('m4a_nosub.sorted.txt', sep='\\t')\n",
    "dfsorted.set_index(OTUhead, inplace=True)\n",
    "\n",
    "#now we want to drop last column\n",
    "header = ['m4A-301-1', 'm4A-301-2', 'm4A-500-1', 'm4A-500-2', 'm4A-712-1', \n",
    "           'm4A-712-2', 'm4A-736-1', 'm4A-736-2', 'm4A-744-1', 'm4A-744-2', \n",
    "           'm4A-755-1', 'm4A-755-2', 'm4A-757-1', 'm4A-757-2', 'm4A-766-1', 'm4A-766-2']\n",
    "dfcleaned = dfsorted.reindex(columns=header, index=natsorted(dfsorted.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "fs = dfcleaned.sum(axis=0)\n",
    "filtered = pd.DataFrame(dfcleaned, columns=fs.index)\n",
    "filt2 = filtered.loc[(filtered != 0).any(1)]\n",
    "print len(dfsorted), len(dfsorted.columns.values), len(filt2), len(filt2.columns.values)\n",
    "filt2.to_csv('m4a.sorted.nomock.txt', sep='\\t')\n",
    "\n",
    "#now convert to biom\n",
    "!amptk taxonomy -f m4a.cluster.otus.fa -o m4a_raw -i m4a.sorted.nomock.txt -d ITS2 -m m4a.mapping_file.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-2\n",
      "[1] ‘1.19.1’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘1.8.4’\n",
      "[1] ‘2.4.2’\n",
      "There were 27 warnings (use warnings() to see them)\n",
      "Square root transformation\n",
      "Wisconsin double standardization\n",
      "Run 0 stress 0.1594448 \n",
      "Run 1 stress 0.1531269 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1510224  max resid 0.3898098 \n",
      "Run 2 stress 0.1609605 \n",
      "Run 3 stress 0.1609604 \n",
      "Run 4 stress 0.1527909 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1883387  max resid 0.5795488 \n",
      "Run 5 stress 0.2179276 \n",
      "Run 6 stress 0.153127 \n",
      "... Procrustes: rmse 0.1883577  max resid 0.5697554 \n",
      "Run 7 stress 0.1531268 \n",
      "... Procrustes: rmse 0.1883261  max resid 0.5697011 \n",
      "Run 8 stress 0.158719 \n",
      "Run 9 stress 0.1523922 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.01659797  max resid 0.04438921 \n",
      "Run 10 stress 0.153127 \n",
      "Run 11 stress 0.1531269 \n",
      "Run 12 stress 0.1609603 \n",
      "Run 13 stress 0.1526138 \n",
      "... Procrustes: rmse 0.01092925  max resid 0.0299032 \n",
      "Run 14 stress 0.2184018 \n",
      "Run 15 stress 0.158719 \n",
      "Run 16 stress 0.1526139 \n",
      "... Procrustes: rmse 0.01097011  max resid 0.03011438 \n",
      "Run 17 stress 0.1508456 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.05988788  max resid 0.172692 \n",
      "Run 18 stress 0.1508456 \n",
      "... Procrustes: rmse 2.794942e-05  max resid 7.577251e-05 \n",
      "... Similar to previous best\n",
      "Run 19 stress 0.354378 \n",
      "Run 20 stress 0.163577 \n",
      "*** Solution reached\n",
      "Run 0 stress 8.932657e-05 \n",
      "Run 1 stress 9.418908e-05 \n",
      "... Procrustes: rmse 0.1856147  max resid 0.4503677 \n",
      "Run 2 stress 9.731021e-05 \n",
      "... Procrustes: rmse 0.2147511  max resid 0.5403825 \n",
      "Run 3 stress 9.135042e-05 \n",
      "... Procrustes: rmse 0.2066926  max resid 0.5267655 \n",
      "Run 4 stress 5.156467e-05 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1806738  max resid 0.4367512 \n",
      "Run 5 stress 0.0526654 \n",
      "Run 6 stress 8.868657e-05 \n",
      "... Procrustes: rmse 0.2169906  max resid 0.3494044 \n",
      "Run 7 stress 9.039424e-05 \n",
      "... Procrustes: rmse 0.2050313  max resid 0.3195469 \n",
      "Run 8 stress 3.658539e-05 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.2311844  max resid 0.3519405 \n",
      "Run 9 stress 0.0504489 \n",
      "Run 10 stress 8.602354e-05 \n",
      "... Procrustes: rmse 0.2088585  max resid 0.3016683 \n",
      "Run 11 stress 9.686611e-05 \n",
      "... Procrustes: rmse 0.173573  max resid 0.3064846 \n",
      "Run 12 stress 0.05059266 \n",
      "Run 13 stress 9.787179e-05 \n",
      "... Procrustes: rmse 0.2086018  max resid 0.3736841 \n",
      "Run 14 stress 1.657075e-05 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1798458  max resid 0.2937998 \n",
      "Run 15 stress 8.096041e-05 \n",
      "... Procrustes: rmse 0.1915524  max resid 0.3354759 \n",
      "Run 16 stress 0.04996306 \n",
      "Run 17 stress 9.983946e-05 \n",
      "... Procrustes: rmse 0.2296799  max resid 0.3875559 \n",
      "Run 18 stress 2.270878e-05 \n",
      "... Procrustes: rmse 0.1773819  max resid 0.2938049 \n",
      "Run 19 stress 9.437371e-05 \n",
      "... Procrustes: rmse 0.228008  max resid 0.3202213 \n",
      "Run 20 stress 0.05063371 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "     4: no. of iterations >= maxit\n",
      "    15: stress < smin\n",
      "     1: stress ratio > sratmax\n",
      "Warning message:\n",
      "In metaMDS(veganifyOTU(physeq), distance, ...) :\n",
      "  Stress is (nearly) zero - you may have insufficient data\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 32 rows containing missing values (geom_errorbar). \n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n"
     ]
    }
   ],
   "source": [
    "#run phyloseq stats\n",
    "!/Users/jon/amptk/util/phyloseq_graphs.R m4a_raw.biom m4a_raw.tree.phy m4a_raw"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Treatment adonis using raupcrick\n",
    "```\n",
    "Call:\n",
    "adonis(formula = rp ~ metadata[[treatments[y]]], permutations = 9999) \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs  MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1   0.24175 0.241748    3.22 0.18699 0.1635\n",
    "Residuals                 14   1.05108 0.075077         0.81301       \n",
    "Total                     15   1.29283                  1.00000    \n",
    "```\n",
    "###NMDS using raup\n",
    "![alt text](m4a_raw.Treatment.raup.pdf)\n",
    "###Treatment adonis using braycurtis\n",
    "```\n",
    "Call:\n",
    "adonis(formula = voturaw ~ metadata[[treatments[y]]], permutations = 9999,      method = \"bray\") \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1    0.4465 0.44648   1.004 0.06691 0.4731\n",
    "Residuals                 14    6.2259 0.44471         0.93309       \n",
    "Total                     15    6.6724                 1.00000   \n",
    "```\n",
    "###NMDS using braycurtis\n",
    "![alt text](m4a_raw.Treatment.braycurtis.pdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[05:08:58 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[05:08:58 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[05:08:58 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[05:08:58 PM]\u001b[0m: 1,752 OTUs\n",
      "\u001b[92m[05:08:58 PM]\u001b[0m: Global alignment OTUs with usearch_global (USEARCH)\n",
      "\u001b[92m[05:09:14 PM]\u001b[0m: Classifying OTUs with UTAX (USEARCH)\n",
      "\u001b[92m[05:09:15 PM]\u001b[0m: Classifying OTUs with SINTAX (USEARCH)\n",
      "\u001b[92m[05:13:37 PM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[05:13:37 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[05:13:43 PM]\u001b[0m: Taxonomy finished: m4a_nosub.taxonomy.txt\n",
      "\u001b[92m[05:13:43 PM]\u001b[0m: Classic OTU table with taxonomy: m4a_nosub.otu_table.taxonomy.txt\n",
      "\u001b[92m[05:13:45 PM]\u001b[0m: BIOM OTU table created: m4a_nosub.biom\n",
      "\u001b[92m[05:13:45 PM]\u001b[0m: OTUs with taxonomy: m4a_nosub.otus.taxonomy.fa\n",
      "\u001b[92m[05:13:45 PM]\u001b[0m: OTU phylogeny: m4a_nosub.tree.phy\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#convert no subtraction to biom\n",
    "!amptk taxonomy -f m4a_nosub.filtered.otus.fa -o m4a_nosub -i m4a_nosub.final.txt -d ITS2 -m m4a.mapping_file.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-2\n",
      "[1] ‘1.19.1’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘1.8.4’\n",
      "[1] ‘2.4.2’\n",
      "Square root transformation\n",
      "Wisconsin double standardization\n",
      "Run 0 stress 0.1536324 \n",
      "Run 1 stress 0.1546032 \n",
      "Run 2 stress 0.1608357 \n",
      "Run 3 stress 0.1536345 \n",
      "... Procrustes: rmse 0.0008581549  max resid 0.001926481 \n",
      "... Similar to previous best\n",
      "Run 4 stress 0.1546048 \n",
      "Run 5 stress 0.1906837 \n",
      "Run 6 stress 0.1536372 \n",
      "... Procrustes: rmse 0.001780999  max resid 0.004182533 \n",
      "... Similar to previous best\n",
      "Run 7 stress 0.2052105 \n",
      "Run 8 stress 0.1698431 \n",
      "Run 9 stress 0.1546039 \n",
      "Run 10 stress 0.1672919 \n",
      "Run 11 stress 0.1536324 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0003885178  max resid 0.0006440897 \n",
      "... Similar to previous best\n",
      "Run 12 stress 0.1672919 \n",
      "Run 13 stress 0.1536322 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0003446835  max resid 0.0007768493 \n",
      "... Similar to previous best\n",
      "Run 14 stress 0.1536329 \n",
      "... Procrustes: rmse 0.0005360825  max resid 0.001616579 \n",
      "... Similar to previous best\n",
      "Run 15 stress 0.1594974 \n",
      "Run 16 stress 0.1608357 \n",
      "Run 17 stress 0.1608357 \n",
      "Run 18 stress 0.1546058 \n",
      "Run 19 stress 0.1981391 \n",
      "Run 20 stress 0.1672919 \n",
      "*** Solution reached\n",
      "Warning message:\n",
      "In postMDS(out$points, dis, plot = max(0, plot - 1), ...) :\n",
      "  skipping half-change scaling: too few points below threshold\n",
      "Run 0 stress 0.1039525 \n",
      "Run 1 stress 0.2061116 \n",
      "Run 2 stress 0.2211578 \n",
      "Run 3 stress 0.1756093 \n",
      "Run 4 stress 0.1434199 \n",
      "Run 5 stress 0.2072676 \n",
      "Run 6 stress 0.2075873 \n",
      "Run 7 stress 0.1096197 \n",
      "Run 8 stress 0.2331496 \n",
      "Run 9 stress 0.2113876 \n",
      "Run 10 stress 0.175433 \n",
      "Run 11 stress 0.1894909 \n",
      "Run 12 stress 0.2062546 \n",
      "Run 13 stress 0.1775609 \n",
      "Run 14 stress 0.1859756 \n",
      "Run 15 stress 0.1614917 \n",
      "Run 16 stress 0.1661777 \n",
      "Run 17 stress 0.1648384 \n",
      "Run 18 stress 0.1661604 \n",
      "Run 19 stress 0.1520038 \n",
      "Run 20 stress 0.1843096 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    17: stress ratio > sratmax\n",
      "     3: scale factor of the gradient < sfgrmin\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: In estimate_richness(physeq, split = TRUE, measures = measures) :\n",
      "  The data you have provided does not have\n",
      "any singletons. This is highly suspicious. Results of richness\n",
      "estimates (for example) are probably unreliable, or wrong, if you have already\n",
      "trimmed low-abundance taxa from the data.\n",
      "\n",
      "We recommended that you find the un-trimmed data and retry.\n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n",
      "3: In estimate_richness(physeq, split = TRUE, measures = measures) :\n",
      "  The data you have provided does not have\n",
      "any singletons. This is highly suspicious. Results of richness\n",
      "estimates (for example) are probably unreliable, or wrong, if you have already\n",
      "trimmed low-abundance taxa from the data.\n",
      "\n",
      "We recommended that you find the un-trimmed data and retry.\n",
      "4: Removed 32 rows containing missing values (geom_errorbar). \n"
     ]
    }
   ],
   "source": [
    "#run phyloseq stats\n",
    "!/Users/jon/amptk/util/phyloseq_graphs.R m4a_nosub.biom m4a_nosub.tree.phy m4a_nosub"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Treatment adonis using raupcrick & 0.1% default index bleed filtering\n",
    "```\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)  \n",
    "metadata[[treatments[y]]]  1   0.52174 0.52174  5.1197 0.26777 0.0459 *\n",
    "Residuals                 14   1.42672 0.10191         0.73223         \n",
    "Total                     15   1.94847                 1.00000         \n",
    "---\n",
    "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
    "```\n",
    "###Treatment adonis using raupcrick & 0.5% index bleed filtering\n",
    "```\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1   0.39228 0.39228  2.5804 0.15563 0.1536\n",
    "Residuals                 14   2.12830 0.15202         0.84437       \n",
    "Total                     15   2.52058                 1.00000 \n",
    "```\n",
    "\n",
    "###NMDS using raup\n",
    "![alt text](m4a_nosub.Treatment.raup.pdf)\n",
    "###Treatment adonis using braycurtis & 0.1% index bleed filter\n",
    "```\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1    0.4461 0.44611  1.0011 0.06673 0.4874\n",
    "Residuals                 14    6.2388 0.44563         0.93327       \n",
    "Total                     15    6.6850                 1.00000  \n",
    "```\n",
    "###Treatment adonis using braycurtis & 0.5% index bleed filter\n",
    "```\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1    0.4458 0.44585 0.99818 0.06655 0.4935\n",
    "Residuals                 14    6.2532 0.44666         0.93345       \n",
    "Total                     15    6.6991                 1.00000   \n",
    "```\n",
    "###NMDS using braycurtis\n",
    "![alt text](m4a_nosub.Treatment.braycurtis.pdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[05:16:34 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[05:16:34 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[05:16:34 PM]\u001b[0m: Loading OTU table: m4a.otu_table.txt\n",
      "\u001b[92m[05:16:34 PM]\u001b[0m: OTU table contains 1798 OTUs\n",
      "\u001b[92m[05:16:34 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[05:16:34 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[05:16:34 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[05:16:34 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[05:16:34 PM]\u001b[0m: Index bleed, samples into mock: 0.054001%.\n",
      "\u001b[92m[05:16:34 PM]\u001b[0m: Overwriting auto detect index-bleed, setting to 0.500000%\n",
      "\u001b[92m[05:16:34 PM]\u001b[0m: Auto subtract filter set to 25\n",
      "\u001b[92m[05:16:34 PM]\u001b[0m: Subtracting 25 from OTU table\n",
      "\u001b[92m[05:16:34 PM]\u001b[0m: Filtering OTU table down to 436 OTUs\n",
      "\u001b[92m[05:16:34 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      m4a.stats.txt\n",
      "Sorted OTU table:     m4a.sorted.txt\n",
      "Normalized/filter:    m4a.normalized.txt\n",
      "Final Binary table:   m4a.final.binary.txt\n",
      "Final OTU table:      m4a.final.txt\n",
      "Filtered OTUs:        m4a.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f m4a.filtered.otus.fa -i m4a.final.txt -m mapping_file.txt -d ITS2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#filter with subtraction\n",
    "!amptk filter -i m4a.otu_table.txt -f m4a.cluster.otus.fa -b m4A-mock3-16000 -m mock3 -c in -o m4a -s auto -p 0.005"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[05:16:48 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[05:16:48 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[05:16:48 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[05:16:48 PM]\u001b[0m: 436 OTUs\n",
      "\u001b[92m[05:16:48 PM]\u001b[0m: Global alignment OTUs with usearch_global (USEARCH)\n",
      "\u001b[92m[05:16:54 PM]\u001b[0m: Classifying OTUs with UTAX (USEARCH)\n",
      "\u001b[92m[05:16:54 PM]\u001b[0m: Classifying OTUs with SINTAX (USEARCH)\n",
      "\u001b[92m[05:18:30 PM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[05:18:30 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[05:18:30 PM]\u001b[0m: Taxonomy finished: m4a.taxonomy.txt\n",
      "\u001b[92m[05:18:30 PM]\u001b[0m: Classic OTU table with taxonomy: m4a.otu_table.taxonomy.txt\n",
      "\u001b[92m[05:18:31 PM]\u001b[0m: BIOM OTU table created: m4a.biom\n",
      "\u001b[92m[05:18:31 PM]\u001b[0m: OTUs with taxonomy: m4a.otus.taxonomy.fa\n",
      "\u001b[92m[05:18:31 PM]\u001b[0m: OTU phylogeny: m4a.tree.phy\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#convert to biom\n",
    "!amptk taxonomy -f m4a.filtered.otus.fa -o m4a -i m4a.final.txt -d ITS2 -m m4a.mapping_file.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-2\n",
      "[1] ‘1.19.1’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘1.8.4’\n",
      "[1] ‘2.4.2’\n",
      "Square root transformation\n",
      "Wisconsin double standardization\n",
      "Run 0 stress 0.2025629 \n",
      "Run 1 stress 0.2333422 \n",
      "Run 2 stress 0.2136412 \n",
      "Run 3 stress 0.2186079 \n",
      "Run 4 stress 0.2462678 \n",
      "Run 5 stress 0.2365634 \n",
      "Run 6 stress 0.2025629 \n",
      "... Procrustes: rmse 1.153241e-05  max resid 3.076636e-05 \n",
      "... Similar to previous best\n",
      "Run 7 stress 0.2541619 \n",
      "Run 8 stress 0.2025629 \n",
      "... New best solution\n",
      "... Procrustes: rmse 2.990079e-06  max resid 6.854737e-06 \n",
      "... Similar to previous best\n",
      "Run 9 stress 0.2468372 \n",
      "Run 10 stress 0.2147798 \n",
      "Run 11 stress 0.2197034 \n",
      "Run 12 stress 0.2157279 \n",
      "Run 13 stress 0.2087092 \n",
      "Run 14 stress 0.2288752 \n",
      "Run 15 stress 0.2148893 \n",
      "Run 16 stress 0.2157279 \n",
      "Run 17 stress 0.2025629 \n",
      "... Procrustes: rmse 2.017526e-06  max resid 4.177743e-06 \n",
      "... Similar to previous best\n",
      "Run 18 stress 0.2150232 \n",
      "Run 19 stress 0.2324361 \n",
      "Run 20 stress 0.2087091 \n",
      "*** Solution reached\n",
      "Warning message:\n",
      "In postMDS(out$points, dis, plot = max(0, plot - 1), ...) :\n",
      "  skipping half-change scaling: too few points below threshold\n",
      "Run 0 stress 0.1773823 \n",
      "Run 1 stress 0.2125229 \n",
      "Run 2 stress 0.1773823 \n",
      "... New best solution\n",
      "... Procrustes: rmse 2.642132e-05  max resid 5.430953e-05 \n",
      "... Similar to previous best\n",
      "Run 3 stress 0.1995036 \n",
      "Run 4 stress 0.2403845 \n",
      "Run 5 stress 0.2026803 \n",
      "Run 6 stress 0.1814685 \n",
      "Run 7 stress 0.2355654 \n",
      "Run 8 stress 0.2178012 \n",
      "Run 9 stress 0.1773824 \n",
      "... Procrustes: rmse 0.0001912085  max resid 0.0005015676 \n",
      "... Similar to previous best\n",
      "Run 10 stress 0.2237052 \n",
      "Run 11 stress 0.1773824 \n",
      "... Procrustes: rmse 0.0001081131  max resid 0.0002890835 \n",
      "... Similar to previous best\n",
      "Run 12 stress 0.2095109 \n",
      "Run 13 stress 0.1773823 \n",
      "... Procrustes: rmse 1.457634e-05  max resid 4.002937e-05 \n",
      "... Similar to previous best\n",
      "Run 14 stress 0.2125233 \n",
      "Run 15 stress 0.2215528 \n",
      "Run 16 stress 0.225865 \n",
      "Run 17 stress 0.1773823 \n",
      "... Procrustes: rmse 1.661419e-05  max resid 3.328948e-05 \n",
      "... Similar to previous best\n",
      "Run 18 stress 0.2336674 \n",
      "Run 19 stress 0.2136768 \n",
      "Run 20 stress 0.1773823 \n",
      "... Procrustes: rmse 7.935507e-06  max resid 1.709075e-05 \n",
      "... Similar to previous best\n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: In estimate_richness(physeq, split = TRUE, measures = measures) :\n",
      "  The data you have provided does not have\n",
      "any singletons. This is highly suspicious. Results of richness\n",
      "estimates (for example) are probably unreliable, or wrong, if you have already\n",
      "trimmed low-abundance taxa from the data.\n",
      "\n",
      "We recommended that you find the un-trimmed data and retry.\n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n",
      "3: In estimate_richness(physeq, split = TRUE, measures = measures) :\n",
      "  The data you have provided does not have\n",
      "any singletons. This is highly suspicious. Results of richness\n",
      "estimates (for example) are probably unreliable, or wrong, if you have already\n",
      "trimmed low-abundance taxa from the data.\n",
      "\n",
      "We recommended that you find the un-trimmed data and retry.\n",
      "4: Removed 32 rows containing missing values (geom_errorbar). \n"
     ]
    }
   ],
   "source": [
    "#run phyloseq stats\n",
    "!/Users/jon/amptk/util/phyloseq_graphs.R m4a.biom m4a.tree.phy m4a"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Treatment adonis using raupcrick & 0.1% index bleed\n",
    "```\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)  \n",
    "metadata[[treatments[y]]]  1   0.58272 0.58272  3.3115 0.19129 0.0416 *\n",
    "Residuals                 14   2.46355 0.17597         0.80871         \n",
    "Total                     15   3.04627                 1.00000         \n",
    "---\n",
    "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
    "```\n",
    "###Treatment adonis using raupcrick & 0.5% index bleed\n",
    "```\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)  \n",
    "metadata[[treatments[y]]]  1   0.59365 0.59365  3.3044 0.19096 0.0418 *\n",
    "Residuals                 14   2.51518 0.17966         0.80904         \n",
    "Total                     15   3.10883                 1.00000         \n",
    "---\n",
    "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
    "```\n",
    "###NMDS using raup\n",
    "![alt text](m4a.Treatment.raup.pdf)\n",
    "###Treatment adonis using braycurtis & 0.1% index bleed\n",
    "```\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1    0.4473 0.44733 0.99773 0.06653  0.481\n",
    "Residuals                 14    6.2769 0.44835         0.93347       \n",
    "Total                     15    6.7242                 1.00000 \n",
    "```\n",
    "###Treatment adonis using braycurtis & 0.5% index bleed\n",
    "```\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1    0.4471 0.44709 0.99609 0.06642  0.493\n",
    "Residuals                 14    6.2838 0.44884         0.93358       \n",
    "Total                     15    6.7309                 1.00000  \n",
    "```\n",
    "###NMDS using braycurtis\n",
    "![alt text](m4a.Treatment.braycurtis.pdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removed 3 samples\n",
      "Kept 5159926 reads out of 6069913 total reads\n",
      "----------------------------------\n",
      "Found 17 barcoded samples\n",
      "              Sample:  Count\n",
      "           m4C-757-2:  390122\n",
      "           m4C-301-2:  385896\n",
      "           m4C-755-2:  351946\n",
      "           m4C-736-2:  314792\n",
      "           m4C-301-1:  312792\n",
      "           m4C-757-1:  310890\n",
      "           m4C-712-1:  306776\n",
      "           m4C-712-2:  300925\n",
      "     m4C-mock3-16000:  299132\n",
      "           m4C-500-1:  299131\n",
      "           m4C-500-2:  297249\n",
      "           m4C-736-1:  288079\n",
      "           m4C-766-1:  284263\n",
      "           m4C-755-1:  280033\n",
      "           m4C-744-1:  262240\n",
      "           m4C-766-2:  255491\n",
      "           m4C-744-2:  220169\n",
      "----------------------------------\n",
      "Read count: 5159926\n",
      "Read Length: 250 bp\n",
      "----------------------------------\n"
     ]
    }
   ],
   "source": [
    "#now run miseqC run\n",
    "#now lets go back and process each of these runs separately, i.e. akin to how you would normally run amptk\n",
    "#first select the reads you want from original demuxing\n",
    "!amptk remove -i m4c.demux.fq -l m4C-mock3-32000b m4C-mock3-32000a m4C-stds -o miseq4c.cleaned.demux.fq\n",
    "!amptk show -i miseq4c.cleaned.demux.fq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[04:11:20 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[04:11:21 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[04:11:21 PM]\u001b[0m: Loading FASTQ Records\n",
      "\u001b[92m[04:11:24 PM]\u001b[0m: 5,159,926 reads (2.6 GB)\n",
      "\u001b[92m[04:11:24 PM]\u001b[0m: Quality Filtering, expected errors < 1.0\n",
      "\u001b[92m[04:13:23 PM]\u001b[0m: 3,324,712 reads passed\n",
      "\u001b[92m[04:13:23 PM]\u001b[0m: De-replication (remove duplicate reads)\n",
      "\u001b[92m[04:13:34 PM]\u001b[0m: 178,499 reads passed\n",
      "\u001b[92m[04:13:35 PM]\u001b[0m: Clustering OTUs (UPARSE)\n",
      "\u001b[92m[04:13:48 PM]\u001b[0m: 1,623 OTUs\n",
      "\u001b[92m[04:13:48 PM]\u001b[0m: Cleaning up padding from OTUs\n",
      "\u001b[92m[04:13:48 PM]\u001b[0m: Mapping Reads to OTUs and Building OTU table\n",
      "\u001b[92m[04:24:15 PM]\u001b[0m: 3,878,003 reads mapped to OTUs (75%)\n",
      "-------------------------------------------------------\n",
      "OTU Clustering Script has Finished Successfully\n",
      "-------------------------------------------------------\n",
      "Clustered OTUs: /Users/jon/projects/synmock_paper/m4c.cluster.otus.fa\n",
      "OTU Table: /Users/jon/projects/synmock_paper/m4c.otu_table.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk filter -i m4c.otu_table.txt -f m4c.cluster.otus.fa -b <mock barcode>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#now cluster the cleaned up data\n",
    "!amptk cluster -i miseq4c.cleaned.demux.fq -o m4c"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[05:22:14 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[05:22:14 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[05:22:14 PM]\u001b[0m: Loading OTU table: m4c.otu_table.txt\n",
      "\u001b[92m[05:22:14 PM]\u001b[0m: OTU table contains 1623 OTUs\n",
      "\u001b[92m[05:22:14 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[05:22:15 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[05:22:15 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[05:22:15 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[05:22:15 PM]\u001b[0m: Index bleed, samples into mock: 0.050002%.\n",
      "\u001b[92m[05:22:15 PM]\u001b[0m: Overwriting auto detect index-bleed, setting to 0.500000%\n",
      "\u001b[92m[05:22:15 PM]\u001b[0m: Filtering OTU table down to 1593 OTUs\n",
      "\u001b[92m[05:22:15 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      m4c_nosub.stats.txt\n",
      "Sorted OTU table:     m4c_nosub.sorted.txt\n",
      "Normalized/filter:    m4c_nosub.normalized.txt\n",
      "Final Binary table:   m4c_nosub.final.binary.txt\n",
      "Final OTU table:      m4c_nosub.final.txt\n",
      "Filtered OTUs:        m4c_nosub.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f m4c_nosub.filtered.otus.fa -i m4c_nosub.final.txt -m mapping_file.txt -d ITS2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#run filtering with no subtraction first\n",
    "!amptk filter -i m4c.otu_table.txt -f m4c.cluster.otus.fa -b m4C-mock3-16000 -m mock3 -c in -o m4c_nosub -p 0.005"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1623 17 1622 16\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[04:24:18 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[04:24:19 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[04:24:19 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[04:24:19 PM]\u001b[0m: 1,623 OTUs\n",
      "\u001b[92m[04:24:19 PM]\u001b[0m: Global alignment OTUs with usearch_global (USEARCH)\n",
      "\u001b[92m[04:24:36 PM]\u001b[0m: Classifying OTUs with UTAX (USEARCH)\n",
      "\u001b[92m[04:24:36 PM]\u001b[0m: Classifying OTUs with SINTAX (USEARCH)\n",
      "\u001b[92m[04:28:44 PM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[04:28:44 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[04:28:50 PM]\u001b[0m: Taxonomy finished: m4c_raw.taxonomy.txt\n",
      "\u001b[92m[04:28:50 PM]\u001b[0m: Classic OTU table with taxonomy: m4c_raw.otu_table.taxonomy.txt\n",
      "\u001b[92m[04:28:51 PM]\u001b[0m: BIOM OTU table created: m4c_raw.biom\n",
      "\u001b[92m[04:28:51 PM]\u001b[0m: OTUs with taxonomy: m4c_raw.otus.taxonomy.fa\n",
      "\u001b[92m[04:28:51 PM]\u001b[0m: OTU phylogeny: m4c_raw.tree.phy\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#get raw data into biom format\n",
    "import pandas as pd\n",
    "from natsort import natsorted\n",
    "#pull out header for indexing later\n",
    "with open('m4c_nosub.sorted.txt', 'rU') as f:\n",
    "    first_line = f.readline()\n",
    "    OTUhead = first_line.split('\\t')[0]\n",
    "    \n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "dfsorted = pd.read_csv('m4c_nosub.sorted.txt', sep='\\t')\n",
    "dfsorted.set_index(OTUhead, inplace=True)\n",
    "\n",
    "#now we want to drop last column\n",
    "header = ['m4C-301-1', 'm4C-301-2', 'm4C-500-1', 'm4C-500-2', 'm4C-712-1', \n",
    "           'm4C-712-2', 'm4C-736-1', 'm4C-736-2', 'm4C-744-1', 'm4C-744-2', \n",
    "           'm4C-755-1', 'm4C-755-2', 'm4C-757-1', 'm4C-757-2', 'm4C-766-1', 'm4C-766-2']\n",
    "dfcleaned = dfsorted.reindex(columns=header, index=natsorted(dfsorted.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "fs = dfcleaned.sum(axis=0)\n",
    "filtered = pd.DataFrame(dfcleaned, columns=fs.index)\n",
    "filt2 = filtered.loc[(filtered != 0).any(1)]\n",
    "print len(dfsorted), len(dfsorted.columns.values), len(filt2), len(filt2.columns.values)\n",
    "filt2.to_csv('m4c.sorted.nomock.txt', sep='\\t')\n",
    "\n",
    "#now convert to biom\n",
    "!amptk taxonomy -f m4c.cluster.otus.fa -o m4c_raw -i m4c.sorted.nomock.txt -d ITS2 -m m4c.mapping_file.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-2\n",
      "[1] ‘1.19.1’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘1.8.4’\n",
      "[1] ‘2.4.2’\n",
      "There were 26 warnings (use warnings() to see them)\n",
      "Square root transformation\n",
      "Wisconsin double standardization\n",
      "Run 0 stress 0.147573 \n",
      "Run 1 stress 0.1482371 \n",
      "Run 2 stress 0.1797045 \n",
      "Run 3 stress 0.147573 \n",
      "... Procrustes: rmse 2.710139e-05  max resid 7.020399e-05 \n",
      "... Similar to previous best\n",
      "Run 4 stress 0.154546 \n",
      "Run 5 stress 0.1482371 \n",
      "Run 6 stress 0.1484346 \n",
      "Run 7 stress 0.147573 \n",
      "... New best solution\n",
      "... Procrustes: rmse 5.294917e-05  max resid 0.0001201151 \n",
      "... Similar to previous best\n",
      "Run 8 stress 0.1545458 \n",
      "Run 9 stress 0.1527002 \n",
      "Run 10 stress 0.1966284 \n",
      "Run 11 stress 0.1640424 \n",
      "Run 12 stress 0.1734689 \n",
      "Run 13 stress 0.147573 \n",
      "... New best solution\n",
      "... Procrustes: rmse 1.403392e-05  max resid 2.809506e-05 \n",
      "... Similar to previous best\n",
      "Run 14 stress 0.1759795 \n",
      "Run 15 stress 0.1475731 \n",
      "... Procrustes: rmse 8.078043e-05  max resid 0.0001655854 \n",
      "... Similar to previous best\n",
      "Run 16 stress 0.1813567 \n",
      "Run 17 stress 0.1482371 \n",
      "Run 18 stress 0.1545458 \n",
      "Run 19 stress 0.1646716 \n",
      "Run 20 stress 0.1484347 \n",
      "*** Solution reached\n",
      "Run 0 stress 9.881781e-05 \n",
      "Run 1 stress 0.08132617 \n",
      "Run 2 stress 0.0001880611 \n",
      "... Procrustes: rmse 0.1808577  max resid 0.2782869 \n",
      "Run 3 stress 0.06081625 \n",
      "Run 4 stress 0.0001669334 \n",
      "... Procrustes: rmse 0.1821789  max resid 0.2461605 \n",
      "Run 5 stress 9.801206e-05 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.2164926  max resid 0.3376264 \n",
      "Run 6 stress 0.07105745 \n",
      "Run 7 stress 0.07753972 \n",
      "Run 8 stress 9.50042e-05 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.2095837  max resid 0.3989812 \n",
      "Run 9 stress 0.07753394 \n",
      "Run 10 stress 0.06115418 \n",
      "Run 11 stress 0.07615507 \n",
      "Run 12 stress 0.07620133 \n",
      "Run 13 stress 0.07566415 \n",
      "Run 14 stress 0.07044603 \n",
      "Run 15 stress 0.06826443 \n",
      "Run 16 stress 8.244569e-05 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.2072362  max resid 0.3679841 \n",
      "Run 17 stress 0.06065516 \n",
      "Run 18 stress 9.416138e-05 \n",
      "... Procrustes: rmse 0.1832735  max resid 0.4036537 \n",
      "Run 19 stress 0.07119497 \n",
      "Run 20 stress 0.07640896 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "     4: no. of iterations >= maxit\n",
      "     4: stress < smin\n",
      "    12: stress ratio > sratmax\n",
      "Warning message:\n",
      "In metaMDS(veganifyOTU(physeq), distance, ...) :\n",
      "  Stress is (nearly) zero - you may have insufficient data\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 32 rows containing missing values (geom_errorbar). \n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n"
     ]
    }
   ],
   "source": [
    "#run phyloseq stats\n",
    "!/Users/jon/amptk/util/phyloseq_graphs.R m4c_raw.biom m4c_raw.tree.phy m4c_raw"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Treatment adonis using raupcrick\n",
    "```\n",
    "Call:\n",
    "adonis(formula = rp ~ metadata[[treatments[y]]], permutations = 9999) \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)   \n",
    "metadata[[treatments[y]]]  1   0.62465 0.62465  11.135 0.44301  0.008 **\n",
    "Residuals                 14   0.78535 0.05610         0.55699          \n",
    "Total                     15   1.41000                 1.00000          \n",
    "---\n",
    "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
    "```\n",
    "###NMDS using raup\n",
    "![alt text](m4c_raw.Treatment.raup.pdf)\n",
    "###Treatment adonis using braycurtis\n",
    "```\n",
    "Call:\n",
    "adonis(formula = voturaw ~ metadata[[treatments[y]]], permutations = 9999,      method = \"bray\") \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1    0.4464 0.44636 0.99684 0.06647  0.496\n",
    "Residuals                 14    6.2688 0.44777         0.93353       \n",
    "Total                     15    6.7152                 1.00000       \n",
    "\n",
    "```\n",
    "###NMDS using braycurtis\n",
    "![alt text](m4c_raw.Treatment.braycurtis.pdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[05:22:40 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[05:22:40 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[05:22:40 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[05:22:40 PM]\u001b[0m: 1,593 OTUs\n",
      "\u001b[92m[05:22:40 PM]\u001b[0m: Global alignment OTUs with usearch_global (USEARCH)\n",
      "\u001b[92m[05:22:55 PM]\u001b[0m: Classifying OTUs with UTAX (USEARCH)\n",
      "\u001b[92m[05:22:56 PM]\u001b[0m: Classifying OTUs with SINTAX (USEARCH)\n",
      "\u001b[92m[05:26:50 PM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[05:26:50 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[05:26:55 PM]\u001b[0m: Taxonomy finished: m4c_nosub.taxonomy.txt\n",
      "\u001b[92m[05:26:55 PM]\u001b[0m: Classic OTU table with taxonomy: m4c_nosub.otu_table.taxonomy.txt\n",
      "\u001b[92m[05:26:57 PM]\u001b[0m: BIOM OTU table created: m4c_nosub.biom\n",
      "\u001b[92m[05:26:57 PM]\u001b[0m: OTUs with taxonomy: m4c_nosub.otus.taxonomy.fa\n",
      "\u001b[92m[05:26:57 PM]\u001b[0m: OTU phylogeny: m4c_nosub.tree.phy\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#convert no subtraction to biom\n",
    "!amptk taxonomy -f m4c_nosub.filtered.otus.fa -o m4c_nosub -i m4c_nosub.final.txt -d ITS2 -m m4c.mapping_file.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-2\n",
      "[1] ‘1.19.1’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘1.8.4’\n",
      "[1] ‘2.4.2’\n",
      "Square root transformation\n",
      "Wisconsin double standardization\n",
      "Run 0 stress 0.1589888 \n",
      "Run 1 stress 0.1660641 \n",
      "Run 2 stress 0.2309605 \n",
      "Run 3 stress 0.1660634 \n",
      "Run 4 stress 0.2125066 \n",
      "Run 5 stress 0.1589888 \n",
      "... New best solution\n",
      "... Procrustes: rmse 1.124799e-05  max resid 3.178494e-05 \n",
      "... Similar to previous best\n",
      "Run 6 stress 0.1660633 \n",
      "Run 7 stress 0.1707548 \n",
      "Run 8 stress 0.1707548 \n",
      "Run 9 stress 0.1684414 \n",
      "Run 10 stress 0.1825581 \n",
      "Run 11 stress 0.1589888 \n",
      "... Procrustes: rmse 2.971052e-06  max resid 7.156171e-06 \n",
      "... Similar to previous best\n",
      "Run 12 stress 0.2138665 \n",
      "Run 13 stress 0.1589888 \n",
      "... Procrustes: rmse 1.342323e-05  max resid 3.811288e-05 \n",
      "... Similar to previous best\n",
      "Run 14 stress 0.1589888 \n",
      "... Procrustes: rmse 0.0001014341  max resid 0.0002864449 \n",
      "... Similar to previous best\n",
      "Run 15 stress 0.1589888 \n",
      "... Procrustes: rmse 2.122e-05  max resid 5.754015e-05 \n",
      "... Similar to previous best\n",
      "Run 16 stress 0.1802274 \n",
      "Run 17 stress 0.2026761 \n",
      "Run 18 stress 0.1707548 \n",
      "Run 19 stress 0.1923009 \n",
      "Run 20 stress 0.1660633 \n",
      "*** Solution reached\n",
      "Warning message:\n",
      "In postMDS(out$points, dis, plot = max(0, plot - 1), ...) :\n",
      "  skipping half-change scaling: too few points below threshold\n",
      "Run 0 stress 0.09320665 \n",
      "Run 1 stress 0.1483085 \n",
      "Run 2 stress 0.18204 \n",
      "Run 3 stress 0.2184771 \n",
      "Run 4 stress 0.1797825 \n",
      "Run 5 stress 0.2030929 \n",
      "Run 6 stress 0.2340981 \n",
      "Run 7 stress 0.2209414 \n",
      "Run 8 stress 0.201609 \n",
      "Run 9 stress 0.2263512 \n",
      "Run 10 stress 0.1457704 \n",
      "Run 11 stress 0.2146222 \n",
      "Run 12 stress 0.1555118 \n",
      "Run 13 stress 0.1836151 \n",
      "Run 14 stress 0.1811222 \n",
      "Run 15 stress 0.1781505 \n",
      "Run 16 stress 0.1929262 \n",
      "Run 17 stress 0.1950245 \n",
      "Run 18 stress 0.09320443 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0007315527  max resid 0.001802601 \n",
      "... Similar to previous best\n",
      "Run 19 stress 0.148133 \n",
      "Run 20 stress 0.1797965 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 32 rows containing missing values (geom_errorbar). \n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n"
     ]
    }
   ],
   "source": [
    "#run phyloseq stats\n",
    "!/Users/jon/amptk/util/phyloseq_graphs.R m4c_nosub.biom m4c_nosub.tree.phy m4c_nosub"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Treatment adonis using raupcrick\n",
    "```\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)  \n",
    "metadata[[treatments[y]]]  1   0.47165 0.47165  4.3805 0.23832 0.0663 .\n",
    "Residuals                 14   1.50738 0.10767         0.76168         \n",
    "Total                     15   1.97904                 1.00000         \n",
    "---\n",
    "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
    "```\n",
    "\n",
    "```\n",
    "                          Df SumsOfSqs MeanSqs F.Model    R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1   0.27439 0.27439    1.66 0.106 0.2647\n",
    "Residuals                 14   2.31415 0.16530         0.894       \n",
    "Total                     15   2.58854                 1.000 \n",
    "```\n",
    "###NMDS using raup\n",
    "![alt text](m4c_nosub.Treatment.raup.pdf)\n",
    "###Treatment adonis using braycurtis\n",
    "```\n",
    "Call:\n",
    "adonis(formula = voturaw ~ metadata[[treatments[y]]], permutations = 9999,      method = \"bray\") \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1    0.4455 0.44548 0.99282 0.06622 0.4981\n",
    "Residuals                 14    6.2818 0.44870         0.93378       \n",
    "Total                     15    6.7273                 1.00000       \n",
    "\n",
    "```\n",
    "###NMDS using braycurtis\n",
    "![alt text](m4c_nosub.Treatment.braycurtis.pdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[04:34:10 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[04:34:11 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[04:34:11 PM]\u001b[0m: Loading OTU table: m4c.otu_table.txt\n",
      "\u001b[92m[04:34:11 PM]\u001b[0m: OTU table contains 1623 OTUs\n",
      "\u001b[92m[04:34:11 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[04:34:11 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[04:34:11 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[04:34:11 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[04:34:11 PM]\u001b[0m: Index bleed, samples into mock: 0.050002%.\n",
      "\u001b[92m[04:34:11 PM]\u001b[0m: Will use value of 0.100000% for index-bleed OTU filtering.\n",
      "\u001b[92m[04:34:11 PM]\u001b[0m: Auto subtract filter set to 17\n",
      "\u001b[92m[04:34:11 PM]\u001b[0m: Subtracting 17 from OTU table\n",
      "\u001b[92m[04:34:11 PM]\u001b[0m: Filtering OTU table down to 486 OTUs\n",
      "\u001b[92m[04:34:11 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      m4c.stats.txt\n",
      "Sorted OTU table:     m4c.sorted.txt\n",
      "Normalized/filter:    m4c.normalized.txt\n",
      "Final Binary table:   m4c.final.binary.txt\n",
      "Final OTU table:      m4c.final.txt\n",
      "Filtered OTUs:        m4c.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f m4c.filtered.otus.fa -i m4c.final.txt -m mapping_file.txt -d ITS2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#filter with subtraction\n",
    "!amptk filter -i m4c.otu_table.txt -f m4c.cluster.otus.fa -b m4C-mock3-16000 -m mock3 -c in -o m4c -s auto"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[04:34:12 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[04:34:12 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[04:34:12 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[04:34:12 PM]\u001b[0m: 486 OTUs\n",
      "\u001b[92m[04:34:12 PM]\u001b[0m: Global alignment OTUs with usearch_global (USEARCH)\n",
      "\u001b[92m[04:34:18 PM]\u001b[0m: Classifying OTUs with UTAX (USEARCH)\n",
      "\u001b[92m[04:34:18 PM]\u001b[0m: Classifying OTUs with SINTAX (USEARCH)\n",
      "\u001b[92m[04:36:05 PM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[04:36:05 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[04:36:06 PM]\u001b[0m: Taxonomy finished: m4c.taxonomy.txt\n",
      "\u001b[92m[04:36:06 PM]\u001b[0m: Classic OTU table with taxonomy: m4c.otu_table.taxonomy.txt\n",
      "\u001b[92m[04:36:07 PM]\u001b[0m: BIOM OTU table created: m4c.biom\n",
      "\u001b[92m[04:36:07 PM]\u001b[0m: OTUs with taxonomy: m4c.otus.taxonomy.fa\n",
      "\u001b[92m[04:36:07 PM]\u001b[0m: OTU phylogeny: m4c.tree.phy\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#convert subtraction to biom\n",
    "!amptk taxonomy -f m4c.filtered.otus.fa -o m4c -i m4c.final.txt -d ITS2 -m m4c.mapping_file.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-2\n",
      "[1] ‘1.19.1’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘1.8.4’\n",
      "[1] ‘2.4.2’\n",
      "Square root transformation\n",
      "Wisconsin double standardization\n",
      "Run 0 stress 0.2044775 \n",
      "Run 1 stress 0.2188196 \n",
      "Run 2 stress 0.2036105 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1435263  max resid 0.3988433 \n",
      "Run 3 stress 0.2106685 \n",
      "Run 4 stress 0.2106687 \n",
      "Run 5 stress 0.2225292 \n",
      "Run 6 stress 0.2161192 \n",
      "Run 7 stress 0.2168716 \n",
      "Run 8 stress 0.221025 \n",
      "Run 9 stress 0.2169447 \n",
      "Run 10 stress 0.2044787 \n",
      "Run 11 stress 0.21681 \n",
      "Run 12 stress 0.2199021 \n",
      "Run 13 stress 0.2044759 \n",
      "Run 14 stress 0.218723 \n",
      "Run 15 stress 0.2044773 \n",
      "Run 16 stress 0.2220565 \n",
      "Run 17 stress 0.2151821 \n",
      "Run 18 stress 0.2161192 \n",
      "Run 19 stress 0.2242043 \n",
      "Run 20 stress 0.2036109 \n",
      "... Procrustes: rmse 0.0008167425  max resid 0.002575266 \n",
      "... Similar to previous best\n",
      "*** Solution reached\n",
      "Warning message:\n",
      "In postMDS(out$points, dis, plot = max(0, plot - 1), ...) :\n",
      "  skipping half-change scaling: too few points below threshold\n",
      "Run 0 stress 0.1960369 \n",
      "Run 1 stress 0.2454272 \n",
      "Run 2 stress 0.1995554 \n",
      "Run 3 stress 0.1960368 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0001681125  max resid 0.0003865168 \n",
      "... Similar to previous best\n",
      "Run 4 stress 0.2153982 \n",
      "Run 5 stress 0.1983937 \n",
      "Run 6 stress 0.2157854 \n",
      "Run 7 stress 0.2332689 \n",
      "Run 8 stress 0.1956648 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.03730303  max resid 0.1076106 \n",
      "Run 9 stress 0.2067513 \n",
      "Run 10 stress 0.2153981 \n",
      "Run 11 stress 0.2544006 \n",
      "Run 12 stress 0.1995344 \n",
      "Run 13 stress 0.270423 \n",
      "Run 14 stress 0.2343045 \n",
      "Run 15 stress 0.1995554 \n",
      "Run 16 stress 0.2160538 \n",
      "Run 17 stress 0.2004561 \n",
      "Run 18 stress 0.1994099 \n",
      "Run 19 stress 0.1982172 \n",
      "Run 20 stress 0.1995341 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: In estimate_richness(physeq, split = TRUE, measures = measures) :\n",
      "  The data you have provided does not have\n",
      "any singletons. This is highly suspicious. Results of richness\n",
      "estimates (for example) are probably unreliable, or wrong, if you have already\n",
      "trimmed low-abundance taxa from the data.\n",
      "\n",
      "We recommended that you find the un-trimmed data and retry.\n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n",
      "3: In estimate_richness(physeq, split = TRUE, measures = measures) :\n",
      "  The data you have provided does not have\n",
      "any singletons. This is highly suspicious. Results of richness\n",
      "estimates (for example) are probably unreliable, or wrong, if you have already\n",
      "trimmed low-abundance taxa from the data.\n",
      "\n",
      "We recommended that you find the un-trimmed data and retry.\n",
      "4: Removed 32 rows containing missing values (geom_errorbar). \n"
     ]
    }
   ],
   "source": [
    "#run phyloseq stats\n",
    "!/Users/jon/amptk/util/phyloseq_graphs.R m4c.biom m4c.tree.phy m4c"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Treatment adonis using raupcrick\n",
    "```\n",
    "Call:\n",
    "adonis(formula = rp ~ metadata[[treatments[y]]], permutations = 9999) \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)  \n",
    "metadata[[treatments[y]]]  1   0.58201 0.58201  3.7066 0.20933 0.0362 *\n",
    "Residuals                 14   2.19831 0.15702         0.79067         \n",
    "Total                     15   2.78032                 1.00000         \n",
    "---\n",
    "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
    "```\n",
    "###NMDS using raup\n",
    "![alt text](m4c_nosub.Treatment.raup.pdf)\n",
    "###Treatment adonis using braycurtis\n",
    "```\n",
    "Call:\n",
    "adonis(formula = voturaw ~ metadata[[treatments[y]]], permutations = 9999,      method = \"bray\") \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1    0.4467 0.44667 0.99108 0.06611 0.4972\n",
    "Residuals                 14    6.3097 0.45069         0.93389       \n",
    "Total                     15    6.7563                 1.00000   \n",
    "```\n",
    "###NMDS using braycurtis\n",
    "![alt text](m4c_nosub.Treatment.braycurtis.pdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removed 3 samples\n",
      "Kept 1914908 reads out of 2227883 total reads\n",
      "----------------------------------\n",
      "Found 17 barcoded samples\n",
      "              Sample:  Count\n",
      "            m5-712-2:  232495\n",
      "            m5-301-2:  186706\n",
      "            m5-301-1:  173161\n",
      "            m5-500-1:  159088\n",
      "            m5-744-2:  115363\n",
      "            m5-736-2:  110748\n",
      "            m5-757-1:  108256\n",
      "            m5-757-2:  107142\n",
      "            m5-755-1:  104765\n",
      "            m5-736-1:  93187\n",
      "            m5-766-1:  86246\n",
      "            m5-755-2:  81537\n",
      "            m5-766-2:  79478\n",
      "            m5-744-1:  79267\n",
      "    m5-SynMock2-8000:  71677\n",
      "            m5-712-1:  71344\n",
      "            m5-500-2:  54448\n",
      "----------------------------------\n",
      "Read count: 1914908\n",
      "Read Length: 250 bp\n",
      "----------------------------------\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[04:39:33 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[04:39:33 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[04:39:33 PM]\u001b[0m: Loading FASTQ Records\n",
      "\u001b[92m[04:39:35 PM]\u001b[0m: 1,914,908 reads (982.1 MB)\n",
      "\u001b[92m[04:39:35 PM]\u001b[0m: Quality Filtering, expected errors < 1.0\n",
      "\u001b[92m[04:40:18 PM]\u001b[0m: 931,994 reads passed\n",
      "\u001b[92m[04:40:18 PM]\u001b[0m: De-replication (remove duplicate reads)\n",
      "\u001b[92m[04:40:21 PM]\u001b[0m: 32,945 reads passed\n",
      "\u001b[92m[04:40:21 PM]\u001b[0m: Clustering OTUs (UPARSE)\n",
      "\u001b[92m[04:40:27 PM]\u001b[0m: 936 OTUs\n",
      "\u001b[92m[04:40:27 PM]\u001b[0m: Cleaning up padding from OTUs\n",
      "\u001b[92m[04:40:27 PM]\u001b[0m: Mapping Reads to OTUs and Building OTU table\n",
      "\u001b[92m[04:43:44 PM]\u001b[0m: 1,639,498 reads mapped to OTUs (86%)\n",
      "-------------------------------------------------------\n",
      "OTU Clustering Script has Finished Successfully\n",
      "-------------------------------------------------------\n",
      "Clustered OTUs: /Users/jon/projects/synmock_paper/i5.cluster.otus.fa\n",
      "OTU Table: /Users/jon/projects/synmock_paper/i5.otu_table.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk filter -i i5.otu_table.txt -f i5.cluster.otus.fa -b <mock barcode>\n",
      "\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[04:43:45 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[04:43:45 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[04:43:45 PM]\u001b[0m: Loading OTU table: i5.otu_table.txt\n",
      "\u001b[92m[04:43:45 PM]\u001b[0m: OTU table contains 936 OTUs\n",
      "\u001b[92m[04:43:45 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[04:43:46 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[04:43:46 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[04:43:46 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[04:43:46 PM]\u001b[0m: Index bleed, mock into samples: 0.178757%.  Index bleed, samples into mock: 0.044999%.\n",
      "\u001b[92m[04:43:46 PM]\u001b[0m: Will use value of 0.200000% for index-bleed OTU filtering.\n",
      "\u001b[92m[04:43:46 PM]\u001b[0m: Filtering OTU table down to 924 OTUs\n",
      "\u001b[92m[04:43:46 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      i5_nosub.stats.txt\n",
      "Sorted OTU table:     i5_nosub.sorted.txt\n",
      "Normalized/filter:    i5_nosub.normalized.txt\n",
      "Final Binary table:   i5_nosub.final.binary.txt\n",
      "Final OTU table:      i5_nosub.final.txt\n",
      "Filtered OTUs:        i5_nosub.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f i5_nosub.filtered.otus.fa -i i5_nosub.final.txt -m mapping_file.txt -d ITS2\n",
      "\n",
      "936 17 936 16\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[04:43:47 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[04:43:47 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[04:43:47 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[04:43:47 PM]\u001b[0m: 936 OTUs\n",
      "\u001b[92m[04:43:47 PM]\u001b[0m: Global alignment OTUs with usearch_global (USEARCH)\n",
      "\u001b[92m[04:43:57 PM]\u001b[0m: Classifying OTUs with UTAX (USEARCH)\n",
      "\u001b[92m[04:43:57 PM]\u001b[0m: Classifying OTUs with SINTAX (USEARCH)\n",
      "\u001b[92m[04:46:29 PM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[04:46:30 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[04:46:32 PM]\u001b[0m: Taxonomy finished: i5_raw.taxonomy.txt\n",
      "\u001b[92m[04:46:32 PM]\u001b[0m: Classic OTU table with taxonomy: i5_raw.otu_table.taxonomy.txt\n",
      "\u001b[92m[04:46:33 PM]\u001b[0m: BIOM OTU table created: i5_raw.biom\n",
      "\u001b[92m[04:46:33 PM]\u001b[0m: OTUs with taxonomy: i5_raw.otus.taxonomy.fa\n",
      "\u001b[92m[04:46:33 PM]\u001b[0m: OTU phylogeny: i5_raw.tree.phy\n",
      "-------------------------------------------------------\n",
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-2\n",
      "[1] ‘1.19.1’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘1.8.4’\n",
      "[1] ‘2.4.2’\n",
      "There were 12 warnings (use warnings() to see them)\n",
      "Square root transformation\n",
      "Wisconsin double standardization\n",
      "Run 0 stress 0.1824184 \n",
      "Run 1 stress 0.1820035 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.07269525  max resid 0.1641637 \n",
      "Run 2 stress 0.2667666 \n",
      "Run 3 stress 0.1958486 \n",
      "Run 4 stress 0.1925865 \n",
      "Run 5 stress 0.1820038 \n",
      "... Procrustes: rmse 0.0003900958  max resid 0.000815263 \n",
      "... Similar to previous best\n",
      "Run 6 stress 0.1824185 \n",
      "... Procrustes: rmse 0.07264635  max resid 0.1693659 \n",
      "Run 7 stress 0.1820036 \n",
      "... Procrustes: rmse 0.0001370427  max resid 0.0002651007 \n",
      "... Similar to previous best\n",
      "Run 8 stress 0.2060041 \n",
      "Run 9 stress 0.2028449 \n",
      "Run 10 stress 0.1820035 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0001188328  max resid 0.0002485861 \n",
      "... Similar to previous best\n",
      "Run 11 stress 0.1824187 \n",
      "... Procrustes: rmse 0.07272409  max resid 0.1690826 \n",
      "Run 12 stress 0.1925864 \n",
      "Run 13 stress 0.1820038 \n",
      "... Procrustes: rmse 0.0003439897  max resid 0.000899304 \n",
      "... Similar to previous best\n",
      "Run 14 stress 0.1926027 \n",
      "Run 15 stress 0.1926113 \n",
      "Run 16 stress 0.1925861 \n",
      "Run 17 stress 0.1824185 \n",
      "... Procrustes: rmse 0.07261392  max resid 0.1693088 \n",
      "Run 18 stress 0.1824184 \n",
      "... Procrustes: rmse 0.07269877  max resid 0.1691985 \n",
      "Run 19 stress 0.1858806 \n",
      "Run 20 stress 0.1958434 \n",
      "*** Solution reached\n",
      "Run 0 stress 0.06159261 \n",
      "Run 1 stress 0.08225027 \n",
      "Run 2 stress 0.1542336 \n",
      "Run 3 stress 0.1049351 \n",
      "Run 4 stress 0.1453095 \n",
      "Run 5 stress 0.1463486 \n",
      "Run 6 stress 0.07232057 \n",
      "Run 7 stress 0.1415093 \n",
      "Run 8 stress 0.1540144 \n",
      "Run 9 stress 0.1538836 \n",
      "Run 10 stress 0.09061584 \n",
      "Run 11 stress 0.164451 \n",
      "Run 12 stress 0.1557841 \n",
      "Run 13 stress 0.08274381 \n",
      "Run 14 stress 0.1516446 \n",
      "Run 15 stress 0.123095 \n",
      "Run 16 stress 0.1626153 \n",
      "Run 17 stress 0.1586245 \n",
      "Run 18 stress 0.1175704 \n",
      "Run 19 stress 0.1327494 \n",
      "Run 20 stress 0.1341432 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 32 rows containing missing values (geom_errorbar). \n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n",
      "-------------------------------------------------------\n",
      "\u001b[92m[04:46:52 PM]\u001b[0m: OS: MacOSX 10.12.3, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[04:46:52 PM]\u001b[0m: amptk v.0.8.9, USEARCH v9.2.64, VSEARCH v2.4.2\n",
      "\u001b[92m[04:46:52 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[04:46:52 PM]\u001b[0m: 924 OTUs\n",
      "\u001b[92m[04:46:52 PM]\u001b[0m: Global alignment OTUs with usearch_global (USEARCH)\n",
      "\u001b[92m[04:47:01 PM]\u001b[0m: Classifying OTUs with UTAX (USEARCH)\n",
      "\u001b[92m[04:47:02 PM]\u001b[0m: Classifying OTUs with SINTAX (USEARCH)\n",
      "\u001b[92m[04:49:32 PM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[04:49:32 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[04:49:34 PM]\u001b[0m: Taxonomy finished: i5_nosub.taxonomy.txt\n",
      "\u001b[92m[04:49:34 PM]\u001b[0m: Classic OTU table with taxonomy: i5_nosub.otu_table.taxonomy.txt\n",
      "\u001b[92m[04:49:36 PM]\u001b[0m: BIOM OTU table created: i5_nosub.biom\n",
      "\u001b[92m[04:49:36 PM]\u001b[0m: OTUs with taxonomy: i5_nosub.otus.taxonomy.fa\n",
      "\u001b[92m[04:49:36 PM]\u001b[0m: OTU phylogeny: i5_nosub.tree.phy\n",
      "-------------------------------------------------------\n",
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-2\n",
      "[1] ‘1.19.1’\n",
      "[1] ‘2.2.1’\n",
      "[1] ‘1.8.4’\n",
      "[1] ‘2.4.2’\n",
      "Square root transformation\n",
      "Wisconsin double standardization\n",
      "Run 0 stress 0.1804848 \n",
      "Run 1 stress 0.1804685 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.07788237  max resid 0.1647463 \n",
      "Run 2 stress 0.200002 \n",
      "Run 3 stress 0.1787185 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.04594231  max resid 0.1499894 \n",
      "Run 4 stress 0.2808804 \n",
      "Run 5 stress 0.1891172 \n",
      "Run 6 stress 0.1897595 \n",
      "Run 7 stress 0.1804685 \n",
      "Run 8 stress 0.2819281 \n",
      "Run 9 stress 0.1787185 \n",
      "... Procrustes: rmse 8.712715e-05  max resid 0.000170463 \n",
      "... Similar to previous best\n",
      "Run 10 stress 0.1787191 \n",
      "... Procrustes: rmse 0.000182275  max resid 0.0004693839 \n",
      "... Similar to previous best\n",
      "Run 11 stress 0.2426145 \n",
      "Run 12 stress 0.1950417 \n",
      "Run 13 stress 0.1804847 \n",
      "Run 14 stress 0.1804686 \n",
      "Run 15 stress 0.1950417 \n",
      "Run 16 stress 0.2059191 \n",
      "Run 17 stress 0.1787185 \n",
      "... Procrustes: rmse 7.714056e-05  max resid 0.0001610748 \n",
      "... Similar to previous best\n",
      "Run 18 stress 0.1804685 \n",
      "Run 19 stress 0.208873 \n",
      "Run 20 stress 0.1891171 \n",
      "*** Solution reached\n",
      "Run 0 stress 0.1331961 \n",
      "Run 1 stress 0.1819634 \n",
      "Run 2 stress 0.1123596 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1772665  max resid 0.4179398 \n",
      "Run 3 stress 0.171106 \n",
      "Run 4 stress 0.1779632 \n",
      "Run 5 stress 0.1424721 \n",
      "Run 6 stress 0.2035493 \n",
      "Run 7 stress 0.1123968 \n",
      "... Procrustes: rmse 0.04540047  max resid 0.172359 \n",
      "Run 8 stress 0.1349471 \n",
      "Run 9 stress 0.1503899 \n",
      "Run 10 stress 0.2142575 \n",
      "Run 11 stress 0.2125274 \n",
      "Run 12 stress 0.1123573 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0008108678  max resid 0.002573084 \n",
      "... Similar to previous best\n",
      "Run 13 stress 0.209925 \n",
      "Run 14 stress 0.1123581 \n",
      "... Procrustes: rmse 0.0004028055  max resid 0.001282905 \n",
      "... Similar to previous best\n",
      "Run 15 stress 0.1289845 \n",
      "Run 16 stress 0.1973106 \n",
      "Run 17 stress 0.1263462 \n",
      "Run 18 stress 0.1426934 \n",
      "Run 19 stress 0.2184578 \n",
      "Run 20 stress 0.2455886 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 32 rows containing missing values (geom_errorbar). \n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n"
     ]
    }
   ],
   "source": [
    "#now onto ion5 synmock run\n",
    "#now lets go back and process each of these runs separately, i.e. akin to how you would normally run amptk\n",
    "#first select the reads you want from original demuxing\n",
    "!amptk remove -i ion5.demux.fq -l m5-SynMock1-8000 m5-mock3.5-16000 m5-mock3.0-16000 -o ion5.cleaned.demux.fq\n",
    "!amptk show -i ion5.cleaned.demux.fq\n",
    "!amptk cluster -i ion5.cleaned.demux.fq -o i5\n",
    "!amptk filter -i i5.otu_table.txt -f i5.cluster.otus.fa -b m5-SynMock2-8000 -m synmock -o i5_nosub\n",
    "\n",
    "#get raw data into biom format\n",
    "import pandas as pd\n",
    "from natsort import natsorted\n",
    "#pull out header for indexing later\n",
    "with open('i5_nosub.sorted.txt', 'rU') as f:\n",
    "    first_line = f.readline()\n",
    "    OTUhead = first_line.split('\\t')[0]\n",
    "    \n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "dfsorted = pd.read_csv('i5_nosub.sorted.txt', sep='\\t')\n",
    "dfsorted.set_index(OTUhead, inplace=True)\n",
    "\n",
    "#now we want to drop last column\n",
    "header = ['m5-301-1', 'm5-301-2', 'm5-500-1', 'm5-500-2', 'm5-712-1', \n",
    "        'm5-712-2', 'm5-736-1', 'm5-736-2', 'm5-744-1', 'm5-744-2', \n",
    "        'm5-755-1', 'm5-755-2', 'm5-757-1', 'm5-757-2', 'm5-766-1', 'm5-766-2']\n",
    "dfcleaned = dfsorted.reindex(columns=header, index=natsorted(dfsorted.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "fs = dfcleaned.sum(axis=0)\n",
    "filtered = pd.DataFrame(dfcleaned, columns=fs.index)\n",
    "filt2 = filtered.loc[(filtered != 0).any(1)]\n",
    "print len(dfsorted), len(dfsorted.columns.values), len(filt2), len(filt2.columns.values)\n",
    "filt2.to_csv('i5.sorted.nomock.txt', sep='\\t')\n",
    "\n",
    "#now convert to biom\n",
    "!amptk taxonomy -f i5.cluster.otus.fa -o i5_raw -i i5.sorted.nomock.txt -d ITS2 -m ion5.mapping_file.txt\n",
    "\n",
    "#run phyloseq stats\n",
    "!/Users/jon/amptk/util/phyloseq_graphs.R i5_raw.biom i5_raw.tree.phy i5_raw\n",
    "\n",
    "#convert no subtraction to biom\n",
    "!amptk taxonomy -f i5_nosub.filtered.otus.fa -o i5_nosub -i i5_nosub.final.txt -d ITS2 -m ion5.mapping_file.txt\n",
    "\n",
    "#run phyloseq stats\n",
    "!/Users/jon/amptk/util/phyloseq_graphs.R i5_nosub.biom i5_nosub.tree.phy i5_nosub\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Treatment adonis using raupcrick\n",
    "```\n",
    "Call:\n",
    "adonis(formula = rp ~ metadata[[treatments[y]]], permutations = 9999) \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs  MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1   0.24521 0.245208   3.296 0.19057 0.1604\n",
    "Residuals                 14   1.04153 0.074395         0.80943       \n",
    "Total                     15   1.28674                  1.00000    \n",
    "```\n",
    "###NMDS using raup\n",
    "![alt text](i5_raw.Treatment.raup.pdf)\n",
    "###Treatment adonis using braycurtis\n",
    "```\n",
    "Call:\n",
    "adonis(formula = voturaw ~ metadata[[treatments[y]]], permutations = 9999,      method = \"bray\") \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1    0.4415 0.44148 0.96188 0.06429 0.5508\n",
    "Residuals                 14    6.4257 0.45898         0.93571       \n",
    "Total                     15    6.8672                 1.00000     \n",
    "```\n",
    "###NMDS using braycurtis\n",
    "![alt text](i5_raw.Treatment.braycurtis.pdf)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Treatment adonis using raupcrick\n",
    "```\n",
    "Call:\n",
    "adonis(formula = rp ~ metadata[[treatments[y]]], permutations = 9999) \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)  \n",
    "metadata[[treatments[y]]]  1     0.346  0.3460  3.8232 0.21451 0.0948 .\n",
    "Residuals                 14     1.267  0.0905         0.78549         \n",
    "Total                     15     1.613                 1.00000         \n",
    "---\n",
    "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
    "\n",
    "```\n",
    "###NMDS using raup\n",
    "![alt text](i5_nosub.Treatment.raup.pdf)\n",
    "###Treatment adonis using braycurtis\n",
    "```\n",
    "Call:\n",
    "adonis(formula = voturaw ~ metadata[[treatments[y]]], permutations = 9999,      method = \"bray\") \n",
    "\n",
    "Permutation: free\n",
    "Number of permutations: 9999\n",
    "\n",
    "Terms added sequentially (first to last)\n",
    "\n",
    "                          Df SumsOfSqs MeanSqs F.Model     R2 Pr(>F)\n",
    "metadata[[treatments[y]]]  1    0.4420 0.44197 0.96205 0.0643 0.5583\n",
    "Residuals                 14    6.4317 0.45941         0.9357       \n",
    "Total                     15    6.8737                 1.0000       \n",
    "\n",
    "```\n",
    "###NMDS using braycurtis\n",
    "![alt text](i5_nosub.Treatment.braycurtis.pdf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
