{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### PreProcessing Reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[08:48:15 PM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[08:48:15 PM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[08:48:16 PM]\u001b[0m: Foward primer: AGTGARTCATCGAATCTTTG,  Rev comp'd rev primer: GCATATCAATAAGCGGAGGA\n",
      "\u001b[92m[08:48:16 PM]\u001b[0m: Loading FASTQ Records\n",
      "\u001b[92m[08:48:21 PM]\u001b[0m: 4,332,502 reads (3.3 GB)\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[08:51:11 PM]\u001b[0m: Concatenating Demuxed Files\n",
      "\u001b[92m[08:51:46 PM]\u001b[0m: 4,332,502 total reads\n",
      "\u001b[92m[08:51:46 PM]\u001b[0m: 3,115,600 valid Barcode\n",
      "\u001b[92m[08:51:46 PM]\u001b[0m: 3,091,409 Fwd Primer found, 2,544,919 Rev Primer found\n",
      "\u001b[92m[08:51:46 PM]\u001b[0m: 67,482 discarded too short (< 100 bp)\n",
      "\u001b[92m[08:51:46 PM]\u001b[0m: 3,023,927 valid output reads\n",
      "\u001b[92m[08:51:56 PM]\u001b[0m: Found 19 barcoded samples\n",
      "                Sample:  Count\n",
      "              m3-766-1:  216946\n",
      "       m3-mock3-32000b:  195110\n",
      "        m3-mock3-16000:  190562\n",
      "        m3-mock3-3200a:  188878\n",
      "              m3-301-2:  177773\n",
      "              m3-744-2:  167910\n",
      "              m3-500-1:  164907\n",
      "              m3-500-2:  161973\n",
      "              m3-766-2:  160797\n",
      "              m3-755-1:  159585\n",
      "              m3-301-1:  156611\n",
      "              m3-736-1:  156075\n",
      "                 755-2:  147023\n",
      "              m3-736-2:  144136\n",
      "              m3-757-2:  142626\n",
      "              m3-744-1:  138267\n",
      "              m3-712-1:  131286\n",
      "              m3-712-2:  116228\n",
      "               m3-stds:  107234\n",
      "\u001b[92m[08:51:56 PM]\u001b[0m: Output file:  ion3.demux.fq (1.7 GB)\n",
      "\u001b[92m[08:51:56 PM]\u001b[0m: Mapping file: ion3.mapping_file.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd: \u001b[0mamptk cluster -i ion3.demux.fq -o out\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!amptk ion -i mock3.ion.fastq --barcode_fasta mock3.barcodes.fa --min_len 100 -o ion3 -l 300"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[08:51:57 PM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[08:51:57 PM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[08:51:57 PM]\u001b[0m: Merging Overlaping Pairs using USEARCH\n",
      "\u001b[92m[08:51:57 PM]\u001b[0m: working on sample m4A-301-1 (Read Length: 300)\n",
      "\u001b[92m[08:52:10 PM]\u001b[0m: 270,478 reads passed (100.0%)\n",
      "\u001b[92m[08:52:10 PM]\u001b[0m: working on sample m4A-301-2 (Read Length: 300)\n",
      "\u001b[92m[08:52:22 PM]\u001b[0m: 296,225 reads passed (100.0%)\n",
      "\u001b[92m[08:52:22 PM]\u001b[0m: working on sample m4A-500-1 (Read Length: 300)\n",
      "\u001b[92m[08:52:37 PM]\u001b[0m: 334,852 reads passed (100.0%)\n",
      "\u001b[92m[08:52:37 PM]\u001b[0m: working on sample m4A-500-2 (Read Length: 300)\n",
      "\u001b[92m[08:52:50 PM]\u001b[0m: 275,990 reads passed (100.0%)\n",
      "\u001b[92m[08:52:50 PM]\u001b[0m: working on sample m4A-712-1 (Read Length: 300)\n",
      "\u001b[92m[08:53:01 PM]\u001b[0m: 265,983 reads passed (100.0%)\n",
      "\u001b[92m[08:53:01 PM]\u001b[0m: working on sample m4A-712-2 (Read Length: 300)\n",
      "\u001b[92m[08:53:13 PM]\u001b[0m: 242,159 reads passed (100.0%)\n",
      "\u001b[92m[08:53:13 PM]\u001b[0m: working on sample m4A-736-1 (Read Length: 300)\n",
      "\u001b[92m[08:53:26 PM]\u001b[0m: 279,894 reads passed (100.0%)\n",
      "\u001b[92m[08:53:26 PM]\u001b[0m: working on sample m4A-736-2 (Read Length: 300)\n",
      "\u001b[92m[08:53:37 PM]\u001b[0m: 264,055 reads passed (100.0%)\n",
      "\u001b[92m[08:53:37 PM]\u001b[0m: working on sample m4A-744-1 (Read Length: 300)\n",
      "\u001b[92m[08:53:51 PM]\u001b[0m: 284,151 reads passed (100.0%)\n",
      "\u001b[92m[08:53:51 PM]\u001b[0m: working on sample m4A-744-2 (Read Length: 300)\n",
      "\u001b[92m[08:54:03 PM]\u001b[0m: 258,138 reads passed (100.0%)\n",
      "\u001b[92m[08:54:03 PM]\u001b[0m: working on sample m4A-755-1 (Read Length: 300)\n",
      "\u001b[92m[08:54:16 PM]\u001b[0m: 305,617 reads passed (100.0%)\n",
      "\u001b[92m[08:54:16 PM]\u001b[0m: working on sample m4A-755-2 (Read Length: 300)\n",
      "\u001b[92m[08:54:30 PM]\u001b[0m: 296,389 reads passed (100.0%)\n",
      "\u001b[92m[08:54:30 PM]\u001b[0m: working on sample m4A-757-1 (Read Length: 300)\n",
      "\u001b[92m[08:54:42 PM]\u001b[0m: 282,080 reads passed (100.0%)\n",
      "\u001b[92m[08:54:42 PM]\u001b[0m: working on sample m4A-757-2 (Read Length: 300)\n",
      "\u001b[92m[08:54:54 PM]\u001b[0m: 277,106 reads passed (100.0%)\n",
      "\u001b[92m[08:54:54 PM]\u001b[0m: working on sample m4A-766-1 (Read Length: 300)\n",
      "\u001b[92m[08:55:08 PM]\u001b[0m: 302,344 reads passed (100.0%)\n",
      "\u001b[92m[08:55:08 PM]\u001b[0m: working on sample m4A-766-2 (Read Length: 300)\n",
      "\u001b[92m[08:55:23 PM]\u001b[0m: 324,123 reads passed (100.0%)\n",
      "\u001b[92m[08:55:23 PM]\u001b[0m: working on sample m4A-mock3-16000 (Read Length: 300)\n",
      "\u001b[92m[08:55:36 PM]\u001b[0m: 284,299 reads passed (100.0%)\n",
      "\u001b[92m[08:55:36 PM]\u001b[0m: working on sample m4A-mock3-32000a (Read Length: 300)\n",
      "\u001b[92m[08:55:50 PM]\u001b[0m: 296,807 reads passed (100.0%)\n",
      "\u001b[92m[08:55:50 PM]\u001b[0m: working on sample m4A-mock3-32000b (Read Length: 300)\n",
      "\u001b[92m[08:56:03 PM]\u001b[0m: 290,916 reads passed (100.0%)\n",
      "\u001b[92m[08:56:03 PM]\u001b[0m: working on sample m4A-stds (Read Length: 300)\n",
      "\u001b[92m[08:56:14 PM]\u001b[0m: 237,349 reads passed (100.0%)\n",
      "\u001b[92m[08:56:14 PM]\u001b[0m: Stripping primers and trim to 300 bp\n",
      "\u001b[92m[08:56:14 PM]\u001b[0m: splitting the job over 8 cpus, but this may still take awhile\n",
      "\u001b[92m[08:56:14 PM]\u001b[0m: Foward primer: GTGARTCATCGAATCTTTG,  Rev comp'd rev primer: GCATATCAATAAGCGGAGGA\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[08:57:08 PM]\u001b[0m: Concatenating Demuxed Files\n",
      "\u001b[92m[08:57:24 PM]\u001b[0m: 5,668,955 total reads\n",
      "\u001b[92m[08:57:24 PM]\u001b[0m: 5,640,712 Fwd Primer found, 5,244,971 Rev Primer found\n",
      "\u001b[92m[08:57:24 PM]\u001b[0m: 760 discarded too short (< 100 bp)\n",
      "\u001b[92m[08:57:24 PM]\u001b[0m: 5,653,466 valid output reads\n",
      "\u001b[92m[08:57:44 PM]\u001b[0m: Found 20 barcoded samples\n",
      "                        Sample:  Count\n",
      "                     m4A-500-1:  333978\n",
      "                     m4A-766-2:  323283\n",
      "                     m4A-755-1:  304594\n",
      "                     m4A-766-1:  302237\n",
      "              m4A-mock3-32000a:  295819\n",
      "                     m4A-755-2:  295414\n",
      "                     m4A-301-2:  295320\n",
      "              m4A-mock3-32000b:  290116\n",
      "               m4A-mock3-16000:  283459\n",
      "                     m4A-744-1:  283153\n",
      "                     m4A-757-1:  281403\n",
      "                     m4A-736-1:  279014\n",
      "                     m4A-757-2:  276314\n",
      "                     m4A-500-2:  275242\n",
      "                     m4A-301-1:  269663\n",
      "                     m4A-712-1:  265548\n",
      "                     m4A-736-2:  263264\n",
      "                     m4A-744-2:  257491\n",
      "                     m4A-712-2:  241446\n",
      "                      m4A-stds:  236708\n",
      "\u001b[92m[08:57:44 PM]\u001b[0m: Output file:  m4a.demux.fq (3.1 GB)\n",
      "\u001b[92m[08:57:44 PM]\u001b[0m: Mapping file: m4a.mapping_file.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd: \u001b[0mamptk cluster -i m4a.demux.fq -o out\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#Now process first illumina run containing termX samples + biomock\n",
    "!amptk illumina -i termX_miseqA -o m4a --min_len 100 --read_length 300 --cleanup -l 300"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[08:57:45 PM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[08:57:45 PM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[08:57:45 PM]\u001b[0m: Merging Overlaping Pairs using USEARCH\n",
      "\u001b[92m[08:57:45 PM]\u001b[0m: working on sample m4a-1mm-301-1 (Read Length: 300)\n",
      "\u001b[92m[08:57:58 PM]\u001b[0m: 283,491 reads passed (100.0%)\n",
      "\u001b[92m[08:57:58 PM]\u001b[0m: working on sample m4a-1mm-301-2 (Read Length: 300)\n",
      "\u001b[92m[08:58:12 PM]\u001b[0m: 310,134 reads passed (100.0%)\n",
      "\u001b[92m[08:58:12 PM]\u001b[0m: working on sample m4a-1mm-500-1 (Read Length: 300)\n",
      "\u001b[92m[08:58:28 PM]\u001b[0m: 351,168 reads passed (100.0%)\n",
      "\u001b[92m[08:58:28 PM]\u001b[0m: working on sample m4a-1mm-500-2 (Read Length: 300)\n",
      "\u001b[92m[08:58:41 PM]\u001b[0m: 289,626 reads passed (100.0%)\n",
      "\u001b[92m[08:58:41 PM]\u001b[0m: working on sample m4a-1mm-712-1 (Read Length: 300)\n",
      "\u001b[92m[08:58:53 PM]\u001b[0m: 277,126 reads passed (100.0%)\n",
      "\u001b[92m[08:58:53 PM]\u001b[0m: working on sample m4a-1mm-712-2 (Read Length: 300)\n",
      "\u001b[92m[08:59:05 PM]\u001b[0m: 259,413 reads passed (100.0%)\n",
      "\u001b[92m[08:59:05 PM]\u001b[0m: working on sample m4a-1mm-736-1 (Read Length: 300)\n",
      "\u001b[92m[08:59:17 PM]\u001b[0m: 293,621 reads passed (100.0%)\n",
      "\u001b[92m[08:59:17 PM]\u001b[0m: working on sample m4a-1mm-736-2 (Read Length: 300)\n",
      "\u001b[92m[08:59:30 PM]\u001b[0m: 279,281 reads passed (100.0%)\n",
      "\u001b[92m[08:59:30 PM]\u001b[0m: working on sample m4a-1mm-744-1 (Read Length: 300)\n",
      "\u001b[92m[08:59:43 PM]\u001b[0m: 296,922 reads passed (100.0%)\n",
      "\u001b[92m[08:59:43 PM]\u001b[0m: working on sample m4a-1mm-744-2 (Read Length: 300)\n",
      "\u001b[92m[08:59:56 PM]\u001b[0m: 276,399 reads passed (100.0%)\n",
      "\u001b[92m[08:59:56 PM]\u001b[0m: working on sample m4a-1mm-755-1 (Read Length: 300)\n",
      "\u001b[92m[09:00:12 PM]\u001b[0m: 324,458 reads passed (100.0%)\n",
      "\u001b[92m[09:00:12 PM]\u001b[0m: working on sample m4a-1mm-755-2 (Read Length: 300)\n",
      "\u001b[92m[09:00:26 PM]\u001b[0m: 309,193 reads passed (100.0%)\n",
      "\u001b[92m[09:00:26 PM]\u001b[0m: working on sample m4a-1mm-757-1 (Read Length: 300)\n",
      "\u001b[92m[09:00:40 PM]\u001b[0m: 298,960 reads passed (100.0%)\n",
      "\u001b[92m[09:00:40 PM]\u001b[0m: working on sample m4a-1mm-757-2 (Read Length: 300)\n",
      "\u001b[92m[09:00:54 PM]\u001b[0m: 288,560 reads passed (100.0%)\n",
      "\u001b[92m[09:00:54 PM]\u001b[0m: working on sample m4a-1mm-766-1 (Read Length: 300)\n",
      "\u001b[92m[09:01:08 PM]\u001b[0m: 317,525 reads passed (100.0%)\n",
      "\u001b[92m[09:01:08 PM]\u001b[0m: working on sample m4a-1mm-766-2 (Read Length: 300)\n",
      "\u001b[92m[09:01:23 PM]\u001b[0m: 342,197 reads passed (100.0%)\n",
      "\u001b[92m[09:01:23 PM]\u001b[0m: working on sample m4a-1mm-mock3-16000 (Read Length: 300)\n",
      "\u001b[92m[09:01:36 PM]\u001b[0m: 296,983 reads passed (100.0%)\n",
      "\u001b[92m[09:01:36 PM]\u001b[0m: working on sample m4a-1mm-mock3-32000a (Read Length: 300)\n",
      "\u001b[92m[09:01:50 PM]\u001b[0m: 312,451 reads passed (100.0%)\n",
      "\u001b[92m[09:01:50 PM]\u001b[0m: working on sample m4a-1mm-mock3-32000b (Read Length: 300)\n",
      "\u001b[92m[09:02:03 PM]\u001b[0m: 302,332 reads passed (100.0%)\n",
      "\u001b[92m[09:02:03 PM]\u001b[0m: working on sample m4a-1mm-stds (Read Length: 300)\n",
      "\u001b[92m[09:02:15 PM]\u001b[0m: 253,158 reads passed (100.0%)\n",
      "\u001b[92m[09:02:15 PM]\u001b[0m: Stripping primers and trim to 300 bp\n",
      "\u001b[92m[09:02:15 PM]\u001b[0m: splitting the job over 8 cpus, but this may still take awhile\n",
      "\u001b[92m[09:02:15 PM]\u001b[0m: Foward primer: GTGARTCATCGAATCTTTG,  Rev comp'd rev primer: GCATATCAATAAGCGGAGGA\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[09:03:11 PM]\u001b[0m: Concatenating Demuxed Files\n",
      "\u001b[92m[09:03:29 PM]\u001b[0m: 5,962,998 total reads\n",
      "\u001b[92m[09:03:29 PM]\u001b[0m: 5,929,630 Fwd Primer found, 5,426,997 Rev Primer found\n",
      "\u001b[92m[09:03:29 PM]\u001b[0m: 789 discarded too short (< 100 bp)\n",
      "\u001b[92m[09:03:29 PM]\u001b[0m: 5,946,978 valid output reads\n",
      "\u001b[92m[09:03:49 PM]\u001b[0m: Found 20 barcoded samples\n",
      "                        Sample:  Count\n",
      "                 m4a-1mm-500-1:  350272\n",
      "                 m4a-1mm-766-2:  341327\n",
      "                 m4a-1mm-755-1:  323393\n",
      "                 m4a-1mm-766-1:  317412\n",
      "          m4a-1mm-mock3-32000a:  311414\n",
      "                 m4a-1mm-301-2:  309206\n",
      "                 m4a-1mm-755-2:  308190\n",
      "          m4a-1mm-mock3-32000b:  301510\n",
      "                 m4a-1mm-757-1:  298258\n",
      "           m4a-1mm-mock3-16000:  296113\n",
      "                 m4a-1mm-744-1:  295907\n",
      "                 m4a-1mm-736-1:  292717\n",
      "                 m4a-1mm-500-2:  288860\n",
      "                 m4a-1mm-757-2:  287744\n",
      "                 m4a-1mm-301-1:  282647\n",
      "                 m4a-1mm-736-2:  278465\n",
      "                 m4a-1mm-712-1:  276684\n",
      "                 m4a-1mm-744-2:  275726\n",
      "                 m4a-1mm-712-2:  258655\n",
      "                  m4a-1mm-stds:  252478\n",
      "\u001b[92m[09:03:49 PM]\u001b[0m: Output file:  m4a1m.demux.fq (3.3 GB)\n",
      "\u001b[92m[09:03:49 PM]\u001b[0m: Mapping file: m4a1m.mapping_file.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd: \u001b[0mamptk cluster -i m4a1m.demux.fq -o out\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#Now process first illumina run containing termX samples + biomock but run with 1 mismatch binning\n",
    "!amptk illumina -i miseqA_1mm -o m4a1m --min_len 100 --read_length 300 --cleanup -l 300"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[09:03:50 PM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[09:03:51 PM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[09:03:51 PM]\u001b[0m: Merging Overlaping Pairs using USEARCH\n",
      "\u001b[92m[09:03:51 PM]\u001b[0m: working on sample m4b-744-1 (Read Length: 300)\n",
      "\u001b[92m[09:03:59 PM]\u001b[0m: 191,554 reads passed (100.0%)\n",
      "\u001b[92m[09:03:59 PM]\u001b[0m: working on sample m4b-744-2 (Read Length: 300)\n",
      "\u001b[92m[09:04:06 PM]\u001b[0m: 156,694 reads passed (100.0%)\n",
      "\u001b[92m[09:04:06 PM]\u001b[0m: working on sample m4b-766-1 (Read Length: 300)\n",
      "\u001b[92m[09:04:14 PM]\u001b[0m: 165,809 reads passed (100.0%)\n",
      "\u001b[92m[09:04:14 PM]\u001b[0m: working on sample m4b-766-2 (Read Length: 300)\n",
      "\u001b[92m[09:04:21 PM]\u001b[0m: 145,681 reads passed (100.0%)\n",
      "\u001b[92m[09:04:21 PM]\u001b[0m: Stripping primers and trim to 300 bp\n",
      "\u001b[92m[09:04:21 PM]\u001b[0m: splitting the job over 8 cpus, but this may still take awhile\n",
      "\u001b[92m[09:04:21 PM]\u001b[0m: Foward primer: GTGARTCATCGAATCTTTG,  Rev comp'd rev primer: GCATATCAATAAGCGGAGGA\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[09:04:29 PM]\u001b[0m: Concatenating Demuxed Files\n",
      "\u001b[92m[09:04:31 PM]\u001b[0m: 659,738 total reads\n",
      "\u001b[92m[09:04:31 PM]\u001b[0m: 656,519 Fwd Primer found, 624,757 Rev Primer found\n",
      "\u001b[92m[09:04:31 PM]\u001b[0m: 21 discarded too short (< 100 bp)\n",
      "\u001b[92m[09:04:31 PM]\u001b[0m: 658,482 valid output reads\n",
      "\u001b[92m[09:04:33 PM]\u001b[0m: Found 4 barcoded samples\n",
      "                        Sample:  Count\n",
      "                     m4b-744-1:  190984\n",
      "                     m4b-766-1:  165760\n",
      "                     m4b-744-2:  156397\n",
      "                     m4b-766-2:  145341\n",
      "\u001b[92m[09:04:33 PM]\u001b[0m: Output file:  m4b.demux.fq (366.3 MB)\n",
      "\u001b[92m[09:04:33 PM]\u001b[0m: Mapping file: m4b.mapping_file.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd: \u001b[0mamptk cluster -i m4b.demux.fq -o out\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#process illumina run B; this was the same 4 libraries run on a different run, shows no index bleed\n",
    "!amptk illumina -i termX_miseqB -o m4b --min_len 100 --read_length 300 --cleanup -l 300"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[09:04:33 PM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[09:04:34 PM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[09:04:34 PM]\u001b[0m: Merging Overlaping Pairs using USEARCH\n",
      "\u001b[92m[09:04:34 PM]\u001b[0m: working on sample m4C-301-1 (Read Length: 300)\n",
      "\u001b[92m[09:04:48 PM]\u001b[0m: 314,811 reads passed (100.0%)\n",
      "\u001b[92m[09:04:48 PM]\u001b[0m: working on sample m4C-301-2 (Read Length: 300)\n",
      "\u001b[92m[09:05:05 PM]\u001b[0m: 388,350 reads passed (100.0%)\n",
      "\u001b[92m[09:05:05 PM]\u001b[0m: working on sample m4C-500-1 (Read Length: 300)\n",
      "\u001b[92m[09:05:18 PM]\u001b[0m: 300,487 reads passed (100.0%)\n",
      "\u001b[92m[09:05:18 PM]\u001b[0m: working on sample m4C-500-2 (Read Length: 300)\n",
      "\u001b[92m[09:05:31 PM]\u001b[0m: 298,857 reads passed (100.0%)\n",
      "\u001b[92m[09:05:31 PM]\u001b[0m: working on sample m4C-712-1 (Read Length: 300)\n",
      "\u001b[92m[09:05:45 PM]\u001b[0m: 307,691 reads passed (100.0%)\n",
      "\u001b[92m[09:05:45 PM]\u001b[0m: working on sample m4C-712-2 (Read Length: 300)\n",
      "\u001b[92m[09:05:59 PM]\u001b[0m: 302,846 reads passed (100.0%)\n",
      "\u001b[92m[09:05:59 PM]\u001b[0m: working on sample m4C-736-1 (Read Length: 300)\n",
      "\u001b[92m[09:06:13 PM]\u001b[0m: 290,077 reads passed (100.0%)\n",
      "\u001b[92m[09:06:13 PM]\u001b[0m: working on sample m4C-736-2 (Read Length: 300)\n",
      "\u001b[92m[09:06:26 PM]\u001b[0m: 316,912 reads passed (100.0%)\n",
      "\u001b[92m[09:06:26 PM]\u001b[0m: working on sample m4C-744-1 (Read Length: 300)\n",
      "\u001b[92m[09:06:39 PM]\u001b[0m: 264,148 reads passed (100.0%)\n",
      "\u001b[92m[09:06:39 PM]\u001b[0m: working on sample m4C-744-2 (Read Length: 300)\n",
      "\u001b[92m[09:06:49 PM]\u001b[0m: 221,193 reads passed (100.0%)\n",
      "\u001b[92m[09:06:49 PM]\u001b[0m: working on sample m4C-755-1 (Read Length: 300)\n",
      "\u001b[92m[09:07:04 PM]\u001b[0m: 281,982 reads passed (100.0%)\n",
      "\u001b[92m[09:07:04 PM]\u001b[0m: working on sample m4C-755-2 (Read Length: 300)\n",
      "\u001b[92m[09:07:21 PM]\u001b[0m: 354,160 reads passed (100.0%)\n",
      "\u001b[92m[09:07:21 PM]\u001b[0m: working on sample m4C-757-1 (Read Length: 300)\n",
      "\u001b[92m[09:07:35 PM]\u001b[0m: 312,287 reads passed (100.0%)\n",
      "\u001b[92m[09:07:35 PM]\u001b[0m: working on sample m4C-757-2 (Read Length: 300)\n",
      "\u001b[92m[09:07:53 PM]\u001b[0m: 392,268 reads passed (100.0%)\n",
      "\u001b[92m[09:07:53 PM]\u001b[0m: working on sample m4C-766-1 (Read Length: 300)\n",
      "\u001b[92m[09:08:06 PM]\u001b[0m: 284,457 reads passed (100.0%)\n",
      "\u001b[92m[09:08:06 PM]\u001b[0m: working on sample m4C-766-2 (Read Length: 300)\n",
      "\u001b[92m[09:08:18 PM]\u001b[0m: 256,774 reads passed (100.0%)\n",
      "\u001b[92m[09:08:18 PM]\u001b[0m: working on sample m4C-mock3-16000 (Read Length: 300)\n",
      "\u001b[92m[09:08:33 PM]\u001b[0m: 300,909 reads passed (100.0%)\n",
      "\u001b[92m[09:08:33 PM]\u001b[0m: working on sample m4C-mock3-32000a (Read Length: 300)\n",
      "\u001b[92m[09:08:50 PM]\u001b[0m: 350,309 reads passed (100.0%)\n",
      "\u001b[92m[09:08:50 PM]\u001b[0m: working on sample m4C-mock3-32000b (Read Length: 300)\n",
      "\u001b[92m[09:09:03 PM]\u001b[0m: 306,879 reads passed (100.0%)\n",
      "\u001b[92m[09:09:03 PM]\u001b[0m: working on sample m4C-stds (Read Length: 300)\n",
      "\u001b[92m[09:09:15 PM]\u001b[0m: 258,283 reads passed (100.0%)\n",
      "\u001b[92m[09:09:15 PM]\u001b[0m: Stripping primers and trim to 300 bp\n",
      "\u001b[92m[09:09:15 PM]\u001b[0m: splitting the job over 8 cpus, but this may still take awhile\n",
      "\u001b[92m[09:09:15 PM]\u001b[0m: Foward primer: GTGARTCATCGAATCTTTG,  Rev comp'd rev primer: GCATATCAATAAGCGGAGGA\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[09:10:08 PM]\u001b[0m: Concatenating Demuxed Files\n",
      "\u001b[92m[09:10:26 PM]\u001b[0m: 6,103,680 total reads\n",
      "\u001b[92m[09:10:26 PM]\u001b[0m: 6,087,516 Fwd Primer found, 4,171,645 Rev Primer found\n",
      "\u001b[92m[09:10:26 PM]\u001b[0m: 5,509 discarded too short (< 100 bp)\n",
      "\u001b[92m[09:10:26 PM]\u001b[0m: 6,094,324 valid output reads\n",
      "\u001b[92m[09:10:47 PM]\u001b[0m: Found 20 barcoded samples\n",
      "                        Sample:  Count\n",
      "                     m4C-757-2:  392019\n",
      "                     m4C-301-2:  388026\n",
      "                     m4C-755-2:  353925\n",
      "              m4C-mock3-32000a:  346977\n",
      "                     m4C-736-2:  316678\n",
      "                     m4C-301-1:  314526\n",
      "                     m4C-757-1:  312058\n",
      "                     m4C-712-1:  307591\n",
      "              m4C-mock3-32000b:  306381\n",
      "                     m4C-712-2:  302558\n",
      "                     m4C-500-1:  300337\n",
      "               m4C-mock3-16000:  299107\n",
      "                     m4C-500-2:  298694\n",
      "                     m4C-736-1:  289749\n",
      "                     m4C-766-1:  284432\n",
      "                     m4C-755-1:  281738\n",
      "                     m4C-744-1:  263892\n",
      "                      m4C-stds:  258001\n",
      "                     m4C-766-2:  256575\n",
      "                     m4C-744-2:  221060\n",
      "\u001b[92m[09:10:47 PM]\u001b[0m: Output file:  m4c.demux.fq (3.4 GB)\n",
      "\u001b[92m[09:10:47 PM]\u001b[0m: Mapping file: m4c.mapping_file.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd: \u001b[0mamptk cluster -i m4c.demux.fq -o out\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#now demux miseq run 3, remade libraries from same amplicons with different barcodes\n",
    "!amptk illumina -i termX_miseqC -o m4c --min_len 100 --read_length 300 --cleanup -l 300"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[09:10:48 PM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[09:10:48 PM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[09:10:48 PM]\u001b[0m: Foward primer: AGTGARTCATCGAATCTTTG,  Rev comp'd rev primer: GCATATCAATAAGCGGAGGA\n",
      "\u001b[92m[09:10:48 PM]\u001b[0m: Loading FASTQ Records\n",
      "\u001b[92m[09:10:53 PM]\u001b[0m: 4,341,392 reads (3.3 GB)\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[09:14:05 PM]\u001b[0m: Concatenating Demuxed Files\n",
      "\u001b[92m[09:14:44 PM]\u001b[0m: 4,341,392 total reads\n",
      "\u001b[92m[09:14:44 PM]\u001b[0m: 3,230,560 valid Barcode\n",
      "\u001b[92m[09:14:44 PM]\u001b[0m: 3,204,111 Fwd Primer found, 2,790,457 Rev Primer found\n",
      "\u001b[92m[09:14:44 PM]\u001b[0m: 52,420 discarded too short (< 100 bp)\n",
      "\u001b[92m[09:14:44 PM]\u001b[0m: 3,151,691 valid output reads\n",
      "\u001b[92m[09:14:54 PM]\u001b[0m: Found 24 barcoded samples\n",
      "                Sample:  Count\n",
      "   m5-SynMock1-mock3.0:  294636\n",
      "              m5-712-2:  254027\n",
      "              m5-301-2:  201097\n",
      "m5-SynMock1.736-1.736-2:  187364\n",
      "              m5-301-1:  187165\n",
      "              m5-500-1:  172804\n",
      "      m5-SynMock1-8000:  159110\n",
      "m5-SynMock2.757-1.757-2:  137856\n",
      "              m5-744-2:  125328\n",
      "              m5-736-2:  122586\n",
      "              m5-757-2:  118441\n",
      "              m5-755-1:  117294\n",
      "              m5-757-1:  116860\n",
      "              m5-736-1:  100436\n",
      "   m5-SynMock1-mock3.5:  99179\n",
      "              m5-766-1:  94901\n",
      "      m5-mock3.5-16000:  94408\n",
      "              m5-755-2:  91144\n",
      "              m5-744-1:  88849\n",
      "      m5-mock3.0-16000:  87295\n",
      "              m5-766-2:  86565\n",
      "      m5-SynMock2-8000:  78357\n",
      "              m5-712-1:  76703\n",
      "              m5-500-2:  59286\n",
      "\u001b[92m[09:14:54 PM]\u001b[0m: Output file:  ion5.demux.fq (1.8 GB)\n",
      "\u001b[92m[09:14:54 PM]\u001b[0m: Mapping file: ion5.mapping_file.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd: \u001b[0mamptk cluster -i ion5.demux.fq -o out\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#now finally one more run, where we used SynMock on same samples using Ion torrent\n",
    "!amptk ion -i mock5.ion.fastq --barcode_fasta mock5.barcodes.fa --min_len 100 -o ion5 -l 300"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[05:01:08 PM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[05:01:08 PM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[05:01:08 PM]\u001b[0m: Merging Overlaping Pairs using USEARCH\n",
      "\u001b[92m[05:01:08 PM]\u001b[0m: working on sample m6-301-1 (Read Length: 300)\n",
      "^C\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/jon/amptk/amptk\", line 2, in <module>\n",
      "    import amptk\n",
      "  File \"/Users/jon/amptk/amptk.py\", line 220, in <module>\n",
      "    subprocess.call(arguments)\n",
      "  File \"/Users/jon/miniconda2/lib/python2.7/subprocess.py\", line 523, in call\n",
      "    return Popen(*popenargs, **kwargs).wait()\n",
      "  File \"/Users/jon/miniconda2/lib/python2.7/subprocess.py\", line 1392, in wait\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/jon/amptk/bin/amptk-process_illumina_folder.py\", line 285, in <module>\n",
      "    amptklib.MergeReads(for_reads, rev_reads, args.out, name+'.fq', read_length, args.min_len, args.usearch, args.rescue_forward, args.merge_method, Index, args.barcode_mismatch)\n",
      "  File \"/Users/jon/amptk/lib/amptklib.py\", line 219, in MergeReads\n",
      "    trim3prime(R1, read_length, pretrim_R1, removelist)\n",
      "  File \"/Users/jon/amptk/lib/amptklib.py\", line 177, in trim3prime\n",
      "    outfile.write(\"@%s\\n%s\\n+\\n%s\\n\" % (title, Seq, Qual))\n",
      "KeyboardInterrupt\n",
      "    pid, sts = _eintr_retry_call(os.waitpid, self.pid, 0)\n",
      "  File \"/Users/jon/miniconda2/lib/python2.7/subprocess.py\", line 476, in _eintr_retry_call\n",
      "    return func(*args)\n",
      "KeyboardInterrupt\n"
     ]
    }
   ],
   "source": [
    "#demux the 4th miseq run - which has SynMock added, will remove barcode mismatches\n",
    "!amptk illumina -i mock6 -o m6 --min_len 100 --read_length 300 --cleanup -l 300 --barcode_mismatch 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[05:01:52 PM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[05:01:53 PM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[05:01:53 PM]\u001b[0m: Merging Overlaping Pairs using USEARCH\n",
      "\u001b[92m[05:01:53 PM]\u001b[0m: working on sample m6-301-1 (Read Length: 300)\n",
      "\u001b[92m[05:02:35 PM]\u001b[0m: 890,433 reads passed (100.0%)\n",
      "\u001b[92m[05:02:35 PM]\u001b[0m: working on sample m6-301-2 (Read Length: 300)\n",
      "\u001b[92m[05:03:31 PM]\u001b[0m: 1,181,538 reads passed (100.0%)\n",
      "\u001b[92m[05:03:31 PM]\u001b[0m: working on sample m6-500-1 (Read Length: 300)\n",
      "\u001b[92m[05:04:16 PM]\u001b[0m: 911,764 reads passed (100.0%)\n",
      "\u001b[92m[05:04:16 PM]\u001b[0m: working on sample m6-500-2 (Read Length: 300)\n",
      "\u001b[92m[05:04:59 PM]\u001b[0m: 879,717 reads passed (100.0%)\n",
      "\u001b[92m[05:04:59 PM]\u001b[0m: working on sample m6-712-1 (Read Length: 300)\n",
      "\u001b[92m[05:05:42 PM]\u001b[0m: 897,131 reads passed (100.0%)\n",
      "\u001b[92m[05:05:42 PM]\u001b[0m: working on sample m6-712-2 (Read Length: 300)\n",
      "\u001b[92m[05:06:25 PM]\u001b[0m: 872,239 reads passed (100.0%)\n",
      "\u001b[92m[05:06:25 PM]\u001b[0m: working on sample m6-736-1 (Read Length: 300)\n",
      "\u001b[92m[05:07:12 PM]\u001b[0m: 998,788 reads passed (100.0%)\n",
      "\u001b[92m[05:07:12 PM]\u001b[0m: working on sample m6-736-2 (Read Length: 300)\n",
      "\u001b[92m[05:07:57 PM]\u001b[0m: 919,332 reads passed (100.0%)\n",
      "\u001b[92m[05:07:57 PM]\u001b[0m: working on sample m6-744-1 (Read Length: 300)\n",
      "\u001b[92m[05:08:35 PM]\u001b[0m: 778,031 reads passed (100.0%)\n",
      "\u001b[92m[05:08:35 PM]\u001b[0m: working on sample m6-744-2 (Read Length: 300)\n",
      "\u001b[92m[05:09:10 PM]\u001b[0m: 704,152 reads passed (100.0%)\n",
      "\u001b[92m[05:09:10 PM]\u001b[0m: working on sample m6-755-1 (Read Length: 300)\n",
      "\u001b[92m[05:09:57 PM]\u001b[0m: 957,047 reads passed (100.0%)\n",
      "\u001b[92m[05:09:57 PM]\u001b[0m: working on sample m6-755-2 (Read Length: 300)\n",
      "\u001b[92m[05:10:52 PM]\u001b[0m: 1,071,790 reads passed (100.0%)\n",
      "\u001b[92m[05:10:52 PM]\u001b[0m: working on sample m6-757-1 (Read Length: 300)\n",
      "\u001b[92m[05:11:37 PM]\u001b[0m: 903,852 reads passed (100.0%)\n",
      "\u001b[92m[05:11:37 PM]\u001b[0m: working on sample m6-757-2 (Read Length: 300)\n",
      "\u001b[92m[05:12:37 PM]\u001b[0m: 1,210,564 reads passed (100.0%)\n",
      "\u001b[92m[05:12:37 PM]\u001b[0m: working on sample m6-766-1 (Read Length: 300)\n",
      "\u001b[92m[05:13:19 PM]\u001b[0m: 840,038 reads passed (100.0%)\n",
      "\u001b[92m[05:13:19 PM]\u001b[0m: working on sample m6-766-2 (Read Length: 300)\n",
      "\u001b[92m[05:13:57 PM]\u001b[0m: 785,737 reads passed (100.0%)\n",
      "\u001b[92m[05:13:57 PM]\u001b[0m: working on sample m6-SynMock (Read Length: 300)\n",
      "\u001b[92m[05:15:01 PM]\u001b[0m: 1,299,197 reads passed (100.0%)\n",
      "\u001b[92m[05:15:01 PM]\u001b[0m: working on sample m6-mock3-16000 (Read Length: 300)\n",
      "\u001b[92m[05:15:48 PM]\u001b[0m: 922,307 reads passed (100.0%)\n",
      "\u001b[92m[05:15:48 PM]\u001b[0m: working on sample m6-mock3-32000a (Read Length: 300)\n",
      "\u001b[92m[05:16:46 PM]\u001b[0m: 1,125,040 reads passed (100.0%)\n",
      "\u001b[92m[05:16:46 PM]\u001b[0m: working on sample m6-mock3-32000b (Read Length: 300)\n",
      "\u001b[92m[05:17:33 PM]\u001b[0m: 943,739 reads passed (100.0%)\n",
      "\u001b[92m[05:17:33 PM]\u001b[0m: working on sample m6-stds (Read Length: 300)\n",
      "\u001b[92m[05:18:15 PM]\u001b[0m: 817,737 reads passed (100.0%)\n",
      "\u001b[92m[05:18:15 PM]\u001b[0m: Stripping primers and trim to 300 bp\n",
      "\u001b[92m[05:18:15 PM]\u001b[0m: splitting the job over 8 cpus, but this may still take awhile\n",
      "\u001b[92m[05:18:15 PM]\u001b[0m: Foward primer: GTGARTCATCGAATCTTTG,  Rev comp'd rev primer: GCATATCAATAAGCGGAGGA\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[05:21:25 PM]\u001b[0m: Concatenating Demuxed Files\n",
      "\u001b[92m[05:22:20 PM]\u001b[0m: 19,910,173 total reads\n",
      "\u001b[92m[05:22:20 PM]\u001b[0m: 19,851,958 Fwd Primer found, 16,197,790 Rev Primer found\n",
      "\u001b[92m[05:22:20 PM]\u001b[0m: 19,161 discarded too short (< 100 bp)\n",
      "\u001b[92m[05:22:20 PM]\u001b[0m: 19,874,272 valid output reads\n",
      "\u001b[92m[05:23:23 PM]\u001b[0m: Found 21 barcoded samples\n",
      "                        Sample:  Count\n",
      "                    m6-SynMock:  1298575\n",
      "                      m6-757-2:  1209349\n",
      "                      m6-301-2:  1180350\n",
      "               m6-mock3-32000a:  1112864\n",
      "                      m6-755-2:  1070659\n",
      "                      m6-736-1:  997483\n",
      "                      m6-755-1:  956055\n",
      "               m6-mock3-32000b:  941928\n",
      "                      m6-736-2:  918396\n",
      "                m6-mock3-16000:  916228\n",
      "                      m6-500-1:  911045\n",
      "                      m6-757-1:  903045\n",
      "                      m6-712-1:  896696\n",
      "                      m6-301-1:  889328\n",
      "                      m6-500-2:  878942\n",
      "                      m6-712-2:  871182\n",
      "                      m6-766-1:  839912\n",
      "                       m6-stds:  816686\n",
      "                      m6-766-2:  784950\n",
      "                      m6-744-1:  777062\n",
      "                      m6-744-2:  703537\n",
      "\u001b[92m[05:23:23 PM]\u001b[0m: Output file:  m61mm.demux.fq (10.9 GB)\n",
      "\u001b[92m[05:23:23 PM]\u001b[0m: Mapping file: m61mm.mapping_file.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd: \u001b[0mamptk cluster -i m61mm.demux.fq -o out\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#demux the 4th miseq run but leave 1 mismatch in index reads\n",
    "!amptk illumina -i mock6 -o m6-1mm --min_len 100 --read_length 300 --cleanup -l 300"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#concatenate all miseq runs\n",
    "!cat m4a.demux.fq m4b.demux.fq m4c.demux.fq m6.demux.fq > miseq.fq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----------------------------------\n",
      "Found 65 barcoded samples\n",
      "              Sample:  Count\n",
      "          m6-SynMock:  1167383\n",
      "            m6-757-2:  1099049\n",
      "            m6-301-2:  1059574\n",
      "     m6-mock3-32000a:  990581\n",
      "            m6-755-2:  988744\n",
      "            m6-736-1:  902612\n",
      "            m6-755-1:  870731\n",
      "     m6-mock3-32000b:  843586\n",
      "            m6-736-2:  839082\n",
      "            m6-500-1:  831847\n",
      "            m6-757-1:  825361\n",
      "      m6-mock3-16000:  818345\n",
      "            m6-301-1:  815157\n",
      "            m6-712-1:  812102\n",
      "            m6-500-2:  808875\n",
      "            m6-712-2:  795405\n",
      "            m6-766-1:  759364\n",
      "             m6-stds:  713488\n",
      "            m6-766-2:  704442\n",
      "            m6-744-1:  703396\n",
      "            m6-744-2:  622853\n",
      "           m4C-757-2:  392019\n",
      "           m4C-301-2:  388026\n",
      "           m4C-755-2:  353925\n",
      "    m4C-mock3-32000a:  346977\n",
      "           m4A-500-1:  333978\n",
      "           m4A-766-2:  323283\n",
      "           m4C-736-2:  316678\n",
      "           m4C-301-1:  314526\n",
      "           m4C-757-1:  312058\n",
      "           m4C-712-1:  307591\n",
      "    m4C-mock3-32000b:  306381\n",
      "           m4A-755-1:  304594\n",
      "           m4C-712-2:  302558\n",
      "           m4A-766-1:  302237\n",
      "           m4C-500-1:  300337\n",
      "     m4C-mock3-16000:  299107\n",
      "           m4C-500-2:  298694\n",
      "    m4A-mock3-32000a:  295819\n",
      "           m4A-755-2:  295414\n",
      "           m4A-301-2:  295320\n",
      "    m4A-mock3-32000b:  290116\n",
      "           m4C-736-1:  289749\n",
      "           m4C-766-1:  284432\n",
      "     m4A-mock3-16000:  283459\n",
      "           m4A-744-1:  283153\n",
      "           m4C-755-1:  281738\n",
      "           m4A-757-1:  281403\n",
      "           m4A-736-1:  279014\n",
      "           m4A-757-2:  276314\n",
      "           m4A-500-2:  275242\n",
      "           m4A-301-1:  269663\n",
      "           m4A-712-1:  265548\n",
      "           m4C-744-1:  263892\n",
      "           m4A-736-2:  263264\n",
      "            m4C-stds:  258001\n",
      "           m4A-744-2:  257491\n",
      "           m4C-766-2:  256575\n",
      "           m4A-712-2:  241446\n",
      "            m4A-stds:  236708\n",
      "           m4C-744-2:  221060\n",
      "           m4b-744-1:  190984\n",
      "           m4b-766-1:  165760\n",
      "           m4b-744-2:  156397\n",
      "           m4b-766-2:  145341\n",
      "----------------------------------\n",
      "Read count: 30378249\n",
      "Read length average: 202\n",
      "Read length range: 100 - 300 bp\n",
      "----------------------------------\n"
     ]
    }
   ],
   "source": [
    "!amptk show -i miseq.fq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Run UPARSE on entire Miseq dataset and then Ion datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[10:51:49 AM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[10:51:50 AM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[10:51:50 AM]\u001b[0m: Loading FASTQ Records\n",
      "\u001b[92m[10:52:18 AM]\u001b[0m: 30,378,249 reads (16.7 GB)\n",
      "\u001b[92m[10:52:18 AM]\u001b[0m: Quality Filtering, expected errors < 1.0\n",
      "\u001b[92m[11:16:20 AM]\u001b[0m: 24,508,786 reads passed\n",
      "\u001b[92m[11:16:20 AM]\u001b[0m: De-replication (remove duplicate reads)\n",
      "\u001b[92m[11:18:25 AM]\u001b[0m: 819,429 reads passed\n",
      "\u001b[92m[11:18:28 AM]\u001b[0m: Clustering OTUs (UPARSE)\n",
      "\u001b[92m[11:19:45 AM]\u001b[0m: 2,816 OTUs\n",
      "\u001b[92m[11:19:45 AM]\u001b[0m: Cleaning up padding from OTUs\n",
      "\u001b[92m[11:19:46 AM]\u001b[0m: Mapping Reads to OTUs and Building OTU table\n",
      "\u001b[92m[12:47:28 PM]\u001b[0m: 26,392,942 reads mapped to OTUs (87%)\n",
      "-------------------------------------------------------\n",
      "OTU Clustering Script has Finished Successfully\n",
      "-------------------------------------------------------\n",
      "Clustered OTUs: /Users/jon/projects/synmock_paper/miseq.cluster.otus.fa\n",
      "OTU Table: /Users/jon/projects/synmock_paper/miseq.otu_table.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk filter -i miseq.otu_table.txt -f miseq.cluster.otus.fa -b <mock barcode>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#cluster the miseq data\n",
    "!amptk cluster -i miseq.fq -o miseq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[12:50:20 PM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[12:50:20 PM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[12:50:20 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[12:50:20 PM]\u001b[0m: 2,816 OTUs\n",
      "\u001b[92m[12:50:20 PM]\u001b[0m: Global alignment OTUs with usearch_global (USEARCH)\n",
      "\u001b[92m[12:51:04 PM]\u001b[0m: Classifying OTUs with UTAX (USEARCH)\n",
      "\u001b[92m[12:51:06 PM]\u001b[0m: Classifying OTUs with SINTAX (USEARCH)\n",
      "\u001b[92m[12:59:10 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[12:59:34 PM]\u001b[0m: Taxonomy finished: miseq.taxonomy.txt\n",
      "\u001b[92m[12:59:34 PM]\u001b[0m: OTUs with taxonomy: miseq.otus.taxonomy.fa\n",
      "\u001b[92m[12:59:34 PM]\u001b[0m: OTU phylogeny: miseq.tree.phy\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#get taxonomy for miseq OTUs\n",
    "!amptk taxonomy -f miseq.cluster.otus.fa -o miseq -d ITS2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#concatenate all ion runs\n",
    "!cat ion3.demux.fq ion5.demux.fq > ion.fq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[12:15:01 AM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[12:15:02 AM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[12:15:02 AM]\u001b[0m: Loading FASTQ Records\n",
      "\u001b[92m[12:15:08 AM]\u001b[0m: 6,175,618 reads (3.4 GB)\n",
      "\u001b[92m[12:15:08 AM]\u001b[0m: Quality Filtering, expected errors < 1.0\n",
      "\u001b[92m[12:17:39 AM]\u001b[0m: 2,688,022 reads passed\n",
      "\u001b[92m[12:17:39 AM]\u001b[0m: De-replication (remove duplicate reads)\n",
      "\u001b[92m[12:17:49 AM]\u001b[0m: 109,495 reads passed\n",
      "\u001b[92m[12:17:49 AM]\u001b[0m: Clustering OTUs (UPARSE)\n",
      "\u001b[92m[12:18:00 AM]\u001b[0m: 1,310 OTUs\n",
      "\u001b[92m[12:18:00 AM]\u001b[0m: Cleaning up padding from OTUs\n",
      "\u001b[92m[12:18:00 AM]\u001b[0m: Mapping Reads to OTUs and Building OTU table\n",
      "\u001b[92m[12:34:03 AM]\u001b[0m: 4,943,835 reads mapped to OTUs (80%)\n",
      "-------------------------------------------------------\n",
      "OTU Clustering Script has Finished Successfully\n",
      "-------------------------------------------------------\n",
      "Clustered OTUs: /Users/jon/projects/synmock_paper/ion.cluster.otus.fa\n",
      "OTU Table: /Users/jon/projects/synmock_paper/ion.otu_table.txt\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk filter -i ion.otu_table.txt -f ion.cluster.otus.fa -b <mock barcode>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#cluster the ion data\n",
    "!amptk cluster -i ion.fq -o ion"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[12:34:04 AM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[12:34:04 AM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[12:34:04 AM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[12:34:04 AM]\u001b[0m: 1,310 OTUs\n",
      "\u001b[92m[12:34:04 AM]\u001b[0m: Global alignment OTUs with usearch_global (USEARCH)\n",
      "\u001b[92m[12:34:21 AM]\u001b[0m: Classifying OTUs with UTAX (USEARCH)\n",
      "\u001b[92m[12:34:22 AM]\u001b[0m: Classifying OTUs with SINTAX (USEARCH)\n",
      "\u001b[92m[12:37:46 AM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[12:37:52 AM]\u001b[0m: Taxonomy finished: ion.taxonomy.txt\n",
      "\u001b[92m[12:37:52 AM]\u001b[0m: OTUs with taxonomy: ion.otus.taxonomy.fa\n",
      "\u001b[92m[12:37:52 AM]\u001b[0m: OTU phylogeny: ion.tree.phy\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#get taxonomy for ion OTUs\n",
    "!amptk taxonomy -f ion.cluster.otus.fa -o ion -d ITS2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Try to Measure index bleed based on the mock communities"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[01:36:23 PM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[01:36:23 PM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[01:36:23 PM]\u001b[0m: Loading OTU table: miseq.otu_table.txt\n",
      "\u001b[92m[01:36:23 PM]\u001b[0m: OTU table contains 2816 OTUs\n",
      "\u001b[92m[01:36:23 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[01:36:23 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[01:36:23 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[01:36:23 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[01:36:23 PM]\u001b[0m: Index bleed, mock into samples: 91.850853%.  Index bleed, samples into mock: 0.360022%.\n",
      "\u001b[92m[01:36:23 PM]\u001b[0m: Overwriting auto detect index-bleed, setting to 0.000000%\n",
      "\u001b[92m[01:36:28 PM]\u001b[0m: Filtering OTU table down to 2471 OTUs\n",
      "\u001b[92m[01:36:28 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      miseq.stats.txt\n",
      "Sorted OTU table:     miseq.sorted.txt\n",
      "Normalized/filter:    miseq.normalized.txt\n",
      "Final Binary table:   miseq.final.binary.txt\n",
      "Final OTU table:      miseq.final.txt\n",
      "Filtered OTUs:        miseq.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f miseq.filtered.otus.fa -i miseq.final.txt -m mapping_file.txt -d ITS2\n",
      "\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[01:36:29 PM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[01:36:29 PM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[01:36:29 PM]\u001b[0m: Loading OTU table: ion.otu_table.txt\n",
      "\u001b[92m[01:36:29 PM]\u001b[0m: OTU table contains 1310 OTUs\n",
      "\u001b[92m[01:36:29 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[01:36:29 PM]\u001b[0m: Mock members not found: mock26 |MD275|Wolfiporia cocos\n",
      "\u001b[92m[01:36:29 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[01:36:29 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[01:36:29 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[01:36:29 PM]\u001b[0m: Index bleed, mock into samples: 84.997400%.  Index bleed, samples into mock: 0.764008%.\n",
      "\u001b[92m[01:36:29 PM]\u001b[0m: Overwriting auto detect index-bleed, setting to 0.000000%\n",
      "\u001b[92m[01:36:31 PM]\u001b[0m: Filtering OTU table down to 1310 OTUs\n",
      "\u001b[92m[01:36:31 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      ion.stats.txt\n",
      "Sorted OTU table:     ion.sorted.txt\n",
      "Normalized/filter:    ion.normalized.txt\n",
      "Final Binary table:   ion.final.binary.txt\n",
      "Final OTU table:      ion.final.txt\n",
      "Filtered OTUs:        ion.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f ion.filtered.otus.fa -i ion.final.txt -m mapping_file.txt -d ITS2\n",
      "\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[01:36:32 PM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[01:36:32 PM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[01:36:32 PM]\u001b[0m: Loading OTU table: miseq.otu_table.txt\n",
      "\u001b[92m[01:36:32 PM]\u001b[0m: OTU table contains 2816 OTUs\n",
      "\u001b[92m[01:36:32 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[01:36:32 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[01:36:32 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[01:36:32 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[01:36:32 PM]\u001b[0m: Index bleed, mock into samples: 0.041021%.  Index bleed, samples into mock: 0.084007%.\n",
      "\u001b[92m[01:36:32 PM]\u001b[0m: Overwriting auto detect index-bleed, setting to 0.000000%\n",
      "\u001b[92m[01:36:37 PM]\u001b[0m: Filtering OTU table down to 2471 OTUs\n",
      "\u001b[92m[01:36:37 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      miseq_syn.stats.txt\n",
      "Sorted OTU table:     miseq_syn.sorted.txt\n",
      "Normalized/filter:    miseq_syn.normalized.txt\n",
      "Final Binary table:   miseq_syn.final.binary.txt\n",
      "Final OTU table:      miseq_syn.final.txt\n",
      "Filtered OTUs:        miseq_syn.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f miseq_syn.filtered.otus.fa -i miseq_syn.final.txt -m mapping_file.txt -d ITS2\n",
      "\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[01:36:38 PM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[01:36:38 PM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[01:36:38 PM]\u001b[0m: Loading OTU table: ion.otu_table.txt\n",
      "\u001b[92m[01:36:38 PM]\u001b[0m: OTU table contains 1310 OTUs\n",
      "\u001b[92m[01:36:38 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[01:36:38 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[01:36:38 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[01:36:38 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[01:36:38 PM]\u001b[0m: Index bleed, mock into samples: 76.565042%.  Index bleed, samples into mock: 1.913904%.\n",
      "\u001b[92m[01:36:38 PM]\u001b[0m: Overwriting auto detect index-bleed, setting to 0.000000%\n",
      "\u001b[92m[01:36:40 PM]\u001b[0m: Filtering OTU table down to 1310 OTUs\n",
      "\u001b[92m[01:36:40 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      ion_syn.stats.txt\n",
      "Sorted OTU table:     ion_syn.sorted.txt\n",
      "Normalized/filter:    ion_syn.normalized.txt\n",
      "Final Binary table:   ion_syn.final.binary.txt\n",
      "Final OTU table:      ion_syn.final.txt\n",
      "Filtered OTUs:        ion_syn.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f ion_syn.filtered.otus.fa -i ion_syn.final.txt -m mapping_file.txt -d ITS2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#measure index bleed\n",
    "#clean up the miseq and ion combined runs to look for index bleed, filter without removing anything\n",
    "!amptk filter -i miseq.otu_table.txt -f miseq.cluster.otus.fa -p 0 -o miseq -b m4A-stds -m mock2 --keep_mock\n",
    "!amptk filter -i ion.otu_table.txt -f ion.cluster.otus.fa -p 0 -o ion -b m3-stds -m mock2 --keep_mock\n",
    "!amptk filter -i miseq.otu_table.txt -f miseq.cluster.otus.fa -p 0 -o miseq_syn -b m6-SynMock -m synmock --keep_mock\n",
    "!amptk filter -i ion.otu_table.txt -f ion.cluster.otus.fa -p 0 -o ion_syn -b m5-SynMock2-8000 -m synmock --keep_mock"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index-bleed out of mock: 0.025337% (114 out of 449930)\n",
      "Index-bleed into mock: 0.018041% (191 out of 1058702)\n",
      "Total index-bleed: 0.043378%\n"
     ]
    }
   ],
   "source": [
    "#need to measure index bleed, lets start with ion dataset\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from natsort import natsorted\n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "df = pd.read_csv('ion.final.txt', sep='\\t', index_col=0)\n",
    "#print list(df.columns.values)\n",
    "#print len(df)\n",
    "#okay, so 1310 OTUs in entire data set, lets trim this down to only run 3, as that one is labeled here with biomock\n",
    "keepers = ['m3-301-1', 'm3-301-2', 'm3-500-1', 'm3-500-2', 'm3-712-1', 'm3-712-2', \n",
    "           'm3-736-1', 'm3-736-2', 'm3-744-1', 'm3-744-2', 'm3-755-1', \n",
    "           'm3-757-2', 'm3-766-1', 'm3-766-2', 'm3-mock3-3200a', 'm3-mock3-16000', 'm3-mock3-32000b', 'm3-stds']\n",
    "mock_samples = ['m3-mock3-3200a', 'm3-mock3-16000', 'm3-mock3-32000b', 'm3-stds']\n",
    "samples = [item for item in keepers if item not in mock_samples]\n",
    "#print samples\n",
    "df2 = df.reindex(columns=keepers, index=natsorted(df.index))\n",
    "df2.to_csv('ion3.bleed.otu_table.txt', sep='\\t')\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "df3 = df2.loc[(df2 != 0).any(1)]\n",
    "#print df3\n",
    "mock_otus = []\n",
    "for x in df3.index:\n",
    "    if x.startswith('mock'):\n",
    "        if not x.startswith('mock3'): #mock3 is actually in some of the samples so can't use for index-bleed measurment\n",
    "            mock_otus.append(x)\n",
    "#print mock_otus\n",
    "#now slice into two tables containing only mocks and calculate index-bleed\n",
    "bleed_counts = df3.reindex(columns=samples, index=mock_otus)\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "real_counts = df3.reindex(columns=mock_samples, index=mock_otus)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed out of mock: %f%% (%i out of %i)\" % (index_bleed, bleed_sum, real_sum)\n",
    "\n",
    "#now try to to do the opposite, count number of times that samples move into mock\n",
    "#first generate list of mock otus\n",
    "sample_otus = []\n",
    "for x in df3.index:\n",
    "    if not x.startswith('mock'):\n",
    "        sample_otus.append(x)\n",
    "#slice first to samples, further trim list of only things found in mocks, would be spurious or chimeras\n",
    "sampledf = df3.reindex(columns=samples, index=sample_otus)\n",
    "sampledf = sampledf.loc[(sampledf != 0).any(1)]\n",
    "cleaned_otus = sampledf.index\n",
    "#print len(cleaned_otus), len(sample_otus)\n",
    "\n",
    "#now using new list grab counts in mock\n",
    "bleed_counts = df3.reindex(columns=mock_samples, index=cleaned_otus)\n",
    "bleed_counts = bleed_counts.loc[(bleed_counts != 0).any(1)]\n",
    "#print bleed_counts\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "#print len(bleed_counts.index), bleed_sum\n",
    "\n",
    "#finally go back to samples and get read counts for the ovlerlapping otus\n",
    "real_counts = df3.reindex(columns=samples, index=bleed_counts.index)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed2 = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed into mock: %f%% (%i out of %i)\" % (index_bleed2, bleed_sum, real_sum)\n",
    "print \"Total index-bleed: %f%%\" % (index_bleed + index_bleed2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index-bleed out of mock: 0.040805% (226 out of 553853)\n",
      "Index-bleed into mock: 0.028418% (91 out of 320223)\n",
      "Total index-bleed: 0.069223%\n"
     ]
    }
   ],
   "source": [
    "#now try on the ion5 dataset\n",
    "#need to measure index bleed, lets start with ion dataset\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from natsort import natsorted\n",
    "import sys\n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "df = pd.read_csv('ion_syn.final.txt', sep='\\t', index_col=0)\n",
    "#print list(df.columns.values)\n",
    "#sys.exit(1)\n",
    "#print len(df)\n",
    "#okay, so 1367 OTUs in entire data set, lets trim this down to only run 3, as that one is labeled here with biomock\n",
    "keepers = ['m5-301-1', 'm5-301-2', 'm5-500-1', 'm5-500-2', 'm5-712-1', 'm5-712-2', 'm5-736-1', 'm5-736-2', \n",
    "           'm5-744-1', 'm5-744-2', 'm5-755-1', 'm5-755-2', 'm5-757-1', 'm5-757-2', 'm5-766-1', 'm5-766-2', \n",
    "           'm5-SynMock1-8000', 'm5-SynMock1-mock3.0', 'm5-SynMock1-mock3.5', 'm5-SynMock1.736-1.736-2', \n",
    "           'm5-SynMock2-8000', 'm5-SynMock2.757-1.757-2', 'm5-mock3.0-16000', 'm5-mock3.5-16000']\n",
    "mock_samples = ['m5-SynMock2-8000', 'm5-SynMock2.757-1.757-2', 'm5-SynMock1-8000', \n",
    "                'm5-SynMock1-mock3.0', 'm5-SynMock1-mock3.5', 'm5-SynMock1.736-1.736-2']\n",
    "samples = [item for item in keepers if item not in mock_samples]\n",
    "#print samples\n",
    "df2 = df.reindex(columns=keepers, index=natsorted(df.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "df3 = df2.loc[(df2 != 0).any(1)]\n",
    "mock_otus = []\n",
    "for x in df3.index:\n",
    "    if 'mock' in x: #note this is different here as I'm also including mock chimeras as based on synthetic mock\n",
    "        mock_otus.append(x)\n",
    "#now slice into two tables containing only mocks and calculate index-bleed\n",
    "bleed_counts = df3.reindex(columns=samples, index=mock_otus)\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "real_counts = df3.reindex(columns=mock_samples, index=mock_otus)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed out of mock: %f%% (%i out of %i)\" % (index_bleed, bleed_sum, real_sum)\n",
    "\n",
    "#now try to to do the opposite, count number of times that samples move into mock\n",
    "#first generate list of mock otus\n",
    "sample_otus = []\n",
    "for x in df3.index:\n",
    "    if not 'mock' in x: #note this is different here as I'm also including mock chimeras as based on synthetic mock\n",
    "        sample_otus.append(x)\n",
    "clean_mocks = ['m5-SynMock2-8000', 'm5-SynMock1-8000',]\n",
    "#slice first to samples, further trim list of only things found in mocks, would be spurious or chimeras\n",
    "sampledf = df3.reindex(columns=samples, index=sample_otus)\n",
    "sampledf = sampledf.loc[(sampledf != 0).any(1)]\n",
    "cleaned_otus = sampledf.index\n",
    "#print len(cleaned_otus), len(sample_otus)\n",
    "\n",
    "#now using new list grab counts in mock\n",
    "bleed_counts = df3.reindex(columns=clean_mocks, index=cleaned_otus)\n",
    "bleed_counts = bleed_counts.loc[(bleed_counts != 0).any(1)]\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "#print len(bleed_counts.index), bleed_sum\n",
    "\n",
    "#finally go back to samples and get read counts for the ovlerlapping otus\n",
    "real_counts = df3.reindex(columns=samples, index=bleed_counts.index)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed2 = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed into mock: %f%% (%i out of %i)\" % (index_bleed2, bleed_sum, real_sum)\n",
    "print \"Total index-bleed: %f%%\" % (index_bleed + index_bleed2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index-bleed out of mock: 0.248094% (2377 out of 958106)\n",
      "Index-bleed into mock: 0.014913% (403 out of 2702316)\n",
      "Total index-bleed: 0.263007%\n"
     ]
    }
   ],
   "source": [
    "#need to measure index bleed of miseq data now, trying 0 mismatch miseqA dataset\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from natsort import natsorted\n",
    "import sys\n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "df = pd.read_csv('miseq.final.txt', sep='\\t', index_col=0)\n",
    "#print list(df.columns.values)\n",
    "#sys.exit(1)\n",
    "#print len(df)\n",
    "#okay, so 1367 OTUs in entire data set, lets trim this down to only run 3, as that one is labeled here with biomock\n",
    "keepers = ['m4A-301-1', 'm4A-301-2', 'm4A-500-1', 'm4A-500-2', 'm4A-712-1', 'm4A-712-2', 'm4A-736-1', \n",
    "           'm4A-736-2', 'm4A-744-1', 'm4A-744-2', 'm4A-755-1', 'm4A-755-2', 'm4A-757-1', 'm4A-757-2', \n",
    "           'm4A-766-1', 'm4A-766-2', 'm4A-mock3-16000', 'm4A-mock3-32000a', 'm4A-mock3-32000b', 'm4A-stds']\n",
    "mock_samples = ['m4A-mock3-16000', 'm4A-mock3-32000a', 'm4A-mock3-32000b', 'm4A-stds']\n",
    "samples = [item for item in keepers if item not in mock_samples]\n",
    "#print samples\n",
    "df2 = df.reindex(columns=keepers, index=natsorted(df.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "df3 = df2.loc[(df2 != 0).any(1)]\n",
    "mock_otus = []\n",
    "for x in df3.index:\n",
    "    if x.startswith('mock'):\n",
    "        if not x.startswith('mock3'): #mock3 is actually in some of the samples so can't use for index-bleed measurment\n",
    "            mock_otus.append(x)\n",
    "#now slice into two tables containing only mocks and calculate index-bleed\n",
    "bleed_counts = df3.reindex(columns=samples, index=mock_otus)\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "real_counts = df3.reindex(columns=mock_samples, index=mock_otus)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed out of mock: %f%% (%i out of %i)\" % (index_bleed, bleed_sum, real_sum)\n",
    "\n",
    "#now try to to do the opposite, count number of times that samples move into mock\n",
    "#first generate list of mock otus\n",
    "sample_otus = []\n",
    "for x in df3.index:\n",
    "    if not x.startswith('mock'):\n",
    "        sample_otus.append(x)\n",
    "#slice first to samples, further trim list of only things found in mocks, would be spurious or chimeras\n",
    "sampledf = df3.reindex(columns=samples, index=sample_otus)\n",
    "sampledf = sampledf.loc[(sampledf != 0).any(1)]\n",
    "cleaned_otus = sampledf.index\n",
    "#print len(cleaned_otus), len(sample_otus)\n",
    "\n",
    "#now using new list grab counts in mock\n",
    "bleed_counts = df3.reindex(columns=mock_samples, index=cleaned_otus)\n",
    "bleed_counts = bleed_counts.loc[(bleed_counts != 0).any(1)]\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "#print len(bleed_counts.index), bleed_sum\n",
    "\n",
    "#finally go back to samples and get read counts for the ovlerlapping otus\n",
    "real_counts = df3.reindex(columns=samples, index=bleed_counts.index)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed2 = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed into mock: %f%% (%i out of %i)\" % (index_bleed2, bleed_sum, real_sum)\n",
    "print \"Total index-bleed: %f%%\" % (index_bleed + index_bleed2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index-bleed out of mock: 0.216339% (1762 out of 814461)\n",
      "Index-bleed into mock: 0.012101% (294 out of 2429463)\n",
      "Total index-bleed: 0.228441%\n"
     ]
    }
   ],
   "source": [
    "#need to measure index bleed of miseq data now, trying 0 mismatch miseqC dataset first\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from natsort import natsorted\n",
    "import sys\n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "df = pd.read_csv('miseq.final.txt', sep='\\t', index_col=0)\n",
    "#print list(df.columns.values)\n",
    "#sys.exit(1)\n",
    "#print len(df)\n",
    "#okay, so 1367 OTUs in entire data set, lets trim this down to only run 3, as that one is labeled here with biomock\n",
    "keepers = ['m4C-301-1', 'm4C-301-2', 'm4C-500-1', 'm4C-500-2', 'm4C-712-1', 'm4C-712-2', \n",
    "           'm4C-736-1', 'm4C-736-2', 'm4C-744-1', 'm4C-744-2', 'm4C-755-1', 'm4C-755-2', \n",
    "           'm4C-757-1', 'm4C-757-2', 'm4C-766-1', 'm4C-766-2', \n",
    "           'm4C-mock3-16000', 'm4C-mock3-32000a', 'm4C-mock3-32000b', 'm4C-stds']\n",
    "mock_samples = ['m4C-mock3-16000', 'm4C-mock3-32000a', 'm4C-mock3-32000b', 'm4C-stds']\n",
    "samples = [item for item in keepers if item not in mock_samples]\n",
    "#print samples\n",
    "df2 = df.reindex(columns=keepers, index=natsorted(df.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "df3 = df2.loc[(df2 != 0).any(1)]\n",
    "mock_otus = []\n",
    "for x in df3.index:\n",
    "    if x.startswith('mock'):\n",
    "        if not x.startswith('mock3'): #mock3 is actually in some of the samples so can't use for index-bleed measurment\n",
    "            mock_otus.append(x)\n",
    "#now slice into two tables containing only mocks and calculate index-bleed\n",
    "bleed_counts = df3.reindex(columns=samples, index=mock_otus)\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "real_counts = df3.reindex(columns=mock_samples, index=mock_otus)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed out of mock: %f%% (%i out of %i)\" % (index_bleed, bleed_sum, real_sum)\n",
    "\n",
    "#now try to to do the opposite, count number of times that samples move into mock\n",
    "#first generate list of mock otus\n",
    "sample_otus = []\n",
    "for x in df3.index:\n",
    "    if not x.startswith('mock'):\n",
    "        sample_otus.append(x)\n",
    "#slice first to samples, further trim list of only things found in mocks, would be spurious or chimeras\n",
    "sampledf = df3.reindex(columns=samples, index=sample_otus)\n",
    "sampledf = sampledf.loc[(sampledf != 0).any(1)]\n",
    "cleaned_otus = sampledf.index\n",
    "#print len(cleaned_otus), len(sample_otus)\n",
    "\n",
    "#now using new list grab counts in mock\n",
    "bleed_counts = df3.reindex(columns=mock_samples, index=cleaned_otus)\n",
    "bleed_counts = bleed_counts.loc[(bleed_counts != 0).any(1)]\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "#print len(bleed_counts.index), bleed_sum\n",
    "\n",
    "#finally go back to samples and get read counts for the ovlerlapping otus\n",
    "real_counts = df3.reindex(columns=samples, index=bleed_counts.index)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed2 = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed into mock: %f%% (%i out of %i)\" % (index_bleed2, bleed_sum, real_sum)\n",
    "print \"Total index-bleed: %f%%\" % (index_bleed + index_bleed2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index-bleed out of mock: 0.244371% (6671 out of 2729864)\n",
      "Index-bleed into mock: 0.032036% (3027 out of 9448644)\n",
      "Total index-bleed: 0.276407%\n"
     ]
    }
   ],
   "source": [
    "#now measure index bleed using the biomock for m6\n",
    "#need to measure index bleed of miseq data now, trying 0 mismatch miseqC dataset first\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from natsort import natsorted\n",
    "import sys\n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "df = pd.read_csv('miseq.final.txt', sep='\\t', index_col=0)\n",
    "#print list(df.columns.values)\n",
    "#sys.exit(1)\n",
    "#print len(df)\n",
    "#okay, so 1367 OTUs in entire data set, lets trim this down to only run 3, as that one is labeled here with biomock\n",
    "keepers = ['m6-301-1', 'm6-301-2', 'm6-500-1', 'm6-500-2', 'm6-712-1', 'm6-712-2', \n",
    "           'm6-736-1', 'm6-736-2', 'm6-744-1', 'm6-744-2', 'm6-755-1', 'm6-755-2', \n",
    "           'm6-757-1', 'm6-757-2', 'm6-766-1', 'm6-766-2', \n",
    "           'm6-mock3-16000', 'm6-mock3-32000a', 'm6-mock3-32000b', 'm6-stds', 'm6-SynMock']\n",
    "mock_samples = ['m6-mock3-16000', 'm6-mock3-32000a', 'm6-mock3-32000b', 'm6-stds']\n",
    "samples = [item for item in keepers if item not in mock_samples]\n",
    "#print samples\n",
    "df2 = df.reindex(columns=keepers, index=natsorted(df.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "df3 = df2.loc[(df2 != 0).any(1)]\n",
    "mock_otus = []\n",
    "for x in df3.index:\n",
    "    if x.startswith('mock'):\n",
    "        if not x.startswith('mock3'): #mock3 is actually in some of the samples so can't use for index-bleed measurment\n",
    "            mock_otus.append(x)\n",
    "#now slice into two tables containing only mocks and calculate index-bleed\n",
    "bleed_counts = df3.reindex(columns=samples, index=mock_otus)\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "real_counts = df3.reindex(columns=mock_samples, index=mock_otus)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed out of mock: %f%% (%i out of %i)\" % (index_bleed, bleed_sum, real_sum)\n",
    "\n",
    "#now try to to do the opposite, count number of times that samples move into mock\n",
    "#first generate list of mock otus\n",
    "sample_otus = []\n",
    "for x in df3.index:\n",
    "    if not x.startswith('mock'):\n",
    "        sample_otus.append(x)\n",
    "#slice first to samples, further trim list of only things found in mocks, would be spurious or chimeras\n",
    "sampledf = df3.reindex(columns=samples, index=sample_otus)\n",
    "sampledf = sampledf.loc[(sampledf != 0).any(1)]\n",
    "cleaned_otus = sampledf.index\n",
    "#print len(cleaned_otus), len(sample_otus)\n",
    "\n",
    "#now using new list grab counts in mock\n",
    "bleed_counts = df3.reindex(columns=mock_samples, index=cleaned_otus)\n",
    "bleed_counts = bleed_counts.loc[(bleed_counts != 0).any(1)]\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "#print len(bleed_counts.index), bleed_sum\n",
    "\n",
    "#finally go back to samples and get read counts for the ovlerlapping otus\n",
    "real_counts = df3.reindex(columns=samples, index=bleed_counts.index)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed2 = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed into mock: %f%% (%i out of %i)\" % (index_bleed2, bleed_sum, real_sum)\n",
    "print \"Total index-bleed: %f%%\" % (index_bleed + index_bleed2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index-bleed out of mock: 0.032520% (340 out of 1045519)\n",
      "Index-bleed into mock: 0.013675% (683 out of 4994664)\n",
      "Total index-bleed: 0.046194%\n"
     ]
    }
   ],
   "source": [
    "#now measure index bleed using synmock data set in m6 run\n",
    "#need to measure index bleed of miseq data now, trying 0 mismatch miseqC dataset first\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from natsort import natsorted\n",
    "import sys\n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "df = pd.read_csv('miseq_syn.final.txt', sep='\\t', index_col=0)\n",
    "#print list(df.columns.values)\n",
    "#sys.exit(1)\n",
    "#print len(df)\n",
    "#okay, so 1367 OTUs in entire data set, lets trim this down to only run 3, as that one is labeled here with biomock\n",
    "keepers = ['m6-301-1', 'm6-301-2', 'm6-500-1', 'm6-500-2', 'm6-712-1', 'm6-712-2', \n",
    "           'm6-736-1', 'm6-736-2', 'm6-744-1', 'm6-744-2', 'm6-755-1', 'm6-755-2', \n",
    "           'm6-757-1', 'm6-757-2', 'm6-766-1', 'm6-766-2', \n",
    "           'm6-mock3-16000', 'm6-mock3-32000a', 'm6-mock3-32000b', 'm6-stds', 'm6-SynMock']\n",
    "mock_samples = ['m6-SynMock']\n",
    "samples = [item for item in keepers if item not in mock_samples]\n",
    "#print samples\n",
    "df2 = df.reindex(columns=keepers, index=natsorted(df.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "df3 = df2.loc[(df2 != 0).any(1)]\n",
    "mock_otus = []\n",
    "for x in df3.index:\n",
    "    if x.startswith('mock'):\n",
    "        if not x.startswith('mock3'): #mock3 is actually in some of the samples so can't use for index-bleed measurment\n",
    "            mock_otus.append(x)\n",
    "#now slice into two tables containing only mocks and calculate index-bleed\n",
    "bleed_counts = df3.reindex(columns=samples, index=mock_otus)\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "real_counts = df3.reindex(columns=mock_samples, index=mock_otus)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed out of mock: %f%% (%i out of %i)\" % (index_bleed, bleed_sum, real_sum)\n",
    "\n",
    "#now try to to do the opposite, count number of times that samples move into mock\n",
    "#first generate list of mock otus\n",
    "sample_otus = []\n",
    "for x in df3.index:\n",
    "    if not x.startswith('mock'):\n",
    "        sample_otus.append(x)\n",
    "#slice first to samples, further trim list of only things found in mocks, would be spurious or chimeras\n",
    "sampledf = df3.reindex(columns=samples, index=sample_otus)\n",
    "sampledf = sampledf.loc[(sampledf != 0).any(1)]\n",
    "cleaned_otus = sampledf.index\n",
    "#print len(cleaned_otus), len(sample_otus)\n",
    "\n",
    "#now using new list grab counts in mock\n",
    "bleed_counts = df3.reindex(columns=mock_samples, index=cleaned_otus)\n",
    "bleed_counts = bleed_counts.loc[(bleed_counts != 0).any(1)]\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "#print len(bleed_counts.index), bleed_sum\n",
    "\n",
    "#finally go back to samples and get read counts for the ovlerlapping otus\n",
    "real_counts = df3.reindex(columns=samples, index=bleed_counts.index)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed2 = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed into mock: %f%% (%i out of %i)\" % (index_bleed2, bleed_sum, real_sum)\n",
    "print \"Total index-bleed: %f%%\" % (index_bleed + index_bleed2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "vsearch v2.4.3_osx_x86_64, 16.0GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading input file 100%                                                                                                                                                                                                                                                                                                                                                                                                              \n",
      "5946978 sequences kept (of which 0 truncated), 0 sequences discarded.\n",
      "vsearch v2.4.3_osx_x86_64, 16.0GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file miseq.cluster.otus.fa 100%  \n",
      "744596 nt in 2816 seqs, min 100, max 300, avg 264\n",
      "Masking 100%  \n",
      "Counting unique k-mers 100%  \n",
      "Creating index of unique k-mers 100%  \n",
      "Searching 100%                                                                                                                                                                                                                                                                                                                                                                                                                \n",
      "Matching query sequences: 5573556 of 5946978 (93.72%)\n",
      "Writing OTU table (classic) 100%   \n"
     ]
    }
   ],
   "source": [
    "#want to measure index bleed, also have 1 mismatch dataset, so generate OTU table for that dataset\n",
    "!vsearch --fastq_filter m4a1m.demux.fq -fastaout m4a1m.demux.fa\n",
    "!vsearch --usearch_global m4a1m.demux.fa --db miseq.cluster.otus.fa --id 0.97 --strand plus --otutabout miseq4a1m.otu_table.txt\n",
    "!rm m4a1m.demux.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[11:29:52 PM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[11:29:52 PM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[11:29:52 PM]\u001b[0m: Loading OTU table: miseq4a1m.otu_table.txt\n",
      "\u001b[92m[11:29:52 PM]\u001b[0m: OTU table contains 2399 OTUs\n",
      "\u001b[92m[11:29:52 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[11:29:52 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[11:29:52 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[11:29:52 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[11:29:52 PM]\u001b[0m: Index bleed, mock into samples: 75.585340%.  Index bleed, samples into mock: 0.389027%.\n",
      "\u001b[92m[11:29:52 PM]\u001b[0m: Overwriting auto detect index-bleed, setting to 0.000000%\n",
      "\u001b[92m[11:29:54 PM]\u001b[0m: Filtering OTU table down to 1989 OTUs\n",
      "\u001b[92m[11:29:54 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      miseq4a1m.stats.txt\n",
      "Sorted OTU table:     miseq4a1m.sorted.txt\n",
      "Normalized/filter:    miseq4a1m.normalized.txt\n",
      "Final Binary table:   miseq4a1m.final.binary.txt\n",
      "Final OTU table:      miseq4a1m.final.txt\n",
      "Filtered OTUs:        miseq4a1m.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f miseq4a1m.filtered.otus.fa -i miseq4a1m.final.txt -m mapping_file.txt -d ITS2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#run through filtering so it is sorted etc\n",
    "!amptk filter -i miseq4a1m.otu_table.txt -f miseq.cluster.otus.fa -p 0 -b m4a-1mm-stds -m mock2 --keep_mock -o miseq4a1m"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index-bleed out of mock: 0.323377% (3206 out of 991411)\n",
      "Index-bleed into mock: 0.033259% (1038 out of 3120998)\n",
      "Total index-bleed: 0.356636%\n"
     ]
    }
   ],
   "source": [
    "#need to measure index bleed of miseq data now, trying 0 mismatch miseqA dataset\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from natsort import natsorted\n",
    "import sys\n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "df = pd.read_csv('miseq4a1m.sorted.txt', sep='\\t', index_col=0)\n",
    "#print list(df.columns.values)\n",
    "#sys.exit(1)\n",
    "#print len(df)\n",
    "#okay, so 1367 OTUs in entire data set, lets trim this down to only run 3, as that one is labeled here with biomock\n",
    "keepers = ['m4a-1mm-301-1','m4a-1mm-301-2','m4a-1mm-500-1','m4a-1mm-500-2',\n",
    "           'm4a-1mm-712-1','m4a-1mm-712-2','m4a-1mm-736-1','m4a-1mm-736-2',\n",
    "           'm4a-1mm-744-1','m4a-1mm-744-2','m4a-1mm-755-1','m4a-1mm-755-2',\n",
    "           'm4a-1mm-757-1','m4a-1mm-757-2','m4a-1mm-766-1','m4a-1mm-766-2',\n",
    "           'm4a-1mm-mock3-16000','m4a-1mm-mock3-32000a','m4a-1mm-mock3-32000b','m4a-1mm-stds']\n",
    "mock_samples = ['m4a-1mm-mock3-16000','m4a-1mm-mock3-32000a','m4a-1mm-mock3-32000b','m4a-1mm-stds']\n",
    "samples = [item for item in keepers if item not in mock_samples]\n",
    "#print samples\n",
    "df2 = df.reindex(columns=keepers, index=natsorted(df.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "df3 = df2.loc[(df2 != 0).any(1)]\n",
    "mock_otus = []\n",
    "for x in df3.index:\n",
    "    if x.startswith('mock'):\n",
    "        if not x.startswith('mock3'): #mock3 is actually in some of the samples so can't use for index-bleed measurment\n",
    "            mock_otus.append(x)\n",
    "#now slice into two tables containing only mocks and calculate index-bleed\n",
    "bleed_counts = df3.reindex(columns=samples, index=mock_otus)\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "real_counts = df3.reindex(columns=mock_samples, index=mock_otus)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed out of mock: %f%% (%i out of %i)\" % (index_bleed, bleed_sum, real_sum)\n",
    "\n",
    "#now try to to do the opposite, count number of times that samples move into mock\n",
    "#first generate list of mock otus\n",
    "sample_otus = []\n",
    "for x in df3.index:\n",
    "    if not x.startswith('mock'):\n",
    "        sample_otus.append(x)\n",
    "#slice first to samples, further trim list of only things found in mocks, would be spurious or chimeras\n",
    "sampledf = df3.reindex(columns=samples, index=sample_otus)\n",
    "sampledf = sampledf.loc[(sampledf != 0).any(1)]\n",
    "cleaned_otus = sampledf.index\n",
    "#print len(cleaned_otus), len(sample_otus)\n",
    "\n",
    "#now using new list grab counts in mock\n",
    "bleed_counts = df3.reindex(columns=mock_samples, index=cleaned_otus)\n",
    "bleed_counts = bleed_counts.loc[(bleed_counts != 0).any(1)]\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "#print len(bleed_counts.index), bleed_sum\n",
    "\n",
    "#finally go back to samples and get read counts for the ovlerlapping otus\n",
    "real_counts = df3.reindex(columns=samples, index=bleed_counts.index)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed2 = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed into mock: %f%% (%i out of %i)\" % (index_bleed2, bleed_sum, real_sum)\n",
    "print \"Total index-bleed: %f%%\" % (index_bleed + index_bleed2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "vsearch v2.4.3_osx_x86_64, 16.0GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading input file 100%                                                                                                                                                                                                                                                                                                                                                                                                              \n",
      "19874272 sequences kept (of which 0 truncated), 0 sequences discarded.\n",
      "vsearch v2.4.3_osx_x86_64, 16.0GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file miseq.cluster.otus.fa 100%  \n",
      "744596 nt in 2816 seqs, min 100, max 300, avg 264\n",
      "Masking 100%   \n",
      "Counting unique k-mers 100%  \n",
      "Creating index of unique k-mers 100%  \n",
      "Searching 100%                                                                                                                                                                                                                                                                                                                                                                                                                \n",
      "Matching query sequences: 16720793 of 19874272 (84.13%)\n",
      "Writing OTU table (classic) 100%  \n",
      "-------------------------------------------------------\n",
      "\u001b[92m[12:30:05 AM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[12:30:06 AM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[12:30:06 AM]\u001b[0m: Loading OTU table: miseq61mm.otu_table.txt\n",
      "\u001b[92m[12:30:06 AM]\u001b[0m: OTU table contains 2745 OTUs\n",
      "\u001b[92m[12:30:06 AM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[12:30:06 AM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[12:30:06 AM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[12:30:06 AM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[12:30:06 AM]\u001b[0m: Index bleed, mock into samples: 75.579729%.  Index bleed, samples into mock: 0.487083%.\n",
      "\u001b[92m[12:30:06 AM]\u001b[0m: Overwriting auto detect index-bleed, setting to 0.000000%\n",
      "\u001b[92m[12:30:07 AM]\u001b[0m: Filtering OTU table down to 2050 OTUs\n",
      "\u001b[92m[12:30:07 AM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      miseq61mm.stats.txt\n",
      "Sorted OTU table:     miseq61mm.sorted.txt\n",
      "Normalized/filter:    miseq61mm.normalized.txt\n",
      "Final Binary table:   miseq61mm.final.binary.txt\n",
      "Final OTU table:      miseq61mm.final.txt\n",
      "Filtered OTUs:        miseq61mm.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f miseq61mm.filtered.otus.fa -i miseq61mm.final.txt -m mapping_file.txt -d ITS2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#measure index bleed of miseq61 mismatch dataset, so generate OTU table for that dataset\n",
    "!vsearch --fastq_filter m61mm.demux.fq -fastaout m61mm.demux.fa\n",
    "!vsearch --usearch_global m61mm.demux.fa --db miseq.cluster.otus.fa --id 0.97 --strand plus --otutabout miseq61mm.otu_table.txt\n",
    "!rm m61mm.demux.fa\n",
    "!amptk filter -i miseq61mm.otu_table.txt -f miseq.cluster.otus.fa -p 0 -b m6-stds -m mock2 --keep_mock -o miseq61mm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index-bleed out of mock: 0.302544% (8778 out of 2901396)\n",
      "Index-bleed into mock: 0.035714% (4094 out of 11463401)\n",
      "Total index-bleed: 0.338258%\n"
     ]
    }
   ],
   "source": [
    "#now measure index bleed using synmock data set in m6 run\n",
    "#need to measure index bleed of miseq data now, trying 0 mismatch miseqC dataset first\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from natsort import natsorted\n",
    "import sys\n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "df = pd.read_csv('miseq61mm.final.txt', sep='\\t', index_col=0)\n",
    "#print list(df.columns.values)\n",
    "#sys.exit(1)\n",
    "#print len(df)\n",
    "#okay, so 1367 OTUs in entire data set, lets trim this down to only run 3, as that one is labeled here with biomock\n",
    "keepers = ['m6-301-1', 'm6-301-2', 'm6-500-1', 'm6-500-2', 'm6-712-1', 'm6-712-2', \n",
    "           'm6-736-1', 'm6-736-2', 'm6-744-1', 'm6-744-2', 'm6-755-1', 'm6-755-2', \n",
    "           'm6-757-1', 'm6-757-2', 'm6-766-1', 'm6-766-2','m6-mock3-16000', 'm6-mock3-32000a', 'm6-mock3-32000b', 'm6-stds', \n",
    "           'm6-SynMock']\n",
    "mock_samples = ['m6-mock3-16000', 'm6-mock3-32000a', 'm6-mock3-32000b', 'm6-stds']\n",
    "samples = [item for item in keepers if item not in mock_samples]\n",
    "#print samples\n",
    "df2 = df.reindex(columns=keepers, index=natsorted(df.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "df3 = df2.loc[(df2 != 0).any(1)]\n",
    "mock_otus = []\n",
    "for x in df3.index:\n",
    "    if x.startswith('mock'):\n",
    "        if not x.startswith('mock3'): #mock3 is actually in some of the samples so can't use for index-bleed measurment\n",
    "            mock_otus.append(x)\n",
    "#now slice into two tables containing only mocks and calculate index-bleed\n",
    "bleed_counts = df3.reindex(columns=samples, index=mock_otus)\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "real_counts = df3.reindex(columns=mock_samples, index=mock_otus)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed out of mock: %f%% (%i out of %i)\" % (index_bleed, bleed_sum, real_sum)\n",
    "\n",
    "#now try to to do the opposite, count number of times that samples move into mock\n",
    "#first generate list of mock otus\n",
    "sample_otus = []\n",
    "for x in df3.index:\n",
    "    if not x.startswith('mock'):\n",
    "        sample_otus.append(x)\n",
    "#slice first to samples, further trim list of only things found in mocks, would be spurious or chimeras\n",
    "sampledf = df3.reindex(columns=samples, index=sample_otus)\n",
    "sampledf = sampledf.loc[(sampledf != 0).any(1)]\n",
    "cleaned_otus = sampledf.index\n",
    "#print len(cleaned_otus), len(sample_otus)\n",
    "\n",
    "#now using new list grab counts in mock\n",
    "bleed_counts = df3.reindex(columns=mock_samples, index=cleaned_otus)\n",
    "bleed_counts = bleed_counts.loc[(bleed_counts != 0).any(1)]\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "#print len(bleed_counts.index), bleed_sum\n",
    "\n",
    "#finally go back to samples and get read counts for the ovlerlapping otus\n",
    "real_counts = df3.reindex(columns=samples, index=bleed_counts.index)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed2 = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed into mock: %f%% (%i out of %i)\" % (index_bleed2, bleed_sum, real_sum)\n",
    "print \"Total index-bleed: %f%%\" % (index_bleed + index_bleed2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[07:58:17 AM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[07:58:17 AM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[07:58:17 AM]\u001b[0m: Loading OTU table: miseq61mm.otu_table.txt\n",
      "\u001b[92m[07:58:17 AM]\u001b[0m: OTU table contains 2745 OTUs\n",
      "\u001b[92m[07:58:17 AM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[07:58:17 AM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[07:58:17 AM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[07:58:17 AM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[07:58:17 AM]\u001b[0m: Index bleed, mock into samples: 0.074052%.  Index bleed, samples into mock: 0.133015%.\n",
      "\u001b[92m[07:58:17 AM]\u001b[0m: Overwriting auto detect index-bleed, setting to 0.000000%\n",
      "\u001b[92m[07:58:18 AM]\u001b[0m: Filtering OTU table down to 2050 OTUs\n",
      "\u001b[92m[07:58:18 AM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      miseq61mm_syn.stats.txt\n",
      "Sorted OTU table:     miseq61mm_syn.sorted.txt\n",
      "Normalized/filter:    miseq61mm_syn.normalized.txt\n",
      "Final Binary table:   miseq61mm_syn.final.binary.txt\n",
      "Final OTU table:      miseq61mm_syn.final.txt\n",
      "Filtered OTUs:        miseq61mm_syn.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f miseq61mm_syn.filtered.otus.fa -i miseq61mm_syn.final.txt -m mapping_file.txt -d ITS2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#now we want to measure index bleed using synmock\n",
    "!amptk filter -i miseq61mm.otu_table.txt -f miseq.cluster.otus.fa -p 0 -b m6-SynMock -m synmock --keep_mock -o miseq61mm_syn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index-bleed out of mock: 0.054651% (603 out of 1103361)\n",
      "Index-bleed into mock: 0.017822% (1238 out of 6946547)\n",
      "Total index-bleed: 0.072473%\n"
     ]
    }
   ],
   "source": [
    "#now measure index bleed using synmock data set in m6 run\n",
    "#need to measure index bleed of miseq data now, trying 0 mismatch miseqC dataset first\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from natsort import natsorted\n",
    "import sys\n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "df = pd.read_csv('miseq61mm_syn.final.txt', sep='\\t', index_col=0)\n",
    "#print list(df.columns.values)\n",
    "#sys.exit(1)\n",
    "#print len(df)\n",
    "#okay, so 1367 OTUs in entire data set, lets trim this down to only run 3, as that one is labeled here with biomock\n",
    "keepers = ['m6-301-1', 'm6-301-2', 'm6-500-1', 'm6-500-2', 'm6-712-1', 'm6-712-2', \n",
    "           'm6-736-1', 'm6-736-2', 'm6-744-1', 'm6-744-2', 'm6-755-1', 'm6-755-2', \n",
    "           'm6-757-1', 'm6-757-2', 'm6-766-1', 'm6-766-2','m6-mock3-16000', 'm6-mock3-32000a', 'm6-mock3-32000b', 'm6-stds', \n",
    "           'm6-SynMock']\n",
    "mock_samples = ['m6-SynMock']\n",
    "samples = [item for item in keepers if item not in mock_samples]\n",
    "#print samples\n",
    "df2 = df.reindex(columns=keepers, index=natsorted(df.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "df3 = df2.loc[(df2 != 0).any(1)]\n",
    "mock_otus = []\n",
    "for x in df3.index:\n",
    "    if x.startswith('mock'):\n",
    "        mock_otus.append(x)\n",
    "#now slice into two tables containing only mocks and calculate index-bleed\n",
    "bleed_counts = df3.reindex(columns=samples, index=mock_otus)\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "real_counts = df3.reindex(columns=mock_samples, index=mock_otus)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed out of mock: %f%% (%i out of %i)\" % (index_bleed, bleed_sum, real_sum)\n",
    "\n",
    "#now try to to do the opposite, count number of times that samples move into mock\n",
    "#first generate list of mock otus\n",
    "sample_otus = []\n",
    "for x in df3.index:\n",
    "    if not x.startswith('mock'):\n",
    "        sample_otus.append(x)\n",
    "#slice first to samples, further trim list of only things found in mocks, would be spurious or chimeras\n",
    "sampledf = df3.reindex(columns=samples, index=sample_otus)\n",
    "sampledf = sampledf.loc[(sampledf != 0).any(1)]\n",
    "cleaned_otus = sampledf.index\n",
    "#print len(cleaned_otus), len(sample_otus)\n",
    "\n",
    "#now using new list grab counts in mock\n",
    "bleed_counts = df3.reindex(columns=mock_samples, index=cleaned_otus)\n",
    "bleed_counts = bleed_counts.loc[(bleed_counts != 0).any(1)]\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "#print len(bleed_counts.index), bleed_sum\n",
    "\n",
    "#finally go back to samples and get read counts for the ovlerlapping otus\n",
    "real_counts = df3.reindex(columns=samples, index=bleed_counts.index)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed2 = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed into mock: %f%% (%i out of %i)\" % (index_bleed2, bleed_sum, real_sum)\n",
    "print \"Total index-bleed: %f%%\" % (index_bleed + index_bleed2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index-bleed out of mock: 0.0%\n"
     ]
    },
    {
     "ename": "SystemExit",
     "evalue": "1",
     "output_type": "error",
     "traceback": [
      "An exception has occurred, use %tb to see the full traceback.\n",
      "\u001b[0;31mSystemExit\u001b[0m\u001b[0;31m:\u001b[0m 1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/jon/miniconda2/lib/python2.7/site-packages/IPython/core/interactiveshell.py:2889: UserWarning: To exit: use 'exit', 'quit', or Ctrl-D.\n",
      "  warn(\"To exit: use 'exit', 'quit', or Ctrl-D.\", stacklevel=1)\n"
     ]
    }
   ],
   "source": [
    "#need to measure index bleed of miseqB dataset, this one slightly differnt as can only measure index out of mock\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from natsort import natsorted\n",
    "import sys\n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "df = pd.read_csv('miseq.final.txt', sep='\\t', index_col=0)\n",
    "#print list(df.columns.values)\n",
    "#sys.exit(1)\n",
    "#print len(df)\n",
    "#okay, so 1367 OTUs in entire data set, lets trim this down to only run 3, as that one is labeled here with biomock\n",
    "keepers = ['m4b-744-1', 'm4b-744-2', 'm4b-766-1', 'm4b-766-2']\n",
    "mock_samples = ['m4C-mock3-16000', 'm4C-mock3-32000a', 'm4C-mock3-32000b', 'm4C-stds']\n",
    "samples = [item for item in keepers if item not in mock_samples]\n",
    "#print samples\n",
    "df2 = df.reindex(columns=keepers, index=natsorted(df.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "df3 = df2.loc[(df2 != 0).any(1)]\n",
    "mock_otus = []\n",
    "for x in df3.index: #get index from total table\n",
    "    if x.startswith('mock'):\n",
    "        if not x.startswith('mock3'): #mock3 is actually in some of the samples so can't use for index-bleed measurment\n",
    "            mock_otus.append(x)\n",
    "if len(mock_otus) < 1: #this means that no read counts from mock members were found, so index-bleed is 0\n",
    "    print \"Index-bleed out of mock: 0.0%\"\n",
    "    sys.exit(1)\n",
    "#now slice into two tables containing only mocks and calculate index-bleed\n",
    "bleed_counts = df3.reindex(columns=samples, index=mock_otus)\n",
    "bleed_sum = bleed_counts.values.sum()\n",
    "real_counts = df3.reindex(columns=mock_samples, index=mock_otus)\n",
    "real_sum = real_counts.values.sum()\n",
    "index_bleed = int(bleed_sum) / float(real_sum) * 100\n",
    "print \"Index-bleed out of mock: %f%% (%i out of %i)\" % (index_bleed, bleed_sum, real_sum)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "| Run    | BC mismatch |  Mock   | Index-bleed out | Index-bleed in | Total index bleed |\n",
    "| ------ | :---------: | :-----: | :-------------: | :------------: | :---------------: |\n",
    "| Ion3   |      0      | biomock |     0.025%      |     0.018%     |      0.043%       |\n",
    "| Ion5   |      0      | synmock |     0.041%      |     0.028%     |      0.069%       |\n",
    "| MiseqA |      0      | biomock |     0.248%      |     0.015%     |      0.263%       |\n",
    "| MiseqA |      1      | biomock |     0.323%      |     0.033%     |      0.356%       |\n",
    "| MiSeqC |      0      | biomock |     0.216%      |     0.012%     |      0.228%       |\n",
    "| Miseq6 |      0      | biomock |     0.244%      |     0.032%     |      0.276%       |\n",
    "| Miseq6 |      0      | synmock |     0.033%      |     0.014%     |      0.046%       |\n",
    "| Miseq6 |      1      | biomock |     0.303%      |     0.035%     |      0.338%       |\n",
    "| Miseq6 |      1      | synmock |     0.055%      |     0.018%     |      0.073%       |\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "These data suggest that measuring index-bleed with biomock is over-estimating the cross-over.  These data also suggest that one can limit some index bleed by demuxing the data using a BC (index) mismatch of 0 instead of the default of 1 on MiSeq runs.  On average, there is a 10% reduction in the number of reads retained if BC mismatch is set to 0 versus 1 (default) on the Miseq6 run, however there was an increase in 63% in total index-bleed (0.046% to 0.073%), suggesting that a larger percentage of the reads containing BC mismatches are sample mis-classified."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Index Bleeds to use for filtering, round up to nearest 10th percentage. will use max read count for threshold for each\n",
    "Ion3: 0.001\n",
    "\n",
    "Ion5: 0.001\n",
    "\n",
    "MiSeqA: 0.003\n",
    "\n",
    "MiSeqC: 0.003\n",
    "\n",
    "MiSeq6: 0.001"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#generate OTU table for each of the runs using the combined OTUs from either miseq or ion\n",
    "#now slice the OTU table for each of the runs, Ion3, Miseq4A, Miseq4C, Ion5\n",
    "#load original table, not the name changed version for normal processing\n",
    "import pandas as pd\n",
    "from natsort import natsorted\n",
    "\n",
    "miseqdf = pd.read_csv('miseq.otu_table.txt', sep='\\t', index_col=0)\n",
    "iondf = pd.read_csv('ion.otu_table.txt', sep='\\t', index_col=0)\n",
    "\n",
    "ion3 = ['m3-301-1', 'm3-301-2', 'm3-500-1', 'm3-500-2', 'm3-712-1', 'm3-712-2', \n",
    "        'm3-736-1', 'm3-736-2', 'm3-744-1', 'm3-744-2', 'm3-755-1',\n",
    "        'm3-757-2', 'm3-766-1', 'm3-766-2', 'm3-mock3-16000', 'm3-mock3-3200a', 'm3-mock3-32000b', 'm3-stds']\n",
    "ion3df = iondf.reindex(columns=ion3, index=natsorted(iondf.index))\n",
    "ion3df.to_csv('ion3.otu_table.txt', sep='\\t')\n",
    "miseq4a = ['m4A-301-1', 'm4A-301-2', 'm4A-500-1', 'm4A-500-2', 'm4A-712-1', \n",
    "           'm4A-712-2', 'm4A-736-1', 'm4A-736-2', 'm4A-744-1', 'm4A-744-2', \n",
    "           'm4A-755-1', 'm4A-755-2', 'm4A-757-1', 'm4A-757-2', 'm4A-766-1', 'm4A-766-2',\n",
    "           'm4A-mock3-16000', 'm4A-mock3-32000a', 'm4A-mock3-32000b', 'm4A-stds']\n",
    "miseq4adf = miseqdf.reindex(columns=miseq4a, index=natsorted(miseqdf.index))\n",
    "miseq4adf.to_csv('miseq4a.otu_table.txt', sep='\\t')\n",
    "miseq4c = ['m4C-301-1', 'm4C-301-2', 'm4C-500-1', 'm4C-500-2', 'm4C-712-1', \n",
    "           'm4C-712-2', 'm4C-736-1', 'm4C-736-2', 'm4C-744-1', 'm4C-744-2', \n",
    "           'm4C-755-1', 'm4C-755-2', 'm4C-757-1', 'm4C-757-2', 'm4C-766-1', 'm4C-766-2',\n",
    "          'm4C-mock3-16000', 'm4C-mock3-32000a', 'm4C-mock3-32000b', 'm4C-stds']\n",
    "miseq4cdf = miseqdf.reindex(columns=miseq4c, index=natsorted(miseqdf.index))\n",
    "miseq4cdf.to_csv('miseq4c.otu_table.txt', sep='\\t')\n",
    "miseq6 = ['m6-301-1', 'm6-301-2', 'm6-500-1', 'm6-500-2', 'm6-712-1', \n",
    "           'm6-712-2', 'm6-736-1', 'm6-736-2', 'm6-744-1', 'm6-744-2', \n",
    "           'm6-755-1', 'm6-755-2', 'm6-757-1', 'm6-757-2', 'm6-766-1', 'm6-766-2', \n",
    "          'm6-mock3-16000', 'm6-mock3-32000a', 'm6-mock3-32000b', 'm6-stds','m6-SynMock']\n",
    "miseq6df = miseqdf.reindex(columns=miseq6, index=natsorted(miseqdf.index))\n",
    "miseq6df.to_csv('miseq6.otu_table.txt', sep='\\t')\n",
    "ion5 = ['m5-301-1', 'm5-301-2', 'm5-500-1', 'm5-500-2', 'm5-712-1', \n",
    "        'm5-712-2', 'm5-736-1', 'm5-736-2', 'm5-744-1', 'm5-744-2', \n",
    "        'm5-755-1', 'm5-755-2', 'm5-757-1', 'm5-757-2', 'm5-766-1', 'm5-766-2',\n",
    "        'm5-SynMock1-8000', 'm5-SynMock1-mock3.0', 'm5-SynMock1-mock3.5', 'm5-SynMock1.736-1.736-2',\n",
    "        'm5-SynMock2-8000', 'm5-SynMock2.757-1.757-2', 'm5-mock3.0-16000', 'm5-mock3.5-16000']\n",
    "ion5df = iondf.reindex(columns=ion5, index=natsorted(iondf.index))\n",
    "ion5df.to_csv('ion5.otu_table.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[02:34:09 PM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[02:34:09 PM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[02:34:09 PM]\u001b[0m: Loading OTU table: miseq4a.otu_table.txt\n",
      "\u001b[92m[02:34:09 PM]\u001b[0m: OTU table contains 2816 OTUs\n",
      "\u001b[92m[02:34:09 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[02:34:09 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[02:34:09 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[02:34:09 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[02:34:09 PM]\u001b[0m: Index bleed, samples into mock: 0.059001%.\n",
      "\u001b[92m[02:34:09 PM]\u001b[0m: Overwriting auto detect index-bleed, setting to 0.300000%\n",
      "\u001b[92m[02:34:09 PM]\u001b[0m: Dropping 3 samples from table: m4A-mock3-32000a, m4A-mock3-32000b, m4A-stds\n",
      "\u001b[92m[02:34:10 PM]\u001b[0m: Filtering OTU table down to 1937 OTUs\n",
      "\u001b[92m[02:34:10 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      miseq4a.stats.txt\n",
      "Sorted OTU table:     miseq4a.sorted.txt\n",
      "Normalized/filter:    miseq4a.normalized.txt\n",
      "Final Binary table:   miseq4a.final.binary.txt\n",
      "Final OTU table:      miseq4a.final.txt\n",
      "Filtered OTUs:        miseq4a.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f miseq4a.filtered.otus.fa -i miseq4a.final.txt -m mapping_file.txt -d ITS2\n",
      "\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[02:34:11 PM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[02:34:11 PM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[02:34:11 PM]\u001b[0m: Loading OTU table: miseq4c.otu_table.txt\n",
      "\u001b[92m[02:34:11 PM]\u001b[0m: OTU table contains 2816 OTUs\n",
      "\u001b[92m[02:34:11 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[02:34:11 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[02:34:11 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[02:34:11 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[02:34:11 PM]\u001b[0m: Index bleed, samples into mock: 0.057004%.\n",
      "\u001b[92m[02:34:11 PM]\u001b[0m: Overwriting auto detect index-bleed, setting to 0.300000%\n",
      "\u001b[92m[02:34:11 PM]\u001b[0m: Dropping 3 samples from table: m4C-mock3-32000a, m4C-mock3-32000, m4C-stds\n",
      "\u001b[92m[02:34:12 PM]\u001b[0m: Filtering OTU table down to 1860 OTUs\n",
      "\u001b[92m[02:34:12 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      miseq4c.stats.txt\n",
      "Sorted OTU table:     miseq4c.sorted.txt\n",
      "Normalized/filter:    miseq4c.normalized.txt\n",
      "Final Binary table:   miseq4c.final.binary.txt\n",
      "Final OTU table:      miseq4c.final.txt\n",
      "Filtered OTUs:        miseq4c.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f miseq4c.filtered.otus.fa -i miseq4c.final.txt -m mapping_file.txt -d ITS2\n",
      "\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[02:34:13 PM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[02:34:13 PM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[02:34:13 PM]\u001b[0m: Loading OTU table: miseq6.otu_table.txt\n",
      "\u001b[92m[02:34:13 PM]\u001b[0m: OTU table contains 2816 OTUs\n",
      "\u001b[92m[02:34:13 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[02:34:13 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[02:34:13 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[02:34:13 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[02:34:13 PM]\u001b[0m: Index bleed, mock into samples: 0.041021%.  Index bleed, samples into mock: 0.084007%.\n",
      "\u001b[92m[02:34:13 PM]\u001b[0m: Overwriting auto detect index-bleed, setting to 0.100000%\n",
      "\u001b[92m[02:34:14 PM]\u001b[0m: Dropping 4 samples from table: m6-mock3-16000, m6-mock3-32000a, m6-mock3-32000b, m6-stds\n",
      "\u001b[92m[02:34:14 PM]\u001b[0m: Filtering OTU table down to 2018 OTUs\n",
      "\u001b[92m[02:34:14 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      miseq6.stats.txt\n",
      "Sorted OTU table:     miseq6.sorted.txt\n",
      "Normalized/filter:    miseq6.normalized.txt\n",
      "Final Binary table:   miseq6.final.binary.txt\n",
      "Final OTU table:      miseq6.final.txt\n",
      "Filtered OTUs:        miseq6.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f miseq6.filtered.otus.fa -i miseq6.final.txt -m mapping_file.txt -d ITS2\n",
      "\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[02:34:15 PM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[02:34:16 PM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[02:34:16 PM]\u001b[0m: Loading OTU table: ion3.otu_table.txt\n",
      "\u001b[92m[02:34:16 PM]\u001b[0m: OTU table contains 1310 OTUs\n",
      "\u001b[92m[02:34:16 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[02:34:16 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[02:34:16 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[02:34:16 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[02:34:16 PM]\u001b[0m: Index bleed, samples into mock: 1.121011%.\n",
      "\u001b[92m[02:34:16 PM]\u001b[0m: Overwriting auto detect index-bleed, setting to 0.100000%\n",
      "\u001b[92m[02:34:16 PM]\u001b[0m: Dropping 3 samples from table: m3-mock3-3200a, m3-mock3-32000b, m3-stds\n",
      "\u001b[92m[02:34:16 PM]\u001b[0m: Filtering OTU table down to 1071 OTUs\n",
      "\u001b[92m[02:34:16 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      ion3.stats.txt\n",
      "Sorted OTU table:     ion3.sorted.txt\n",
      "Normalized/filter:    ion3.normalized.txt\n",
      "Final Binary table:   ion3.final.binary.txt\n",
      "Final OTU table:      ion3.final.txt\n",
      "Filtered OTUs:        ion3.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f ion3.filtered.otus.fa -i ion3.final.txt -m mapping_file.txt -d ITS2\n",
      "\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[02:34:17 PM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[02:34:17 PM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[02:34:17 PM]\u001b[0m: Loading OTU table: ion5.otu_table.txt\n",
      "\u001b[92m[02:34:17 PM]\u001b[0m: OTU table contains 1310 OTUs\n",
      "\u001b[92m[02:34:17 PM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[02:34:17 PM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[02:34:17 PM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[02:34:17 PM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[02:34:17 PM]\u001b[0m: Index bleed, mock into samples: 76.565042%.  Index bleed, samples into mock: 1.913904%.\n",
      "\u001b[92m[02:34:17 PM]\u001b[0m: Overwriting auto detect index-bleed, setting to 0.100000%\n",
      "\u001b[92m[02:34:17 PM]\u001b[0m: Dropping 7 samples from table: m5-SynMock1-8000, m5-SynMock1-mock3.0, m5-SynMock1-mock3.5, m5-SynMock1.736-1.736-2, m5-SynMock2.757-1.757-2, m5-mock3.0-16000, m5-mock3.5-16000\n",
      "\u001b[92m[02:34:18 PM]\u001b[0m: Filtering OTU table down to 1115 OTUs\n",
      "\u001b[92m[02:34:18 PM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      ion5.stats.txt\n",
      "Sorted OTU table:     ion5.sorted.txt\n",
      "Normalized/filter:    ion5.normalized.txt\n",
      "Final Binary table:   ion5.final.binary.txt\n",
      "Final OTU table:      ion5.final.txt\n",
      "Filtered OTUs:        ion5.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f ion5.filtered.otus.fa -i ion5.final.txt -m mapping_file.txt -d ITS2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#now filter these datasets for raw and filtered (no subtraction, index-bleed based on calculated values above)\n",
    "!amptk filter -f miseq.cluster.otus.fa -i miseq4a.otu_table.txt -o miseq4a -b m4A-mock3-16000 -m mock3 -p 0.003 -c in -d m4A-mock3-32000a m4A-mock3-32000b m4A-stds \n",
    "!amptk filter -f miseq.cluster.otus.fa -i miseq4c.otu_table.txt -o miseq4c -b m4C-mock3-16000 -m mock3 -p 0.003 -c in -d m4C-mock3-32000a m4C-mock3-32000 m4C-stds\n",
    "!amptk filter -f miseq.cluster.otus.fa -i miseq6.otu_table.txt -o miseq6 -b m6-SynMock -m synmock -p 0.001 -d m6-mock3-16000 m6-mock3-32000a m6-mock3-32000b m6-stds\n",
    "!amptk filter -f ion.cluster.otus.fa -i ion3.otu_table.txt -o ion3 -b m3-mock3-16000 -m mock3 -p 0.001 -c in -d m3-mock3-3200a m3-mock3-32000b m3-stds\n",
    "!amptk filter -f ion.cluster.otus.fa -i ion5.otu_table.txt -o ion5 -b m5-SynMock2-8000 -m synmock -p 0.001 -d m5-SynMock1-8000 m5-SynMock1-mock3.0 m5-SynMock1-mock3.5 m5-SynMock1.736-1.736-2 m5-SynMock2.757-1.757-2 m5-mock3.0-16000 m5-mock3.5-16000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[10:45:47 AM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[10:45:47 AM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[10:45:47 AM]\u001b[0m: Loading OTU table: miseq6_biomock.otu_table.txt\n",
      "\u001b[92m[10:45:47 AM]\u001b[0m: OTU table contains 2816 OTUs\n",
      "\u001b[92m[10:45:47 AM]\u001b[0m: Mapping OTUs to Mock Community (USEARCH)\n",
      "\u001b[92m[10:45:47 AM]\u001b[0m: Sorting OTU table naturally\n",
      "\u001b[92m[10:45:48 AM]\u001b[0m: Removing OTUs according to --min_reads_otu: (OTUs with less than 2 reads from all samples)\n",
      "\u001b[92m[10:45:48 AM]\u001b[0m: Normalizing OTU table to number of reads per sample\n",
      "\u001b[92m[10:45:48 AM]\u001b[0m: Index bleed, mock into samples: 7.510277%.  Index bleed, samples into mock: 0.094006%.\n",
      "\u001b[92m[10:45:48 AM]\u001b[0m: Overwriting auto detect index-bleed, setting to 0.280000%\n",
      "\u001b[92m[10:45:48 AM]\u001b[0m: Filtering OTU table down to 2015 OTUs\n",
      "\u001b[92m[10:45:48 AM]\u001b[0m: Filtering valid OTUs\n",
      "-------------------------------------------------------\n",
      "OTU Table filtering finished\n",
      "-------------------------------------------------------\n",
      "OTU Table Stats:      miseq6_biomock.stats.txt\n",
      "Sorted OTU table:     miseq6_biomock.sorted.txt\n",
      "Normalized/filter:    miseq6_biomock.normalized.txt\n",
      "Final Binary table:   miseq6_biomock.final.binary.txt\n",
      "Final OTU table:      miseq6_biomock.final.txt\n",
      "Filtered OTUs:        miseq6_biomock.filtered.otus.fa\n",
      "-------------------------------------------------------\n",
      "\u001b[93m\n",
      "Example of next cmd:\u001b[0m amptk taxonomy -f miseq6_biomock.filtered.otus.fa -i miseq6_biomock.final.txt -m mapping_file.txt -d ITS2\n",
      "\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[10:45:49 AM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[10:45:49 AM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[10:45:49 AM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[10:45:49 AM]\u001b[0m: 2,015 OTUs\n",
      "\u001b[92m[10:45:49 AM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[10:45:49 AM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[10:46:01 AM]\u001b[0m: Taxonomy finished: miseq.taxonomy.txt\n",
      "\u001b[92m[10:46:01 AM]\u001b[0m: Classic OTU table with taxonomy: miseq6_biomock.otu_table.taxonomy.txt\n",
      "\u001b[92m[10:46:03 AM]\u001b[0m: BIOM OTU table created: miseq6_biomock.biom\n",
      "\u001b[92m[10:46:03 AM]\u001b[0m: OTUs with taxonomy: miseq6_biomock.otus.taxonomy.fa\n",
      "\u001b[92m[10:46:03 AM]\u001b[0m: OTU phylogeny: miseq6_biomock.tree.phy\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#try to filter the miseq6 dataset based on the biomock index bleed as well.\n",
    "#first create OTU table\n",
    "import pandas as pd\n",
    "from natsort import natsorted\n",
    "miseqdf = pd.read_csv('miseq.otu_table.txt', sep='\\t', index_col=0)\n",
    "miseq6 = ['m6-301-1', 'm6-301-2', 'm6-500-1', 'm6-500-2', 'm6-712-1', \n",
    "           'm6-712-2', 'm6-736-1', 'm6-736-2', 'm6-744-1', 'm6-744-2', \n",
    "           'm6-755-1', 'm6-755-2', 'm6-757-1', 'm6-757-2', 'm6-766-1', 'm6-766-2', 'm6-mock3-16000']\n",
    "miseq6df = miseqdf.reindex(columns=miseq6, index=natsorted(miseqdf.index))\n",
    "miseq6df.to_csv('miseq6_biomock.otu_table.txt', sep='\\t')\n",
    "!amptk filter -f miseq.cluster.otus.fa -i miseq6_biomock.otu_table.txt -o miseq6_biomock -b m6-mock3-16000 -m mock3 -p 0.0028\n",
    "!amptk taxonomy -f miseq6_biomock.filtered.otus.fa -i miseq6_biomock.final.txt -o miseq6_biomock -d ITS2 -t miseq.taxonomy.txt -m mapping/miseq6.mapping_file.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1310 15 1158 14\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#now generate the raw OTU table from the sorted (original output)\n",
    "import pandas as pd\n",
    "from natsort import natsorted\n",
    "import subprocess\n",
    "#pull out header for indexing later\n",
    "base = 'ion3'\n",
    "platform = 'ion'\n",
    "    \n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "sort = pd.read_csv(base+'.sorted.txt', sep='\\t', index_col=0)\n",
    "\n",
    "#now we want to drop last column\n",
    "header = ['m3-301-1', 'm3-301-2', 'm3-500-1', 'm3-500-2', 'm3-712-1', 'm3-712-2', \n",
    "        'm3-736-1', 'm3-736-2', 'm3-744-1', 'm3-744-2', 'm3-755-1',\n",
    "        'm3-757-2', 'm3-766-1', 'm3-766-2']\n",
    "cleaned = sort.reindex(columns=header, index=natsorted(sort.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "fs = cleaned.sum(axis=0)\n",
    "filtered = pd.DataFrame(cleaned, columns=fs.index)\n",
    "filt2 = filtered.loc[(filtered != 0).any(1)]\n",
    "print len(sort), len(sort.columns.values), len(filt2), len(filt2.columns.values)\n",
    "filt2.to_csv(base+'.sorted.nomock.txt', sep='\\t')\n",
    "\n",
    "#now convert to biom\n",
    "cmd = ['amptk', 'taxonomy', '-f', platform+'.cluster.otus.fa', '-o', base+'_raw', '-i', base+'.sorted.nomock.txt', '-d', 'ITS2','-m','mapping/'+base+'.mapping_file.txt', '-t', platform+'.taxonomy.txt']\n",
    "subprocess.call(cmd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1310 17 1179 16\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#now generate the raw OTU table from the sorted (original output)\n",
    "import pandas as pd\n",
    "from natsort import natsorted\n",
    "import subprocess\n",
    "#pull out header for indexing later\n",
    "base = 'ion5'\n",
    "platform = 'ion'\n",
    "    \n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "sort = pd.read_csv(base+'.sorted.txt', sep='\\t', index_col=0)\n",
    "\n",
    "#now we want to drop last column\n",
    "header = ['m5-301-1', 'm5-301-2', 'm5-500-1', 'm5-500-2', 'm5-712-1', \n",
    "        'm5-712-2', 'm5-736-1', 'm5-736-2', 'm5-744-1', 'm5-744-2', \n",
    "        'm5-755-1', 'm5-755-2', 'm5-757-1', 'm5-757-2', 'm5-766-1', 'm5-766-2']\n",
    "cleaned = sort.reindex(columns=header, index=natsorted(sort.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "fs = cleaned.sum(axis=0)\n",
    "filtered = pd.DataFrame(cleaned, columns=fs.index)\n",
    "filt2 = filtered.loc[(filtered != 0).any(1)]\n",
    "print len(sort), len(sort.columns.values), len(filt2), len(filt2.columns.values)\n",
    "filt2.to_csv(base+'.sorted.nomock.txt', sep='\\t')\n",
    "\n",
    "#now convert to biom\n",
    "cmd = ['amptk', 'taxonomy', '-f', platform+'.cluster.otus.fa', '-o', base+'_raw', '-i', base+'.sorted.nomock.txt', '-d', 'ITS2','-m','mapping/'+base+'.mapping_file.txt', '-t', platform+'.taxonomy.txt']\n",
    "subprocess.call(cmd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2816 17 2368 16\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#now generate the raw OTU table from the sorted (original output)\n",
    "import pandas as pd\n",
    "from natsort import natsorted\n",
    "import subprocess\n",
    "#pull out header for indexing later\n",
    "base = 'miseq4a'\n",
    "platform = 'miseq'\n",
    "    \n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "sort = pd.read_csv(base+'.sorted.txt', sep='\\t', index_col=0)\n",
    "\n",
    "#now we want to drop last column\n",
    "header = ['m4A-301-1', 'm4A-301-2', 'm4A-500-1', 'm4A-500-2', 'm4A-712-1', \n",
    "           'm4A-712-2', 'm4A-736-1', 'm4A-736-2', 'm4A-744-1', 'm4A-744-2', \n",
    "           'm4A-755-1', 'm4A-755-2', 'm4A-757-1', 'm4A-757-2', 'm4A-766-1', 'm4A-766-2']\n",
    "cleaned = sort.reindex(columns=header, index=natsorted(sort.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "fs = cleaned.sum(axis=0)\n",
    "filtered = pd.DataFrame(cleaned, columns=fs.index)\n",
    "filt2 = filtered.loc[(filtered != 0).any(1)]\n",
    "print len(sort), len(sort.columns.values), len(filt2), len(filt2.columns.values)\n",
    "filt2.to_csv(base+'.sorted.nomock.txt', sep='\\t')\n",
    "\n",
    "#now convert to biom\n",
    "cmd = ['amptk', 'taxonomy', '-f', platform+'.cluster.otus.fa', '-o', base+'_raw', '-i', base+'.sorted.nomock.txt', '-d', 'ITS2','-m','mapping/'+base+'.mapping_file.txt', '-t', platform+'.taxonomy.txt']\n",
    "subprocess.call(cmd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2816 17 2271 16\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#now generate the raw OTU table from the sorted (original output)\n",
    "import pandas as pd\n",
    "from natsort import natsorted\n",
    "import subprocess\n",
    "#pull out header for indexing later\n",
    "base = 'miseq4c'\n",
    "platform = 'miseq'\n",
    "    \n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "sort = pd.read_csv(base+'.sorted.txt', sep='\\t', index_col=0)\n",
    "\n",
    "#now we want to drop last column\n",
    "header = ['m4C-301-1', 'm4C-301-2', 'm4C-500-1', 'm4C-500-2', 'm4C-712-1', \n",
    "           'm4C-712-2', 'm4C-736-1', 'm4C-736-2', 'm4C-744-1', 'm4C-744-2', \n",
    "           'm4C-755-1', 'm4C-755-2', 'm4C-757-1', 'm4C-757-2', 'm4C-766-1', 'm4C-766-2']\n",
    "cleaned = sort.reindex(columns=header, index=natsorted(sort.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "fs = cleaned.sum(axis=0)\n",
    "filtered = pd.DataFrame(cleaned, columns=fs.index)\n",
    "filt2 = filtered.loc[(filtered != 0).any(1)]\n",
    "print len(sort), len(sort.columns.values), len(filt2), len(filt2.columns.values)\n",
    "filt2.to_csv(base+'.sorted.nomock.txt', sep='\\t')\n",
    "\n",
    "#now convert to biom\n",
    "cmd = ['amptk', 'taxonomy', '-f', platform+'.cluster.otus.fa', '-o', base+'_raw', '-i', base+'.sorted.nomock.txt', '-d', 'ITS2','-m','mapping/'+base+'.mapping_file.txt', '-t', platform+'.taxonomy.txt']\n",
    "subprocess.call(cmd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2816 17 2699 16\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#now generate the raw OTU table from the sorted (original output)\n",
    "import pandas as pd\n",
    "from natsort import natsorted\n",
    "import subprocess\n",
    "#pull out header for indexing later\n",
    "base = 'miseq6'\n",
    "platform = 'miseq'\n",
    "    \n",
    "#now load sorted \"raw\" OTU table into dataframe\n",
    "sort = pd.read_csv(base+'.sorted.txt', sep='\\t', index_col=0)\n",
    "\n",
    "#now we want to drop last column\n",
    "header = ['m6-301-1', 'm6-301-2', 'm6-500-1', 'm6-500-2', 'm6-712-1', \n",
    "           'm6-712-2', 'm6-736-1', 'm6-736-2', 'm6-744-1', 'm6-744-2', \n",
    "           'm6-755-1', 'm6-755-2', 'm6-757-1', 'm6-757-2', 'm6-766-1', 'm6-766-2']\n",
    "cleaned = sort.reindex(columns=header, index=natsorted(sort.index))\n",
    "\n",
    "#now drop OTUs where they are all zero\n",
    "fs = cleaned.sum(axis=0)\n",
    "filtered = pd.DataFrame(cleaned, columns=fs.index)\n",
    "filt2 = filtered.loc[(filtered != 0).any(1)]\n",
    "print len(sort), len(sort.columns.values), len(filt2), len(filt2.columns.values)\n",
    "filt2.to_csv(base+'.sorted.nomock.txt', sep='\\t')\n",
    "\n",
    "#now convert to biom\n",
    "cmd = ['amptk', 'taxonomy', '-f', platform+'.cluster.otus.fa', '-o', base+'_raw', '-i', base+'.sorted.nomock.txt', '-d', 'ITS2','-m','mapping/'+base+'.mapping_file.txt', '-t', platform+'.taxonomy.txt']\n",
    "subprocess.call(cmd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------\n",
      "\u001b[92m[02:35:07 PM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[02:35:07 PM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[02:35:07 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[02:35:07 PM]\u001b[0m: 1,071 OTUs\n",
      "\u001b[92m[02:35:07 PM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[02:35:08 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[02:35:11 PM]\u001b[0m: Taxonomy finished: ion.taxonomy.txt\n",
      "\u001b[92m[02:35:11 PM]\u001b[0m: Classic OTU table with taxonomy: ion3.otu_table.taxonomy.txt\n",
      "\u001b[92m[02:35:12 PM]\u001b[0m: BIOM OTU table created: ion3.biom\n",
      "\u001b[92m[02:35:12 PM]\u001b[0m: OTUs with taxonomy: ion3.otus.taxonomy.fa\n",
      "\u001b[92m[02:35:12 PM]\u001b[0m: OTU phylogeny: ion3.tree.phy\n",
      "-------------------------------------------------------\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[02:35:13 PM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[02:35:13 PM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[02:35:13 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[02:35:13 PM]\u001b[0m: 1,115 OTUs\n",
      "\u001b[92m[02:35:13 PM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[02:35:13 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[02:35:17 PM]\u001b[0m: Taxonomy finished: ion.taxonomy.txt\n",
      "\u001b[92m[02:35:17 PM]\u001b[0m: Classic OTU table with taxonomy: ion5.otu_table.taxonomy.txt\n",
      "\u001b[92m[02:35:18 PM]\u001b[0m: BIOM OTU table created: ion5.biom\n",
      "\u001b[92m[02:35:18 PM]\u001b[0m: OTUs with taxonomy: ion5.otus.taxonomy.fa\n",
      "\u001b[92m[02:35:18 PM]\u001b[0m: OTU phylogeny: ion5.tree.phy\n",
      "-------------------------------------------------------\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[02:35:19 PM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[02:35:19 PM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[02:35:19 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[02:35:19 PM]\u001b[0m: 1,937 OTUs\n",
      "\u001b[92m[02:35:19 PM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[02:35:19 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[02:35:30 PM]\u001b[0m: Taxonomy finished: miseq.taxonomy.txt\n",
      "\u001b[92m[02:35:30 PM]\u001b[0m: Classic OTU table with taxonomy: miseq4a.otu_table.taxonomy.txt\n",
      "\u001b[92m[02:35:31 PM]\u001b[0m: BIOM OTU table created: miseq4a.biom\n",
      "\u001b[92m[02:35:31 PM]\u001b[0m: OTUs with taxonomy: miseq4a.otus.taxonomy.fa\n",
      "\u001b[92m[02:35:31 PM]\u001b[0m: OTU phylogeny: miseq4a.tree.phy\n",
      "-------------------------------------------------------\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[02:35:32 PM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[02:35:32 PM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[02:35:32 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[02:35:32 PM]\u001b[0m: 1,860 OTUs\n",
      "\u001b[92m[02:35:32 PM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[02:35:32 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[02:35:42 PM]\u001b[0m: Taxonomy finished: miseq.taxonomy.txt\n",
      "\u001b[92m[02:35:42 PM]\u001b[0m: Classic OTU table with taxonomy: miseq4c.otu_table.taxonomy.txt\n",
      "\u001b[92m[02:35:43 PM]\u001b[0m: BIOM OTU table created: miseq4c.biom\n",
      "\u001b[92m[02:35:43 PM]\u001b[0m: OTUs with taxonomy: miseq4c.otus.taxonomy.fa\n",
      "\u001b[92m[02:35:43 PM]\u001b[0m: OTU phylogeny: miseq4c.tree.phy\n",
      "-------------------------------------------------------\n",
      "-------------------------------------------------------\n",
      "\u001b[92m[02:35:44 PM]\u001b[0m: OS: MacOSX 10.12.5, 8 cores, ~ 16 GB RAM. Python: 2.7.12\n",
      "\u001b[92m[02:35:44 PM]\u001b[0m: AMPtk v.0.10.0, USEARCH v9.2.64, VSEARCH v2.4.3\n",
      "\u001b[92m[02:35:44 PM]\u001b[0m: Loading FASTA Records\n",
      "\u001b[92m[02:35:44 PM]\u001b[0m: 2,018 OTUs\n",
      "\u001b[92m[02:35:44 PM]\u001b[0m: Appending taxonomy to OTU table and OTUs\n",
      "\u001b[92m[02:35:44 PM]\u001b[0m: Generating phylogenetic tree\n",
      "\u001b[92m[02:35:56 PM]\u001b[0m: Taxonomy finished: miseq.taxonomy.txt\n",
      "\u001b[92m[02:35:56 PM]\u001b[0m: Classic OTU table with taxonomy: miseq6.otu_table.taxonomy.txt\n",
      "\u001b[92m[02:35:58 PM]\u001b[0m: BIOM OTU table created: miseq6.biom\n",
      "\u001b[92m[02:35:58 PM]\u001b[0m: OTUs with taxonomy: miseq6.otus.taxonomy.fa\n",
      "\u001b[92m[02:35:58 PM]\u001b[0m: OTU phylogeny: miseq6.tree.phy\n",
      "-------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "#run taxonomy command on all output\n",
    "!amptk taxonomy -f ion3.filtered.otus.fa -i ion3.final.txt -o ion3 -d ITS2 -t ion.taxonomy.txt -m mapping/ion3.mapping_file.txt\n",
    "!amptk taxonomy -f ion5.filtered.otus.fa -i ion5.final.txt -o ion5 -d ITS2 -t ion.taxonomy.txt -m mapping/ion5.mapping_file.txt\n",
    "!amptk taxonomy -f miseq4a.filtered.otus.fa -i miseq4a.final.txt -o miseq4a -d ITS2 -t miseq.taxonomy.txt -m mapping/miseq4a.mapping_file.txt\n",
    "!amptk taxonomy -f miseq4c.filtered.otus.fa -i miseq4c.final.txt -o miseq4c -d ITS2 -t miseq.taxonomy.txt -m mapping/miseq4c.mapping_file.txt\n",
    "!amptk taxonomy -f miseq6.filtered.otus.fa -i miseq6.final.txt -o miseq6 -d ITS2 -t miseq.taxonomy.txt -m mapping/miseq6.mapping_file.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Working on complete OTU table: found 1158 OTUs\n",
      "ERROR m4C-301-1, m4C-301-2, m4C-500-1, m4C-500-2, m4C-712-1, m4C-712-2, m4C-736-1, m4C-736-2, m4C-744-1, m4C-744-2, m4C-755-1, m4C-755-2, m4C-757-1, m4C-757-2, m4C-766-1, m4C-766-2, m4C-mock3-16000, m4C-mock3-32000a, m4C-mock3-32000b, m4C-stds, m4A-301-1, m4A-301-2, m4A-500-1, m4A-500-2, m4A-712-1, m4A-712-2, m4A-736-1, m4A-736-2, m4A-744-1, m4A-744-2, m4A-755-1, m4A-755-2, m4A-757-1, m4A-757-2, m4A-766-1, m4A-766-2, m4A-mock3-16000, m4A-mock3-32000a, m4A-mock3-32000b, m4A-stds, m3-755-2, m3-mock3-3200a, m3-mock3-16000, m3-mock3-32000b, m3-stds, m5-301-1, m5-301-2, m5-500-1, m5-500-2, m5-712-1, m5-712-2, m5-736-1, m5-736-2, m5-744-1, m5-744-2, m5-755-1, m5-755-2, m5-757-1, m5-757-2, m5-766-1, m5-766-2, m5-SynMock1-8000, m5-SynMock2-8000, m5-mock3.0-16000, m5-mock3.5-16000, m6-301-1, m6-301-2, m6-500-1, m6-500-2, m6-712-1, m6-712-2, m6-736-1, m6-736-2, m6-744-1, m6-744-2, m6-755-1, m6-755-2, m6-757-1, m6-757-2, m6-766-1, m6-766-2, m6-mock3-16000, m6-mock3-32000a, m6-mock3-32000b, m6-stds, m6-SynMock: not found in OTU table. Names must max exactly.\n",
      "Working on complete OTU table: found 1071 OTUs\n",
      "ERROR m4C-301-1, m4C-301-2, m4C-500-1, m4C-500-2, m4C-712-1, m4C-712-2, m4C-736-1, m4C-736-2, m4C-744-1, m4C-744-2, m4C-755-1, m4C-755-2, m4C-757-1, m4C-757-2, m4C-766-1, m4C-766-2, m4C-mock3-16000, m4C-mock3-32000a, m4C-mock3-32000b, m4C-stds, m4A-301-1, m4A-301-2, m4A-500-1, m4A-500-2, m4A-712-1, m4A-712-2, m4A-736-1, m4A-736-2, m4A-744-1, m4A-744-2, m4A-755-1, m4A-755-2, m4A-757-1, m4A-757-2, m4A-766-1, m4A-766-2, m4A-mock3-16000, m4A-mock3-32000a, m4A-mock3-32000b, m4A-stds, m3-755-2, m3-mock3-3200a, m3-mock3-16000, m3-mock3-32000b, m3-stds, m5-301-1, m5-301-2, m5-500-1, m5-500-2, m5-712-1, m5-712-2, m5-736-1, m5-736-2, m5-744-1, m5-744-2, m5-755-1, m5-755-2, m5-757-1, m5-757-2, m5-766-1, m5-766-2, m5-SynMock1-8000, m5-SynMock2-8000, m5-mock3.0-16000, m5-mock3.5-16000, m6-301-1, m6-301-2, m6-500-1, m6-500-2, m6-712-1, m6-712-2, m6-736-1, m6-736-2, m6-744-1, m6-744-2, m6-755-1, m6-755-2, m6-757-1, m6-757-2, m6-766-1, m6-766-2, m6-mock3-16000, m6-mock3-32000a, m6-mock3-32000b, m6-stds, m6-SynMock: not found in OTU table. Names must max exactly.\n",
      "Working on complete OTU table: found 1179 OTUs\n",
      "ERROR m4C-301-1, m4C-301-2, m4C-500-1, m4C-500-2, m4C-712-1, m4C-712-2, m4C-736-1, m4C-736-2, m4C-744-1, m4C-744-2, m4C-755-1, m4C-755-2, m4C-757-1, m4C-757-2, m4C-766-1, m4C-766-2, m4C-mock3-16000, m4C-mock3-32000a, m4C-mock3-32000b, m4C-stds, m4A-301-1, m4A-301-2, m4A-500-1, m4A-500-2, m4A-712-1, m4A-712-2, m4A-736-1, m4A-736-2, m4A-744-1, m4A-744-2, m4A-755-1, m4A-755-2, m4A-757-1, m4A-757-2, m4A-766-1, m4A-766-2, m4A-mock3-16000, m4A-mock3-32000a, m4A-mock3-32000b, m4A-stds, m3-301-1, m3-301-2, m3-500-1, m3-500-2, m3-712-1, m3-712-2, m3-736-1, m3-736-2, m3-744-1, m3-744-2, m3-755-1, m3-755-2, m3-757-2, m3-766-1, m3-766-2, m3-mock3-3200a, m3-mock3-16000, m3-mock3-32000b, m3-stds, m5-SynMock1-8000, m5-SynMock2-8000, m5-mock3.0-16000, m5-mock3.5-16000, m6-301-1, m6-301-2, m6-500-1, m6-500-2, m6-712-1, m6-712-2, m6-736-1, m6-736-2, m6-744-1, m6-744-2, m6-755-1, m6-755-2, m6-757-1, m6-757-2, m6-766-1, m6-766-2, m6-mock3-16000, m6-mock3-32000a, m6-mock3-32000b, m6-stds, m6-SynMock: not found in OTU table. Names must max exactly.\n",
      "Working on complete OTU table: found 1115 OTUs\n",
      "ERROR m4C-301-1, m4C-301-2, m4C-500-1, m4C-500-2, m4C-712-1, m4C-712-2, m4C-736-1, m4C-736-2, m4C-744-1, m4C-744-2, m4C-755-1, m4C-755-2, m4C-757-1, m4C-757-2, m4C-766-1, m4C-766-2, m4C-mock3-16000, m4C-mock3-32000a, m4C-mock3-32000b, m4C-stds, m4A-301-1, m4A-301-2, m4A-500-1, m4A-500-2, m4A-712-1, m4A-712-2, m4A-736-1, m4A-736-2, m4A-744-1, m4A-744-2, m4A-755-1, m4A-755-2, m4A-757-1, m4A-757-2, m4A-766-1, m4A-766-2, m4A-mock3-16000, m4A-mock3-32000a, m4A-mock3-32000b, m4A-stds, m3-301-1, m3-301-2, m3-500-1, m3-500-2, m3-712-1, m3-712-2, m3-736-1, m3-736-2, m3-744-1, m3-744-2, m3-755-1, m3-755-2, m3-757-2, m3-766-1, m3-766-2, m3-mock3-3200a, m3-mock3-16000, m3-mock3-32000b, m3-stds, m5-SynMock1-8000, m5-SynMock2-8000, m5-mock3.0-16000, m5-mock3.5-16000, m6-301-1, m6-301-2, m6-500-1, m6-500-2, m6-712-1, m6-712-2, m6-736-1, m6-736-2, m6-744-1, m6-744-2, m6-755-1, m6-755-2, m6-757-1, m6-757-2, m6-766-1, m6-766-2, m6-mock3-16000, m6-mock3-32000a, m6-mock3-32000b, m6-stds, m6-SynMock: not found in OTU table. Names must max exactly.\n",
      "Working on complete OTU table: found 2368 OTUs\n",
      "ERROR m4C-301-1, m4C-301-2, m4C-500-1, m4C-500-2, m4C-712-1, m4C-712-2, m4C-736-1, m4C-736-2, m4C-744-1, m4C-744-2, m4C-755-1, m4C-755-2, m4C-757-1, m4C-757-2, m4C-766-1, m4C-766-2, m4C-mock3-16000, m4C-mock3-32000a, m4C-mock3-32000b, m4C-stds, m4A-mock3-16000, m4A-mock3-32000a, m4A-mock3-32000b, m4A-stds, m3-301-1, m3-301-2, m3-500-1, m3-500-2, m3-712-1, m3-712-2, m3-736-1, m3-736-2, m3-744-1, m3-744-2, m3-755-1, m3-755-2, m3-757-2, m3-766-1, m3-766-2, m3-mock3-3200a, m3-mock3-16000, m3-mock3-32000b, m3-stds, m5-301-1, m5-301-2, m5-500-1, m5-500-2, m5-712-1, m5-712-2, m5-736-1, m5-736-2, m5-744-1, m5-744-2, m5-755-1, m5-755-2, m5-757-1, m5-757-2, m5-766-1, m5-766-2, m5-SynMock1-8000, m5-SynMock2-8000, m5-mock3.0-16000, m5-mock3.5-16000, m6-301-1, m6-301-2, m6-500-1, m6-500-2, m6-712-1, m6-712-2, m6-736-1, m6-736-2, m6-744-1, m6-744-2, m6-755-1, m6-755-2, m6-757-1, m6-757-2, m6-766-1, m6-766-2, m6-mock3-16000, m6-mock3-32000a, m6-mock3-32000b, m6-stds, m6-SynMock: not found in OTU table. Names must max exactly.\n",
      "Working on complete OTU table: found 1937 OTUs\n",
      "ERROR m4C-301-1, m4C-301-2, m4C-500-1, m4C-500-2, m4C-712-1, m4C-712-2, m4C-736-1, m4C-736-2, m4C-744-1, m4C-744-2, m4C-755-1, m4C-755-2, m4C-757-1, m4C-757-2, m4C-766-1, m4C-766-2, m4C-mock3-16000, m4C-mock3-32000a, m4C-mock3-32000b, m4C-stds, m4A-mock3-16000, m4A-mock3-32000a, m4A-mock3-32000b, m4A-stds, m3-301-1, m3-301-2, m3-500-1, m3-500-2, m3-712-1, m3-712-2, m3-736-1, m3-736-2, m3-744-1, m3-744-2, m3-755-1, m3-755-2, m3-757-2, m3-766-1, m3-766-2, m3-mock3-3200a, m3-mock3-16000, m3-mock3-32000b, m3-stds, m5-301-1, m5-301-2, m5-500-1, m5-500-2, m5-712-1, m5-712-2, m5-736-1, m5-736-2, m5-744-1, m5-744-2, m5-755-1, m5-755-2, m5-757-1, m5-757-2, m5-766-1, m5-766-2, m5-SynMock1-8000, m5-SynMock2-8000, m5-mock3.0-16000, m5-mock3.5-16000, m6-301-1, m6-301-2, m6-500-1, m6-500-2, m6-712-1, m6-712-2, m6-736-1, m6-736-2, m6-744-1, m6-744-2, m6-755-1, m6-755-2, m6-757-1, m6-757-2, m6-766-1, m6-766-2, m6-mock3-16000, m6-mock3-32000a, m6-mock3-32000b, m6-stds, m6-SynMock: not found in OTU table. Names must max exactly.\n",
      "Working on complete OTU table: found 2271 OTUs\n",
      "ERROR m4C-mock3-16000, m4C-mock3-32000a, m4C-mock3-32000b, m4C-stds, m4A-301-1, m4A-301-2, m4A-500-1, m4A-500-2, m4A-712-1, m4A-712-2, m4A-736-1, m4A-736-2, m4A-744-1, m4A-744-2, m4A-755-1, m4A-755-2, m4A-757-1, m4A-757-2, m4A-766-1, m4A-766-2, m4A-mock3-16000, m4A-mock3-32000a, m4A-mock3-32000b, m4A-stds, m3-301-1, m3-301-2, m3-500-1, m3-500-2, m3-712-1, m3-712-2, m3-736-1, m3-736-2, m3-744-1, m3-744-2, m3-755-1, m3-755-2, m3-757-2, m3-766-1, m3-766-2, m3-mock3-3200a, m3-mock3-16000, m3-mock3-32000b, m3-stds, m5-301-1, m5-301-2, m5-500-1, m5-500-2, m5-712-1, m5-712-2, m5-736-1, m5-736-2, m5-744-1, m5-744-2, m5-755-1, m5-755-2, m5-757-1, m5-757-2, m5-766-1, m5-766-2, m5-SynMock1-8000, m5-SynMock2-8000, m5-mock3.0-16000, m5-mock3.5-16000, m6-301-1, m6-301-2, m6-500-1, m6-500-2, m6-712-1, m6-712-2, m6-736-1, m6-736-2, m6-744-1, m6-744-2, m6-755-1, m6-755-2, m6-757-1, m6-757-2, m6-766-1, m6-766-2, m6-mock3-16000, m6-mock3-32000a, m6-mock3-32000b, m6-stds, m6-SynMock: not found in OTU table. Names must max exactly.\n",
      "Working on complete OTU table: found 1860 OTUs\n",
      "ERROR m4C-mock3-16000, m4C-mock3-32000a, m4C-stds, m4A-301-1, m4A-301-2, m4A-500-1, m4A-500-2, m4A-712-1, m4A-712-2, m4A-736-1, m4A-736-2, m4A-744-1, m4A-744-2, m4A-755-1, m4A-755-2, m4A-757-1, m4A-757-2, m4A-766-1, m4A-766-2, m4A-mock3-16000, m4A-mock3-32000a, m4A-mock3-32000b, m4A-stds, m3-301-1, m3-301-2, m3-500-1, m3-500-2, m3-712-1, m3-712-2, m3-736-1, m3-736-2, m3-744-1, m3-744-2, m3-755-1, m3-755-2, m3-757-2, m3-766-1, m3-766-2, m3-mock3-3200a, m3-mock3-16000, m3-mock3-32000b, m3-stds, m5-301-1, m5-301-2, m5-500-1, m5-500-2, m5-712-1, m5-712-2, m5-736-1, m5-736-2, m5-744-1, m5-744-2, m5-755-1, m5-755-2, m5-757-1, m5-757-2, m5-766-1, m5-766-2, m5-SynMock1-8000, m5-SynMock2-8000, m5-mock3.0-16000, m5-mock3.5-16000, m6-301-1, m6-301-2, m6-500-1, m6-500-2, m6-712-1, m6-712-2, m6-736-1, m6-736-2, m6-744-1, m6-744-2, m6-755-1, m6-755-2, m6-757-1, m6-757-2, m6-766-1, m6-766-2, m6-mock3-16000, m6-mock3-32000a, m6-mock3-32000b, m6-stds, m6-SynMock: not found in OTU table. Names must max exactly.\n",
      "Working on complete OTU table: found 2699 OTUs\n",
      "ERROR m4C-301-1, m4C-301-2, m4C-500-1, m4C-500-2, m4C-712-1, m4C-712-2, m4C-736-1, m4C-736-2, m4C-744-1, m4C-744-2, m4C-755-1, m4C-755-2, m4C-757-1, m4C-757-2, m4C-766-1, m4C-766-2, m4C-mock3-16000, m4C-mock3-32000a, m4C-mock3-32000b, m4C-stds, m4A-301-1, m4A-301-2, m4A-500-1, m4A-500-2, m4A-712-1, m4A-712-2, m4A-736-1, m4A-736-2, m4A-744-1, m4A-744-2, m4A-755-1, m4A-755-2, m4A-757-1, m4A-757-2, m4A-766-1, m4A-766-2, m4A-mock3-16000, m4A-mock3-32000a, m4A-mock3-32000b, m4A-stds, m3-301-1, m3-301-2, m3-500-1, m3-500-2, m3-712-1, m3-712-2, m3-736-1, m3-736-2, m3-744-1, m3-744-2, m3-755-1, m3-755-2, m3-757-2, m3-766-1, m3-766-2, m3-mock3-3200a, m3-mock3-16000, m3-mock3-32000b, m3-stds, m5-301-1, m5-301-2, m5-500-1, m5-500-2, m5-712-1, m5-712-2, m5-736-1, m5-736-2, m5-744-1, m5-744-2, m5-755-1, m5-755-2, m5-757-1, m5-757-2, m5-766-1, m5-766-2, m5-SynMock1-8000, m5-SynMock2-8000, m5-mock3.0-16000, m5-mock3.5-16000, m6-mock3-16000, m6-mock3-32000a, m6-mock3-32000b, m6-stds, m6-SynMock: not found in OTU table. Names must max exactly.\n",
      "Working on complete OTU table: found 2018 OTUs\n",
      "ERROR m4C-301-1, m4C-301-2, m4C-500-1, m4C-500-2, m4C-712-1, m4C-712-2, m4C-736-1, m4C-736-2, m4C-744-1, m4C-744-2, m4C-755-1, m4C-755-2, m4C-757-1, m4C-757-2, m4C-766-1, m4C-766-2, m4C-mock3-16000, m4C-mock3-32000a, m4C-mock3-32000b, m4C-stds, m4A-301-1, m4A-301-2, m4A-500-1, m4A-500-2, m4A-712-1, m4A-712-2, m4A-736-1, m4A-736-2, m4A-744-1, m4A-744-2, m4A-755-1, m4A-755-2, m4A-757-1, m4A-757-2, m4A-766-1, m4A-766-2, m4A-mock3-16000, m4A-mock3-32000a, m4A-mock3-32000b, m4A-stds, m3-301-1, m3-301-2, m3-500-1, m3-500-2, m3-712-1, m3-712-2, m3-736-1, m3-736-2, m3-744-1, m3-744-2, m3-755-1, m3-755-2, m3-757-2, m3-766-1, m3-766-2, m3-mock3-3200a, m3-mock3-16000, m3-mock3-32000b, m3-stds, m5-301-1, m5-301-2, m5-500-1, m5-500-2, m5-712-1, m5-712-2, m5-736-1, m5-736-2, m5-744-1, m5-744-2, m5-755-1, m5-755-2, m5-757-1, m5-757-2, m5-766-1, m5-766-2, m5-SynMock1-8000, m5-SynMock2-8000, m5-mock3.0-16000, m5-mock3.5-16000, m6-mock3-16000, m6-mock3-32000a, m6-mock3-32000b, m6-stds, m6-SynMock: not found in OTU table. Names must max exactly.\n"
     ]
    }
   ],
   "source": [
    "#now convert all to meta transposition for R (vegan) scripts\n",
    "!amptk meta -i ion3_raw.otu_table.taxonomy.txt -m all_data.meta.csv -o ion3_raw.meta.csv\n",
    "!amptk meta -i ion3.otu_table.taxonomy.txt -m all_data.meta.csv -o ion3_filtered.meta.csv\n",
    "!amptk meta -i ion5_raw.otu_table.taxonomy.txt -m all_data.meta.csv -o ion5_raw.meta.csv\n",
    "!amptk meta -i ion5.otu_table.taxonomy.txt -m all_data.meta.csv -o ion5_filtered.meta.csv\n",
    "!amptk meta -i miseq4a_raw.otu_table.taxonomy.txt -m all_data.meta.csv -o miseq4a_raw.meta.csv\n",
    "!amptk meta -i miseq4a.otu_table.taxonomy.txt -m all_data.meta.csv -o miseq4a_filtered.meta.csv\n",
    "!amptk meta -i miseq4c_raw.otu_table.taxonomy.txt -m all_data.meta.csv -o miseq4c_raw.meta.csv\n",
    "!amptk meta -i miseq4c.otu_table.taxonomy.txt -m all_data.meta.csv -o miseq4c_filtered.meta.csv\n",
    "!amptk meta -i miseq6_raw.otu_table.taxonomy.txt -m all_data.meta.csv -o miseq6_raw.meta.csv\n",
    "!amptk meta -i miseq6.otu_table.taxonomy.txt -m all_data.meta.csv -o miseq6_filtered.meta.csv\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 196,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "vsearch v2.4.2_osx_x86_64, 16.0GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading input file 100%                                                                                                                                                                                                                                                                                                                                                                                                              \n",
      "18273964 sequences kept (of which 0 truncated), 0 sequences discarded.\n",
      "vsearch v2.4.2_osx_x86_64, 16.0GB RAM, 8 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file /Users/jon/amptk/DB/amptk_mock2.fa 100%  \n",
      "14847 nt in 26 seqs, min 335, max 680, avg 571\n",
      "Masking 100% \n",
      "Counting unique k-mers 100% \n",
      "Creating index of unique k-mers 100% \n",
      "Searching 100%                                                                                                                                                                                                                                                                                                                                                                                                                \n",
      "Matching query sequences: 3182927 of 18273964 (17.42%)\n",
      "Writing OTU table (classic) 100%  \n"
     ]
    }
   ],
   "source": [
    "#map the miseq reads to the mock community fasta file\n",
    "!vsearch --fastq_filter miseq.fq --fastaout miseq.fa\n",
    "!vsearch --usearch_global miseq.fa --id 0.97 --db /Users/jon/amptk/DB/amptk_mock2.fa --otutabout miseq.mock.otu_table.txt\n",
    "!rm miseq.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#now split OTU table into the different parts/segments needed for different figures\n",
    "#lets do this in python/pandas\n",
    "import pandas as pd\n",
    "from natsort import natsorted\n",
    "\n",
    "#now load OTU table into dataframe\n",
    "df = pd.read_csv('miseq.mock.otu_table.txt', sep='\\t', index_col=0)\n",
    "#print df.columns.values\n",
    "#headers for \n",
    "figureOne = ['m3-stds','m3-mock3-16000','m3-mock3-3200a','m3-mock3-32000b',\n",
    "             'm4A-stds','m4A-mock3-16000','m4A-mock3-32000a','m4A-mock3-32000b',\n",
    "             'm4C-stds','m4C-mock3-16000','m4C-mock3-32000a','m4C-mock3-32000b',\n",
    "            'm6-stds', 'm6-mock3-16000', 'm6-mock3-32000a', 'm6-mock3-32000b']\n",
    "\n",
    "mocks = []\n",
    "for x in df.index:\n",
    "    if x.startswith('mock'):\n",
    "        mocks.append(x)\n",
    "print mocks\n",
    "fig1 = df.reindex(columns=figureOne, index=natsorted(mocks))\n",
    "fig1 = fig1.fillna(value=0).astype(int)\n",
    "#print fig1\n",
    "#write to file\n",
    "fig1.to_csv('synmock.figure1.abundances.txt', sep='\\t')\n",
    "#then generate heatmap\n",
    "!amptk heatmap -i synmock.figure1.abundances.txt -d tsv -o figure1.pdf -m heatmap --annotate --color YlOrRd --vmax 50000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "m4A-stds             31\n",
      "m4A-mock3-16000      18\n",
      "m4A-mock3-32000a     25\n",
      "m4A-mock3-32000b     33\n",
      "m4A-755-1           375\n",
      "m4A-755-2           353\n",
      "m4A-744-1           281\n",
      "m4A-744-2           361\n",
      "m4A-766-1           445\n",
      "m4A-766-2           276\n",
      "m4b-744-1           239\n",
      "m4b-744-2           315\n",
      "m4b-766-1           349\n",
      "m4b-766-2           235\n",
      "Name: Number of OTUs, dtype: int64\n",
      "                                                    m4A-stds  m4A-mock3-16000  \\\n",
      "#OTU ID                                                                         \n",
      "mock1 |FP170182|Phialocephala fusca_pident=99.6...      8655              727   \n",
      "mock2 |RF10JR|unknown Ascomycete_pident=100.0_O...      9176            20769   \n",
      "mock3 |FP170134|Phialocephala lagerbergii_piden...      8652            28522   \n",
      "mock4 |FP170235|unknown Ascomycete_pident=100.0...      9051            27727   \n",
      "mock5 |RF1JR|Aspergillus candidus_pident=100.0_...      8989              147   \n",
      "mock6 |RF3JR|Bjerkandara adusta_pident=100.0_OTU20     13565            13080   \n",
      "mock7 |GDL-1|Laetiporus caribensis_pident=100.0...      9391            15008   \n",
      "mock8 |RF5JR|Trametes gibbosa_pident=100.0_OTU26        8125            13797   \n",
      "mock9 |OR2|Laetiporus gilbertsonii_pident=100.0...      8734            22808   \n",
      "mock10 |MR5-1|Gloeoprus pannocinctus_pident=100...      9696            11119   \n",
      "mock11 |FP72612|Wolfiporia dilatohypha_pident=1...      8760              243   \n",
      "mock12 |FP170198|Schizopora sp_pident=100.0_OTU27       8659            12827   \n",
      "mock13 |FP170231|Fomitopsis ochracea_pident=100...      9444             5515   \n",
      "mock14 |L34-2|Laetiporus cremeiporus_pident=100...      8246            16414   \n",
      "mock15 |RF9JR|Phanerochaete laevis_pident=100.0...      9221            15644   \n",
      "mock16 |DA37|Laetiporus cincinnatus_pident=100....      7662            16785   \n",
      "mock17 |RF7JR|Puntularia strigooszonata_pident=...      7643            10673   \n",
      "mock18 |IN4-1|Phellinus cinereus_pident=100.0_O...      9793             6308   \n",
      "mock19 |MR5-3|Antrodiellla semisupina_pident=10...      9286             9335   \n",
      "mock20 |TJV93-174|Leptoporus mollis_pident=100....      9033             8834   \n",
      "mock21 |RLG7163|Leptoporus mollis_pident=99.3_O...      9293              404   \n",
      "mock22 |FP170186|Mortierelllales sp_pident=100....      9135            10845   \n",
      "mock23 |HHB9564|Laetiporus percisinus_pident=10...      6477              487   \n",
      "mock24 |FP170215|Penicillium nothofagi_pident=1...      8260                0   \n",
      "mock25 |FP170177|Pochonia suchlasporia_pident=1...      2043                0   \n",
      "mock26 |MD275|Wolfiporia cocos_pident=100.0_OTU179      5984                1   \n",
      "Other OTUs                                               815              165   \n",
      "\n",
      "                                                    m4A-mock3-32000a  \\\n",
      "#OTU ID                                                                \n",
      "mock1 |FP170182|Phialocephala fusca_pident=99.6...               331   \n",
      "mock2 |RF10JR|unknown Ascomycete_pident=100.0_O...             26139   \n",
      "mock3 |FP170134|Phialocephala lagerbergii_piden...             29490   \n",
      "mock4 |FP170235|unknown Ascomycete_pident=100.0...             32578   \n",
      "mock5 |RF1JR|Aspergillus candidus_pident=100.0_...               122   \n",
      "mock6 |RF3JR|Bjerkandara adusta_pident=100.0_OTU20             13836   \n",
      "mock7 |GDL-1|Laetiporus caribensis_pident=100.0...             16588   \n",
      "mock8 |RF5JR|Trametes gibbosa_pident=100.0_OTU26               14546   \n",
      "mock9 |OR2|Laetiporus gilbertsonii_pident=100.0...             21612   \n",
      "mock10 |MR5-1|Gloeoprus pannocinctus_pident=100...             11769   \n",
      "mock11 |FP72612|Wolfiporia dilatohypha_pident=1...               224   \n",
      "mock12 |FP170198|Schizopora sp_pident=100.0_OTU27              13921   \n",
      "mock13 |FP170231|Fomitopsis ochracea_pident=100...              5422   \n",
      "mock14 |L34-2|Laetiporus cremeiporus_pident=100...             16362   \n",
      "mock15 |RF9JR|Phanerochaete laevis_pident=100.0...             15512   \n",
      "mock16 |DA37|Laetiporus cincinnatus_pident=100....             16124   \n",
      "mock17 |RF7JR|Puntularia strigooszonata_pident=...             11558   \n",
      "mock18 |IN4-1|Phellinus cinereus_pident=100.0_O...              5936   \n",
      "mock19 |MR5-3|Antrodiellla semisupina_pident=10...              8974   \n",
      "mock20 |TJV93-174|Leptoporus mollis_pident=100....              8729   \n",
      "mock21 |RLG7163|Leptoporus mollis_pident=99.3_O...               308   \n",
      "mock22 |FP170186|Mortierelllales sp_pident=100....              9709   \n",
      "mock23 |HHB9564|Laetiporus percisinus_pident=10...               420   \n",
      "mock24 |FP170215|Penicillium nothofagi_pident=1...                 0   \n",
      "mock25 |FP170177|Pochonia suchlasporia_pident=1...                 0   \n",
      "mock26 |MD275|Wolfiporia cocos_pident=100.0_OTU179                 1   \n",
      "Other OTUs                                                       116   \n",
      "\n",
      "                                                    m4A-mock3-32000b  \\\n",
      "#OTU ID                                                                \n",
      "mock1 |FP170182|Phialocephala fusca_pident=99.6...              3352   \n",
      "mock2 |RF10JR|unknown Ascomycete_pident=100.0_O...             18343   \n",
      "mock3 |FP170134|Phialocephala lagerbergii_piden...             21271   \n",
      "mock4 |FP170235|unknown Ascomycete_pident=100.0...             24280   \n",
      "mock5 |RF1JR|Aspergillus candidus_pident=100.0_...               269   \n",
      "mock6 |RF3JR|Bjerkandara adusta_pident=100.0_OTU20             15038   \n",
      "mock7 |GDL-1|Laetiporus caribensis_pident=100.0...             16354   \n",
      "mock8 |RF5JR|Trametes gibbosa_pident=100.0_OTU26               14764   \n",
      "mock9 |OR2|Laetiporus gilbertsonii_pident=100.0...             20628   \n",
      "mock10 |MR5-1|Gloeoprus pannocinctus_pident=100...             14233   \n",
      "mock11 |FP72612|Wolfiporia dilatohypha_pident=1...               194   \n",
      "mock12 |FP170198|Schizopora sp_pident=100.0_OTU27              14834   \n",
      "mock13 |FP170231|Fomitopsis ochracea_pident=100...              6874   \n",
      "mock14 |L34-2|Laetiporus cremeiporus_pident=100...             16758   \n",
      "mock15 |RF9JR|Phanerochaete laevis_pident=100.0...             18121   \n",
      "mock16 |DA37|Laetiporus cincinnatus_pident=100....             14582   \n",
      "mock17 |RF7JR|Puntularia strigooszonata_pident=...             11645   \n",
      "mock18 |IN4-1|Phellinus cinereus_pident=100.0_O...              7486   \n",
      "mock19 |MR5-3|Antrodiellla semisupina_pident=10...             11570   \n",
      "mock20 |TJV93-174|Leptoporus mollis_pident=100....              9965   \n",
      "mock21 |RLG7163|Leptoporus mollis_pident=99.3_O...               412   \n",
      "mock22 |FP170186|Mortierelllales sp_pident=100....             13338   \n",
      "mock23 |HHB9564|Laetiporus percisinus_pident=10...               517   \n",
      "mock24 |FP170215|Penicillium nothofagi_pident=1...                 8   \n",
      "mock25 |FP170177|Pochonia suchlasporia_pident=1...                 2   \n",
      "mock26 |MD275|Wolfiporia cocos_pident=100.0_OTU179                 2   \n",
      "Other OTUs                                                       154   \n",
      "\n",
      "                                                    m4A-755-1  m4A-755-2  \\\n",
      "#OTU ID                                                                    \n",
      "mock1 |FP170182|Phialocephala fusca_pident=99.6...          0          0   \n",
      "mock2 |RF10JR|unknown Ascomycete_pident=100.0_O...          0          0   \n",
      "mock3 |FP170134|Phialocephala lagerbergii_piden...          6          0   \n",
      "mock4 |FP170235|unknown Ascomycete_pident=100.0...          0          0   \n",
      "mock5 |RF1JR|Aspergillus candidus_pident=100.0_...          0          0   \n",
      "mock6 |RF3JR|Bjerkandara adusta_pident=100.0_OTU20          0          1   \n",
      "mock7 |GDL-1|Laetiporus caribensis_pident=100.0...          0          0   \n",
      "mock8 |RF5JR|Trametes gibbosa_pident=100.0_OTU26            0          0   \n",
      "mock9 |OR2|Laetiporus gilbertsonii_pident=100.0...          0          0   \n",
      "mock10 |MR5-1|Gloeoprus pannocinctus_pident=100...          0          0   \n",
      "mock11 |FP72612|Wolfiporia dilatohypha_pident=1...          0          0   \n",
      "mock12 |FP170198|Schizopora sp_pident=100.0_OTU27           0          0   \n",
      "mock13 |FP170231|Fomitopsis ochracea_pident=100...          0          0   \n",
      "mock14 |L34-2|Laetiporus cremeiporus_pident=100...          0          0   \n",
      "mock15 |RF9JR|Phanerochaete laevis_pident=100.0...          0          0   \n",
      "mock16 |DA37|Laetiporus cincinnatus_pident=100....          0          0   \n",
      "mock17 |RF7JR|Puntularia strigooszonata_pident=...          0          0   \n",
      "mock18 |IN4-1|Phellinus cinereus_pident=100.0_O...          0          2   \n",
      "mock19 |MR5-3|Antrodiellla semisupina_pident=10...          0          0   \n",
      "mock20 |TJV93-174|Leptoporus mollis_pident=100....          0          0   \n",
      "mock21 |RLG7163|Leptoporus mollis_pident=99.3_O...          0          0   \n",
      "mock22 |FP170186|Mortierelllales sp_pident=100....          0          0   \n",
      "mock23 |HHB9564|Laetiporus percisinus_pident=10...          0          0   \n",
      "mock24 |FP170215|Penicillium nothofagi_pident=1...          0          0   \n",
      "mock25 |FP170177|Pochonia suchlasporia_pident=1...          0          0   \n",
      "mock26 |MD275|Wolfiporia cocos_pident=100.0_OTU179          0          0   \n",
      "Other OTUs                                             292810     283423   \n",
      "\n",
      "                                                    m4A-744-1  m4A-744-2  \\\n",
      "#OTU ID                                                                    \n",
      "mock1 |FP170182|Phialocephala fusca_pident=99.6...          9         24   \n",
      "mock2 |RF10JR|unknown Ascomycete_pident=100.0_O...         42         25   \n",
      "mock3 |FP170134|Phialocephala lagerbergii_piden...         74         18   \n",
      "mock4 |FP170235|unknown Ascomycete_pident=100.0...         60         17   \n",
      "mock5 |RF1JR|Aspergillus candidus_pident=100.0_...          2         17   \n",
      "mock6 |RF3JR|Bjerkandara adusta_pident=100.0_OTU20         32         24   \n",
      "mock7 |GDL-1|Laetiporus caribensis_pident=100.0...         42         28   \n",
      "mock8 |RF5JR|Trametes gibbosa_pident=100.0_OTU26           31         13   \n",
      "mock9 |OR2|Laetiporus gilbertsonii_pident=100.0...         46         24   \n",
      "mock10 |MR5-1|Gloeoprus pannocinctus_pident=100...         33         18   \n",
      "mock11 |FP72612|Wolfiporia dilatohypha_pident=1...          0         11   \n",
      "mock12 |FP170198|Schizopora sp_pident=100.0_OTU27          40         16   \n",
      "mock13 |FP170231|Fomitopsis ochracea_pident=100...         16         18   \n",
      "mock14 |L34-2|Laetiporus cremeiporus_pident=100...         42         18   \n",
      "mock15 |RF9JR|Phanerochaete laevis_pident=100.0...         32         18   \n",
      "mock16 |DA37|Laetiporus cincinnatus_pident=100....         28         20   \n",
      "mock17 |RF7JR|Puntularia strigooszonata_pident=...         31         10   \n",
      "mock18 |IN4-1|Phellinus cinereus_pident=100.0_O...         28         11   \n",
      "mock19 |MR5-3|Antrodiellla semisupina_pident=10...         26         23   \n",
      "mock20 |TJV93-174|Leptoporus mollis_pident=100....         28         19   \n",
      "mock21 |RLG7163|Leptoporus mollis_pident=99.3_O...          1         18   \n",
      "mock22 |FP170186|Mortierelllales sp_pident=100....         24         23   \n",
      "mock23 |HHB9564|Laetiporus percisinus_pident=10...          0         19   \n",
      "mock24 |FP170215|Penicillium nothofagi_pident=1...          0         13   \n",
      "mock25 |FP170177|Pochonia suchlasporia_pident=1...          0          6   \n",
      "mock26 |MD275|Wolfiporia cocos_pident=100.0_OTU179          0         12   \n",
      "Other OTUs                                             267638     247235   \n",
      "\n",
      "                                                    m4A-766-1  m4A-766-2  \\\n",
      "#OTU ID                                                                    \n",
      "mock1 |FP170182|Phialocephala fusca_pident=99.6...          2          0   \n",
      "mock2 |RF10JR|unknown Ascomycete_pident=100.0_O...         57         55   \n",
      "mock3 |FP170134|Phialocephala lagerbergii_piden...         61        579   \n",
      "mock4 |FP170235|unknown Ascomycete_pident=100.0...         54         65   \n",
      "mock5 |RF1JR|Aspergillus candidus_pident=100.0_...          0          0   \n",
      "mock6 |RF3JR|Bjerkandara adusta_pident=100.0_OTU20         27         30   \n",
      "mock7 |GDL-1|Laetiporus caribensis_pident=100.0...         32         33   \n",
      "mock8 |RF5JR|Trametes gibbosa_pident=100.0_OTU26           26         30   \n",
      "mock9 |OR2|Laetiporus gilbertsonii_pident=100.0...         55         43   \n",
      "mock10 |MR5-1|Gloeoprus pannocinctus_pident=100...         22         20   \n",
      "mock11 |FP72612|Wolfiporia dilatohypha_pident=1...          1          0   \n",
      "mock12 |FP170198|Schizopora sp_pident=100.0_OTU27          23         25   \n",
      "mock13 |FP170231|Fomitopsis ochracea_pident=100...          7         11   \n",
      "mock14 |L34-2|Laetiporus cremeiporus_pident=100...         42         42   \n",
      "mock15 |RF9JR|Phanerochaete laevis_pident=100.0...         35         26   \n",
      "mock16 |DA37|Laetiporus cincinnatus_pident=100....         32         34   \n",
      "mock17 |RF7JR|Puntularia strigooszonata_pident=...         19         23   \n",
      "mock18 |IN4-1|Phellinus cinereus_pident=100.0_O...         16         15   \n",
      "mock19 |MR5-3|Antrodiellla semisupina_pident=10...         14         28   \n",
      "mock20 |TJV93-174|Leptoporus mollis_pident=100....         18         23   \n",
      "mock21 |RLG7163|Leptoporus mollis_pident=99.3_O...          1          1   \n",
      "mock22 |FP170186|Mortierelllales sp_pident=100....          9         30   \n",
      "mock23 |HHB9564|Laetiporus percisinus_pident=10...          0          0   \n",
      "mock24 |FP170215|Penicillium nothofagi_pident=1...          0          0   \n",
      "mock25 |FP170177|Pochonia suchlasporia_pident=1...          0          0   \n",
      "mock26 |MD275|Wolfiporia cocos_pident=100.0_OTU179          0          0   \n",
      "Other OTUs                                             292230     301915   \n",
      "\n",
      "                                                    m4b-744-1  m4b-744-2  \\\n",
      "#OTU ID                                                                    \n",
      "mock1 |FP170182|Phialocephala fusca_pident=99.6...          0          0   \n",
      "mock2 |RF10JR|unknown Ascomycete_pident=100.0_O...          0          0   \n",
      "mock3 |FP170134|Phialocephala lagerbergii_piden...         12          4   \n",
      "mock4 |FP170235|unknown Ascomycete_pident=100.0...          0          0   \n",
      "mock5 |RF1JR|Aspergillus candidus_pident=100.0_...          0          0   \n",
      "mock6 |RF3JR|Bjerkandara adusta_pident=100.0_OTU20          0          0   \n",
      "mock7 |GDL-1|Laetiporus caribensis_pident=100.0...          0          0   \n",
      "mock8 |RF5JR|Trametes gibbosa_pident=100.0_OTU26            0          0   \n",
      "mock9 |OR2|Laetiporus gilbertsonii_pident=100.0...          0          0   \n",
      "mock10 |MR5-1|Gloeoprus pannocinctus_pident=100...          0          0   \n",
      "mock11 |FP72612|Wolfiporia dilatohypha_pident=1...          0          0   \n",
      "mock12 |FP170198|Schizopora sp_pident=100.0_OTU27           0          0   \n",
      "mock13 |FP170231|Fomitopsis ochracea_pident=100...          0          0   \n",
      "mock14 |L34-2|Laetiporus cremeiporus_pident=100...          0          0   \n",
      "mock15 |RF9JR|Phanerochaete laevis_pident=100.0...          0          0   \n",
      "mock16 |DA37|Laetiporus cincinnatus_pident=100....          0          0   \n",
      "mock17 |RF7JR|Puntularia strigooszonata_pident=...          0          0   \n",
      "mock18 |IN4-1|Phellinus cinereus_pident=100.0_O...          0          0   \n",
      "mock19 |MR5-3|Antrodiellla semisupina_pident=10...          0          0   \n",
      "mock20 |TJV93-174|Leptoporus mollis_pident=100....          0          0   \n",
      "mock21 |RLG7163|Leptoporus mollis_pident=99.3_O...          0          0   \n",
      "mock22 |FP170186|Mortierelllales sp_pident=100....          0          0   \n",
      "mock23 |HHB9564|Laetiporus percisinus_pident=10...          0          0   \n",
      "mock24 |FP170215|Penicillium nothofagi_pident=1...          0          0   \n",
      "mock25 |FP170177|Pochonia suchlasporia_pident=1...          0          0   \n",
      "mock26 |MD275|Wolfiporia cocos_pident=100.0_OTU179          0          0   \n",
      "Other OTUs                                             182246     152655   \n",
      "\n",
      "                                                    m4b-766-1  m4b-766-2  \n",
      "#OTU ID                                                                   \n",
      "mock1 |FP170182|Phialocephala fusca_pident=99.6...          0          0  \n",
      "mock2 |RF10JR|unknown Ascomycete_pident=100.0_O...          0          0  \n",
      "mock3 |FP170134|Phialocephala lagerbergii_piden...          0        246  \n",
      "mock4 |FP170235|unknown Ascomycete_pident=100.0...          0          0  \n",
      "mock5 |RF1JR|Aspergillus candidus_pident=100.0_...          0          0  \n",
      "mock6 |RF3JR|Bjerkandara adusta_pident=100.0_OTU20          0          0  \n",
      "mock7 |GDL-1|Laetiporus caribensis_pident=100.0...          0          0  \n",
      "mock8 |RF5JR|Trametes gibbosa_pident=100.0_OTU26            0          0  \n",
      "mock9 |OR2|Laetiporus gilbertsonii_pident=100.0...          0          0  \n",
      "mock10 |MR5-1|Gloeoprus pannocinctus_pident=100...          0          0  \n",
      "mock11 |FP72612|Wolfiporia dilatohypha_pident=1...          0          0  \n",
      "mock12 |FP170198|Schizopora sp_pident=100.0_OTU27           0          0  \n",
      "mock13 |FP170231|Fomitopsis ochracea_pident=100...          0          0  \n",
      "mock14 |L34-2|Laetiporus cremeiporus_pident=100...          0          0  \n",
      "mock15 |RF9JR|Phanerochaete laevis_pident=100.0...          0          0  \n",
      "mock16 |DA37|Laetiporus cincinnatus_pident=100....          0          0  \n",
      "mock17 |RF7JR|Puntularia strigooszonata_pident=...          0          0  \n",
      "mock18 |IN4-1|Phellinus cinereus_pident=100.0_O...          0          0  \n",
      "mock19 |MR5-3|Antrodiellla semisupina_pident=10...          0          0  \n",
      "mock20 |TJV93-174|Leptoporus mollis_pident=100....          0          0  \n",
      "mock21 |RLG7163|Leptoporus mollis_pident=99.3_O...          0          0  \n",
      "mock22 |FP170186|Mortierelllales sp_pident=100....          0          0  \n",
      "mock23 |HHB9564|Laetiporus percisinus_pident=10...          0          0  \n",
      "mock24 |FP170215|Penicillium nothofagi_pident=1...          0          0  \n",
      "mock25 |FP170177|Pochonia suchlasporia_pident=1...          0          0  \n",
      "mock26 |MD275|Wolfiporia cocos_pident=100.0_OTU179          0          0  \n",
      "Other OTUs                                             162290     138802  \n",
      "m4A-stds            223788\n",
      "m4A-mock3-16000     268184\n",
      "m4A-mock3-32000a    280327\n",
      "m4A-mock3-32000b    274994\n",
      "m4A-755-1           292816\n",
      "m4A-755-2           283426\n",
      "m4A-744-1           268305\n",
      "m4A-744-2           247698\n",
      "m4A-766-1           292783\n",
      "m4A-766-2           303028\n",
      "m4b-744-1           182258\n",
      "m4b-744-2           152659\n",
      "m4b-766-1           162290\n",
      "m4b-766-2           139048\n",
      "dtype: int64\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "from natsort import natsorted\n",
    "#load the miseq table, no filtering version\n",
    "df = pd.read_csv('miseq.sorted.txt', sep='\\t', index_col=0)\n",
    "#Figure 2 should be about index bleed, so generate a table containing the samples that were run in all runs\n",
    "figureTwo = ['m4A-stds','m4A-mock3-16000','m4A-mock3-32000a','m4A-mock3-32000b','m4A-755-1','m4A-755-2',\n",
    "             'm4A-744-1','m4A-744-2','m4A-766-1','m4A-766-2',\n",
    "             'm4b-744-1','m4b-744-2','m4b-766-1','m4b-766-2' ]\n",
    "df2 = df.reindex(columns=figureTwo, index=natsorted(df.index))\n",
    "mocks = []\n",
    "for x in df2.index:\n",
    "    if x.startswith('mock'):\n",
    "        mocks.append(x)\n",
    "otus = [item for item in df2.index if item not in mocks]\n",
    "\n",
    "df3 = df2.reindex(columns=figureTwo, index=mocks)\n",
    "df3.to_csv('figure2.onlymocks.txt', sep='\\t')\n",
    "df4 = df2.reindex(columns=figureTwo, index=otus)\n",
    "#get number of OTUs from df4 as well as total reads\n",
    "reads = df4.sum(axis=0)\n",
    "reads.name =\"Other OTUs\"\n",
    "otu_count = df4[df4 > 0].count(axis=0, numeric_only=True)\n",
    "otu_count.name = \"Number of OTUs\"\n",
    "#print reads\n",
    "print otu_count\n",
    "#now append the read numbers to df3 dataframe\n",
    "df3 = df3.append(reads)\n",
    "print df3\n",
    "total_reads = df3.sum(axis=0)\n",
    "print total_reads\n",
    "df3.to_csv('figure2.miseqAB_only2.txt', sep='\\t')\n",
    "\n",
    "\n",
    "!amptk heatmap -i figure2.miseqAB_only2.txt -d tsv -o figure2-2.pdf --format pdf -m heatmap --annotate --color YlOrBr --vmax 30000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "m6-301-1           732\n",
      "m6-301-2           657\n",
      "m6-500-1           605\n",
      "m6-500-2           658\n",
      "m6-712-1           625\n",
      "m6-712-2           708\n",
      "m6-736-1           475\n",
      "m6-736-2           479\n",
      "m6-744-1           396\n",
      "m6-744-2           497\n",
      "m6-755-1           441\n",
      "m6-755-2           437\n",
      "m6-757-1           336\n",
      "m6-757-2           599\n",
      "m6-766-1           532\n",
      "m6-766-2           363\n",
      "m6-mock3-16000      90\n",
      "m6-mock3-32000a     89\n",
      "m6-mock3-32000b    102\n",
      "m6-stds            100\n",
      "m6-SynMock          99\n",
      "Name: Number of OTUs, dtype: int64\n",
      "m6-301-1            713497\n",
      "m6-301-2            953709\n",
      "m6-500-1            723217\n",
      "m6-500-2            718754\n",
      "m6-712-1            708178\n",
      "m6-712-2            693457\n",
      "m6-736-1            817881\n",
      "m6-736-2            740933\n",
      "m6-744-1            612510\n",
      "m6-744-2            561815\n",
      "m6-755-1            775334\n",
      "m6-755-2            891265\n",
      "m6-757-1            733736\n",
      "m6-757-2            961336\n",
      "m6-766-1            694195\n",
      "m6-766-2            602907\n",
      "m6-mock3-16000      729561\n",
      "m6-mock3-32000a     875430\n",
      "m6-mock3-32000b     748482\n",
      "m6-stds             635597\n",
      "m6-SynMock         1046497\n",
      "dtype: int64\n"
     ]
    }
   ],
   "source": [
    "#now want to take a look at synmock distribution in entire miseq6 run, see if bleed is only happening when there\n",
    "#is a shared index or not.\n",
    "\n",
    "import pandas as pd\n",
    "from natsort import natsorted\n",
    "#load the miseq table, no filtering version\n",
    "df = pd.read_csv('miseq_syn.sorted.txt', sep='\\t', index_col=0)\n",
    "#Figure 2 should be about index bleed, so generate a table containing the samples that were run in all runs\n",
    "keepers = ['m6-301-1', 'm6-301-2', 'm6-500-1', 'm6-500-2', 'm6-712-1', 'm6-712-2', \n",
    "           'm6-736-1', 'm6-736-2', 'm6-744-1', 'm6-744-2', 'm6-755-1', 'm6-755-2', \n",
    "           'm6-757-1', 'm6-757-2', 'm6-766-1', 'm6-766-2', \n",
    "           'm6-mock3-16000', 'm6-mock3-32000a', 'm6-mock3-32000b', 'm6-stds', 'm6-SynMock']\n",
    "df2 = df.reindex(columns=keepers, index=natsorted(df.index))\n",
    "mocks = []\n",
    "for x in df2.index:\n",
    "    if x.startswith('mock'):\n",
    "        mocks.append(x)\n",
    "otus = [item for item in df2.index if item not in mocks]\n",
    "\n",
    "df3 = df2.reindex(columns=keepers, index=mocks)\n",
    "df3.to_csv('figure2.onlymocks.txt', sep='\\t')\n",
    "df4 = df2.reindex(columns=keepers, index=otus)\n",
    "#get number of OTUs from df4 as well as total reads\n",
    "reads = df4.sum(axis=0)\n",
    "reads.name =\"Other OTUs\"\n",
    "otu_count = df4[df4 > 0].count(axis=0, numeric_only=True)\n",
    "otu_count.name = \"Number of OTUs\"\n",
    "#print reads\n",
    "print otu_count\n",
    "#now append the read numbers to df3 dataframe\n",
    "df3 = df3.append(reads)\n",
    "#print df3\n",
    "total_reads = df3.sum(axis=0)\n",
    "print total_reads\n",
    "df3.to_csv('figure3.miseq_synmock.txt', sep='\\t')\n",
    "\n",
    "\n",
    "!amptk heatmap -i figure3.miseq_synmock.txt -d tsv -o figure3.pdf --format pdf -m heatmap --annotate --color YlOrBr --vmax 120000"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Lets check raw versus filtered for treatment effects"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ion3.biom\r\n",
      "ion3_raw.biom\r\n",
      "ion5.biom\r\n",
      "ion5_raw.biom\r\n",
      "miseq4a.biom\r\n",
      "miseq4a_raw.biom\r\n",
      "miseq4c.biom\r\n",
      "miseq4c_raw.biom\r\n",
      "miseq6.biom\r\n",
      "miseq6_raw.biom\r\n"
     ]
    }
   ],
   "source": [
    "!ls | grep '.biom'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-3\n",
      "Loading required package: RColorBrewer\n",
      "[1] 1.19.1\n",
      "[1] 2.2.1\n",
      "[1] 1.8.4\n",
      "[1] 2.4.3\n",
      "[1] 1.1.2\n",
      "There were 26 warnings (use warnings() to see them)\n",
      "[1] \"LinkerPrimerSequence\" \"phinchID\"             \"Treatment\"           \n",
      "[4] \"BarcodeSequence\"      \"ReversePrimer\"        \"Terrain\"             \n",
      "[1] \"Treatment\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 2342 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 2342 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 2342 tips and 2341 internal nodes ]\n",
      "Run 0 stress 0.09775419 \n",
      "Run 1 stress 0.1187046 \n",
      "Run 2 stress 0.09561271 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1136169  max resid 0.3028549 \n",
      "Run 3 stress 0.2480941 \n",
      "Run 4 stress 0.09800281 \n",
      "Run 5 stress 0.2709404 \n",
      "Run 6 stress 0.09800227 \n",
      "Run 7 stress 0.1910293 \n",
      "Run 8 stress 0.151092 \n",
      "Run 9 stress 0.1786326 \n",
      "Run 10 stress 0.2655566 \n",
      "Run 11 stress 0.1854815 \n",
      "Run 12 stress 0.1956449 \n",
      "Run 13 stress 0.0980021 \n",
      "Run 14 stress 0.0980021 \n",
      "Run 15 stress 0.09561271 \n",
      "... New best solution\n",
      "... Procrustes: rmse 4.247942e-05  max resid 0.0001055793 \n",
      "... Similar to previous best\n",
      "Run 16 stress 0.0977541 \n",
      "Run 17 stress 0.1510922 \n",
      "Run 18 stress 0.09800252 \n",
      "Run 19 stress 0.1511668 \n",
      "Run 20 stress 0.2048576 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "[1] \"Terrain\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 2342 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 2342 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 2342 tips and 2341 internal nodes ]\n",
      "Run 0 stress 0.09933379 \n",
      "Run 1 stress 0.1876642 \n",
      "Run 2 stress 0.1062334 \n",
      "Run 3 stress 0.1671029 \n",
      "Run 4 stress 0.1667213 \n",
      "Run 5 stress 0.1587295 \n",
      "Run 6 stress 0.1866779 \n",
      "Run 7 stress 0.1816253 \n",
      "Run 8 stress 0.1988716 \n",
      "Run 9 stress 0.2563433 \n",
      "Run 10 stress 0.1560276 \n",
      "Run 11 stress 0.1620895 \n",
      "Run 12 stress 0.1926401 \n",
      "Run 13 stress 0.2453816 \n",
      "Run 14 stress 0.2926473 \n",
      "Run 15 stress 0.1924144 \n",
      "Run 16 stress 0.1715196 \n",
      "Run 17 stress 0.1503748 \n",
      "Run 18 stress 0.1560278 \n",
      "Run 19 stress 0.2042746 \n",
      "Run 20 stress 0.1897502 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 32 rows containing missing values (geom_errorbar). \n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n",
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-3\n",
      "Loading required package: RColorBrewer\n",
      "[1] 1.19.1\n",
      "[1] 2.2.1\n",
      "[1] 1.8.4\n",
      "[1] 2.4.3\n",
      "[1] 1.1.2\n",
      "Warning messages:\n",
      "1: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "2: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "3: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "4: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "[1] \"LinkerPrimerSequence\" \"phinchID\"             \"Treatment\"           \n",
      "[4] \"BarcodeSequence\"      \"ReversePrimer\"        \"Terrain\"             \n",
      "[1] \"Treatment\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 1936 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 1936 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 1936 tips and 1935 internal nodes ]\n",
      "Run 0 stress 0.1246509 \n",
      "Run 1 stress 0.1546024 \n",
      "Run 2 stress 0.1370029 \n",
      "Run 3 stress 0.137003 \n",
      "Run 4 stress 0.1370028 \n",
      "Run 5 stress 0.1370028 \n",
      "Run 6 stress 0.1246508 \n",
      "... New best solution\n",
      "... Procrustes: rmse 7.118717e-05  max resid 0.000166498 \n",
      "... Similar to previous best\n",
      "Run 7 stress 0.1370028 \n",
      "Run 8 stress 0.1370029 \n",
      "Run 9 stress 0.1370028 \n",
      "Run 10 stress 0.1370029 \n",
      "Run 11 stress 0.1246508 \n",
      "... Procrustes: rmse 4.096051e-06  max resid 6.163174e-06 \n",
      "... Similar to previous best\n",
      "Run 12 stress 0.1460213 \n",
      "Run 13 stress 0.1246509 \n",
      "... Procrustes: rmse 0.0001260836  max resid 0.0002918611 \n",
      "... Similar to previous best\n",
      "Run 14 stress 0.1370028 \n",
      "Run 15 stress 0.1370028 \n",
      "Run 16 stress 0.1246508 \n",
      "... Procrustes: rmse 8.98841e-06  max resid 1.900638e-05 \n",
      "... Similar to previous best\n",
      "Run 17 stress 0.1370033 \n",
      "Run 18 stress 0.1246509 \n",
      "... Procrustes: rmse 2.790156e-05  max resid 6.205426e-05 \n",
      "... Similar to previous best\n",
      "Run 19 stress 0.1546024 \n",
      "Run 20 stress 0.1246508 \n",
      "... New best solution\n",
      "... Procrustes: rmse 4.022962e-06  max resid 7.995197e-06 \n",
      "... Similar to previous best\n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "[1] \"Terrain\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 1936 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 1936 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 1936 tips and 1935 internal nodes ]\n",
      "Run 0 stress 0.1276732 \n",
      "Run 1 stress 0.1890528 \n",
      "Run 2 stress 0.1276732 \n",
      "... New best solution\n",
      "... Procrustes: rmse 6.104734e-05  max resid 0.0001438889 \n",
      "... Similar to previous best\n",
      "Run 3 stress 0.1363872 \n",
      "Run 4 stress 0.2914369 \n",
      "Run 5 stress 0.1905811 \n",
      "Run 6 stress 0.1408713 \n",
      "Run 7 stress 0.1968335 \n",
      "Run 8 stress 0.1276733 \n",
      "... Procrustes: rmse 8.169558e-05  max resid 0.0001919944 \n",
      "... Similar to previous best\n",
      "Run 9 stress 0.1276734 \n",
      "... Procrustes: rmse 0.0001817093  max resid 0.000426366 \n",
      "... Similar to previous best\n",
      "Run 10 stress 0.1276732 \n",
      "... Procrustes: rmse 1.708864e-05  max resid 4.027867e-05 \n",
      "... Similar to previous best\n",
      "Run 11 stress 0.1276732 \n",
      "... Procrustes: rmse 7.780817e-05  max resid 0.0001809789 \n",
      "... Similar to previous best\n",
      "Run 12 stress 0.1276734 \n",
      "... Procrustes: rmse 0.000217712  max resid 0.0005085387 \n",
      "... Similar to previous best\n",
      "Run 13 stress 0.140875 \n",
      "Run 14 stress 0.1276732 \n",
      "... New best solution\n",
      "... Procrustes: rmse 3.61186e-05  max resid 8.42227e-05 \n",
      "... Similar to previous best\n",
      "Run 15 stress 0.1868556 \n",
      "Run 16 stress 0.1276732 \n",
      "... Procrustes: rmse 5.970369e-06  max resid 1.245569e-05 \n",
      "... Similar to previous best\n",
      "Run 17 stress 0.1972419 \n",
      "Run 18 stress 0.1276732 \n",
      "... New best solution\n",
      "... Procrustes: rmse 1.778938e-06  max resid 4.814609e-06 \n",
      "... Similar to previous best\n",
      "Run 19 stress 0.1276732 \n",
      "... Procrustes: rmse 2.136275e-05  max resid 5.145319e-05 \n",
      "... Similar to previous best\n",
      "Run 20 stress 0.1363873 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 1 rows containing missing values (geom_errorbarh). \n",
      "2: In estimate_richness(physeq, split = TRUE, measures = measures) :\n",
      "  The data you have provided does not have\n",
      "any singletons. This is highly suspicious. Results of richness\n",
      "estimates (for example) are probably unreliable, or wrong, if you have already\n",
      "trimmed low-abundance taxa from the data.\n",
      "\n",
      "We recommended that you find the un-trimmed data and retry.\n",
      "3: Removed 32 rows containing missing values (geom_errorbar). \n",
      "4: In estimate_richness(physeq, split = TRUE, measures = measures) :\n",
      "  The data you have provided does not have\n",
      "any singletons. This is highly suspicious. Results of richness\n",
      "estimates (for example) are probably unreliable, or wrong, if you have already\n",
      "trimmed low-abundance taxa from the data.\n",
      "\n",
      "We recommended that you find the un-trimmed data and retry.\n",
      "5: Removed 32 rows containing missing values (geom_errorbar). \n",
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-3\n",
      "Loading required package: RColorBrewer\n",
      "[1] 1.19.1\n",
      "[1] 2.2.1\n",
      "[1] 1.8.4\n",
      "[1] 2.4.3\n",
      "[1] 1.1.2\n",
      "There were 27 warnings (use warnings() to see them)\n",
      "[1] \"LinkerPrimerSequence\" \"phinchID\"             \"Treatment\"           \n",
      "[4] \"BarcodeSequence\"      \"ReversePrimer\"        \"Terrain\"             \n",
      "[1] \"Treatment\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 1837 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 1837 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 1837 tips and 1836 internal nodes ]\n",
      "Run 0 stress 0.107331 \n",
      "Run 1 stress 0.107331 \n",
      "... New best solution\n",
      "... Procrustes: rmse 5.671896e-06  max resid 1.461001e-05 \n",
      "... Similar to previous best\n",
      "Run 2 stress 0.2751294 \n",
      "Run 3 stress 0.107331 \n",
      "... Procrustes: rmse 1.584726e-05  max resid 3.92791e-05 \n",
      "... Similar to previous best\n",
      "Run 4 stress 0.1748292 \n",
      "Run 5 stress 0.1848127 \n",
      "Run 6 stress 0.107331 \n",
      "... Procrustes: rmse 3.405764e-05  max resid 8.37256e-05 \n",
      "... Similar to previous best\n",
      "Run 7 stress 0.107331 \n",
      "... Procrustes: rmse 2.825892e-05  max resid 7.159612e-05 \n",
      "... Similar to previous best\n",
      "Run 8 stress 0.2083458 \n",
      "Run 9 stress 0.107331 \n",
      "... Procrustes: rmse 5.950467e-05  max resid 0.0001452658 \n",
      "... Similar to previous best\n",
      "Run 10 stress 0.107331 \n",
      "... Procrustes: rmse 5.013735e-05  max resid 0.0001243581 \n",
      "... Similar to previous best\n",
      "Run 11 stress 0.1692922 \n",
      "Run 12 stress 0.107331 \n",
      "... Procrustes: rmse 2.522994e-05  max resid 6.353275e-05 \n",
      "... Similar to previous best\n",
      "Run 13 stress 0.1748291 \n",
      "Run 14 stress 0.107331 \n",
      "... New best solution\n",
      "... Procrustes: rmse 5.345439e-06  max resid 1.309523e-05 \n",
      "... Similar to previous best\n",
      "Run 15 stress 0.107331 \n",
      "... Procrustes: rmse 3.39922e-05  max resid 8.422243e-05 \n",
      "... Similar to previous best\n",
      "Run 16 stress 0.107331 \n",
      "... Procrustes: rmse 1.510402e-06  max resid 3.516691e-06 \n",
      "... Similar to previous best\n",
      "Run 17 stress 0.107331 \n",
      "... Procrustes: rmse 1.659118e-06  max resid 3.749744e-06 \n",
      "... Similar to previous best\n",
      "Run 18 stress 0.107331 \n",
      "... Procrustes: rmse 4.808572e-06  max resid 1.221913e-05 \n",
      "... Similar to previous best\n",
      "Run 19 stress 0.107331 \n",
      "... Procrustes: rmse 3.483672e-06  max resid 9.618885e-06 \n",
      "... Similar to previous best\n",
      "Run 20 stress 0.1748285 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "[1] \"Terrain\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 1837 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 1837 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 1837 tips and 1836 internal nodes ]\n",
      "Run 0 stress 0.1078736 \n",
      "Run 1 stress 0.1316338 \n",
      "Run 2 stress 0.1316338 \n",
      "Run 3 stress 0.1078737 \n",
      "... Procrustes: rmse 2.806513e-05  max resid 5.620622e-05 \n",
      "... Similar to previous best\n",
      "Run 4 stress 0.1078736 \n",
      "... New best solution\n",
      "... Procrustes: rmse 2.3755e-06  max resid 5.902078e-06 \n",
      "... Similar to previous best\n",
      "Run 5 stress 0.1078736 \n",
      "... Procrustes: rmse 5.225553e-06  max resid 1.173491e-05 \n",
      "... Similar to previous best\n",
      "Run 6 stress 0.1316339 \n",
      "Run 7 stress 0.1811707 \n",
      "Run 8 stress 0.2067901 \n",
      "Run 9 stress 0.1811707 \n",
      "Run 10 stress 0.1831013 \n",
      "Run 11 stress 0.1078736 \n",
      "... Procrustes: rmse 3.003431e-05  max resid 5.183423e-05 \n",
      "... Similar to previous best\n",
      "Run 12 stress 0.1078736 \n",
      "... Procrustes: rmse 1.369911e-05  max resid 3.219497e-05 \n",
      "... Similar to previous best\n",
      "Run 13 stress 0.1078736 \n",
      "... Procrustes: rmse 2.799626e-05  max resid 7.758105e-05 \n",
      "... Similar to previous best\n",
      "Run 14 stress 0.1078737 \n",
      "... Procrustes: rmse 3.377159e-05  max resid 6.76103e-05 \n",
      "... Similar to previous best\n",
      "Run 15 stress 0.1316339 \n",
      "Run 16 stress 0.1078736 \n",
      "... Procrustes: rmse 1.744296e-05  max resid 4.151722e-05 \n",
      "... Similar to previous best\n",
      "Run 17 stress 0.1078737 \n",
      "... Procrustes: rmse 4.695497e-05  max resid 0.000103967 \n",
      "... Similar to previous best\n",
      "Run 18 stress 0.1078737 \n",
      "... Procrustes: rmse 6.860238e-05  max resid 0.0001911756 \n",
      "... Similar to previous best\n",
      "Run 19 stress 0.1316338 \n",
      "Run 20 stress 0.1316338 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 32 rows containing missing values (geom_errorbar). \n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n",
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-3\n",
      "Loading required package: RColorBrewer\n",
      "[1] 1.19.1\n",
      "[1] 2.2.1\n",
      "[1] 1.8.4\n",
      "[1] 2.4.3\n",
      "[1] 1.1.2\n",
      "There were 26 warnings (use warnings() to see them)\n",
      "[1] \"LinkerPrimerSequence\" \"phinchID\"             \"Treatment\"           \n",
      "[4] \"BarcodeSequence\"      \"ReversePrimer\"        \"Terrain\"             \n",
      "[1] \"Treatment\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 2245 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 2245 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 2245 tips and 2244 internal nodes ]\n",
      "Run 0 stress 0.08471697 \n",
      "Run 1 stress 0.08471634 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0002112666  max resid 0.0005277124 \n",
      "... Similar to previous best\n",
      "Run 2 stress 0.08471564 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0003313074  max resid 0.0008245734 \n",
      "... Similar to previous best\n",
      "Run 3 stress 0.08471544 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.00014828  max resid 0.0003731586 \n",
      "... Similar to previous best\n",
      "Run 4 stress 0.08471537 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0005231261  max resid 0.001306905 \n",
      "... Similar to previous best\n",
      "Run 5 stress 0.08471561 \n",
      "... Procrustes: rmse 0.0001881518  max resid 0.0004549216 \n",
      "... Similar to previous best\n",
      "Run 6 stress 0.2536414 \n",
      "Run 7 stress 0.1856008 \n",
      "Run 8 stress 0.08471551 \n",
      "... Procrustes: rmse 0.000113006  max resid 0.0002788381 \n",
      "... Similar to previous best\n",
      "Run 9 stress 0.08471558 \n",
      "... Procrustes: rmse 0.0001242595  max resid 0.000292084 \n",
      "... Similar to previous best\n",
      "Run 10 stress 0.08471563 \n",
      "... Procrustes: rmse 0.0006507429  max resid 0.001624744 \n",
      "... Similar to previous best\n",
      "Run 11 stress 0.1513041 \n",
      "Run 12 stress 0.1900088 \n",
      "Run 13 stress 0.227291 \n",
      "Run 14 stress 0.08471584 \n",
      "... Procrustes: rmse 0.0007739061  max resid 0.001884853 \n",
      "... Similar to previous best\n",
      "Run 15 stress 0.2394903 \n",
      "Run 16 stress 0.08471545 \n",
      "... Procrustes: rmse 8.207752e-05  max resid 0.0001988101 \n",
      "... Similar to previous best\n",
      "Run 17 stress 0.2607105 \n",
      "Run 18 stress 0.2116608 \n",
      "Run 19 stress 0.2099276 \n",
      "Run 20 stress 0.1539159 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "[1] \"Terrain\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 2245 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 2245 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 2245 tips and 2244 internal nodes ]\n",
      "Run 0 stress 0.08497874 \n",
      "Run 1 stress 0.08497886 \n",
      "... Procrustes: rmse 0.0001365677  max resid 0.0004015472 \n",
      "... Similar to previous best\n",
      "Run 2 stress 0.208322 \n",
      "Run 3 stress 0.08497913 \n",
      "... Procrustes: rmse 0.000295008  max resid 0.000811589 \n",
      "... Similar to previous best\n",
      "Run 4 stress 0.08497878 \n",
      "... Procrustes: rmse 0.0001411374  max resid 0.0003595524 \n",
      "... Similar to previous best\n",
      "Run 5 stress 0.1582333 \n",
      "Run 6 stress 0.0849788 \n",
      "... Procrustes: rmse 0.000152046  max resid 0.000422967 \n",
      "... Similar to previous best\n",
      "Run 7 stress 0.2563634 \n",
      "Run 8 stress 0.1589744 \n",
      "Run 9 stress 0.2245777 \n",
      "Run 10 stress 0.08497965 \n",
      "... Procrustes: rmse 0.0004635716  max resid 0.001355684 \n",
      "... Similar to previous best\n",
      "Run 11 stress 0.08497875 \n",
      "... Procrustes: rmse 0.0001057391  max resid 0.0003006223 \n",
      "... Similar to previous best\n",
      "Run 12 stress 0.2267188 \n",
      "Run 13 stress 0.08497952 \n",
      "... Procrustes: rmse 0.0004006459  max resid 0.001171349 \n",
      "... Similar to previous best\n",
      "Run 14 stress 0.2336541 \n",
      "Run 15 stress 0.1561476 \n",
      "Run 16 stress 0.08497881 \n",
      "... Procrustes: rmse 0.0001828929  max resid 0.0004744296 \n",
      "... Similar to previous best\n",
      "Run 17 stress 0.08497931 \n",
      "... Procrustes: rmse 0.0004444883  max resid 0.001302235 \n",
      "... Similar to previous best\n",
      "Run 18 stress 0.08497931 \n",
      "... Procrustes: rmse 0.0004348206  max resid 0.00132125 \n",
      "... Similar to previous best\n",
      "Run 19 stress 0.1582333 \n",
      "Run 20 stress 0.08497878 \n",
      "... Procrustes: rmse 0.0001100924  max resid 0.0003064675 \n",
      "... Similar to previous best\n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 32 rows containing missing values (geom_errorbar). \n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n",
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-3\n",
      "Loading required package: RColorBrewer\n",
      "[1] 1.19.1\n",
      "[1] 2.2.1\n",
      "[1] 1.8.4\n",
      "[1] 2.4.3\n",
      "[1] 1.1.2\n",
      "[1] \"LinkerPrimerSequence\" \"phinchID\"             \"Treatment\"           \n",
      "[4] \"BarcodeSequence\"      \"ReversePrimer\"        \"Terrain\"             \n",
      "[1] \"Treatment\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 2018 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 2018 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 2018 tips and 2017 internal nodes ]\n",
      "Run 0 stress 0.1420524 \n",
      "Run 1 stress 0.1372091 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0550187  max resid 0.1655183 \n",
      "Run 2 stress 0.2487231 \n",
      "Run 3 stress 0.1883968 \n",
      "Run 4 stress 0.1372108 \n",
      "... Procrustes: rmse 0.0008256764  max resid 0.001764846 \n",
      "... Similar to previous best\n",
      "Run 5 stress 0.3074491 \n",
      "Run 6 stress 0.1420524 \n",
      "Run 7 stress 0.1420526 \n",
      "Run 8 stress 0.1372103 \n",
      "... Procrustes: rmse 0.0006811921  max resid 0.001483564 \n",
      "... Similar to previous best\n",
      "Run 9 stress 0.1894724 \n",
      "Run 10 stress 0.1372092 \n",
      "... Procrustes: rmse 8.808809e-05  max resid 0.0001707314 \n",
      "... Similar to previous best\n",
      "Run 11 stress 0.1372133 \n",
      "... Procrustes: rmse 0.001868464  max resid 0.004688547 \n",
      "... Similar to previous best\n",
      "Run 12 stress 0.2003361 \n",
      "Run 13 stress 0.1420524 \n",
      "Run 14 stress 0.1372091 \n",
      "... New best solution\n",
      "... Procrustes: rmse 2.733462e-05  max resid 7.148108e-05 \n",
      "... Similar to previous best\n",
      "Run 15 stress 0.1372092 \n",
      "... Procrustes: rmse 0.0001586371  max resid 0.000379793 \n",
      "... Similar to previous best\n",
      "Run 16 stress 0.1372096 \n",
      "... Procrustes: rmse 0.0003283269  max resid 0.0007103907 \n",
      "... Similar to previous best\n",
      "Run 17 stress 0.1372092 \n",
      "... Procrustes: rmse 0.0004192351  max resid 0.001111433 \n",
      "... Similar to previous best\n",
      "Run 18 stress 0.1372097 \n",
      "... Procrustes: rmse 0.0003697791  max resid 0.0009535531 \n",
      "... Similar to previous best\n",
      "Run 19 stress 0.1562063 \n",
      "Run 20 stress 0.1372098 \n",
      "... Procrustes: rmse 0.0005279284  max resid 0.001235773 \n",
      "... Similar to previous best\n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "[1] \"Terrain\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 2018 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 2018 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 2018 tips and 2017 internal nodes ]\n",
      "Run 0 stress 0.1426937 \n",
      "Run 1 stress 0.1379471 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.05697906  max resid 0.1813407 \n",
      "Run 2 stress 0.1446042 \n",
      "Run 3 stress 0.1379503 \n",
      "... Procrustes: rmse 0.002425382  max resid 0.00593647 \n",
      "... Similar to previous best\n",
      "Run 4 stress 0.1379477 \n",
      "... Procrustes: rmse 0.0001420971  max resid 0.0003273532 \n",
      "... Similar to previous best\n",
      "Run 5 stress 0.137948 \n",
      "... Procrustes: rmse 0.0004356436  max resid 0.00106407 \n",
      "... Similar to previous best\n",
      "Run 6 stress 0.1379481 \n",
      "... Procrustes: rmse 0.000177346  max resid 0.0005195356 \n",
      "... Similar to previous best\n",
      "Run 7 stress 0.1379501 \n",
      "... Procrustes: rmse 0.002379906  max resid 0.005852272 \n",
      "... Similar to previous best\n",
      "Run 8 stress 0.1379512 \n",
      "... Procrustes: rmse 0.002618668  max resid 0.006403613 \n",
      "... Similar to previous best\n",
      "Run 9 stress 0.1446042 \n",
      "Run 10 stress 0.1930904 \n",
      "Run 11 stress 0.1924695 \n",
      "Run 12 stress 0.137947 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.001217813  max resid 0.003003581 \n",
      "... Similar to previous best\n",
      "Run 13 stress 0.1379485 \n",
      "... Procrustes: rmse 0.0005375862  max resid 0.001279309 \n",
      "... Similar to previous best\n",
      "Run 14 stress 0.261782 \n",
      "Run 15 stress 0.1932311 \n",
      "Run 16 stress 0.2014455 \n",
      "Run 17 stress 0.1987888 \n",
      "Run 18 stress 0.1379483 \n",
      "... Procrustes: rmse 0.001750564  max resid 0.004205864 \n",
      "... Similar to previous best\n",
      "Run 19 stress 0.1446041 \n",
      "Run 20 stress 0.1706064 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: In estimate_richness(physeq, split = TRUE, measures = measures) :\n",
      "  The data you have provided does not have\n",
      "any singletons. This is highly suspicious. Results of richness\n",
      "estimates (for example) are probably unreliable, or wrong, if you have already\n",
      "trimmed low-abundance taxa from the data.\n",
      "\n",
      "We recommended that you find the un-trimmed data and retry.\n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n",
      "3: In estimate_richness(physeq, split = TRUE, measures = measures) :\n",
      "  The data you have provided does not have\n",
      "any singletons. This is highly suspicious. Results of richness\n",
      "estimates (for example) are probably unreliable, or wrong, if you have already\n",
      "trimmed low-abundance taxa from the data.\n",
      "\n",
      "We recommended that you find the un-trimmed data and retry.\n",
      "4: Removed 32 rows containing missing values (geom_errorbar). \n",
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-3\n",
      "Loading required package: RColorBrewer\n",
      "[1] 1.19.1\n",
      "[1] 2.2.1\n",
      "[1] 1.8.4\n",
      "[1] 2.4.3\n",
      "[1] 1.1.2\n",
      "There were 12 warnings (use warnings() to see them)\n",
      "[1] \"LinkerPrimerSequence\" \"phinchID\"             \"Treatment\"           \n",
      "[4] \"BarcodeSequence\"      \"ReversePrimer\"        \"Terrain\"             \n",
      "[1] \"Treatment\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 2687 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 2687 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 2687 tips and 2686 internal nodes ]\n",
      "Run 0 stress 0.08013366 \n",
      "Run 1 stress 0.0801336 \n",
      "... New best solution\n",
      "... Procrustes: rmse 9.609461e-05  max resid 0.0002692205 \n",
      "... Similar to previous best\n",
      "Run 2 stress 0.1658965 \n",
      "Run 3 stress 0.1834043 \n",
      "Run 4 stress 0.2064054 \n",
      "Run 5 stress 0.1893945 \n",
      "Run 6 stress 0.1990574 \n",
      "Run 7 stress 0.2274841 \n",
      "Run 8 stress 0.1936611 \n",
      "Run 9 stress 0.223877 \n",
      "Run 10 stress 0.08013383 \n",
      "... Procrustes: rmse 0.0002014919  max resid 0.0006234852 \n",
      "... Similar to previous best\n",
      "Run 11 stress 0.0801336 \n",
      "... New best solution\n",
      "... Procrustes: rmse 2.361424e-05  max resid 5.91962e-05 \n",
      "... Similar to previous best\n",
      "Run 12 stress 0.0801336 \n",
      "... New best solution\n",
      "... Procrustes: rmse 6.759484e-05  max resid 0.0002059301 \n",
      "... Similar to previous best\n",
      "Run 13 stress 0.151261 \n",
      "Run 14 stress 0.2059523 \n",
      "Run 15 stress 0.1674891 \n",
      "Run 16 stress 0.223345 \n",
      "Run 17 stress 0.08013382 \n",
      "... Procrustes: rmse 0.0001935126  max resid 0.0005827137 \n",
      "... Similar to previous best\n",
      "Run 18 stress 0.08013363 \n",
      "... Procrustes: rmse 0.0001181073  max resid 0.0003651962 \n",
      "... Similar to previous best\n",
      "Run 19 stress 0.2074249 \n",
      "Run 20 stress 0.2050311 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "[1] \"Terrain\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 2687 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 2687 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 2687 tips and 2686 internal nodes ]\n",
      "Run 0 stress 0.09689281 \n",
      "Run 1 stress 0.199578 \n",
      "Run 2 stress 0.1932161 \n",
      "Run 3 stress 0.176059 \n",
      "Run 4 stress 0.1909245 \n",
      "Run 5 stress 0.1979959 \n",
      "Run 6 stress 0.1031374 \n",
      "Run 7 stress 0.1909202 \n",
      "Run 8 stress 0.09689281 \n",
      "... Procrustes: rmse 9.49716e-06  max resid 2.615263e-05 \n",
      "... Similar to previous best\n",
      "Run 9 stress 0.09689282 \n",
      "... Procrustes: rmse 2.711314e-05  max resid 6.693551e-05 \n",
      "... Similar to previous best\n",
      "Run 10 stress 0.1342942 \n",
      "Run 11 stress 0.1748149 \n",
      "Run 12 stress 0.09689281 \n",
      "... Procrustes: rmse 1.367991e-05  max resid 3.070288e-05 \n",
      "... Similar to previous best\n",
      "Run 13 stress 0.2087414 \n",
      "Run 14 stress 0.2120374 \n",
      "Run 15 stress 0.09689282 \n",
      "... Procrustes: rmse 1.775755e-05  max resid 4.100827e-05 \n",
      "... Similar to previous best\n",
      "Run 16 stress 0.2348823 \n",
      "Run 17 stress 0.1985472 \n",
      "Run 18 stress 0.2063423 \n",
      "Run 19 stress 0.2107049 \n",
      "Run 20 stress 0.09689281 \n",
      "... Procrustes: rmse 1.187002e-05  max resid 2.620133e-05 \n",
      "... Similar to previous best\n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 32 rows containing missing values (geom_errorbar). \n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n",
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-3\n",
      "Loading required package: RColorBrewer\n",
      "[1] 1.19.1\n",
      "[1] 2.2.1\n",
      "[1] 1.8.4\n",
      "[1] 2.4.3\n",
      "[1] 1.1.2\n",
      "Warning messages:\n",
      "1: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "2: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "3: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "4: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "[1] \"LinkerPrimerSequence\" \"phinchID\"             \"Treatment\"           \n",
      "[4] \"BarcodeSequence\"      \"ReversePrimer\"        \"Terrain\"             \n",
      "[1] \"Treatment\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 1070 taxa and 14 samples ]\n",
      "sample_data() Sample Data:       [ 14 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 1070 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 1070 tips and 1069 internal nodes ]\n",
      "Run 0 stress 0.1299128 \n",
      "Run 1 stress 0.1299128 \n",
      "... Procrustes: rmse 4.91979e-05  max resid 0.0001074902 \n",
      "... Similar to previous best\n",
      "Run 2 stress 0.1648282 \n",
      "Run 3 stress 0.1299128 \n",
      "... New best solution\n",
      "... Procrustes: rmse 2.467163e-06  max resid 4.887341e-06 \n",
      "... Similar to previous best\n",
      "Run 4 stress 0.1818289 \n",
      "Run 5 stress 0.2948725 \n",
      "Run 6 stress 0.1299128 \n",
      "... Procrustes: rmse 2.369325e-06  max resid 3.911439e-06 \n",
      "... Similar to previous best\n",
      "Run 7 stress 0.1986785 \n",
      "Run 8 stress 0.1299128 \n",
      "... New best solution\n",
      "... Procrustes: rmse 2.585531e-06  max resid 6.154813e-06 \n",
      "... Similar to previous best\n",
      "Run 9 stress 0.1921844 \n",
      "Run 10 stress 0.1299128 \n",
      "... Procrustes: rmse 1.356556e-05  max resid 2.9344e-05 \n",
      "... Similar to previous best\n",
      "Run 11 stress 0.1299128 \n",
      "... Procrustes: rmse 2.698042e-06  max resid 4.510177e-06 \n",
      "... Similar to previous best\n",
      "Run 12 stress 0.1299128 \n",
      "... Procrustes: rmse 2.111647e-05  max resid 4.655708e-05 \n",
      "... Similar to previous best\n",
      "Run 13 stress 0.1299128 \n",
      "... Procrustes: rmse 2.684774e-05  max resid 5.599938e-05 \n",
      "... Similar to previous best\n",
      "Run 14 stress 0.1299128 \n",
      "... Procrustes: rmse 1.600308e-05  max resid 3.342213e-05 \n",
      "... Similar to previous best\n",
      "Run 15 stress 0.2979189 \n",
      "Run 16 stress 0.1299128 \n",
      "... Procrustes: rmse 7.05525e-06  max resid 1.509696e-05 \n",
      "... Similar to previous best\n",
      "Run 17 stress 0.1299128 \n",
      "... Procrustes: rmse 1.700927e-05  max resid 3.968201e-05 \n",
      "... Similar to previous best\n",
      "Run 18 stress 0.1538194 \n",
      "Run 19 stress 0.1299128 \n",
      "... Procrustes: rmse 4.269858e-06  max resid 8.898805e-06 \n",
      "... Similar to previous best\n",
      "Run 20 stress 0.1299128 \n",
      "... New best solution\n",
      "... Procrustes: rmse 1.487789e-06  max resid 3.102675e-06 \n",
      "... Similar to previous best\n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "[1] \"Terrain\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 1070 taxa and 14 samples ]\n",
      "sample_data() Sample Data:       [ 14 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 1070 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 1070 tips and 1069 internal nodes ]\n",
      "Run 0 stress 0.130755 \n",
      "Run 1 stress 0.1912253 \n",
      "Run 2 stress 0.130755 \n",
      "... New best solution\n",
      "... Procrustes: rmse 1.181159e-05  max resid 3.356531e-05 \n",
      "... Similar to previous best\n",
      "Run 3 stress 0.130755 \n",
      "... Procrustes: rmse 6.116048e-06  max resid 1.732276e-05 \n",
      "... Similar to previous best\n",
      "Run 4 stress 0.2734896 \n",
      "Run 5 stress 0.130755 \n",
      "... Procrustes: rmse 7.822981e-06  max resid 2.378494e-05 \n",
      "... Similar to previous best\n",
      "Run 6 stress 0.1590736 \n",
      "Run 7 stress 0.1995452 \n",
      "Run 8 stress 0.1556732 \n",
      "Run 9 stress 0.130755 \n",
      "... Procrustes: rmse 1.979693e-06  max resid 4.033166e-06 \n",
      "... Similar to previous best\n",
      "Run 10 stress 0.130755 \n",
      "... Procrustes: rmse 6.655655e-06  max resid 1.417889e-05 \n",
      "... Similar to previous best\n",
      "Run 11 stress 0.1908268 \n",
      "Run 12 stress 0.130755 \n",
      "... Procrustes: rmse 1.231824e-06  max resid 3.566731e-06 \n",
      "... Similar to previous best\n",
      "Run 13 stress 0.1672279 \n",
      "Run 14 stress 0.130755 \n",
      "... Procrustes: rmse 5.283945e-06  max resid 1.384718e-05 \n",
      "... Similar to previous best\n",
      "Run 15 stress 0.130755 \n",
      "... Procrustes: rmse 2.588829e-06  max resid 5.583016e-06 \n",
      "... Similar to previous best\n",
      "Run 16 stress 0.130755 \n",
      "... Procrustes: rmse 1.580867e-05  max resid 4.735138e-05 \n",
      "... Similar to previous best\n",
      "Run 17 stress 0.130755 \n",
      "... Procrustes: rmse 2.235827e-06  max resid 3.996071e-06 \n",
      "... Similar to previous best\n",
      "Run 18 stress 0.130755 \n",
      "... Procrustes: rmse 8.415623e-07  max resid 1.611893e-06 \n",
      "... Similar to previous best\n",
      "Run 19 stress 0.130755 \n",
      "... Procrustes: rmse 4.942596e-06  max resid 1.464667e-05 \n",
      "... Similar to previous best\n",
      "Run 20 stress 0.1813724 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 28 rows containing missing values (geom_errorbar). \n",
      "2: Removed 28 rows containing missing values (geom_errorbar). \n",
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-3\n",
      "Loading required package: RColorBrewer\n",
      "[1] 1.19.1\n",
      "[1] 2.2.1\n",
      "[1] 1.8.4\n",
      "[1] 2.4.3\n",
      "[1] 1.1.2\n",
      "There were 21 warnings (use warnings() to see them)\n",
      "[1] \"LinkerPrimerSequence\" \"phinchID\"             \"Treatment\"           \n",
      "[4] \"BarcodeSequence\"      \"ReversePrimer\"        \"Terrain\"             \n",
      "[1] \"Treatment\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 1137 taxa and 14 samples ]\n",
      "sample_data() Sample Data:       [ 14 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 1137 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 1137 tips and 1136 internal nodes ]\n",
      "Run 0 stress 0.1295228 \n",
      "Run 1 stress 0.1295228 \n",
      "... Procrustes: rmse 1.972648e-05  max resid 4.999532e-05 \n",
      "... Similar to previous best\n",
      "Run 2 stress 0.1468968 \n",
      "Run 3 stress 0.240269 \n",
      "Run 4 stress 0.1692039 \n",
      "Run 5 stress 0.2896335 \n",
      "Run 6 stress 0.1601902 \n",
      "Run 7 stress 0.1468968 \n",
      "Run 8 stress 0.1302709 \n",
      "Run 9 stress 0.1440397 \n",
      "Run 10 stress 0.1367386 \n",
      "Run 11 stress 0.1367385 \n",
      "Run 12 stress 0.2029614 \n",
      "Run 13 stress 0.2921231 \n",
      "Run 14 stress 0.1468968 \n",
      "Run 15 stress 0.1468969 \n",
      "Run 16 stress 0.1468974 \n",
      "Run 17 stress 0.1599522 \n",
      "Run 18 stress 0.1367387 \n",
      "Run 19 stress 0.1302708 \n",
      "Run 20 stress 0.1467106 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "[1] \"Terrain\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 1137 taxa and 14 samples ]\n",
      "sample_data() Sample Data:       [ 14 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 1137 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 1137 tips and 1136 internal nodes ]\n",
      "Run 0 stress 0.1252428 \n",
      "Run 1 stress 0.1252428 \n",
      "... Procrustes: rmse 8.778314e-05  max resid 0.0002551057 \n",
      "... Similar to previous best\n",
      "Run 2 stress 0.125243 \n",
      "... Procrustes: rmse 8.8101e-05  max resid 0.000139284 \n",
      "... Similar to previous best\n",
      "Run 3 stress 0.1490936 \n",
      "Run 4 stress 0.1674485 \n",
      "Run 5 stress 0.1471358 \n",
      "Run 6 stress 0.1490936 \n",
      "Run 7 stress 0.2752592 \n",
      "Run 8 stress 0.1252443 \n",
      "... Procrustes: rmse 0.0006525361  max resid 0.002163921 \n",
      "... Similar to previous best\n",
      "Run 9 stress 0.1674486 \n",
      "Run 10 stress 0.1463289 \n",
      "Run 11 stress 0.1252426 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0001385214  max resid 0.0004344837 \n",
      "... Similar to previous best\n",
      "Run 12 stress 0.1471358 \n",
      "Run 13 stress 0.1471358 \n",
      "Run 14 stress 0.1960947 \n",
      "Run 15 stress 0.1463289 \n",
      "Run 16 stress 0.1252429 \n",
      "... Procrustes: rmse 0.0002410257  max resid 0.0007959792 \n",
      "... Similar to previous best\n",
      "Run 17 stress 0.2971021 \n",
      "Run 18 stress 0.1490936 \n",
      "Run 19 stress 0.125243 \n",
      "... Procrustes: rmse 0.0002794823  max resid 0.0009253009 \n",
      "... Similar to previous best\n",
      "Run 20 stress 0.2938201 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 28 rows containing missing values (geom_errorbar). \n",
      "2: Removed 28 rows containing missing values (geom_errorbar). \n",
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-3\n",
      "Loading required package: RColorBrewer\n",
      "[1] 1.19.1\n",
      "[1] 2.2.1\n",
      "[1] 1.8.4\n",
      "[1] 2.4.3\n",
      "[1] 1.1.2\n",
      "Warning messages:\n",
      "1: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "2: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "3: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "4: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "[1] \"LinkerPrimerSequence\" \"phinchID\"             \"Treatment\"           \n",
      "[4] \"BarcodeSequence\"      \"ReversePrimer\"        \"Terrain\"             \n",
      "[1] \"Treatment\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 1115 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 1115 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 1115 tips and 1114 internal nodes ]\n",
      "Run 0 stress 0.1437464 \n",
      "Run 1 stress 0.1896367 \n",
      "Run 2 stress 0.1791763 \n",
      "Run 3 stress 0.1437473 \n",
      "... Procrustes: rmse 0.0005422484  max resid 0.001240051 \n",
      "... Similar to previous best\n",
      "Run 4 stress 0.2418282 \n",
      "Run 5 stress 0.3085947 \n",
      "Run 6 stress 0.1437466 \n",
      "... Procrustes: rmse 0.0002018403  max resid 0.000463978 \n",
      "... Similar to previous best\n",
      "Run 7 stress 0.1915384 \n",
      "Run 8 stress 0.1896367 \n",
      "Run 9 stress 0.1869265 \n",
      "Run 10 stress 0.1518855 \n",
      "Run 11 stress 0.1437464 \n",
      "... Procrustes: rmse 7.004148e-06  max resid 1.505281e-05 \n",
      "... Similar to previous best\n",
      "Run 12 stress 0.2154867 \n",
      "Run 13 stress 0.142972 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.05036971  max resid 0.1720458 \n",
      "Run 14 stress 0.1532549 \n",
      "Run 15 stress 0.1864439 \n",
      "Run 16 stress 0.1871372 \n",
      "Run 17 stress 0.142972 \n",
      "... Procrustes: rmse 4.944693e-05  max resid 0.0001233497 \n",
      "... Similar to previous best\n",
      "Run 18 stress 0.142972 \n",
      "... New best solution\n",
      "... Procrustes: rmse 4.211211e-05  max resid 0.0001189018 \n",
      "... Similar to previous best\n",
      "Run 19 stress 0.1864437 \n",
      "Run 20 stress 0.1869274 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "[1] \"Terrain\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 1115 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 1115 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 1115 tips and 1114 internal nodes ]\n",
      "Run 0 stress 0.1384662 \n",
      "Run 1 stress 0.147465 \n",
      "Run 2 stress 0.1782556 \n",
      "Run 3 stress 0.1881402 \n",
      "Run 4 stress 0.1948932 \n",
      "Run 5 stress 0.1823252 \n",
      "Run 6 stress 0.1729865 \n",
      "Run 7 stress 0.1500399 \n",
      "Run 8 stress 0.1384662 \n",
      "... Procrustes: rmse 1.517357e-05  max resid 3.149722e-05 \n",
      "... Similar to previous best\n",
      "Run 9 stress 0.1528482 \n",
      "Run 10 stress 0.1500398 \n",
      "Run 11 stress 0.1782555 \n",
      "Run 12 stress 0.1408666 \n",
      "Run 13 stress 0.1879575 \n",
      "Run 14 stress 0.147466 \n",
      "Run 15 stress 0.17356 \n",
      "Run 16 stress 0.1782555 \n",
      "Run 17 stress 0.1782556 \n",
      "Run 18 stress 0.1500398 \n",
      "Run 19 stress 0.1384662 \n",
      "... Procrustes: rmse 4.076851e-05  max resid 8.397506e-05 \n",
      "... Similar to previous best\n",
      "Run 20 stress 0.1474651 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 32 rows containing missing values (geom_errorbar). \n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n",
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-3\n",
      "Loading required package: RColorBrewer\n",
      "[1] 1.19.1\n",
      "[1] 2.2.1\n",
      "[1] 1.8.4\n",
      "[1] 2.4.3\n",
      "[1] 1.1.2\n",
      "There were 15 warnings (use warnings() to see them)\n",
      "[1] \"LinkerPrimerSequence\" \"phinchID\"             \"Treatment\"           \n",
      "[4] \"BarcodeSequence\"      \"ReversePrimer\"        \"Terrain\"             \n",
      "[1] \"Treatment\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 1164 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 1164 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 1164 tips and 1163 internal nodes ]\n",
      "Run 0 stress 0.1239369 \n",
      "Run 1 stress 0.1251939 \n",
      "Run 2 stress 0.1813429 \n",
      "Run 3 stress 0.1239373 \n",
      "... Procrustes: rmse 0.0007990179  max resid 0.001982577 \n",
      "... Similar to previous best\n",
      "Run 4 stress 0.1486806 \n",
      "Run 5 stress 0.123937 \n",
      "... Procrustes: rmse 0.0006775263  max resid 0.001679679 \n",
      "... Similar to previous best\n",
      "Run 6 stress 0.1486805 \n",
      "Run 7 stress 0.212001 \n",
      "Run 8 stress 0.2653463 \n",
      "Run 9 stress 0.1678504 \n",
      "Run 10 stress 0.1281991 \n",
      "Run 11 stress 0.1718861 \n",
      "Run 12 stress 0.2716617 \n",
      "Run 13 stress 0.1396779 \n",
      "Run 14 stress 0.1569634 \n",
      "Run 15 stress 0.2452706 \n",
      "Run 16 stress 0.1355699 \n",
      "Run 17 stress 0.1239375 \n",
      "... Procrustes: rmse 0.0002899797  max resid 0.0007133238 \n",
      "... Similar to previous best\n",
      "Run 18 stress 0.1545476 \n",
      "Run 19 stress 0.167851 \n",
      "Run 20 stress 0.1251938 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "[1] \"Terrain\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 1164 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 1164 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 1164 tips and 1163 internal nodes ]\n",
      "Run 0 stress 0.1280426 \n",
      "Run 1 stress 0.1551914 \n",
      "Run 2 stress 0.1821399 \n",
      "Run 3 stress 0.131672 \n",
      "Run 4 stress 0.114078 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.06394173  max resid 0.2076829 \n",
      "Run 5 stress 0.1917388 \n",
      "Run 6 stress 0.131672 \n",
      "Run 7 stress 0.1938004 \n",
      "Run 8 stress 0.131672 \n",
      "Run 9 stress 0.1992888 \n",
      "Run 10 stress 0.114078 \n",
      "... New best solution\n",
      "... Procrustes: rmse 1.854917e-05  max resid 4.00632e-05 \n",
      "... Similar to previous best\n",
      "Run 11 stress 0.1425267 \n",
      "Run 12 stress 0.131672 \n",
      "Run 13 stress 0.1316721 \n",
      "Run 14 stress 0.1831567 \n",
      "Run 15 stress 0.1665706 \n",
      "Run 16 stress 0.1551904 \n",
      "Run 17 stress 0.1696622 \n",
      "Run 18 stress 0.2289724 \n",
      "Run 19 stress 0.2208318 \n",
      "Run 20 stress 0.1293668 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 32 rows containing missing values (geom_errorbar). \n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n",
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-3\n",
      "Loading required package: RColorBrewer\n",
      "[1] 1.19.1\n",
      "[1] 2.2.1\n",
      "[1] 1.8.4\n",
      "[1] 2.4.3\n",
      "[1] 1.1.2\n",
      "There were 26 warnings (use warnings() to see them)\n",
      "[1] \"LinkerPrimerSequence\" \"phinchID\"             \"Treatment\"           \n",
      "[4] \"BarcodeSequence\"      \"ReversePrimer\"        \"Terrain\"             \n",
      "[1] \"Treatment\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 2342 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 2342 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 2342 tips and 2341 internal nodes ]\n",
      "Run 0 stress 0.2228892 \n",
      "Run 1 stress 0.2237818 \n",
      "Run 2 stress 0.208109 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.138589  max resid 0.2851836 \n",
      "Run 3 stress 0.2081106 \n",
      "... Procrustes: rmse 0.001496872  max resid 0.003827723 \n",
      "... Similar to previous best\n",
      "Run 4 stress 0.2170575 \n",
      "Run 5 stress 0.2377281 \n",
      "Run 6 stress 0.2081089 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0002709746  max resid 0.0006801573 \n",
      "... Similar to previous best\n",
      "Run 7 stress 0.227651 \n",
      "Run 8 stress 0.2229071 \n",
      "Run 9 stress 0.2377281 \n",
      "Run 10 stress 0.2169278 \n",
      "Run 11 stress 0.2193019 \n",
      "Run 12 stress 0.2081091 \n",
      "... Procrustes: rmse 0.0004256167  max resid 0.001091475 \n",
      "... Similar to previous best\n",
      "Run 13 stress 0.237729 \n",
      "Run 14 stress 0.2186511 \n",
      "Run 15 stress 0.2165791 \n",
      "Run 16 stress 0.2237817 \n",
      "Run 17 stress 0.2499787 \n",
      "Run 18 stress 0.2215645 \n",
      "Run 19 stress 0.2229843 \n",
      "Run 20 stress 0.2439762 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "[1] \"Terrain\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 2342 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 2342 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 2342 tips and 2341 internal nodes ]\n",
      "Run 0 stress 0.2228892 \n",
      "Run 1 stress 0.285159 \n",
      "Run 2 stress 0.2209655 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1084603  max resid 0.3432917 \n",
      "Run 3 stress 0.2186511 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1924223  max resid 0.4591817 \n",
      "Run 4 stress 0.2186511 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0002640725  max resid 0.0005285517 \n",
      "... Similar to previous best\n",
      "Run 5 stress 0.2202479 \n",
      "Run 6 stress 0.2209643 \n",
      "Run 7 stress 0.3486744 \n",
      "Run 8 stress 0.218651 \n",
      "... New best solution\n",
      "... Procrustes: rmse 6.875612e-05  max resid 0.0001351218 \n",
      "... Similar to previous best\n",
      "Run 9 stress 0.235747 \n",
      "Run 10 stress 0.2248948 \n",
      "Run 11 stress 0.220802 \n",
      "Run 12 stress 0.2170575 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1654304  max resid 0.3473685 \n",
      "Run 13 stress 0.2081094 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1922705  max resid 0.4658287 \n",
      "Run 14 stress 0.2229846 \n",
      "Run 15 stress 0.2226598 \n",
      "Run 16 stress 0.3539493 \n",
      "Run 17 stress 0.216579 \n",
      "Run 18 stress 0.2228849 \n",
      "Run 19 stress 0.2202593 \n",
      "Run 20 stress 0.2216975 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 32 rows containing missing values (geom_errorbar). \n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n",
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-3\n",
      "Loading required package: RColorBrewer\n",
      "[1] 1.19.1\n",
      "[1] 2.2.1\n",
      "[1] 1.8.4\n",
      "[1] 2.4.3\n",
      "[1] 1.1.2\n",
      "Warning messages:\n",
      "1: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "2: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "3: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "4: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "[1] \"LinkerPrimerSequence\" \"phinchID\"             \"Treatment\"           \n",
      "[4] \"BarcodeSequence\"      \"ReversePrimer\"        \"Terrain\"             \n",
      "[1] \"Treatment\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 1936 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 1936 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 1936 tips and 1935 internal nodes ]\n",
      "Run 0 stress 0.222858 \n",
      "Run 1 stress 0.2189999 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1958557  max resid 0.4543116 \n",
      "Run 2 stress 0.2078702 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1441668  max resid 0.4373273 \n",
      "Run 3 stress 0.226008 \n",
      "Run 4 stress 0.2372416 \n",
      "Run 5 stress 0.217879 \n",
      "Run 6 stress 0.2219094 \n",
      "Run 7 stress 0.2296746 \n",
      "Run 8 stress 0.2220787 \n",
      "Run 9 stress 0.2214765 \n",
      "Run 10 stress 0.2306875 \n",
      "Run 11 stress 0.2296741 \n",
      "Run 12 stress 0.2623224 \n",
      "Run 13 stress 0.2228313 \n",
      "Run 14 stress 0.2296741 \n",
      "Run 15 stress 0.2179005 \n",
      "Run 16 stress 0.273576 \n",
      "Run 17 stress 0.2487191 \n",
      "Run 18 stress 0.2218536 \n",
      "Run 19 stress 0.3087361 \n",
      "Run 20 stress 0.226008 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "[1] \"Terrain\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 1936 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 1936 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 1936 tips and 1935 internal nodes ]\n",
      "Run 0 stress 0.222858 \n",
      "Run 1 stress 0.22248 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1755777  max resid 0.4910684 \n",
      "Run 2 stress 0.2219298 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1044195  max resid 0.2711207 \n",
      "Run 3 stress 0.221384 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1640227  max resid 0.4960293 \n",
      "Run 4 stress 0.2219293 \n",
      "Run 5 stress 0.2395735 \n",
      "Run 6 stress 0.2191688 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1714869  max resid 0.4820724 \n",
      "Run 7 stress 0.207864 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1439033  max resid 0.4264945 \n",
      "Run 8 stress 0.2306873 \n",
      "Run 9 stress 0.2622735 \n",
      "Run 10 stress 0.217879 \n",
      "Run 11 stress 0.2245432 \n",
      "Run 12 stress 0.2257937 \n",
      "Run 13 stress 0.2729565 \n",
      "Run 14 stress 0.2207569 \n",
      "Run 15 stress 0.2257662 \n",
      "Run 16 stress 0.2193024 \n",
      "Run 17 stress 0.2209394 \n",
      "Run 18 stress 0.2372935 \n",
      "Run 19 stress 0.2855788 \n",
      "Run 20 stress 0.2682211 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: In estimate_richness(physeq, split = TRUE, measures = measures) :\n",
      "  The data you have provided does not have\n",
      "any singletons. This is highly suspicious. Results of richness\n",
      "estimates (for example) are probably unreliable, or wrong, if you have already\n",
      "trimmed low-abundance taxa from the data.\n",
      "\n",
      "We recommended that you find the un-trimmed data and retry.\n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n",
      "3: In estimate_richness(physeq, split = TRUE, measures = measures) :\n",
      "  The data you have provided does not have\n",
      "any singletons. This is highly suspicious. Results of richness\n",
      "estimates (for example) are probably unreliable, or wrong, if you have already\n",
      "trimmed low-abundance taxa from the data.\n",
      "\n",
      "We recommended that you find the un-trimmed data and retry.\n",
      "4: Removed 32 rows containing missing values (geom_errorbar). \n",
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-3\n",
      "Loading required package: RColorBrewer\n",
      "[1] 1.19.1\n",
      "[1] 2.2.1\n",
      "[1] 1.8.4\n",
      "[1] 2.4.3\n",
      "[1] 1.1.2\n",
      "There were 27 warnings (use warnings() to see them)\n",
      "[1] \"LinkerPrimerSequence\" \"phinchID\"             \"Treatment\"           \n",
      "[4] \"BarcodeSequence\"      \"ReversePrimer\"        \"Terrain\"             \n",
      "[1] \"Treatment\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 1837 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 1837 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 1837 tips and 1836 internal nodes ]\n",
      "Run 0 stress 0.216079 \n",
      "Run 1 stress 0.2097447 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1776696  max resid 0.3784214 \n",
      "Run 2 stress 0.2247719 \n",
      "Run 3 stress 0.2097434 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.001524468  max resid 0.004045987 \n",
      "... Similar to previous best\n",
      "Run 4 stress 0.2297339 \n",
      "Run 5 stress 0.209741 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.001173094  max resid 0.003082072 \n",
      "... Similar to previous best\n",
      "Run 6 stress 0.2206398 \n",
      "Run 7 stress 0.218001 \n",
      "Run 8 stress 0.2166116 \n",
      "Run 9 stress 0.2312629 \n",
      "Run 10 stress 0.2380523 \n",
      "Run 11 stress 0.2165922 \n",
      "Run 12 stress 0.2097415 \n",
      "... Procrustes: rmse 0.0001430018  max resid 0.0003512033 \n",
      "... Similar to previous best\n",
      "Run 13 stress 0.218001 \n",
      "Run 14 stress 0.2375179 \n",
      "Run 15 stress 0.209741 \n",
      "... Procrustes: rmse 0.000179502  max resid 0.0004230197 \n",
      "... Similar to previous best\n",
      "Run 16 stress 0.2174128 \n",
      "Run 17 stress 0.209741 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.000112718  max resid 0.000306728 \n",
      "... Similar to previous best\n",
      "Run 18 stress 0.2231275 \n",
      "Run 19 stress 0.2231275 \n",
      "Run 20 stress 0.2097412 \n",
      "... Procrustes: rmse 0.0003864128  max resid 0.0007088657 \n",
      "... Similar to previous best\n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "[1] \"Terrain\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 1837 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 1837 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 1837 tips and 1836 internal nodes ]\n",
      "Run 0 stress 0.216079 \n",
      "Run 1 stress 0.2297339 \n",
      "Run 2 stress 0.2297342 \n",
      "Run 3 stress 0.2231275 \n",
      "Run 4 stress 0.2420042 \n",
      "Run 5 stress 0.3045651 \n",
      "Run 6 stress 0.2257093 \n",
      "Run 7 stress 0.2201522 \n",
      "Run 8 stress 0.2166117 \n",
      "Run 9 stress 0.209741 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1775828  max resid 0.3779741 \n",
      "Run 10 stress 0.2215762 \n",
      "Run 11 stress 0.209741 \n",
      "... Procrustes: rmse 0.0002366778  max resid 0.0004512875 \n",
      "... Similar to previous best\n",
      "Run 12 stress 0.2293627 \n",
      "Run 13 stress 0.2165922 \n",
      "Run 14 stress 0.221575 \n",
      "Run 15 stress 0.2271196 \n",
      "Run 16 stress 0.221374 \n",
      "Run 17 stress 0.3056085 \n",
      "Run 18 stress 0.2213742 \n",
      "Run 19 stress 0.2165922 \n",
      "Run 20 stress 0.2415258 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 32 rows containing missing values (geom_errorbar). \n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n",
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-3\n",
      "Loading required package: RColorBrewer\n",
      "[1] 1.19.1\n",
      "[1] 2.2.1\n",
      "[1] 1.8.4\n",
      "[1] 2.4.3\n",
      "[1] 1.1.2\n",
      "There were 26 warnings (use warnings() to see them)\n",
      "[1] \"LinkerPrimerSequence\" \"phinchID\"             \"Treatment\"           \n",
      "[4] \"BarcodeSequence\"      \"ReversePrimer\"        \"Terrain\"             \n",
      "[1] \"Treatment\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 2245 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 2245 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 2245 tips and 2244 internal nodes ]\n",
      "Run 0 stress 0.2168399 \n",
      "Run 1 stress 0.2137511 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1709766  max resid 0.4230488 \n",
      "Run 2 stress 0.2064481 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.06260356  max resid 0.2017631 \n",
      "Run 3 stress 0.2167022 \n",
      "Run 4 stress 0.217494 \n",
      "Run 5 stress 0.2203785 \n",
      "Run 6 stress 0.2168399 \n",
      "Run 7 stress 0.2217252 \n",
      "Run 8 stress 0.2133872 \n",
      "Run 9 stress 0.231919 \n",
      "Run 10 stress 0.2209369 \n",
      "Run 11 stress 0.2795618 \n",
      "Run 12 stress 0.2163055 \n",
      "Run 13 stress 0.2133872 \n",
      "Run 14 stress 0.206448 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.000221645  max resid 0.0006128575 \n",
      "... Similar to previous best\n",
      "Run 15 stress 0.2168416 \n",
      "Run 16 stress 0.2209352 \n",
      "Run 17 stress 0.2064481 \n",
      "... Procrustes: rmse 8.180548e-05  max resid 0.0002046508 \n",
      "... Similar to previous best\n",
      "Run 18 stress 0.2163052 \n",
      "Run 19 stress 0.2413743 \n",
      "Run 20 stress 0.2208876 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "[1] \"Terrain\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 2245 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 2245 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 2245 tips and 2244 internal nodes ]\n",
      "Run 0 stress 0.2168399 \n",
      "Run 1 stress 0.206448 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1836689  max resid 0.4225627 \n",
      "Run 2 stress 0.2207717 \n",
      "Run 3 stress 0.230604 \n",
      "Run 4 stress 0.2306043 \n",
      "Run 5 stress 0.224128 \n",
      "Run 6 stress 0.2163057 \n",
      "Run 7 stress 0.2218619 \n",
      "Run 8 stress 0.2155741 \n",
      "Run 9 stress 0.2133872 \n",
      "Run 10 stress 0.2479837 \n",
      "Run 11 stress 0.2125846 \n",
      "Run 12 stress 0.2133872 \n",
      "Run 13 stress 0.2125849 \n",
      "Run 14 stress 0.2234827 \n",
      "Run 15 stress 0.220937 \n",
      "Run 16 stress 0.2455346 \n",
      "Run 17 stress 0.2125847 \n",
      "Run 18 stress 0.2196025 \n",
      "Run 19 stress 0.2196025 \n",
      "Run 20 stress 0.2133872 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 32 rows containing missing values (geom_errorbar). \n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n",
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-3\n",
      "Loading required package: RColorBrewer\n",
      "[1] 1.19.1\n",
      "[1] 2.2.1\n",
      "[1] 1.8.4\n",
      "[1] 2.4.3\n",
      "[1] 1.1.2\n",
      "[1] \"LinkerPrimerSequence\" \"phinchID\"             \"Treatment\"           \n",
      "[4] \"BarcodeSequence\"      \"ReversePrimer\"        \"Terrain\"             \n",
      "[1] \"Treatment\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 2018 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 2018 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 2018 tips and 2017 internal nodes ]\n",
      "Run 0 stress 0.2186613 \n",
      "Run 1 stress 0.2477357 \n",
      "Run 2 stress 0.3368954 \n",
      "Run 3 stress 0.2077481 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.150182  max resid 0.4745556 \n",
      "Run 4 stress 0.2377022 \n",
      "Run 5 stress 0.2077481 \n",
      "... Procrustes: rmse 3.158117e-05  max resid 7.927218e-05 \n",
      "... Similar to previous best\n",
      "Run 6 stress 0.2262956 \n",
      "Run 7 stress 0.2055306 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1845986  max resid 0.4491418 \n",
      "Run 8 stress 0.2315953 \n",
      "Run 9 stress 0.2055304 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0002385094  max resid 0.0004633285 \n",
      "... Similar to previous best\n",
      "Run 10 stress 0.2463676 \n",
      "Run 11 stress 0.2077481 \n",
      "Run 12 stress 0.2209557 \n",
      "Run 13 stress 0.2437049 \n",
      "Run 14 stress 0.2312618 \n",
      "Run 15 stress 0.2265121 \n",
      "Run 16 stress 0.2202505 \n",
      "Run 17 stress 0.2077481 \n",
      "Run 18 stress 0.2055307 \n",
      "... Procrustes: rmse 0.0003958527  max resid 0.0007536999 \n",
      "... Similar to previous best\n",
      "Run 19 stress 0.2156298 \n",
      "Run 20 stress 0.2077481 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "[1] \"Terrain\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 2018 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 2018 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 2018 tips and 2017 internal nodes ]\n",
      "Run 0 stress 0.2186613 \n",
      "Run 1 stress 0.2262953 \n",
      "Run 2 stress 0.2376967 \n",
      "Run 3 stress 0.2055309 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1795604  max resid 0.409606 \n",
      "Run 4 stress 0.2055304 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0003549852  max resid 0.0006590688 \n",
      "... Similar to previous best\n",
      "Run 5 stress 0.2077481 \n",
      "Run 6 stress 0.2336902 \n",
      "Run 7 stress 0.2240608 \n",
      "Run 8 stress 0.2121082 \n",
      "Run 9 stress 0.2156298 \n",
      "Run 10 stress 0.2569727 \n",
      "Run 11 stress 0.2262953 \n",
      "Run 12 stress 0.2055311 \n",
      "... Procrustes: rmse 0.0004427649  max resid 0.0008635218 \n",
      "... Similar to previous best\n",
      "Run 13 stress 0.2167286 \n",
      "Run 14 stress 0.2077482 \n",
      "Run 15 stress 0.2625726 \n",
      "Run 16 stress 0.2077481 \n",
      "Run 17 stress 0.2262953 \n",
      "Run 18 stress 0.2218148 \n",
      "Run 19 stress 0.205531 \n",
      "... Procrustes: rmse 0.0001995639  max resid 0.0003193689 \n",
      "... Similar to previous best\n",
      "Run 20 stress 0.2055305 \n",
      "... Procrustes: rmse 0.0001375587  max resid 0.0002348096 \n",
      "... Similar to previous best\n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: In estimate_richness(physeq, split = TRUE, measures = measures) :\n",
      "  The data you have provided does not have\n",
      "any singletons. This is highly suspicious. Results of richness\n",
      "estimates (for example) are probably unreliable, or wrong, if you have already\n",
      "trimmed low-abundance taxa from the data.\n",
      "\n",
      "We recommended that you find the un-trimmed data and retry.\n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n",
      "3: In estimate_richness(physeq, split = TRUE, measures = measures) :\n",
      "  The data you have provided does not have\n",
      "any singletons. This is highly suspicious. Results of richness\n",
      "estimates (for example) are probably unreliable, or wrong, if you have already\n",
      "trimmed low-abundance taxa from the data.\n",
      "\n",
      "We recommended that you find the un-trimmed data and retry.\n",
      "4: Removed 32 rows containing missing values (geom_errorbar). \n",
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-3\n",
      "Loading required package: RColorBrewer\n",
      "[1] 1.19.1\n",
      "[1] 2.2.1\n",
      "[1] 1.8.4\n",
      "[1] 2.4.3\n",
      "[1] 1.1.2\n",
      "There were 12 warnings (use warnings() to see them)\n",
      "[1] \"LinkerPrimerSequence\" \"phinchID\"             \"Treatment\"           \n",
      "[4] \"BarcodeSequence\"      \"ReversePrimer\"        \"Terrain\"             \n",
      "[1] \"Treatment\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 2687 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 2687 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 2687 tips and 2686 internal nodes ]\n",
      "Run 0 stress 0.21694 \n",
      "Run 1 stress 0.2144888 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1994867  max resid 0.4700255 \n",
      "Run 2 stress 0.2226733 \n",
      "Run 3 stress 0.2204988 \n",
      "Run 4 stress 0.2189726 \n",
      "Run 5 stress 0.221136 \n",
      "Run 6 stress 0.2296345 \n",
      "Run 7 stress 0.2205001 \n",
      "Run 8 stress 0.2047578 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1391797  max resid 0.4234086 \n",
      "Run 9 stress 0.2049882 \n",
      "... Procrustes: rmse 0.03399941  max resid 0.0928285 \n",
      "Run 10 stress 0.2047579 \n",
      "... Procrustes: rmse 0.0001068847  max resid 0.0002125991 \n",
      "... Similar to previous best\n",
      "Run 11 stress 0.2179587 \n",
      "Run 12 stress 0.2144882 \n",
      "Run 13 stress 0.2049789 \n",
      "... Procrustes: rmse 0.03470583  max resid 0.09347218 \n",
      "Run 14 stress 0.2047644 \n",
      "... Procrustes: rmse 0.001850522  max resid 0.004169911 \n",
      "... Similar to previous best\n",
      "Run 15 stress 0.2049808 \n",
      "... Procrustes: rmse 0.0353148  max resid 0.09490932 \n",
      "Run 16 stress 0.3350849 \n",
      "Run 17 stress 0.2205111 \n",
      "Run 18 stress 0.2080173 \n",
      "Run 19 stress 0.2080173 \n",
      "Run 20 stress 0.2222859 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "[1] \"Terrain\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 2687 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 2687 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 2687 tips and 2686 internal nodes ]\n",
      "Run 0 stress 0.21694 \n",
      "Run 1 stress 0.2211806 \n",
      "Run 2 stress 0.2461566 \n",
      "Run 3 stress 0.2047579 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1360075  max resid 0.2586879 \n",
      "Run 4 stress 0.2270945 \n",
      "Run 5 stress 0.2080173 \n",
      "Run 6 stress 0.2179585 \n",
      "Run 7 stress 0.2144882 \n",
      "Run 8 stress 0.2342749 \n",
      "Run 9 stress 0.2080173 \n",
      "Run 10 stress 0.2047582 \n",
      "... Procrustes: rmse 0.0003192569  max resid 0.0008526984 \n",
      "... Similar to previous best\n",
      "Run 11 stress 0.2226732 \n",
      "Run 12 stress 0.2494031 \n",
      "Run 13 stress 0.2123523 \n",
      "Run 14 stress 0.204759 \n",
      "... Procrustes: rmse 0.000244059  max resid 0.000633998 \n",
      "... Similar to previous best\n",
      "Run 15 stress 0.2080173 \n",
      "Run 16 stress 0.2781837 \n",
      "Run 17 stress 0.3420245 \n",
      "Run 18 stress 0.2080173 \n",
      "Run 19 stress 0.2049813 \n",
      "... Procrustes: rmse 0.03358977  max resid 0.09147085 \n",
      "Run 20 stress 0.2392322 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 32 rows containing missing values (geom_errorbar). \n",
      "2: Removed 32 rows containing missing values (geom_errorbar). \n",
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-3\n",
      "Loading required package: RColorBrewer\n",
      "[1] 1.19.1\n",
      "[1] 2.2.1\n",
      "[1] 1.8.4\n",
      "[1] 2.4.3\n",
      "[1] 1.1.2\n",
      "Warning messages:\n",
      "1: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "2: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "3: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "4: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "[1] \"LinkerPrimerSequence\" \"phinchID\"             \"Treatment\"           \n",
      "[4] \"BarcodeSequence\"      \"ReversePrimer\"        \"Terrain\"             \n",
      "[1] \"Treatment\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 1070 taxa and 14 samples ]\n",
      "sample_data() Sample Data:       [ 14 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 1070 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 1070 tips and 1069 internal nodes ]\n",
      "Run 0 stress 0.1961462 \n",
      "Run 1 stress 0.2087399 \n",
      "Run 2 stress 0.18514 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1995389  max resid 0.4447555 \n",
      "Run 3 stress 0.1958863 \n",
      "Run 4 stress 0.2021981 \n",
      "Run 5 stress 0.1851405 \n",
      "... Procrustes: rmse 0.0001542757  max resid 0.0003203161 \n",
      "... Similar to previous best\n",
      "Run 6 stress 0.2210145 \n",
      "Run 7 stress 0.1958863 \n",
      "Run 8 stress 0.18514 \n",
      "... Procrustes: rmse 6.281502e-05  max resid 0.0001131511 \n",
      "... Similar to previous best\n",
      "Run 9 stress 0.2351342 \n",
      "Run 10 stress 0.2021983 \n",
      "Run 11 stress 0.2030233 \n",
      "Run 12 stress 0.2312837 \n",
      "Run 13 stress 0.1924897 \n",
      "Run 14 stress 0.2020315 \n",
      "Run 15 stress 0.1924897 \n",
      "Run 16 stress 0.2436212 \n",
      "Run 17 stress 0.2347685 \n",
      "Run 18 stress 0.1924899 \n",
      "Run 19 stress 0.1961464 \n",
      "Run 20 stress 0.2058312 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "[1] \"Terrain\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 1070 taxa and 14 samples ]\n",
      "sample_data() Sample Data:       [ 14 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 1070 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 1070 tips and 1069 internal nodes ]\n",
      "Run 0 stress 0.1961462 \n",
      "Run 1 stress 0.2018952 \n",
      "Run 2 stress 0.2202009 \n",
      "Run 3 stress 0.215804 \n",
      "Run 4 stress 0.18514 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1995262  max resid 0.4448218 \n",
      "Run 5 stress 0.2098074 \n",
      "Run 6 stress 0.1958863 \n",
      "Run 7 stress 0.2165916 \n",
      "Run 8 stress 0.2020393 \n",
      "Run 9 stress 0.18514 \n",
      "... New best solution\n",
      "... Procrustes: rmse 5.512118e-05  max resid 0.0001005862 \n",
      "... Similar to previous best\n",
      "Run 10 stress 0.2039399 \n",
      "Run 11 stress 0.2033477 \n",
      "Run 12 stress 0.2098074 \n",
      "Run 13 stress 0.2313359 \n",
      "Run 14 stress 0.1958863 \n",
      "Run 15 stress 0.1958863 \n",
      "Run 16 stress 0.3185781 \n",
      "Run 17 stress 0.2165916 \n",
      "Run 18 stress 0.231722 \n",
      "Run 19 stress 0.2033361 \n",
      "Run 20 stress 0.1958863 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 1 rows containing missing values (geom_errorbar). \n",
      "2: Removed 28 rows containing missing values (geom_errorbar). \n",
      "3: Removed 28 rows containing missing values (geom_errorbar). \n",
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-3\n",
      "Loading required package: RColorBrewer\n",
      "[1] 1.19.1\n",
      "[1] 2.2.1\n",
      "[1] 1.8.4\n",
      "[1] 2.4.3\n",
      "[1] 1.1.2\n",
      "There were 21 warnings (use warnings() to see them)\n",
      "[1] \"LinkerPrimerSequence\" \"phinchID\"             \"Treatment\"           \n",
      "[4] \"BarcodeSequence\"      \"ReversePrimer\"        \"Terrain\"             \n",
      "[1] \"Treatment\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 1137 taxa and 14 samples ]\n",
      "sample_data() Sample Data:       [ 14 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 1137 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 1137 tips and 1136 internal nodes ]\n",
      "Run 0 stress 0.1966794 \n",
      "Run 1 stress 0.2176494 \n",
      "Run 2 stress 0.185124 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1981384  max resid 0.443374 \n",
      "Run 3 stress 0.185124 \n",
      "... Procrustes: rmse 1.366156e-05  max resid 3.143177e-05 \n",
      "... Similar to previous best\n",
      "Run 4 stress 0.218854 \n",
      "Run 5 stress 0.1961051 \n",
      "Run 6 stress 0.2093336 \n",
      "Run 7 stress 0.1930191 \n",
      "Run 8 stress 0.1851241 \n",
      "... Procrustes: rmse 0.0001102525  max resid 0.0002049658 \n",
      "... Similar to previous best\n",
      "Run 9 stress 0.2049494 \n",
      "Run 10 stress 0.2076556 \n",
      "Run 11 stress 0.2144372 \n",
      "Run 12 stress 0.2061741 \n",
      "Run 13 stress 0.2347429 \n",
      "Run 14 stress 0.2007895 \n",
      "Run 15 stress 0.2095705 \n",
      "Run 16 stress 0.1976349 \n",
      "Run 17 stress 0.203736 \n",
      "Run 18 stress 0.1961759 \n",
      "Run 19 stress 0.2240642 \n",
      "Run 20 stress 0.1851241 \n",
      "... Procrustes: rmse 6.182843e-05  max resid 0.0001349369 \n",
      "... Similar to previous best\n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "[1] \"Terrain\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 1137 taxa and 14 samples ]\n",
      "sample_data() Sample Data:       [ 14 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 1137 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 1137 tips and 1136 internal nodes ]\n",
      "Run 0 stress 0.1966794 \n",
      "Run 1 stress 0.2007895 \n",
      "Run 2 stress 0.2021774 \n",
      "Run 3 stress 0.185124 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.19815  max resid 0.4433333 \n",
      "Run 4 stress 0.2011892 \n",
      "Run 5 stress 0.2095706 \n",
      "Run 6 stress 0.2144371 \n",
      "Run 7 stress 0.1930192 \n",
      "Run 8 stress 0.185124 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.0001269224  max resid 0.0002271162 \n",
      "... Similar to previous best\n",
      "Run 9 stress 0.2376057 \n",
      "Run 10 stress 0.233406 \n",
      "Run 11 stress 0.2095715 \n",
      "Run 12 stress 0.2411007 \n",
      "Run 13 stress 0.2076555 \n",
      "Run 14 stress 0.2049487 \n",
      "Run 15 stress 0.185124 \n",
      "... New best solution\n",
      "... Procrustes: rmse 7.926077e-05  max resid 0.0001370995 \n",
      "... Similar to previous best\n",
      "Run 16 stress 0.2011875 \n",
      "Run 17 stress 0.2046308 \n",
      "Run 18 stress 0.185124 \n",
      "... Procrustes: rmse 5.206999e-05  max resid 9.038762e-05 \n",
      "... Similar to previous best\n",
      "Run 19 stress 0.2017674 \n",
      "Run 20 stress 0.3331768 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 1 rows containing missing values (geom_errorbar). \n",
      "2: Removed 28 rows containing missing values (geom_errorbar). \n",
      "3: Removed 28 rows containing missing values (geom_errorbar). \n",
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-3\n",
      "Loading required package: RColorBrewer\n",
      "[1] 1.19.1\n",
      "[1] 2.2.1\n",
      "[1] 1.8.4\n",
      "[1] 2.4.3\n",
      "[1] 1.1.2\n",
      "Warning messages:\n",
      "1: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "2: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "3: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "4: In parseFunction(i$metadata$taxonomy) :\n",
      "  Empty taxonomy vector encountered.\n",
      "[1] \"LinkerPrimerSequence\" \"phinchID\"             \"Treatment\"           \n",
      "[4] \"BarcodeSequence\"      \"ReversePrimer\"        \"Terrain\"             \n",
      "[1] \"Treatment\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 1115 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 1115 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 1115 tips and 1114 internal nodes ]\n",
      "Run 0 stress 0.2075469 \n",
      "Run 1 stress 0.2516246 \n",
      "Run 2 stress 0.2390562 \n",
      "Run 3 stress 0.2597498 \n",
      "Run 4 stress 0.2074845 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.02125412  max resid 0.05912106 \n",
      "Run 5 stress 0.226639 \n",
      "Run 6 stress 0.2626123 \n",
      "Run 7 stress 0.1998437 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1371557  max resid 0.4441748 \n",
      "Run 8 stress 0.2570884 \n",
      "Run 9 stress 0.215379 \n",
      "Run 10 stress 0.2498598 \n",
      "Run 11 stress 0.2343558 \n",
      "Run 12 stress 0.2154452 \n",
      "Run 13 stress 0.1998437 \n",
      "... Procrustes: rmse 4.35455e-06  max resid 1.206746e-05 \n",
      "... Similar to previous best\n",
      "Run 14 stress 0.2258861 \n",
      "Run 15 stress 0.2195659 \n",
      "Run 16 stress 0.2023901 \n",
      "Run 17 stress 0.22479 \n",
      "Run 18 stress 0.2074847 \n",
      "Run 19 stress 0.202274 \n",
      "Run 20 stress 0.2364237 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "[1] \"Terrain\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 1115 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 1115 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 1115 tips and 1114 internal nodes ]\n",
      "Run 0 stress 0.2075469 \n",
      "Run 1 stress 0.1992714 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.137405  max resid 0.4471815 \n",
      "Run 2 stress 0.2252009 \n",
      "Run 3 stress 0.2232934 \n",
      "Run 4 stress 0.2392665 \n",
      "Run 5 stress 0.2216141 \n",
      "Run 6 stress 0.2620879 \n",
      "Run 7 stress 0.2301314 \n",
      "Run 8 stress 0.2153788 \n",
      "Run 9 stress 0.2201253 \n",
      "Run 10 stress 0.2256043 \n",
      "Run 11 stress 0.2280324 \n",
      "Run 12 stress 0.1998437 \n",
      "Run 13 stress 0.2753839 \n",
      "Run 14 stress 0.2298626 \n",
      "Run 15 stress 0.229785 \n",
      "Run 16 stress 0.2754947 \n",
      "Run 17 stress 0.2201253 \n",
      "Run 18 stress 0.2271882 \n",
      "Run 19 stress 0.228831 \n",
      "Run 20 stress 0.2243719 \n",
      "*** No convergence -- monoMDS stopping criteria:\n",
      "    20: stress ratio > sratmax\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 32 rows containing missing values (geom_errorbar). \n",
      "2: Removed 1 rows containing missing values (geom_errorbarh). \n",
      "3: Removed 32 rows containing missing values (geom_errorbar). \n",
      "Loading required package: phyloseq\n",
      "Loading required package: methods\n",
      "Loading required package: ggplot2\n",
      "Loading required package: plyr\n",
      "Loading required package: vegan\n",
      "Loading required package: permute\n",
      "Loading required package: lattice\n",
      "This is vegan 2.4-3\n",
      "Loading required package: RColorBrewer\n",
      "[1] 1.19.1\n",
      "[1] 2.2.1\n",
      "[1] 1.8.4\n",
      "[1] 2.4.3\n",
      "[1] 1.1.2\n",
      "There were 15 warnings (use warnings() to see them)\n",
      "[1] \"LinkerPrimerSequence\" \"phinchID\"             \"Treatment\"           \n",
      "[4] \"BarcodeSequence\"      \"ReversePrimer\"        \"Terrain\"             \n",
      "[1] \"Treatment\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 1164 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 1164 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 1164 tips and 1163 internal nodes ]\n",
      "Run 0 stress 0.2071095 \n",
      "Run 1 stress 0.2049085 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1465321  max resid 0.5022419 \n",
      "Run 2 stress 0.2030622 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.06249792  max resid 0.1807088 \n",
      "Run 3 stress 0.2242981 \n",
      "Run 4 stress 0.2247938 \n",
      "Run 5 stress 0.244916 \n",
      "Run 6 stress 0.2178349 \n",
      "Run 7 stress 0.2178359 \n",
      "Run 8 stress 0.2650978 \n",
      "Run 9 stress 0.2071099 \n",
      "Run 10 stress 0.2116269 \n",
      "Run 11 stress 0.2149829 \n",
      "Run 12 stress 0.2333935 \n",
      "Run 13 stress 0.2030622 \n",
      "... Procrustes: rmse 2.56001e-05  max resid 5.394886e-05 \n",
      "... Similar to previous best\n",
      "Run 14 stress 0.2248541 \n",
      "Run 15 stress 0.2196294 \n",
      "Run 16 stress 0.2329458 \n",
      "Run 17 stress 0.2178359 \n",
      "Run 18 stress 0.20711 \n",
      "Run 19 stress 0.2281926 \n",
      "Run 20 stress 0.2178351 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "[1] \"Terrain\"\n",
      "phyloseq-class experiment-level object\n",
      "otu_table()   OTU Table:         [ 1164 taxa and 16 samples ]\n",
      "sample_data() Sample Data:       [ 16 samples by 6 sample variables ]\n",
      "tax_table()   Taxonomy Table:    [ 1164 taxa by 7 taxonomic ranks ]\n",
      "phy_tree()    Phylogenetic Tree: [ 1164 tips and 1163 internal nodes ]\n",
      "Run 0 stress 0.2071095 \n",
      "Run 1 stress 0.2219922 \n",
      "Run 2 stress 0.2123193 \n",
      "Run 3 stress 0.2030622 \n",
      "... New best solution\n",
      "... Procrustes: rmse 0.1307181  max resid 0.4410799 \n",
      "Run 4 stress 0.2064365 \n",
      "Run 5 stress 0.2064365 \n",
      "Run 6 stress 0.2242981 \n",
      "Run 7 stress 0.2043058 \n",
      "Run 8 stress 0.2064366 \n",
      "Run 9 stress 0.2196282 \n",
      "Run 10 stress 0.2178349 \n",
      "Run 11 stress 0.2544797 \n",
      "Run 12 stress 0.2219922 \n",
      "Run 13 stress 0.2111719 \n",
      "Run 14 stress 0.2030623 \n",
      "... Procrustes: rmse 7.360608e-05  max resid 0.0001210134 \n",
      "... Similar to previous best\n",
      "Run 15 stress 0.2310308 \n",
      "Run 16 stress 0.222528 \n",
      "Run 17 stress 0.217835 \n",
      "Run 18 stress 0.236328 \n",
      "Run 19 stress 0.2322431 \n",
      "Run 20 stress 0.2219481 \n",
      "*** Solution reached\n",
      "Saving 7 x 7 in image\n",
      "Saving 7 x 7 in image\n",
      "Warning messages:\n",
      "1: Removed 32 rows containing missing values (geom_errorbar). \n",
      "2: Removed 1 rows containing missing values (geom_errorbarh). \n",
      "3: Removed 32 rows containing missing values (geom_errorbar). \n"
     ]
    }
   ],
   "source": [
    "!$HOME/amptk/util/phyloseq_nmds.R miseq4a_raw.biom miseq4a_raw.tree.phy miseq4a_raw raupcrick\n",
    "!$HOME/amptk/util/phyloseq_nmds.R miseq4a.biom miseq4a.tree.phy miseq4a raupcrick\n",
    "!$HOME/amptk/util/phyloseq_nmds.R miseq4c.biom miseq4c.tree.phy miseq4c raupcrick\n",
    "!$HOME/amptk/util/phyloseq_nmds.R miseq4c_raw.biom miseq4c_raw.tree.phy miseq4c_raw raupcrick\n",
    "!$HOME/amptk/util/phyloseq_nmds.R miseq6.biom miseq6.tree.phy miseq6 raupcrick\n",
    "!$HOME/amptk/util/phyloseq_nmds.R miseq6_raw.biom miseq6_raw.tree.phy miseq6_raw raupcrick\n",
    "!$HOME/amptk/util/phyloseq_nmds.R ion3.biom ion3.tree.phy ion3 raupcrick\n",
    "!$HOME/amptk/util/phyloseq_nmds.R ion3_raw.biom ion3_raw.tree.phy ion3_raw raupcrick\n",
    "!$HOME/amptk/util/phyloseq_nmds.R ion5.biom ion5.tree.phy ion5 raupcrick\n",
    "!$HOME/amptk/util/phyloseq_nmds.R ion5_raw.biom ion5_raw.tree.phy ion5_raw raupcrick\n",
    "!$HOME/amptk/util/phyloseq_nmds.R miseq4a_raw.biom miseq4a_raw.tree.phy miseq4a_raw bray\n",
    "!$HOME/amptk/util/phyloseq_nmds.R miseq4a.biom miseq4a.tree.phy miseq4a bray\n",
    "!$HOME/amptk/util/phyloseq_nmds.R miseq4c.biom miseq4c.tree.phy miseq4c bray\n",
    "!$HOME/amptk/util/phyloseq_nmds.R miseq4c_raw.biom miseq4c_raw.tree.phy miseq4c_raw bray\n",
    "!$HOME/amptk/util/phyloseq_nmds.R miseq6.biom miseq6.tree.phy miseq6 bray\n",
    "!$HOME/amptk/util/phyloseq_nmds.R miseq6_raw.biom miseq6_raw.tree.phy miseq6_raw bray\n",
    "!$HOME/amptk/util/phyloseq_nmds.R ion3.biom ion3.tree.phy ion3 bray\n",
    "!$HOME/amptk/util/phyloseq_nmds.R ion3_raw.biom ion3_raw.tree.phy ion3_raw bray\n",
    "!$HOME/amptk/util/phyloseq_nmds.R ion5.biom ion5.tree.phy ion5 bray\n",
    "!$HOME/amptk/util/phyloseq_nmds.R ion5_raw.biom ion5_raw.tree.phy ion5_raw bray"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "!$HOME/amptk/util/parse_adonis.py stats summary_stats.csv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "| Run         | Distance  | Test       | Terrain | Treatment  |\n",
    "| ----------- | --------- | ---------- | ------- | ---------- |\n",
    "| ion3        | bray      | Adonis     | 0.0928  | 0.7441     |\n",
    "| ion3        | bray      | Betadisper | 0.4653  | 0.6512     |\n",
    "| ion3        | raupcrick | Adonis     | 0.001   | 0.1928     |\n",
    "| ion3        | raupcrick | Betadisper | 0.0039  | 0.5488     |\n",
    "| ion3_raw    | bray      | Adonis     | 0.0998  | 0.7336     |\n",
    "| ion3_raw    | bray      | Betadisper | 0.4579  | 0.6801     |\n",
    "| ion3_raw    | raupcrick | Adonis     | 0.0008  | 0.2255     |\n",
    "| ion3_raw    | raupcrick | Betadisper | 0.0001  | 0.589      |\n",
    "| ion5        | bray      | Adonis     | 0.0313  | 0.4757     |\n",
    "| ion5        | bray      | Betadisper | 0.8302  | 0.8232     |\n",
    "| ion5        | raupcrick | Adonis     | 0.0007  | 0.1047     |\n",
    "| ion5        | raupcrick | Betadisper | 0.0001  | 0.9046     |\n",
    "| ion5_raw    | bray      | Adonis     | 0.0296  | 0.4791     |\n",
    "| ion5_raw    | bray      | Betadisper | 0.8312  | 0.8274     |\n",
    "| ion5_raw    | raupcrick | Adonis     | 0.0004  | 0.2444     |\n",
    "| ion5_raw    | raupcrick | Betadisper | 0.0001  | 0.823      |\n",
    "| miseq4a     | bray      | Adonis     | 0.0186  | 0.4973     |\n",
    "| miseq4a     | bray      | Betadisper | 0.9397  | 0.9681     |\n",
    "| miseq4a     | raupcrick | Adonis     | 0.0002  | 0.1406     |\n",
    "| miseq4a     | raupcrick | Betadisper | 0.0001  | 0.5266     |\n",
    "| miseq4a_raw | bray      | Adonis     | 0.0147  | 0.4904     |\n",
    "| miseq4a_raw | bray      | Betadisper | 0.9304  | 0.9868     |\n",
    "| miseq4a_raw | raupcrick | Adonis     | 0.0004  | 0.0751     |\n",
    "| miseq4a_raw | raupcrick | Betadisper | 0.0001  | 0.4094     |\n",
    "| miseq4c     | bray      | Adonis     | 0.0161  | 0.5189     |\n",
    "| miseq4c     | bray      | Betadisper | 0.8757  | 0.9981     |\n",
    "| miseq4c     | raupcrick | Adonis     | 0.0001  | 0.1175     |\n",
    "| miseq4c     | raupcrick | Betadisper | 0.0001  | 0.3067     |\n",
    "| miseq4c_raw | bray      | Adonis     | 0.014   | 0.494      |\n",
    "| miseq4c_raw | bray      | Betadisper | 0.8587  | 0.9817     |\n",
    "| miseq4c_raw | raupcrick | Adonis     | 0.0015  | **0.0099** |\n",
    "| miseq4c_raw | raupcrick | Betadisper | 0.0001  | **0.0082** |\n",
    "| miseq6      | bray      | Adonis     | 0.0132  | 0.5091     |\n",
    "| miseq6      | bray      | Betadisper | 0.8215  | 0.9792     |\n",
    "| miseq6      | raupcrick | Adonis     | 0.0017  | **0.0124** |\n",
    "| miseq6      | raupcrick | Betadisper | 0.0001  | 0.9145     |\n",
    "| miseq6_raw  | bray      | Adonis     | 0.014   | 0.5079     |\n",
    "| miseq6_raw  | bray      | Betadisper | 0.8237  | 0.9906     |\n",
    "| miseq6_raw  | raupcrick | Adonis     | 0.0011  | 0.0546     |\n",
    "| miseq6_raw  | raupcrick | Betadisper | 0.001   | **0.0001** |"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The miseq6 data here was filtered with SynMock first (0.1%)- that is where we then \"create\" a treatment effect with the filtering, while there appears to be no treatment effect in the raw data (although it is close). If you index-bleed filter to a higher value (0.28%) then you do not see the treatment effect. This is of course only in raupcrick distance metric.  This is so dumb."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
